{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "colab_type": "text",
        "id": "view-in-github"
      },
      "source": [
        "<a href=\"https://polly.elucidata.io/manage/workspaces?action=open_polly_notebook&source=github&path=path_place_holder&kernel=elucidata/Python 3.10&machine=large\" target=\"_parent\"><img src=\"https://elucidatainc.github.io/PublicAssets/open_polly.svg\" alt=\"Open in Polly\"/></a>\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "# Welcome to Polly Python3 Notebook."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 8,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "/bin/bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)\n",
            "Collecting polly-python==0.1.6-featurelvl\n",
            "  Downloading https://elucidatainc.github.io/PublicAssets/builds/polly-python/tests/testpolly/polly_python-0.1.6_featurelvl-py3-none-any.whl (67 kB)\n",
            "\u001b[K     |████████████████████████████████| 67 kB 1.0 MB/s eta 0:00:011\n",
            "\u001b[?25hCollecting black\n",
            "  Downloading black-22.6.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.5 MB)\n",
            "\u001b[K     |████████████████████████████████| 1.5 MB 6.3 MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting pytest\n",
            "  Downloading pytest-7.1.2-py3-none-any.whl (297 kB)\n",
            "\u001b[K     |████████████████████████████████| 297 kB 142.3 MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting python-magic==0.4.24\n",
            "  Downloading python_magic-0.4.24-py2.py3-none-any.whl (12 kB)\n",
            "Collecting Cerberus==1.3.2\n",
            "  Downloading Cerberus-1.3.2.tar.gz (52 kB)\n",
            "\u001b[K     |████████████████████████████████| 52 kB 220 kB/s s eta 0:00:01\n",
            "\u001b[?25hCollecting cmapPy\n",
            "  Downloading cmapPy-4.0.1-py2.py3-none-any.whl (150 kB)\n",
            "\u001b[K     |████████████████████████████████| 150 kB 133.5 MB/s eta 0:00:01\n",
            "\u001b[?25hRequirement already satisfied: chardet==4.0.0 in /usr/local/lib/python3.10/site-packages (from polly-python==0.1.6-featurelvl) (4.0.0)\n",
            "Requirement already satisfied: pandas in /usr/local/lib/python3.10/site-packages (from polly-python==0.1.6-featurelvl) (1.4.1)\n",
            "Collecting requests==2.25.1\n",
            "  Downloading requests-2.25.1-py2.py3-none-any.whl (61 kB)\n",
            "\u001b[K     |████████████████████████████████| 61 kB 2.5 MB/s s eta 0:00:01\n",
            "\u001b[?25hCollecting certifi==2021.10.8\n",
            "  Downloading certifi-2021.10.8-py2.py3-none-any.whl (149 kB)\n",
            "\u001b[K     |████████████████████████████████| 149 kB 135.9 MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting postpy2==0.0.6\n",
            "  Downloading postpy2-0.0.6-py3-none-any.whl (17 kB)\n",
            "Collecting pytz==2021.1\n",
            "  Downloading pytz-2021.1-py2.py3-none-any.whl (510 kB)\n",
            "\u001b[K     |████████████████████████████████| 510 kB 124.9 MB/s eta 0:00:01\n",
            "\u001b[?25hRequirement already satisfied: botocore>=1.20.73 in /usr/local/lib/python3.10/site-packages (from polly-python==0.1.6-featurelvl) (1.24.19)\n",
            "Collecting joblib\n",
            "  Downloading joblib-1.1.0-py2.py3-none-any.whl (306 kB)\n",
            "\u001b[K     |████████████████████████████████| 306 kB 132.3 MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting retrying==1.3.3\n",
            "  Downloading retrying-1.3.3.tar.gz (10 kB)\n",
            "Requirement already satisfied: idna==2.10 in /usr/local/lib/python3.10/site-packages (from polly-python==0.1.6-featurelvl) (2.10)\n",
            "Requirement already satisfied: tqdm in /usr/local/lib/python3.10/site-packages (from polly-python==0.1.6-featurelvl) (4.64.0)\n",
            "Collecting Deprecated\n",
            "  Downloading Deprecated-1.2.13-py2.py3-none-any.whl (9.6 kB)\n",
            "Collecting elucidatacloudpathlib==0.6.6\n",
            "  Downloading elucidatacloudpathlib-0.6.6-py3-none-any.whl (50 kB)\n",
            "\u001b[K     |████████████████████████████████| 50 kB 1.8 MB/s  eta 0:00:01\n",
            "\u001b[?25hRequirement already satisfied: six==1.16.0 in /usr/local/lib/python3.10/site-packages (from polly-python==0.1.6-featurelvl) (1.16.0)\n",
            "Requirement already satisfied: boto3>=1.17.73 in /usr/local/lib/python3.10/site-packages (from polly-python==0.1.6-featurelvl) (1.21.19)\n",
            "Requirement already satisfied: urllib3==1.26.6 in /usr/local/lib/python3.10/site-packages (from polly-python==0.1.6-featurelvl) (1.26.6)\n",
            "Collecting rst2txt\n",
            "  Downloading rst2txt-1.1.0-py2.py3-none-any.whl (12 kB)\n",
            "Requirement already satisfied: python-dateutil==2.8.1 in /usr/local/lib/python3.10/site-packages (from polly-python==0.1.6-featurelvl) (2.8.1)\n",
            "Collecting sqlparse\n",
            "  Downloading sqlparse-0.4.2-py3-none-any.whl (42 kB)\n",
            "\u001b[K     |████████████████████████████████| 42 kB 425 kB/s s eta 0:00:01\n",
            "\u001b[?25hRequirement already satisfied: setuptools in /usr/local/lib/python3.10/site-packages (from Cerberus==1.3.2->polly-python==0.1.6-featurelvl) (58.1.0)\n",
            "Requirement already satisfied: s3transfer<0.6.0,>=0.5.0 in /usr/local/lib/python3.10/site-packages (from boto3>=1.17.73->polly-python==0.1.6-featurelvl) (0.5.2)\n",
            "Requirement already satisfied: jmespath<1.0.0,>=0.7.1 in /usr/local/lib/python3.10/site-packages (from boto3>=1.17.73->polly-python==0.1.6-featurelvl) (0.10.0)\n",
            "Collecting pathspec>=0.9.0\n",
            "  Downloading pathspec-0.9.0-py2.py3-none-any.whl (31 kB)\n",
            "Collecting tomli>=1.1.0\n",
            "  Downloading tomli-2.0.1-py3-none-any.whl (12 kB)\n",
            "Collecting mypy-extensions>=0.4.3\n",
            "  Downloading mypy_extensions-0.4.3-py2.py3-none-any.whl (4.5 kB)\n",
            "Collecting click>=8.0.0\n",
            "  Downloading click-8.1.3-py3-none-any.whl (96 kB)\n",
            "\u001b[K     |████████████████████████████████| 96 kB 1.8 MB/s s eta 0:00:01\n",
            "\u001b[?25hCollecting platformdirs>=2\n",
            "  Downloading platformdirs-2.5.2-py3-none-any.whl (14 kB)\n",
            "Collecting h5py>=2.6.0\n",
            "  Downloading h5py-3.7.0-cp310-cp310-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (4.5 MB)\n",
            "\u001b[K     |████████████████████████████████| 4.5 MB 115.2 MB/s eta 0:00:01\n",
            "\u001b[?25hRequirement already satisfied: numpy>=1.11.2 in /usr/local/lib/python3.10/site-packages (from cmapPy->polly-python==0.1.6-featurelvl) (1.22.3)\n",
            "Collecting wrapt<2,>=1.10\n",
            "  Downloading wrapt-1.14.1-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (77 kB)\n",
            "\u001b[K     |████████████████████████████████| 77 kB 2.1 MB/s s eta 0:00:01\n",
            "\u001b[?25hRequirement already satisfied: packaging in /usr/local/lib/python3.10/site-packages (from pytest->polly-python==0.1.6-featurelvl) (21.3)\n",
            "Collecting py>=1.8.2\n",
            "  Downloading py-1.11.0-py2.py3-none-any.whl (98 kB)\n",
            "\u001b[K     |████████████████████████████████| 98 kB 2.1 MB/s s eta 0:00:01\n",
            "\u001b[?25hRequirement already satisfied: attrs>=19.2.0 in /usr/local/lib/python3.10/site-packages (from pytest->polly-python==0.1.6-featurelvl) (21.2.0)\n",
            "Collecting pluggy<2.0,>=0.12\n",
            "  Downloading pluggy-1.0.0-py2.py3-none-any.whl (13 kB)\n",
            "Collecting iniconfig\n",
            "  Downloading iniconfig-1.1.1-py2.py3-none-any.whl (5.0 kB)\n",
            "Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in /usr/local/lib/python3.10/site-packages (from packaging->pytest->polly-python==0.1.6-featurelvl) (3.0.6)\n",
            "Requirement already satisfied: pygments in /usr/local/lib/python3.10/site-packages (from rst2txt->polly-python==0.1.6-featurelvl) (2.10.0)\n",
            "Requirement already satisfied: docutils>=0.11 in /usr/local/lib/python3.10/site-packages (from rst2txt->polly-python==0.1.6-featurelvl) (0.15.2)\n",
            "Using legacy 'setup.py install' for Cerberus, since package 'wheel' is not installed.\n",
            "Using legacy 'setup.py install' for retrying, since package 'wheel' is not installed.\n",
            "Installing collected packages: pytz, certifi, wrapt, tomli, requests, python-magic, py, pluggy, platformdirs, pathspec, mypy-extensions, iniconfig, h5py, click, sqlparse, rst2txt, retrying, pytest, postpy2, joblib, elucidatacloudpathlib, Deprecated, cmapPy, Cerberus, black, polly-python\n",
            "  Attempting uninstall: pytz\n",
            "    Found existing installation: pytz 2021.3\n",
            "    Uninstalling pytz-2021.3:\n",
            "      Successfully uninstalled pytz-2021.3\n",
            "  Attempting uninstall: certifi\n",
            "    Found existing installation: certifi 2021.5.30\n",
            "    Uninstalling certifi-2021.5.30:\n",
            "      Successfully uninstalled certifi-2021.5.30\n",
            "  Attempting uninstall: requests\n",
            "    Found existing installation: requests 2.27.1\n",
            "    Uninstalling requests-2.27.1:\n",
            "      Successfully uninstalled requests-2.27.1\n",
            "    Running setup.py install for retrying ... \u001b[?25ldone\n",
            "\u001b[?25h  Attempting uninstall: Cerberus\n",
            "    Found existing installation: Cerberus 1.3.4\n",
            "    Uninstalling Cerberus-1.3.4:\n",
            "      Successfully uninstalled Cerberus-1.3.4\n",
            "    Running setup.py install for Cerberus ... \u001b[?25ldone\n",
            "\u001b[?25hSuccessfully installed Cerberus-1.3.2 Deprecated-1.2.13 black-22.6.0 certifi-2021.10.8 click-8.1.3 cmapPy-4.0.1 elucidatacloudpathlib-0.6.6 h5py-3.7.0 iniconfig-1.1.1 joblib-1.1.0 mypy-extensions-0.4.3 pathspec-0.9.0 platformdirs-2.5.2 pluggy-1.0.0 polly-python-0.1.6 postpy2-0.0.6 py-1.11.0 pytest-7.1.2 python-magic-0.4.24 pytz-2021.1 requests-2.25.1 retrying-1.3.3 rst2txt-1.1.0 sqlparse-0.4.2 tomli-2.0.1 wrapt-1.14.1\n",
            "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\n",
            "\u001b[33mWARNING: You are using pip version 21.2.4; however, version 22.2.1 is available.\n",
            "You should consider upgrading via the '/usr/local/bin/python3.10 -m pip install --upgrade pip' command.\u001b[0m\n"
          ]
        }
      ],
      "source": [
        "!sudo pip3 install https://elucidatainc.github.io/PublicAssets/builds/polly-python/tests/testpolly/polly_python-0.1.6_featurelvl-py3-none-any.whl"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 9,
      "metadata": {},
      "outputs": [],
      "source": [
        "import os\n",
        "from polly.auth import Polly\n",
        "AUTH_TOKEN=(os.environ['POLLY_REFRESH_TOKEN']) # Obtain authentication tokens\n",
        "Polly.auth(AUTH_TOKEN)\n",
        "from polly.omixatlas import OmixAtlas\n",
        "omixatlas = OmixAtlas()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 10,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "--schema table key---datasets\n",
            "--schema table key---samples\n",
            "-----dataset payload------\n",
            "   Source Datatype             Field Name  \\\n",
            "0     all      all  data_matrix_available   \n",
            "1     all      all       curated_organism   \n",
            "2     all      all                src_uri   \n",
            "3     all      all      total_num_samples   \n",
            "4     all      all                   year   \n",
            "5     all      all            description   \n",
            "6     all      all      curated_cell_line   \n",
            "7     all      all        data_table_name   \n",
            "8     all      all     data_table_version   \n",
            "9     all      all               platform   \n",
            "10    all      all             timestamp_   \n",
            "11    all      all          subseries_ids   \n",
            "12    all      all              file_type   \n",
            "13    all      all            publication   \n",
            "14    all      all        missing_samples   \n",
            "15    all      all      curated_cell_type   \n",
            "16    all      all                    key   \n",
            "17    all      all                summary   \n",
            "18    all      all               src_repo   \n",
            "19    all      all            drug_smiles   \n",
            "20    all      all                package   \n",
            "21    all      all          file_location   \n",
            "22    all      all                 author   \n",
            "23    all      all             dataset_id   \n",
            "24    all      all        curated_disease   \n",
            "25    all      all           curated_drug   \n",
            "26    all      all           curated_gene   \n",
            "27    all      all               abstract   \n",
            "28    all      all                version   \n",
            "29    all      all         curated_strain   \n",
            "30    all      all                 bucket   \n",
            "31    all      all         curated_tissue   \n",
            "32    all      all         dataset_source   \n",
            "33    all      all              data_type   \n",
            "34    all      all         overall_design   \n",
            "35    all      all             is_current   \n",
            "36    all      all                 region   \n",
            "\n",
            "                                    Field Description Field Type  Is Curated  \\\n",
            "0                               Data Matrix Available       text       False   \n",
            "1         Orgnism from which the samples were derived       text        True   \n",
            "2     Unique URI derived from data file's S3 location       text       False   \n",
            "3                Total number of samples in a dataset    integer       False   \n",
            "4             Year in which the dataset was published       text       False   \n",
            "5                          Description of the dataset       text       False   \n",
            "6   Cell lines from which the samples were derived...       text        True   \n",
            "7    Name of the data table associated with data file       text       False   \n",
            "8   Current version of the data table associated w...    integer       False   \n",
            "9   Platform Accession number signifying the techn...       text        True   \n",
            "10  Unix timestamp denoting time of creation for t...       text       False   \n",
            "11              GSE IDs associated with a SuperSeries       text       False   \n",
            "12                                   Data file's type       text       False   \n",
            "13            Publication associated with the dataset       text       False   \n",
            "14  Additional sample IDs that are available at so...       text       False   \n",
            "15               Types of cell present in the dataset       text        True   \n",
            "16             Key to the data file in source package       text       False   \n",
            "17                               Summary of the paper       text       False   \n",
            "18  Name of the repository this data file originat...       text       False   \n",
            "19                            SMILES code of the drug       text        True   \n",
            "20     Name of the package this data file was part of       text       False   \n",
            "21              Unique S3 location for this data file       text       False   \n",
            "22       Name of the author who published the dataset       text       False   \n",
            "23            Unique ID assocaited with every dataset       text        True   \n",
            "24                Disease associated with the dataset       text        True   \n",
            "25  Drugs administered in the samples belonging to...       text        True   \n",
            "26                        Gene studied in the dataset       text        True   \n",
            "27  Abstract of the publication assocaited with th...       text       False   \n",
            "28                               Version of data file    integer       False   \n",
            "29  Strain of the organism from which samples were...       text        True   \n",
            "30           S3 bucket in which the data file resides       text       False   \n",
            "31        Tissue from which the samples were derivved       text        True   \n",
            "32             Source from where the data was fetched       text        True   \n",
            "33  The type of biomolecular data captured (eg - E...       text        True   \n",
            "34  Overall Design of the experiment as given by t...       text       False   \n",
            "35  Whether this is the current version of the dat...       text       False   \n",
            "36         Region (AWS) in which this data is located       text       False   \n",
            "\n",
            "    Is Array  \n",
            "0      False  \n",
            "1       True  \n",
            "2      False  \n",
            "3      False  \n",
            "4      False  \n",
            "5      False  \n",
            "6       True  \n",
            "7      False  \n",
            "8      False  \n",
            "9      False  \n",
            "10     False  \n",
            "11      True  \n",
            "12     False  \n",
            "13     False  \n",
            "14      True  \n",
            "15      True  \n",
            "16     False  \n",
            "17     False  \n",
            "18     False  \n",
            "19      True  \n",
            "20     False  \n",
            "21     False  \n",
            "22     False  \n",
            "23     False  \n",
            "24      True  \n",
            "25      True  \n",
            "26      True  \n",
            "27     False  \n",
            "28     False  \n",
            "29      True  \n",
            "30     False  \n",
            "31      True  \n",
            "32     False  \n",
            "33     False  \n",
            "34     False  \n",
            "35     False  \n",
            "36     False  \n",
            "\n",
            "\n",
            "-----sample payload-----\n",
            "   Source Datatype                         Field Name  \\\n",
            "0     all      all                growth_protocol_ch1   \n",
            "1     all      all                            src_uri   \n",
            "2     all      all                          sample_id   \n",
            "3     all      all              curated_gene_modified   \n",
            "4     all      all                           dose_ch1   \n",
            "5     all      all                curated_cohort_name   \n",
            "6     all      all                    curated_control   \n",
            "7     all      all                     src_dataset_id   \n",
            "8     all      all               extract_protocol_ch1   \n",
            "9     all      all                characteristics_ch2   \n",
            "10    all      all                characteristics_ch1   \n",
            "11    all      all               extract_protocol_ch2   \n",
            "12    all      all                            age_ch1   \n",
            "13    all      all                  curated_cell_type   \n",
            "14    all      all                            sex_ch1   \n",
            "15    all      all                      treatment_ch1   \n",
            "16    all      all                         dataset_id   \n",
            "17    all      all                          taxid_ch1   \n",
            "18    all      all                       curated_gene   \n",
            "19    all      all                       curated_drug   \n",
            "20    all      all                            version   \n",
            "21    all      all                 label_protocol_ch2   \n",
            "22    all      all                 label_protocol_ch1   \n",
            "23    all      all             treatment_protocol_ch2   \n",
            "24    all      all              characteristics_ch2_1   \n",
            "25    all      all             treatment_protocol_ch1   \n",
            "26    all      all                   instrument_model   \n",
            "27    all      all                               name   \n",
            "28    all      all  curated_genetic_modification_type   \n",
            "29    all      all                            bmi_ch1   \n",
            "30    all      all                    data_processing   \n",
            "31    all      all                  curated_cell_line   \n",
            "32    all      all                        description   \n",
            "33    all      all                              title   \n",
            "34    all      all                         timestamp_   \n",
            "35    all      all                  curated_cohort_id   \n",
            "36    all      all                            data_id   \n",
            "37    all      all                           src_repo   \n",
            "38    all      all                             id_key   \n",
            "39    all      all                         gender_ch1   \n",
            "40    all      all                 smoking_status_ch1   \n",
            "41    all      all                          label_ch2   \n",
            "42    all      all                    curated_disease   \n",
            "43    all      all                          label_ch1   \n",
            "44    all      all                     time_point_ch1   \n",
            "45    all      all              characteristics_ch1_3   \n",
            "46    all      all              characteristics_ch1_2   \n",
            "47    all      all                     curated_tissue   \n",
            "48    all      all           curated_drug_smiles_code   \n",
            "49    all      all                       hyb_protocol   \n",
            "50    all      all                        platform_id   \n",
            "51    all      all                         is_current   \n",
            "52    all      all              characteristics_ch1_1   \n",
            "\n",
            "                                    Field Description Field Type  Is Curated  \\\n",
            "0                                                  NA       text       False   \n",
            "1   Unique URI derived from source data file's S3 ...       text       False   \n",
            "2              Unique ID associated with every sample       text       False   \n",
            "3          Gene modified through genetic modification       text       False   \n",
            "4                                                  NA       text       False   \n",
            "5      Name of the cohort to which the sample belongs       text       False   \n",
            "6   Signifies whether the given sample is a contro...    integer       False   \n",
            "7   Dataset ID of the file this data entity origin...       text       False   \n",
            "8                                                  NA       text       False   \n",
            "9                                                  NA       text       False   \n",
            "10                                                 NA       text       False   \n",
            "11                                                 NA       text       False   \n",
            "12                                                 NA       text       False   \n",
            "13                Types of cell present in the sample       text       False   \n",
            "14                                                 NA       text       False   \n",
            "15                                                 NA       text       False   \n",
            "16  Unique ID of the dataset to which the sample b...       text       False   \n",
            "17                                                 NA       text       False   \n",
            "18                     Gene of interest in the sample       text       False   \n",
            "19                      Drug admistered in the sample       text       False   \n",
            "20                             Version of data entity    integer       False   \n",
            "21                                                 NA       text       False   \n",
            "22                                                 NA       text       False   \n",
            "23                                                 NA       text       False   \n",
            "24                                                 NA       text       False   \n",
            "25                                                 NA       text       False   \n",
            "26                                                 NA       text       False   \n",
            "27  Name of the data entity (local to its source f...       text       False   \n",
            "28  Signifies the kind of genetic modification per...       text       False   \n",
            "29                                                 NA       text       False   \n",
            "30                                                 NA       text       False   \n",
            "31        Cell line from which the sample was derived       text       False   \n",
            "32                          Description of the sample       text       False   \n",
            "33                                                 NA       text       False   \n",
            "34  Unix timestamp denoting time of creation for t...       text       False   \n",
            "35  Numeric ID of the cohort to which the sample b...    integer       False   \n",
            "36            Unique ID for this data entity on Polly       text       False   \n",
            "37  Name of the repository this data entity origin...       text       False   \n",
            "38  Name of the key that was used for creation of ...       text       False   \n",
            "39                                                 NA       text       False   \n",
            "40                                                 NA       text       False   \n",
            "41                                                 NA       text       False   \n",
            "42                 Disease associated with the sample       text       False   \n",
            "43                                                 NA       text       False   \n",
            "44                                                 NA       text       False   \n",
            "45                                                 NA       text       False   \n",
            "46                                                 NA       text       False   \n",
            "47                 Tissue to which the sample belongs       text       False   \n",
            "48   SMILES code of the drug admistered in the sample       text       False   \n",
            "49                                                 NA       text       False   \n",
            "50                                                 NA       text       False   \n",
            "51  Whether this is the current version of the dat...       text       False   \n",
            "52                                                 NA       text       False   \n",
            "\n",
            "    Is Array  \n",
            "0      False  \n",
            "1      False  \n",
            "2      False  \n",
            "3       True  \n",
            "4      False  \n",
            "5      False  \n",
            "6      False  \n",
            "7      False  \n",
            "8      False  \n",
            "9      False  \n",
            "10     False  \n",
            "11     False  \n",
            "12     False  \n",
            "13      True  \n",
            "14     False  \n",
            "15     False  \n",
            "16      True  \n",
            "17     False  \n",
            "18      True  \n",
            "19      True  \n",
            "20     False  \n",
            "21     False  \n",
            "22     False  \n",
            "23     False  \n",
            "24     False  \n",
            "25     False  \n",
            "26     False  \n",
            "27     False  \n",
            "28      True  \n",
            "29     False  \n",
            "30     False  \n",
            "31     False  \n",
            "32     False  \n",
            "33     False  \n",
            "34     False  \n",
            "35     False  \n",
            "36     False  \n",
            "37     False  \n",
            "38     False  \n",
            "39     False  \n",
            "40     False  \n",
            "41     False  \n",
            "42      True  \n",
            "43     False  \n",
            "44     False  \n",
            "45     False  \n",
            "46     False  \n",
            "47     False  \n",
            "48      True  \n",
            "49     False  \n",
            "50     False  \n",
            "51     False  \n",
            "52     False  \n"
          ]
        }
      ],
      "source": [
        "# testing point 1 -> passed\n",
        "# table names passed \n",
        "repo_key_1 = \"geo\"\n",
        "resp = omixatlas.get_schema(repo_key_1, [\"datasets\", \"samples\"], source=\"all\", data_type=\"all\")\n",
        "print(\"-----dataset payload------\")\n",
        "print(resp.datasets)\n",
        "print(\"\\n\")\n",
        "print(\"-----sample payload-----\")\n",
        "print(resp.samples)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 12,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "--schema table key---dataset\n",
            "-----dataset payload------\n",
            "   Source Datatype             Field Name  \\\n",
            "0     all      all  data_matrix_available   \n",
            "1     all      all       curated_organism   \n",
            "2     all      all                src_uri   \n",
            "3     all      all      total_num_samples   \n",
            "4     all      all                   year   \n",
            "5     all      all            description   \n",
            "6     all      all      curated_cell_line   \n",
            "7     all      all        data_table_name   \n",
            "8     all      all     data_table_version   \n",
            "9     all      all               platform   \n",
            "10    all      all             timestamp_   \n",
            "11    all      all          subseries_ids   \n",
            "12    all      all              file_type   \n",
            "13    all      all            publication   \n",
            "14    all      all        missing_samples   \n",
            "15    all      all      curated_cell_type   \n",
            "16    all      all                    key   \n",
            "17    all      all                summary   \n",
            "18    all      all               src_repo   \n",
            "19    all      all            drug_smiles   \n",
            "20    all      all                package   \n",
            "21    all      all          file_location   \n",
            "22    all      all                 author   \n",
            "23    all      all             dataset_id   \n",
            "24    all      all        curated_disease   \n",
            "25    all      all           curated_drug   \n",
            "26    all      all           curated_gene   \n",
            "27    all      all               abstract   \n",
            "28    all      all                version   \n",
            "29    all      all         curated_strain   \n",
            "30    all      all                 bucket   \n",
            "31    all      all         curated_tissue   \n",
            "32    all      all         dataset_source   \n",
            "33    all      all              data_type   \n",
            "34    all      all         overall_design   \n",
            "35    all      all             is_current   \n",
            "36    all      all                 region   \n",
            "\n",
            "                                    Field Description Field Type  Is Curated  \\\n",
            "0                               Data Matrix Available       text       False   \n",
            "1         Orgnism from which the samples were derived       text        True   \n",
            "2     Unique URI derived from data file's S3 location       text       False   \n",
            "3                Total number of samples in a dataset    integer       False   \n",
            "4             Year in which the dataset was published       text       False   \n",
            "5                          Description of the dataset       text       False   \n",
            "6   Cell lines from which the samples were derived...       text        True   \n",
            "7    Name of the data table associated with data file       text       False   \n",
            "8   Current version of the data table associated w...    integer       False   \n",
            "9   Platform Accession number signifying the techn...       text        True   \n",
            "10  Unix timestamp denoting time of creation for t...       text       False   \n",
            "11              GSE IDs associated with a SuperSeries       text       False   \n",
            "12                                   Data file's type       text       False   \n",
            "13            Publication associated with the dataset       text       False   \n",
            "14  Additional sample IDs that are available at so...       text       False   \n",
            "15               Types of cell present in the dataset       text        True   \n",
            "16             Key to the data file in source package       text       False   \n",
            "17                               Summary of the paper       text       False   \n",
            "18  Name of the repository this data file originat...       text       False   \n",
            "19                            SMILES code of the drug       text        True   \n",
            "20     Name of the package this data file was part of       text       False   \n",
            "21              Unique S3 location for this data file       text       False   \n",
            "22       Name of the author who published the dataset       text       False   \n",
            "23            Unique ID assocaited with every dataset       text        True   \n",
            "24                Disease associated with the dataset       text        True   \n",
            "25  Drugs administered in the samples belonging to...       text        True   \n",
            "26                        Gene studied in the dataset       text        True   \n",
            "27  Abstract of the publication assocaited with th...       text       False   \n",
            "28                               Version of data file    integer       False   \n",
            "29  Strain of the organism from which samples were...       text        True   \n",
            "30           S3 bucket in which the data file resides       text       False   \n",
            "31        Tissue from which the samples were derivved       text        True   \n",
            "32             Source from where the data was fetched       text        True   \n",
            "33  The type of biomolecular data captured (eg - E...       text        True   \n",
            "34  Overall Design of the experiment as given by t...       text       False   \n",
            "35  Whether this is the current version of the dat...       text       False   \n",
            "36         Region (AWS) in which this data is located       text       False   \n",
            "\n",
            "    Is Array  \n",
            "0      False  \n",
            "1       True  \n",
            "2      False  \n",
            "3      False  \n",
            "4      False  \n",
            "5      False  \n",
            "6       True  \n",
            "7      False  \n",
            "8      False  \n",
            "9      False  \n",
            "10     False  \n",
            "11      True  \n",
            "12     False  \n",
            "13     False  \n",
            "14      True  \n",
            "15      True  \n",
            "16     False  \n",
            "17     False  \n",
            "18     False  \n",
            "19      True  \n",
            "20     False  \n",
            "21     False  \n",
            "22     False  \n",
            "23     False  \n",
            "24      True  \n",
            "25      True  \n",
            "26      True  \n",
            "27     False  \n",
            "28     False  \n",
            "29      True  \n",
            "30     False  \n",
            "31      True  \n",
            "32     False  \n",
            "33     False  \n",
            "34     False  \n",
            "35     False  \n",
            "36     False  \n"
          ]
        }
      ],
      "source": [
        "# testing point 2 -> passed\n",
        "# valid schema level given as param\n",
        "repo_key_1 = \"geo\"\n",
        "resp = omixatlas.get_schema(repo_key_1, [\"dataset\"], source=\"all\", data_type=\"all\")\n",
        "print(\"-----dataset payload------\")\n",
        "print(resp.dataset)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 13,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "--schema table key---datasets\n",
            "--schema table key---samples\n",
            "--schema table key---features\n",
            "-----dataset payload------\n",
            "   Source Datatype             Field Name  \\\n",
            "0     all      all  data_matrix_available   \n",
            "1     all      all       curated_organism   \n",
            "2     all      all                src_uri   \n",
            "3     all      all      total_num_samples   \n",
            "4     all      all                   year   \n",
            "5     all      all            description   \n",
            "6     all      all      curated_cell_line   \n",
            "7     all      all        data_table_name   \n",
            "8     all      all     data_table_version   \n",
            "9     all      all               platform   \n",
            "10    all      all             timestamp_   \n",
            "11    all      all          subseries_ids   \n",
            "12    all      all              file_type   \n",
            "13    all      all            publication   \n",
            "14    all      all        missing_samples   \n",
            "15    all      all      curated_cell_type   \n",
            "16    all      all                    key   \n",
            "17    all      all                summary   \n",
            "18    all      all               src_repo   \n",
            "19    all      all            drug_smiles   \n",
            "20    all      all                package   \n",
            "21    all      all          file_location   \n",
            "22    all      all                 author   \n",
            "23    all      all             dataset_id   \n",
            "24    all      all        curated_disease   \n",
            "25    all      all           curated_drug   \n",
            "26    all      all           curated_gene   \n",
            "27    all      all               abstract   \n",
            "28    all      all                version   \n",
            "29    all      all         curated_strain   \n",
            "30    all      all                 bucket   \n",
            "31    all      all         curated_tissue   \n",
            "32    all      all         dataset_source   \n",
            "33    all      all              data_type   \n",
            "34    all      all         overall_design   \n",
            "35    all      all             is_current   \n",
            "36    all      all                 region   \n",
            "\n",
            "                                    Field Description Field Type  Is Curated  \\\n",
            "0                               Data Matrix Available       text       False   \n",
            "1         Orgnism from which the samples were derived       text        True   \n",
            "2     Unique URI derived from data file's S3 location       text       False   \n",
            "3                Total number of samples in a dataset    integer       False   \n",
            "4             Year in which the dataset was published       text       False   \n",
            "5                          Description of the dataset       text       False   \n",
            "6   Cell lines from which the samples were derived...       text        True   \n",
            "7    Name of the data table associated with data file       text       False   \n",
            "8   Current version of the data table associated w...    integer       False   \n",
            "9   Platform Accession number signifying the techn...       text        True   \n",
            "10  Unix timestamp denoting time of creation for t...       text       False   \n",
            "11              GSE IDs associated with a SuperSeries       text       False   \n",
            "12                                   Data file's type       text       False   \n",
            "13            Publication associated with the dataset       text       False   \n",
            "14  Additional sample IDs that are available at so...       text       False   \n",
            "15               Types of cell present in the dataset       text        True   \n",
            "16             Key to the data file in source package       text       False   \n",
            "17                               Summary of the paper       text       False   \n",
            "18  Name of the repository this data file originat...       text       False   \n",
            "19                            SMILES code of the drug       text        True   \n",
            "20     Name of the package this data file was part of       text       False   \n",
            "21              Unique S3 location for this data file       text       False   \n",
            "22       Name of the author who published the dataset       text       False   \n",
            "23            Unique ID assocaited with every dataset       text        True   \n",
            "24                Disease associated with the dataset       text        True   \n",
            "25  Drugs administered in the samples belonging to...       text        True   \n",
            "26                        Gene studied in the dataset       text        True   \n",
            "27  Abstract of the publication assocaited with th...       text       False   \n",
            "28                               Version of data file    integer       False   \n",
            "29  Strain of the organism from which samples were...       text        True   \n",
            "30           S3 bucket in which the data file resides       text       False   \n",
            "31        Tissue from which the samples were derivved       text        True   \n",
            "32             Source from where the data was fetched       text        True   \n",
            "33  The type of biomolecular data captured (eg - E...       text        True   \n",
            "34  Overall Design of the experiment as given by t...       text       False   \n",
            "35  Whether this is the current version of the dat...       text       False   \n",
            "36         Region (AWS) in which this data is located       text       False   \n",
            "\n",
            "    Is Array  \n",
            "0      False  \n",
            "1       True  \n",
            "2      False  \n",
            "3      False  \n",
            "4      False  \n",
            "5      False  \n",
            "6       True  \n",
            "7      False  \n",
            "8      False  \n",
            "9      False  \n",
            "10     False  \n",
            "11      True  \n",
            "12     False  \n",
            "13     False  \n",
            "14      True  \n",
            "15      True  \n",
            "16     False  \n",
            "17     False  \n",
            "18     False  \n",
            "19      True  \n",
            "20     False  \n",
            "21     False  \n",
            "22     False  \n",
            "23     False  \n",
            "24      True  \n",
            "25      True  \n",
            "26      True  \n",
            "27     False  \n",
            "28     False  \n",
            "29      True  \n",
            "30     False  \n",
            "31      True  \n",
            "32     False  \n",
            "33     False  \n",
            "34     False  \n",
            "35     False  \n",
            "36     False  \n",
            "------sample payload-----\n",
            "   Source Datatype                         Field Name  \\\n",
            "0     all      all                growth_protocol_ch1   \n",
            "1     all      all                            src_uri   \n",
            "2     all      all                          sample_id   \n",
            "3     all      all              curated_gene_modified   \n",
            "4     all      all                           dose_ch1   \n",
            "5     all      all                curated_cohort_name   \n",
            "6     all      all                    curated_control   \n",
            "7     all      all                     src_dataset_id   \n",
            "8     all      all               extract_protocol_ch1   \n",
            "9     all      all                characteristics_ch2   \n",
            "10    all      all                characteristics_ch1   \n",
            "11    all      all               extract_protocol_ch2   \n",
            "12    all      all                            age_ch1   \n",
            "13    all      all                  curated_cell_type   \n",
            "14    all      all                            sex_ch1   \n",
            "15    all      all                      treatment_ch1   \n",
            "16    all      all                         dataset_id   \n",
            "17    all      all                          taxid_ch1   \n",
            "18    all      all                       curated_gene   \n",
            "19    all      all                       curated_drug   \n",
            "20    all      all                            version   \n",
            "21    all      all                 label_protocol_ch2   \n",
            "22    all      all                 label_protocol_ch1   \n",
            "23    all      all             treatment_protocol_ch2   \n",
            "24    all      all              characteristics_ch2_1   \n",
            "25    all      all             treatment_protocol_ch1   \n",
            "26    all      all                   instrument_model   \n",
            "27    all      all                               name   \n",
            "28    all      all  curated_genetic_modification_type   \n",
            "29    all      all                            bmi_ch1   \n",
            "30    all      all                    data_processing   \n",
            "31    all      all                  curated_cell_line   \n",
            "32    all      all                        description   \n",
            "33    all      all                              title   \n",
            "34    all      all                         timestamp_   \n",
            "35    all      all                  curated_cohort_id   \n",
            "36    all      all                            data_id   \n",
            "37    all      all                           src_repo   \n",
            "38    all      all                             id_key   \n",
            "39    all      all                         gender_ch1   \n",
            "40    all      all                 smoking_status_ch1   \n",
            "41    all      all                          label_ch2   \n",
            "42    all      all                    curated_disease   \n",
            "43    all      all                          label_ch1   \n",
            "44    all      all                     time_point_ch1   \n",
            "45    all      all              characteristics_ch1_3   \n",
            "46    all      all              characteristics_ch1_2   \n",
            "47    all      all                     curated_tissue   \n",
            "48    all      all           curated_drug_smiles_code   \n",
            "49    all      all                       hyb_protocol   \n",
            "50    all      all                        platform_id   \n",
            "51    all      all                         is_current   \n",
            "52    all      all              characteristics_ch1_1   \n",
            "\n",
            "                                    Field Description Field Type  Is Curated  \\\n",
            "0                                                  NA       text       False   \n",
            "1   Unique URI derived from source data file's S3 ...       text       False   \n",
            "2              Unique ID associated with every sample       text       False   \n",
            "3          Gene modified through genetic modification       text       False   \n",
            "4                                                  NA       text       False   \n",
            "5      Name of the cohort to which the sample belongs       text       False   \n",
            "6   Signifies whether the given sample is a contro...    integer       False   \n",
            "7   Dataset ID of the file this data entity origin...       text       False   \n",
            "8                                                  NA       text       False   \n",
            "9                                                  NA       text       False   \n",
            "10                                                 NA       text       False   \n",
            "11                                                 NA       text       False   \n",
            "12                                                 NA       text       False   \n",
            "13                Types of cell present in the sample       text       False   \n",
            "14                                                 NA       text       False   \n",
            "15                                                 NA       text       False   \n",
            "16  Unique ID of the dataset to which the sample b...       text       False   \n",
            "17                                                 NA       text       False   \n",
            "18                     Gene of interest in the sample       text       False   \n",
            "19                      Drug admistered in the sample       text       False   \n",
            "20                             Version of data entity    integer       False   \n",
            "21                                                 NA       text       False   \n",
            "22                                                 NA       text       False   \n",
            "23                                                 NA       text       False   \n",
            "24                                                 NA       text       False   \n",
            "25                                                 NA       text       False   \n",
            "26                                                 NA       text       False   \n",
            "27  Name of the data entity (local to its source f...       text       False   \n",
            "28  Signifies the kind of genetic modification per...       text       False   \n",
            "29                                                 NA       text       False   \n",
            "30                                                 NA       text       False   \n",
            "31        Cell line from which the sample was derived       text       False   \n",
            "32                          Description of the sample       text       False   \n",
            "33                                                 NA       text       False   \n",
            "34  Unix timestamp denoting time of creation for t...       text       False   \n",
            "35  Numeric ID of the cohort to which the sample b...    integer       False   \n",
            "36            Unique ID for this data entity on Polly       text       False   \n",
            "37  Name of the repository this data entity origin...       text       False   \n",
            "38  Name of the key that was used for creation of ...       text       False   \n",
            "39                                                 NA       text       False   \n",
            "40                                                 NA       text       False   \n",
            "41                                                 NA       text       False   \n",
            "42                 Disease associated with the sample       text       False   \n",
            "43                                                 NA       text       False   \n",
            "44                                                 NA       text       False   \n",
            "45                                                 NA       text       False   \n",
            "46                                                 NA       text       False   \n",
            "47                 Tissue to which the sample belongs       text       False   \n",
            "48   SMILES code of the drug admistered in the sample       text       False   \n",
            "49                                                 NA       text       False   \n",
            "50                                                 NA       text       False   \n",
            "51  Whether this is the current version of the dat...       text       False   \n",
            "52                                                 NA       text       False   \n",
            "\n",
            "    Is Array  \n",
            "0      False  \n",
            "1      False  \n",
            "2      False  \n",
            "3       True  \n",
            "4      False  \n",
            "5      False  \n",
            "6      False  \n",
            "7      False  \n",
            "8      False  \n",
            "9      False  \n",
            "10     False  \n",
            "11     False  \n",
            "12     False  \n",
            "13      True  \n",
            "14     False  \n",
            "15     False  \n",
            "16      True  \n",
            "17     False  \n",
            "18      True  \n",
            "19      True  \n",
            "20     False  \n",
            "21     False  \n",
            "22     False  \n",
            "23     False  \n",
            "24     False  \n",
            "25     False  \n",
            "26     False  \n",
            "27     False  \n",
            "28      True  \n",
            "29     False  \n",
            "30     False  \n",
            "31     False  \n",
            "32     False  \n",
            "33     False  \n",
            "34     False  \n",
            "35     False  \n",
            "36     False  \n",
            "37     False  \n",
            "38     False  \n",
            "39     False  \n",
            "40     False  \n",
            "41     False  \n",
            "42      True  \n",
            "43     False  \n",
            "44     False  \n",
            "45     False  \n",
            "46     False  \n",
            "47     False  \n",
            "48      True  \n",
            "49     False  \n",
            "50     False  \n",
            "51     False  \n",
            "52     False  \n",
            "-----features payload-------\n",
            "  Source Datatype      Field Name  \\\n",
            "0    all      all        src_repo   \n",
            "1    all      all          id_key   \n",
            "2    all      all      timestamp_   \n",
            "3    all      all         src_uri   \n",
            "4    all      all         data_id   \n",
            "5    all      all            name   \n",
            "6    all      all          row_id   \n",
            "7    all      all      is_current   \n",
            "8    all      all  src_dataset_id   \n",
            "9    all      all         version   \n",
            "\n",
            "                                   Field Description Field Type  Is Array  \n",
            "0  Name of the repository this data entity origin...       text     False  \n",
            "1  Name of the key that was used for creation of ...       text     False  \n",
            "2  Unix timestamp denoting time of creation for t...       text     False  \n",
            "3  Unique URI derived from source data file's S3 ...       text     False  \n",
            "4            Unique ID for this data entity on Polly       text     False  \n",
            "5  Name of the data entity (local to its source f...       text     False  \n",
            "6                                                 NA       text     False  \n",
            "7  Whether this is the current version of the dat...       text     False  \n",
            "8  Dataset ID of the file this data entity origin...       text     False  \n",
            "9                             Version of data entity    integer     False  \n"
          ]
        }
      ],
      "source": [
        "# testing point 3\n",
        "# neither valid table name or schema level value passed\n",
        "# should take all the table names -> passed\n",
        "repo_key_1 = \"geo\"\n",
        "resp = omixatlas.get_schema(repo_key_1, [], source=\"all\", data_type=\"all\")\n",
        "print(\"-----dataset payload------\")\n",
        "print(resp.datasets)\n",
        "print(\"------sample payload-----\")\n",
        "print(resp.samples)\n",
        "print(\"-----features payload-------\")\n",
        "print(resp.features)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 15,
      "metadata": {
        "scrolled": false
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "--schema table key---samples_singlecell\n",
            "--schema table key---features_singlecell\n",
            "-----samples single cell------\n",
            "   Source Datatype                         Field Name  \\\n",
            "0     all      all                        gene_counts   \n",
            "1     all      all                              organ   \n",
            "2     all      all              curated_raw_cell_type   \n",
            "3     all      all                             gender   \n",
            "4     all      all                            src_uri   \n",
            "5     all      all                          sample_id   \n",
            "6     all      all              curated_gene_modified   \n",
            "7     all      all                  curated_cell_line   \n",
            "8     all      all                              title   \n",
            "9     all      all                     src_dataset_id   \n",
            "10    all      all                         timestamp_   \n",
            "11    all      all                       percent_mito   \n",
            "12    all      all                              donor   \n",
            "13    all      all                            data_id   \n",
            "14    all      all                characteristics_ch1   \n",
            "15    all      all                  curated_cell_type   \n",
            "16    all      all                           clusters   \n",
            "17    all      all           curated_cell_ontology_id   \n",
            "18    all      all                             strain   \n",
            "19    all      all                           src_repo   \n",
            "20    all      all                             id_key   \n",
            "21    all      all                      organism_part   \n",
            "22    all      all                         dataset_id   \n",
            "23    all      all                developmental_stage   \n",
            "24    all      all                    curated_disease   \n",
            "25    all      all                       curated_drug   \n",
            "26    all      all          curated_cell_type_singler   \n",
            "27    all      all                       curated_gene   \n",
            "28    all      all                             sample   \n",
            "29    all      all                            version   \n",
            "30    all      all                         umi_counts   \n",
            "31    all      all                     curated_tissue   \n",
            "32    all      all              curated_marker_absent   \n",
            "33    all      all                               name   \n",
            "34    all      all  curated_genetic_modification_type   \n",
            "35    all      all                           location   \n",
            "36    all      all                         is_current   \n",
            "37    all      all                                age   \n",
            "\n",
            "                                    Field Description Field Type  Is Curated  \\\n",
            "0                                         Gene Counts       text       False   \n",
            "1                                               organ       text       False   \n",
            "2                               Curated raw cell type       text       False   \n",
            "3   Gender of the organism from whom the sample wa...       text       False   \n",
            "4   Unique URI derived from source data file's S3 ...       text       False   \n",
            "5                Unique ID associated with every cell       text       False   \n",
            "6                                      Modified Genes       text       False   \n",
            "7         Cell line from which the sample was derived       text       False   \n",
            "8                                               Title       text       False   \n",
            "9   Dataset ID of the file this data entity origin...       text       False   \n",
            "10  Unix timestamp denoting time of creation for t...       text       False   \n",
            "11  Proportion of total counts for a cell which ar...       text       False   \n",
            "12                                              Donor       text       False   \n",
            "13            Unique ID for this data entity on Polly       text       False   \n",
            "14                                characteristics_ch1       text       False   \n",
            "15                                       kw_cell_type       text       False   \n",
            "16                                  Cell Type Cluster       text       False   \n",
            "17                           Curated Cell Ontology ID       text       False   \n",
            "18                                             Strain       text       False   \n",
            "19  Name of the repository this data entity origin...       text       False   \n",
            "20  Name of the key that was used for creation of ...       text       False   \n",
            "21                                      organism_part       text       False   \n",
            "22  Unique ID of the dataset to which the sample b...       text       False   \n",
            "23                                developmental_stage       text       False   \n",
            "24                 Disease associated with the sample       text       False   \n",
            "25                      Drug admistered in the sample       text       False   \n",
            "26                               kw_cell_type_singler       text       False   \n",
            "27                     Gene of interest in the sample       text       False   \n",
            "28                        ID associated with a sample       text       False   \n",
            "29                             Version of data entity    integer       False   \n",
            "30  Absolute number of observed transcripts for a ...       text       False   \n",
            "31                 Tissue to which the sample belongs       text       False   \n",
            "32                              Curated Marker Absent       text       False   \n",
            "33  Name of the data entity (local to its source f...       text       False   \n",
            "34  Signifies the kind of genetic modification per...       text       False   \n",
            "35                                           Location       text       False   \n",
            "36  Whether this is the current version of the dat...       text       False   \n",
            "37  Age of the organism from whom the sample was t...       text       False   \n",
            "\n",
            "    Is Array  \n",
            "0      False  \n",
            "1      False  \n",
            "2      False  \n",
            "3      False  \n",
            "4      False  \n",
            "5      False  \n",
            "6       True  \n",
            "7       True  \n",
            "8      False  \n",
            "9      False  \n",
            "10     False  \n",
            "11     False  \n",
            "12     False  \n",
            "13     False  \n",
            "14     False  \n",
            "15     False  \n",
            "16     False  \n",
            "17     False  \n",
            "18     False  \n",
            "19     False  \n",
            "20     False  \n",
            "21     False  \n",
            "22      True  \n",
            "23     False  \n",
            "24      True  \n",
            "25      True  \n",
            "26     False  \n",
            "27      True  \n",
            "28     False  \n",
            "29     False  \n",
            "30     False  \n",
            "31      True  \n",
            "32      True  \n",
            "33     False  \n",
            "34      True  \n",
            "35     False  \n",
            "36     False  \n",
            "37     False  \n",
            "------features single cell-----\n",
            "   Source Datatype       Field Name  \\\n",
            "0     all      all       feature_id   \n",
            "1     all      all         src_repo   \n",
            "2     all      all           id_key   \n",
            "3     all      all          src_uri   \n",
            "4     all      all       dataset_id   \n",
            "5     all      all          n_cells   \n",
            "6     all      all   src_dataset_id   \n",
            "7     all      all          version   \n",
            "8     all      all       timestamp_   \n",
            "9     all      all  highly_variable   \n",
            "10    all      all          data_id   \n",
            "11    all      all             name   \n",
            "12    all      all       is_current   \n",
            "\n",
            "                                    Field Description Field Type  Is Curated  \\\n",
            "0   ID of the feature (gene, protein, metabolite e...       text       False   \n",
            "1   Name of the repository this data entity origin...       text       False   \n",
            "2   Name of the key that was used for creation of ...       text       False   \n",
            "3   Unique URI derived from source data file's S3 ...       text       False   \n",
            "4   ID of the dataset in which the feature has bee...       text       False   \n",
            "5                 Number of cells containing the gene       text       False   \n",
            "6   Dataset ID of the file this data entity origin...       text       False   \n",
            "7                              Version of data entity    integer       False   \n",
            "8   Unix timestamp denoting time of creation for t...       text       False   \n",
            "9                     Whether gene is highly variable       text       False   \n",
            "10            Unique ID for this data entity on Polly       text       False   \n",
            "11  Name of the data entity (local to its source f...       text       False   \n",
            "12  Whether this is the current version of the dat...       text       False   \n",
            "\n",
            "    Is Array  \n",
            "0      False  \n",
            "1      False  \n",
            "2      False  \n",
            "3      False  \n",
            "4      False  \n",
            "5      False  \n",
            "6      False  \n",
            "7      False  \n",
            "8      False  \n",
            "9      False  \n",
            "10     False  \n",
            "11     False  \n",
            "12     False  \n"
          ]
        }
      ],
      "source": [
        "# testing point 4\n",
        "# valid table names for sc_data_lake -> passed\n",
        "repo_key_2 = \"sc_data_lake\"\n",
        "resp = omixatlas.get_schema(repo_key_2, [\"samples_singlecell\", \"features_singlecell\"], source=\"all\", data_type=\"all\")\n",
        "print(\"-----samples single cell------\")\n",
        "print(resp.samples_singlecell)\n",
        "print(\"------features single cell-----\")\n",
        "print(resp.features_singlecell)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 17,
      "metadata": {},
      "outputs": [
        {
          "ename": "RequestException",
          "evalue": "('Datatype not found', \"Datatype passed in the query param does not existfor this source: all. You can try these instead ['all']\")",
          "output_type": "error",
          "traceback": [
            "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
            "\u001b[0;31mRequestException\u001b[0m                          Traceback (most recent call last)",
            "Input \u001b[0;32mIn [17]\u001b[0m, in \u001b[0;36m<cell line: 4>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;66;03m# testing point 5\u001b[39;00m\n\u001b[1;32m      2\u001b[0m \u001b[38;5;66;03m# valid schema level sample with single cell\u001b[39;00m\n\u001b[1;32m      3\u001b[0m repo_key_2 \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124msc_data_lake\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m----> 4\u001b[0m resp \u001b[38;5;241m=\u001b[39m \u001b[43momixatlas\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_schema\u001b[49m\u001b[43m(\u001b[49m\u001b[43mrepo_key_2\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43msample\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m]\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43msource\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mall\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mdata_type\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43msingle_cell\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[1;32m      5\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m-----samples single cell------\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m      6\u001b[0m \u001b[38;5;28mprint\u001b[39m(resp\u001b[38;5;241m.\u001b[39msample)\n",
            "File \u001b[0;32m/usr/local/lib/python3.10/site-packages/polly/omixatlas.py:772\u001b[0m, in \u001b[0;36mOmixAtlas.get_schema\u001b[0;34m(self, repo_key, schema_level, source, data_type, return_type)\u001b[0m\n\u001b[1;32m    770\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mreturn_schema_data(schema_payload_dict)\n\u001b[1;32m    771\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m--> 772\u001b[0m         schema \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_schema_from_api\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    773\u001b[0m \u001b[43m            \u001b[49m\u001b[43mrepo_key\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mschema_type_dict\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43msource\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mdata_type\u001b[49m\n\u001b[1;32m    774\u001b[0m \u001b[43m        \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    775\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m schema \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(schema, Dict):\n\u001b[1;32m    776\u001b[0m     \u001b[38;5;28;01mfor\u001b[39;00m key, val \u001b[38;5;129;01min\u001b[39;00m schema\u001b[38;5;241m.\u001b[39mitems():\n",
            "File \u001b[0;32m/usr/local/lib/python3.10/site-packages/polly/omixatlas.py:608\u001b[0m, in \u001b[0;36mOmixAtlas.get_schema_from_api\u001b[0;34m(self, repo_key, schema_type_dict, source, data_type)\u001b[0m\n\u001b[1;32m    606\u001b[0m     dataset_url \u001b[38;5;241m=\u001b[39m \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mschema_base_url\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m/\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mrepo_key\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m/schemas/\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mschema_type\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;132;01m{\u001b[39;00msummary_query_param\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    607\u001b[0m resp \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msession\u001b[38;5;241m.\u001b[39mget(dataset_url)\n\u001b[0;32m--> 608\u001b[0m \u001b[43merror_handler\u001b[49m\u001b[43m(\u001b[49m\u001b[43mresp\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    609\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m--schema table key---\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mschema_table_key\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m    610\u001b[0m resp_dict[schema_table_key] \u001b[38;5;241m=\u001b[39m resp\u001b[38;5;241m.\u001b[39mjson()\n",
            "File \u001b[0;32m/usr/local/lib/python3.10/site-packages/polly/errors.py:261\u001b[0m, in \u001b[0;36merror_handler\u001b[0;34m(response)\u001b[0m\n\u001b[1;32m    259\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m has_error_message(response):\n\u001b[1;32m    260\u001b[0m     title, detail \u001b[38;5;241m=\u001b[39m extract_json_api_error(response)\n\u001b[0;32m--> 261\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m RequestException(title, detail)\n\u001b[1;32m    262\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m response\u001b[38;5;241m.\u001b[39mstatus_code \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m401\u001b[39m:\n\u001b[1;32m    263\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m UnauthorizedException\n",
            "\u001b[0;31mRequestException\u001b[0m: ('Datatype not found', \"Datatype passed in the query param does not existfor this source: all. You can try these instead ['all']\")"
          ]
        }
      ],
      "source": [
        "# testing point 5\n",
        "# valid schema level sample with single cell\n",
        "repo_key_2 = \"sc_data_lake\"\n",
        "resp = omixatlas.get_schema(repo_key_2, [\"sample\"], source=\"all\", data_type=\"single_cell\")\n",
        "print(\"-----samples single cell------\")\n",
        "print(resp.sample)\n",
        "# as data_type is only `all` -> there is no single_cell datatype in the schema\n",
        "# it should return error"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 19,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "-----dataset payload------\n",
            "{'data': {'type': 'schemas', 'id': '9', 'attributes': {'repo_id': '9', 'schema_type': 'files', 'schema': {'all': {'all': {'data_matrix_available': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'data_matrix_available', 'description': 'Data Matrix Available', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Data Matrix Available'}, 'curated_organism': {'is_array': True, 'is_ontology': False, 'is_keyword': True, 'original_name': 'organism', 'description': 'Orgnism from which the samples were derived', 'type': 'text', 'is_filter': True, 'is_column': True, 'is_curated': True, 'filter_size': 1000, 'display_name': 'Organism'}, 'src_uri': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_src_uri', 'description': \"Unique URI derived from data file's S3 location\", 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Source URI'}, 'total_num_samples': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'total_num_samples', 'description': 'Total number of samples in a dataset', 'type': 'integer', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Number of Samples'}, 'year': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'year', 'description': 'Year in which the dataset was published', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Year'}, 'description': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'description', 'description': 'Description of the dataset', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Title'}, 'curated_cell_line': {'is_array': True, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_cell_line', 'description': 'Cell lines from which the samples were derived for this dataset', 'type': 'text', 'is_filter': True, 'is_column': True, 'is_curated': True, 'filter_size': 1000, 'display_name': 'Cell line'}, 'data_table_name': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'data_table_name', 'description': 'Name of the data table associated with data file', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Data-matrix table'}, 'data_table_version': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'data_table_version', 'description': 'Current version of the data table associated with data file', 'type': 'integer', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Data-matrix version'}, 'platform': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'platform', 'description': 'Platform Accession number signifying the technology, molecule types and annotation conventions', 'type': 'text', 'is_filter': True, 'is_column': False, 'is_curated': True, 'filter_size': 1, 'display_name': 'Platform'}, 'timestamp_': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_timestamp_', 'description': 'Unix timestamp denoting time of creation for this entry', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Timestamp'}, 'subseries_ids': {'is_array': True, 'is_ontology': False, 'is_keyword': True, 'original_name': 'subseries_ids', 'description': 'GSE IDs associated with a SuperSeries', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Subseries IDs'}, 'file_type': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_file_type', 'description': \"Data file's type\", 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'File type'}, 'publication': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'publication', 'description': 'Publication associated with the dataset', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Publication'}, 'missing_samples': {'is_array': True, 'is_ontology': False, 'is_keyword': True, 'original_name': 'missing_samples', 'description': 'Additional sample IDs that are available at source', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Missing Samples'}, 'curated_cell_type': {'is_array': True, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_cell_type', 'description': 'Types of cell present in the dataset', 'type': 'text', 'is_filter': False, 'is_column': True, 'is_curated': True, 'filter_size': 1, 'display_name': 'Cell type'}, 'key': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_key', 'description': 'Key to the data file in source package', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Key'}, 'summary': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'summary', 'description': 'Summary of the paper', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Summary'}, 'src_repo': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_src_repo', 'description': 'Name of the repository this data file originated from', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Source repository'}, 'drug_smiles': {'is_array': True, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_smiles', 'description': 'SMILES code of the drug', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': True, 'filter_size': 1, 'display_name': 'Drug (SMILES code)'}, 'package': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_package', 'description': 'Name of the package this data file was part of', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Package'}, 'file_location': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_file_location', 'description': 'Unique S3 location for this data file', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'File location'}, 'author': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'author', 'description': 'Name of the author who published the dataset', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Author'}, 'dataset_id': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'dataset_id', 'description': 'Unique ID assocaited with every dataset', 'type': 'text', 'is_filter': False, 'is_column': True, 'is_curated': True, 'filter_size': 1, 'display_name': 'Dataset ID'}, 'curated_disease': {'is_array': True, 'is_ontology': True, 'is_keyword': True, 'original_name': 'disease', 'description': 'Disease associated with the dataset', 'type': 'text', 'is_filter': True, 'is_column': True, 'is_curated': True, 'filter_size': 3000, 'display_name': 'Disease'}, 'curated_drug': {'is_array': True, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_drug', 'description': 'Drugs administered in the samples belonging to the dataset', 'type': 'text', 'is_filter': True, 'is_column': True, 'is_curated': True, 'filter_size': 3000, 'display_name': 'Drug'}, 'curated_gene': {'is_array': True, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_gene', 'description': 'Gene studied in the dataset', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': True, 'filter_size': 1, 'display_name': 'Gene'}, 'abstract': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'abstract', 'description': 'Abstract of the publication assocaited with the dataset', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Abstract'}, 'version': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'version', 'description': 'Version of data file', 'type': 'integer', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Version'}, 'curated_strain': {'is_array': True, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_strain', 'description': 'Strain of the organism from which samples were derived', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': True, 'filter_size': 1, 'display_name': 'Strain'}, 'bucket': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_bucket', 'description': 'S3 bucket in which the data file resides', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Bucket'}, 'curated_tissue': {'is_array': True, 'is_ontology': True, 'is_keyword': True, 'original_name': 'tissue', 'description': 'Tissue from which the samples were derivved', 'type': 'text', 'is_filter': True, 'is_column': True, 'is_curated': True, 'filter_size': 1000, 'display_name': 'Tissue'}, 'dataset_source': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'dataset_source', 'description': 'Source from where the data was fetched', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': True, 'filter_size': 1, 'display_name': 'Source'}, 'data_type': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_data_type', 'description': 'The type of biomolecular data captured (eg - Exome Sequencing, Transcriptomics)', 'type': 'text', 'is_filter': True, 'is_column': True, 'is_curated': True, 'filter_size': 10, 'display_name': 'Data type'}, 'overall_design': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'overall_design', 'description': 'Overall Design of the experiment as given by the author', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Overall Design'}, 'is_current': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'is_current', 'description': 'Whether this is the current version of the data file or not', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Is current'}, 'region': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_region', 'description': 'Region (AWS) in which this data is located', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Region'}}}}, 'created_time': 1656737601693, 'last_updated_time': 1657881950837, 'enabled': True}}}\n"
          ]
        }
      ],
      "source": [
        "# testing point 6\n",
        "# return type dict for valid schema level names\n",
        "repo_key_1 = \"geo\"\n",
        "resp = omixatlas.get_schema(repo_key_1, [\"dataset\"], source=\"all\", data_type=\"all\", return_type=\"dict\")\n",
        "print(\"-----dataset payload------\")\n",
        "print(resp.dataset)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 22,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "-----dataset payload------\n",
            "{'data': {'type': 'schemas', 'id': '9', 'attributes': {'repo_id': '9', 'schema_type': 'files', 'schema': {'all': {'all': {'data_matrix_available': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'data_matrix_available', 'description': 'Data Matrix Available', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Data Matrix Available'}, 'curated_organism': {'is_array': True, 'is_ontology': False, 'is_keyword': True, 'original_name': 'organism', 'description': 'Orgnism from which the samples were derived', 'type': 'text', 'is_filter': True, 'is_column': True, 'is_curated': True, 'filter_size': 1000, 'display_name': 'Organism'}, 'src_uri': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_src_uri', 'description': \"Unique URI derived from data file's S3 location\", 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Source URI'}, 'total_num_samples': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'total_num_samples', 'description': 'Total number of samples in a dataset', 'type': 'integer', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Number of Samples'}, 'year': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'year', 'description': 'Year in which the dataset was published', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Year'}, 'description': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'description', 'description': 'Description of the dataset', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Title'}, 'curated_cell_line': {'is_array': True, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_cell_line', 'description': 'Cell lines from which the samples were derived for this dataset', 'type': 'text', 'is_filter': True, 'is_column': True, 'is_curated': True, 'filter_size': 1000, 'display_name': 'Cell line'}, 'data_table_name': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'data_table_name', 'description': 'Name of the data table associated with data file', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Data-matrix table'}, 'data_table_version': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'data_table_version', 'description': 'Current version of the data table associated with data file', 'type': 'integer', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Data-matrix version'}, 'platform': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'platform', 'description': 'Platform Accession number signifying the technology, molecule types and annotation conventions', 'type': 'text', 'is_filter': True, 'is_column': False, 'is_curated': True, 'filter_size': 1, 'display_name': 'Platform'}, 'timestamp_': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_timestamp_', 'description': 'Unix timestamp denoting time of creation for this entry', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Timestamp'}, 'subseries_ids': {'is_array': True, 'is_ontology': False, 'is_keyword': True, 'original_name': 'subseries_ids', 'description': 'GSE IDs associated with a SuperSeries', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Subseries IDs'}, 'file_type': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_file_type', 'description': \"Data file's type\", 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'File type'}, 'publication': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'publication', 'description': 'Publication associated with the dataset', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Publication'}, 'missing_samples': {'is_array': True, 'is_ontology': False, 'is_keyword': True, 'original_name': 'missing_samples', 'description': 'Additional sample IDs that are available at source', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Missing Samples'}, 'curated_cell_type': {'is_array': True, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_cell_type', 'description': 'Types of cell present in the dataset', 'type': 'text', 'is_filter': False, 'is_column': True, 'is_curated': True, 'filter_size': 1, 'display_name': 'Cell type'}, 'key': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_key', 'description': 'Key to the data file in source package', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Key'}, 'summary': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'summary', 'description': 'Summary of the paper', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Summary'}, 'src_repo': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_src_repo', 'description': 'Name of the repository this data file originated from', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Source repository'}, 'drug_smiles': {'is_array': True, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_smiles', 'description': 'SMILES code of the drug', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': True, 'filter_size': 1, 'display_name': 'Drug (SMILES code)'}, 'package': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_package', 'description': 'Name of the package this data file was part of', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Package'}, 'file_location': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_file_location', 'description': 'Unique S3 location for this data file', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'File location'}, 'author': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'author', 'description': 'Name of the author who published the dataset', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Author'}, 'dataset_id': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'dataset_id', 'description': 'Unique ID assocaited with every dataset', 'type': 'text', 'is_filter': False, 'is_column': True, 'is_curated': True, 'filter_size': 1, 'display_name': 'Dataset ID'}, 'curated_disease': {'is_array': True, 'is_ontology': True, 'is_keyword': True, 'original_name': 'disease', 'description': 'Disease associated with the dataset', 'type': 'text', 'is_filter': True, 'is_column': True, 'is_curated': True, 'filter_size': 3000, 'display_name': 'Disease'}, 'curated_drug': {'is_array': True, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_drug', 'description': 'Drugs administered in the samples belonging to the dataset', 'type': 'text', 'is_filter': True, 'is_column': True, 'is_curated': True, 'filter_size': 3000, 'display_name': 'Drug'}, 'curated_gene': {'is_array': True, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_gene', 'description': 'Gene studied in the dataset', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': True, 'filter_size': 1, 'display_name': 'Gene'}, 'abstract': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'abstract', 'description': 'Abstract of the publication assocaited with the dataset', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Abstract'}, 'version': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'version', 'description': 'Version of data file', 'type': 'integer', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Version'}, 'curated_strain': {'is_array': True, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_strain', 'description': 'Strain of the organism from which samples were derived', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': True, 'filter_size': 1, 'display_name': 'Strain'}, 'bucket': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_bucket', 'description': 'S3 bucket in which the data file resides', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Bucket'}, 'curated_tissue': {'is_array': True, 'is_ontology': True, 'is_keyword': True, 'original_name': 'tissue', 'description': 'Tissue from which the samples were derivved', 'type': 'text', 'is_filter': True, 'is_column': True, 'is_curated': True, 'filter_size': 1000, 'display_name': 'Tissue'}, 'dataset_source': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'dataset_source', 'description': 'Source from where the data was fetched', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': True, 'filter_size': 1, 'display_name': 'Source'}, 'data_type': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_data_type', 'description': 'The type of biomolecular data captured (eg - Exome Sequencing, Transcriptomics)', 'type': 'text', 'is_filter': True, 'is_column': True, 'is_curated': True, 'filter_size': 10, 'display_name': 'Data type'}, 'overall_design': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'overall_design', 'description': 'Overall Design of the experiment as given by the author', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Overall Design'}, 'is_current': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'is_current', 'description': 'Whether this is the current version of the data file or not', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Is current'}, 'region': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_region', 'description': 'Region (AWS) in which this data is located', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Region'}}}}, 'created_time': 1656737601693, 'last_updated_time': 1657881950837, 'enabled': True}}}\n",
            "\n",
            "\n",
            "-----sample payload-----\n",
            "{'data': {'type': 'schemas', 'id': '9', 'attributes': {'repo_id': '9', 'schema_type': 'gct_metadata', 'schema': {'all': {'all': {'growth_protocol_ch1': {'is_array': False, 'is_keyword': False, 'original_name': 'growth_protocol_ch1', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Growth protocol ch1'}, 'src_uri': {'is_array': False, 'is_keyword': True, 'original_name': 'kw_src_uri', 'description': \"Unique URI derived from source data file's S3 location\", 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Source URI'}, 'sample_id': {'is_array': False, 'is_keyword': True, 'original_name': 'kw_column', 'description': 'Unique ID associated with every sample', 'type': 'text', 'is_filter': False, 'is_column': True, 'is_curated': False, 'filter_size': 1, 'display_name': 'Sample ID'}, 'curated_gene_modified': {'is_array': True, 'is_keyword': True, 'original_name': 'kw_curated_modified_gene', 'description': 'Gene modified through genetic modification', 'type': 'text', 'is_filter': False, 'is_column': True, 'is_curated': False, 'filter_size': 1, 'display_name': 'Gene modified'}, 'dose_ch1': {'is_array': False, 'is_keyword': False, 'original_name': 'dose.ch1', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Dose ch1'}, 'curated_cohort_name': {'is_array': False, 'is_keyword': True, 'original_name': 'curated_cohort_name', 'description': 'Name of the cohort to which the sample belongs', 'type': 'text', 'is_filter': False, 'is_column': True, 'is_curated': False, 'filter_size': 1, 'display_name': 'Cohort name'}, 'curated_control': {'is_array': False, 'is_keyword': True, 'original_name': 'curated_is_control', 'description': 'Signifies whether the given sample is a control or not, 1 signifies control and 0 signifies perturbed sample', 'type': 'integer', 'is_filter': False, 'is_column': True, 'is_curated': False, 'filter_size': 1, 'display_name': 'Control'}, 'src_dataset_id': {'is_array': False, 'is_keyword': True, 'original_name': 'kw_src_dataset_id', 'description': 'Dataset ID of the file this data entity originated from', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Source dataset ID'}, 'extract_protocol_ch1': {'is_array': False, 'is_keyword': False, 'original_name': 'extract_protocol_ch1', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Extract protocol ch1'}, 'characteristics_ch2': {'is_array': False, 'is_keyword': False, 'original_name': 'characteristics_ch2', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Characteristics ch2'}, 'characteristics_ch1': {'is_array': False, 'is_keyword': False, 'original_name': 'characteristics_ch1', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Characteristics ch1'}, 'extract_protocol_ch2': {'is_array': False, 'is_keyword': False, 'original_name': 'extract_protocol_ch2', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Extract protocol ch2'}, 'age_ch1': {'is_array': False, 'is_keyword': False, 'original_name': 'age.ch1', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Age ch1'}, 'curated_cell_type': {'is_array': True, 'is_keyword': True, 'original_name': 'kw_curated_cell_type', 'description': 'Types of cell present in the sample', 'type': 'text', 'is_filter': False, 'is_column': True, 'is_curated': False, 'filter_size': 1, 'display_name': 'Cell type'}, 'sex_ch1': {'is_array': False, 'is_keyword': False, 'original_name': 'sex_ch1', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Sex ch1'}, 'treatment_ch1': {'is_array': False, 'is_keyword': False, 'original_name': 'treatment.ch1', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Treatment ch1'}, 'dataset_id': {'is_array': True, 'is_keyword': True, 'original_name': 'kw_doc_id', 'description': 'Unique ID of the dataset to which the sample belongs', 'type': 'text', 'is_filter': False, 'is_column': True, 'is_curated': False, 'filter_size': 1, 'display_name': 'Dataset ID'}, 'taxid_ch1': {'is_array': False, 'is_keyword': False, 'original_name': 'taxid_ch1', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Taxonomy ID ch1'}, 'curated_gene': {'is_array': True, 'is_keyword': True, 'original_name': 'kw_curated_gene', 'description': 'Gene of interest in the sample', 'type': 'text', 'is_filter': False, 'is_column': True, 'is_curated': False, 'filter_size': 1, 'display_name': 'Gene'}, 'curated_drug': {'is_array': True, 'is_keyword': True, 'original_name': 'kw_curated_drug', 'description': 'Drug admistered in the sample', 'type': 'text', 'is_filter': True, 'is_column': True, 'is_curated': False, 'filter_size': 3000, 'display_name': 'Drug'}, 'version': {'is_array': False, 'is_keyword': True, 'original_name': 'version', 'description': 'Version of data entity', 'type': 'integer', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Version'}, 'label_protocol_ch2': {'is_array': False, 'is_keyword': False, 'original_name': 'label_protocol_ch2', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Label protocol ch2'}, 'label_protocol_ch1': {'is_array': False, 'is_keyword': False, 'original_name': 'label_protocol_ch1', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Label protocol ch1'}, 'treatment_protocol_ch2': {'is_array': False, 'is_keyword': False, 'original_name': 'treatment_protocol_ch2', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Treatment protocol ch2'}, 'characteristics_ch2_1': {'is_array': False, 'is_keyword': False, 'original_name': 'characteristics_ch2_1', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Characteristics ch2_1'}, 'treatment_protocol_ch1': {'is_array': False, 'is_keyword': False, 'original_name': 'treatment_protocol_ch1', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Treatment protocol ch1'}, 'instrument_model': {'is_array': False, 'is_keyword': False, 'original_name': 'instrument_model', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Instrument model'}, 'name': {'is_array': False, 'is_keyword': True, 'original_name': 'kw_name', 'description': 'Name of the data entity (local to its source file)', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Name'}, 'curated_genetic_modification_type': {'is_array': True, 'is_keyword': True, 'original_name': 'kw_curated_genetic_mod_type', 'description': 'Signifies the kind of genetic modification performed on the sample like Knockout or Mutation', 'type': 'text', 'is_filter': False, 'is_column': True, 'is_curated': False, 'filter_size': 1, 'display_name': 'Genetic modification'}, 'bmi_ch1': {'is_array': False, 'is_keyword': False, 'original_name': 'bmi_ch1', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'BMI ch1'}, 'data_processing': {'is_array': False, 'is_keyword': False, 'original_name': 'data_processing', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Data processing'}, 'curated_cell_line': {'is_array': False, 'is_keyword': True, 'original_name': 'kw_curated_cell_line', 'description': 'Cell line from which the sample was derived', 'type': 'text', 'is_filter': True, 'is_column': True, 'is_curated': False, 'filter_size': 1000, 'display_name': 'Cell line'}, 'description': {'is_array': False, 'is_keyword': True, 'original_name': 'description', 'description': 'Description of the sample', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Description'}, 'title': {'is_array': False, 'is_keyword': True, 'original_name': 'title', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Title'}, 'timestamp_': {'is_array': False, 'is_keyword': True, 'original_name': 'kw_timestamp_', 'description': 'Unix timestamp denoting time of creation for this entry', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Timestamp'}, 'curated_cohort_id': {'is_array': False, 'is_keyword': True, 'original_name': 'curated_cohort_id', 'description': 'Numeric ID of the cohort to which the sample belongs', 'type': 'integer', 'is_filter': False, 'is_column': True, 'is_curated': False, 'filter_size': 1, 'display_name': 'Cohort ID'}, 'data_id': {'is_array': False, 'is_keyword': True, 'original_name': 'kw_data_id', 'description': 'Unique ID for this data entity on Polly', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Data ID'}, 'src_repo': {'is_array': False, 'is_keyword': True, 'original_name': 'kw_src_repo', 'description': 'Name of the repository this data entity originated from', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Source repository'}, 'id_key': {'is_array': False, 'is_keyword': True, 'original_name': 'kw_id_key', 'description': 'Name of the key that was used for creation of unique identity', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'ID key'}, 'gender_ch1': {'is_array': False, 'is_keyword': False, 'original_name': 'gender.ch1', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Gender ch1'}, 'smoking_status_ch1': {'is_array': False, 'is_keyword': False, 'original_name': 'smoking_status.ch1', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Smoking status ch1'}, 'label_ch2': {'is_array': False, 'is_keyword': False, 'original_name': 'label_ch2', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Label ch2'}, 'curated_disease': {'is_array': True, 'is_keyword': True, 'original_name': 'kw_curated_disease', 'description': 'Disease associated with the sample', 'type': 'text', 'is_filter': True, 'is_column': True, 'is_curated': False, 'filter_size': 3000, 'display_name': 'Disease'}, 'label_ch1': {'is_array': False, 'is_keyword': False, 'original_name': 'label_ch1', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Label ch1'}, 'time_point_ch1': {'is_array': False, 'is_keyword': False, 'original_name': 'time_point_ch1', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Time point ch1'}, 'characteristics_ch1_3': {'is_array': False, 'is_keyword': False, 'original_name': 'characteristics_ch1.3', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Characteristics ch1_2'}, 'characteristics_ch1_2': {'is_array': False, 'is_keyword': False, 'original_name': 'characteristics_ch1.2', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Characteristics ch1_2'}, 'curated_tissue': {'is_array': False, 'is_keyword': True, 'original_name': 'kw_curated_tissue', 'description': 'Tissue to which the sample belongs', 'type': 'text', 'is_filter': True, 'is_column': True, 'is_curated': False, 'filter_size': 1000, 'display_name': 'Tissue'}, 'curated_drug_smiles_code': {'is_array': True, 'is_keyword': True, 'original_name': 'kw_curated_smiles', 'description': 'SMILES code of the drug admistered in the sample', 'type': 'text', 'is_filter': False, 'is_column': True, 'is_curated': False, 'filter_size': 1, 'display_name': 'Drug (SMILES code)'}, 'hyb_protocol': {'is_array': False, 'is_keyword': False, 'original_name': 'hyb_protocol', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Hybridization protocol'}, 'platform_id': {'is_array': False, 'is_keyword': False, 'original_name': 'platform_id', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Platform ID'}, 'is_current': {'is_array': False, 'is_keyword': True, 'original_name': 'is_current', 'description': 'Whether this is the current version of the data entity or not', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Is current'}, 'characteristics_ch1_1': {'is_array': False, 'is_keyword': False, 'original_name': 'characteristics_ch1.1', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Characteristics ch1_1'}}}}, 'created_time': 1653926604432, 'last_updated_time': 1653926604432, 'enabled': True}}}\n"
          ]
        }
      ],
      "source": [
        "# testing point 7\n",
        "# return type dict for valid schema table names\n",
        "repo_key_1 = \"geo\"\n",
        "resp = omixatlas.get_schema(repo_key_1, [\"datasets\", \"samples\"], source=\"all\", data_type=\"all\", return_type=\"dict\")\n",
        "print(\"-----dataset payload------\")\n",
        "print(resp.datasets)\n",
        "print(\"\\n\")\n",
        "print(\"-----sample payload-----\")\n",
        "print(resp.samples)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 23,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Schema(datasets={'data': {'type': 'schemas', 'id': '9', 'attributes': {'repo_id': '9', 'schema_type': 'files', 'schema': {'all': {'all': {'data_matrix_available': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'data_matrix_available', 'description': 'Data Matrix Available', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Data Matrix Available'}, 'curated_organism': {'is_array': True, 'is_ontology': False, 'is_keyword': True, 'original_name': 'organism', 'description': 'Orgnism from which the samples were derived', 'type': 'text', 'is_filter': True, 'is_column': True, 'is_curated': True, 'filter_size': 1000, 'display_name': 'Organism'}, 'src_uri': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_src_uri', 'description': \"Unique URI derived from data file's S3 location\", 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Source URI'}, 'total_num_samples': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'total_num_samples', 'description': 'Total number of samples in a dataset', 'type': 'integer', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Number of Samples'}, 'year': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'year', 'description': 'Year in which the dataset was published', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Year'}, 'description': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'description', 'description': 'Description of the dataset', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Title'}, 'curated_cell_line': {'is_array': True, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_cell_line', 'description': 'Cell lines from which the samples were derived for this dataset', 'type': 'text', 'is_filter': True, 'is_column': True, 'is_curated': True, 'filter_size': 1000, 'display_name': 'Cell line'}, 'data_table_name': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'data_table_name', 'description': 'Name of the data table associated with data file', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Data-matrix table'}, 'data_table_version': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'data_table_version', 'description': 'Current version of the data table associated with data file', 'type': 'integer', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Data-matrix version'}, 'platform': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'platform', 'description': 'Platform Accession number signifying the technology, molecule types and annotation conventions', 'type': 'text', 'is_filter': True, 'is_column': False, 'is_curated': True, 'filter_size': 1, 'display_name': 'Platform'}, 'timestamp_': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_timestamp_', 'description': 'Unix timestamp denoting time of creation for this entry', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Timestamp'}, 'subseries_ids': {'is_array': True, 'is_ontology': False, 'is_keyword': True, 'original_name': 'subseries_ids', 'description': 'GSE IDs associated with a SuperSeries', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Subseries IDs'}, 'file_type': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_file_type', 'description': \"Data file's type\", 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'File type'}, 'publication': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'publication', 'description': 'Publication associated with the dataset', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Publication'}, 'missing_samples': {'is_array': True, 'is_ontology': False, 'is_keyword': True, 'original_name': 'missing_samples', 'description': 'Additional sample IDs that are available at source', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Missing Samples'}, 'curated_cell_type': {'is_array': True, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_cell_type', 'description': 'Types of cell present in the dataset', 'type': 'text', 'is_filter': False, 'is_column': True, 'is_curated': True, 'filter_size': 1, 'display_name': 'Cell type'}, 'key': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_key', 'description': 'Key to the data file in source package', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Key'}, 'summary': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'summary', 'description': 'Summary of the paper', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Summary'}, 'src_repo': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_src_repo', 'description': 'Name of the repository this data file originated from', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Source repository'}, 'drug_smiles': {'is_array': True, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_smiles', 'description': 'SMILES code of the drug', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': True, 'filter_size': 1, 'display_name': 'Drug (SMILES code)'}, 'package': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_package', 'description': 'Name of the package this data file was part of', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Package'}, 'file_location': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_file_location', 'description': 'Unique S3 location for this data file', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'File location'}, 'author': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'author', 'description': 'Name of the author who published the dataset', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Author'}, 'dataset_id': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'dataset_id', 'description': 'Unique ID assocaited with every dataset', 'type': 'text', 'is_filter': False, 'is_column': True, 'is_curated': True, 'filter_size': 1, 'display_name': 'Dataset ID'}, 'curated_disease': {'is_array': True, 'is_ontology': True, 'is_keyword': True, 'original_name': 'disease', 'description': 'Disease associated with the dataset', 'type': 'text', 'is_filter': True, 'is_column': True, 'is_curated': True, 'filter_size': 3000, 'display_name': 'Disease'}, 'curated_drug': {'is_array': True, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_drug', 'description': 'Drugs administered in the samples belonging to the dataset', 'type': 'text', 'is_filter': True, 'is_column': True, 'is_curated': True, 'filter_size': 3000, 'display_name': 'Drug'}, 'curated_gene': {'is_array': True, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_gene', 'description': 'Gene studied in the dataset', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': True, 'filter_size': 1, 'display_name': 'Gene'}, 'abstract': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'abstract', 'description': 'Abstract of the publication assocaited with the dataset', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Abstract'}, 'version': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'version', 'description': 'Version of data file', 'type': 'integer', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Version'}, 'curated_strain': {'is_array': True, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_strain', 'description': 'Strain of the organism from which samples were derived', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': True, 'filter_size': 1, 'display_name': 'Strain'}, 'bucket': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_bucket', 'description': 'S3 bucket in which the data file resides', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Bucket'}, 'curated_tissue': {'is_array': True, 'is_ontology': True, 'is_keyword': True, 'original_name': 'tissue', 'description': 'Tissue from which the samples were derivved', 'type': 'text', 'is_filter': True, 'is_column': True, 'is_curated': True, 'filter_size': 1000, 'display_name': 'Tissue'}, 'dataset_source': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'dataset_source', 'description': 'Source from where the data was fetched', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': True, 'filter_size': 1, 'display_name': 'Source'}, 'data_type': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_data_type', 'description': 'The type of biomolecular data captured (eg - Exome Sequencing, Transcriptomics)', 'type': 'text', 'is_filter': True, 'is_column': True, 'is_curated': True, 'filter_size': 10, 'display_name': 'Data type'}, 'overall_design': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'overall_design', 'description': 'Overall Design of the experiment as given by the author', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Overall Design'}, 'is_current': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'is_current', 'description': 'Whether this is the current version of the data file or not', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Is current'}, 'region': {'is_array': False, 'is_ontology': False, 'is_keyword': True, 'original_name': 'kw_region', 'description': 'Region (AWS) in which this data is located', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Region'}}}}, 'created_time': 1656737601693, 'last_updated_time': 1657881950837, 'enabled': True}}}, samples={'data': {'type': 'schemas', 'id': '9', 'attributes': {'repo_id': '9', 'schema_type': 'gct_metadata', 'schema': {'all': {'all': {'growth_protocol_ch1': {'is_array': False, 'is_keyword': False, 'original_name': 'growth_protocol_ch1', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Growth protocol ch1'}, 'src_uri': {'is_array': False, 'is_keyword': True, 'original_name': 'kw_src_uri', 'description': \"Unique URI derived from source data file's S3 location\", 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Source URI'}, 'sample_id': {'is_array': False, 'is_keyword': True, 'original_name': 'kw_column', 'description': 'Unique ID associated with every sample', 'type': 'text', 'is_filter': False, 'is_column': True, 'is_curated': False, 'filter_size': 1, 'display_name': 'Sample ID'}, 'curated_gene_modified': {'is_array': True, 'is_keyword': True, 'original_name': 'kw_curated_modified_gene', 'description': 'Gene modified through genetic modification', 'type': 'text', 'is_filter': False, 'is_column': True, 'is_curated': False, 'filter_size': 1, 'display_name': 'Gene modified'}, 'dose_ch1': {'is_array': False, 'is_keyword': False, 'original_name': 'dose.ch1', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Dose ch1'}, 'curated_cohort_name': {'is_array': False, 'is_keyword': True, 'original_name': 'curated_cohort_name', 'description': 'Name of the cohort to which the sample belongs', 'type': 'text', 'is_filter': False, 'is_column': True, 'is_curated': False, 'filter_size': 1, 'display_name': 'Cohort name'}, 'curated_control': {'is_array': False, 'is_keyword': True, 'original_name': 'curated_is_control', 'description': 'Signifies whether the given sample is a control or not, 1 signifies control and 0 signifies perturbed sample', 'type': 'integer', 'is_filter': False, 'is_column': True, 'is_curated': False, 'filter_size': 1, 'display_name': 'Control'}, 'src_dataset_id': {'is_array': False, 'is_keyword': True, 'original_name': 'kw_src_dataset_id', 'description': 'Dataset ID of the file this data entity originated from', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Source dataset ID'}, 'extract_protocol_ch1': {'is_array': False, 'is_keyword': False, 'original_name': 'extract_protocol_ch1', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Extract protocol ch1'}, 'characteristics_ch2': {'is_array': False, 'is_keyword': False, 'original_name': 'characteristics_ch2', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Characteristics ch2'}, 'characteristics_ch1': {'is_array': False, 'is_keyword': False, 'original_name': 'characteristics_ch1', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Characteristics ch1'}, 'extract_protocol_ch2': {'is_array': False, 'is_keyword': False, 'original_name': 'extract_protocol_ch2', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Extract protocol ch2'}, 'age_ch1': {'is_array': False, 'is_keyword': False, 'original_name': 'age.ch1', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Age ch1'}, 'curated_cell_type': {'is_array': True, 'is_keyword': True, 'original_name': 'kw_curated_cell_type', 'description': 'Types of cell present in the sample', 'type': 'text', 'is_filter': False, 'is_column': True, 'is_curated': False, 'filter_size': 1, 'display_name': 'Cell type'}, 'sex_ch1': {'is_array': False, 'is_keyword': False, 'original_name': 'sex_ch1', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Sex ch1'}, 'treatment_ch1': {'is_array': False, 'is_keyword': False, 'original_name': 'treatment.ch1', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Treatment ch1'}, 'dataset_id': {'is_array': True, 'is_keyword': True, 'original_name': 'kw_doc_id', 'description': 'Unique ID of the dataset to which the sample belongs', 'type': 'text', 'is_filter': False, 'is_column': True, 'is_curated': False, 'filter_size': 1, 'display_name': 'Dataset ID'}, 'taxid_ch1': {'is_array': False, 'is_keyword': False, 'original_name': 'taxid_ch1', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Taxonomy ID ch1'}, 'curated_gene': {'is_array': True, 'is_keyword': True, 'original_name': 'kw_curated_gene', 'description': 'Gene of interest in the sample', 'type': 'text', 'is_filter': False, 'is_column': True, 'is_curated': False, 'filter_size': 1, 'display_name': 'Gene'}, 'curated_drug': {'is_array': True, 'is_keyword': True, 'original_name': 'kw_curated_drug', 'description': 'Drug admistered in the sample', 'type': 'text', 'is_filter': True, 'is_column': True, 'is_curated': False, 'filter_size': 3000, 'display_name': 'Drug'}, 'version': {'is_array': False, 'is_keyword': True, 'original_name': 'version', 'description': 'Version of data entity', 'type': 'integer', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Version'}, 'label_protocol_ch2': {'is_array': False, 'is_keyword': False, 'original_name': 'label_protocol_ch2', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Label protocol ch2'}, 'label_protocol_ch1': {'is_array': False, 'is_keyword': False, 'original_name': 'label_protocol_ch1', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Label protocol ch1'}, 'treatment_protocol_ch2': {'is_array': False, 'is_keyword': False, 'original_name': 'treatment_protocol_ch2', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Treatment protocol ch2'}, 'characteristics_ch2_1': {'is_array': False, 'is_keyword': False, 'original_name': 'characteristics_ch2_1', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Characteristics ch2_1'}, 'treatment_protocol_ch1': {'is_array': False, 'is_keyword': False, 'original_name': 'treatment_protocol_ch1', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Treatment protocol ch1'}, 'instrument_model': {'is_array': False, 'is_keyword': False, 'original_name': 'instrument_model', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Instrument model'}, 'name': {'is_array': False, 'is_keyword': True, 'original_name': 'kw_name', 'description': 'Name of the data entity (local to its source file)', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Name'}, 'curated_genetic_modification_type': {'is_array': True, 'is_keyword': True, 'original_name': 'kw_curated_genetic_mod_type', 'description': 'Signifies the kind of genetic modification performed on the sample like Knockout or Mutation', 'type': 'text', 'is_filter': False, 'is_column': True, 'is_curated': False, 'filter_size': 1, 'display_name': 'Genetic modification'}, 'bmi_ch1': {'is_array': False, 'is_keyword': False, 'original_name': 'bmi_ch1', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'BMI ch1'}, 'data_processing': {'is_array': False, 'is_keyword': False, 'original_name': 'data_processing', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Data processing'}, 'curated_cell_line': {'is_array': False, 'is_keyword': True, 'original_name': 'kw_curated_cell_line', 'description': 'Cell line from which the sample was derived', 'type': 'text', 'is_filter': True, 'is_column': True, 'is_curated': False, 'filter_size': 1000, 'display_name': 'Cell line'}, 'description': {'is_array': False, 'is_keyword': True, 'original_name': 'description', 'description': 'Description of the sample', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Description'}, 'title': {'is_array': False, 'is_keyword': True, 'original_name': 'title', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Title'}, 'timestamp_': {'is_array': False, 'is_keyword': True, 'original_name': 'kw_timestamp_', 'description': 'Unix timestamp denoting time of creation for this entry', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Timestamp'}, 'curated_cohort_id': {'is_array': False, 'is_keyword': True, 'original_name': 'curated_cohort_id', 'description': 'Numeric ID of the cohort to which the sample belongs', 'type': 'integer', 'is_filter': False, 'is_column': True, 'is_curated': False, 'filter_size': 1, 'display_name': 'Cohort ID'}, 'data_id': {'is_array': False, 'is_keyword': True, 'original_name': 'kw_data_id', 'description': 'Unique ID for this data entity on Polly', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Data ID'}, 'src_repo': {'is_array': False, 'is_keyword': True, 'original_name': 'kw_src_repo', 'description': 'Name of the repository this data entity originated from', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Source repository'}, 'id_key': {'is_array': False, 'is_keyword': True, 'original_name': 'kw_id_key', 'description': 'Name of the key that was used for creation of unique identity', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'ID key'}, 'gender_ch1': {'is_array': False, 'is_keyword': False, 'original_name': 'gender.ch1', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Gender ch1'}, 'smoking_status_ch1': {'is_array': False, 'is_keyword': False, 'original_name': 'smoking_status.ch1', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Smoking status ch1'}, 'label_ch2': {'is_array': False, 'is_keyword': False, 'original_name': 'label_ch2', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Label ch2'}, 'curated_disease': {'is_array': True, 'is_keyword': True, 'original_name': 'kw_curated_disease', 'description': 'Disease associated with the sample', 'type': 'text', 'is_filter': True, 'is_column': True, 'is_curated': False, 'filter_size': 3000, 'display_name': 'Disease'}, 'label_ch1': {'is_array': False, 'is_keyword': False, 'original_name': 'label_ch1', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Label ch1'}, 'time_point_ch1': {'is_array': False, 'is_keyword': False, 'original_name': 'time_point_ch1', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Time point ch1'}, 'characteristics_ch1_3': {'is_array': False, 'is_keyword': False, 'original_name': 'characteristics_ch1.3', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Characteristics ch1_2'}, 'characteristics_ch1_2': {'is_array': False, 'is_keyword': False, 'original_name': 'characteristics_ch1.2', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Characteristics ch1_2'}, 'curated_tissue': {'is_array': False, 'is_keyword': True, 'original_name': 'kw_curated_tissue', 'description': 'Tissue to which the sample belongs', 'type': 'text', 'is_filter': True, 'is_column': True, 'is_curated': False, 'filter_size': 1000, 'display_name': 'Tissue'}, 'curated_drug_smiles_code': {'is_array': True, 'is_keyword': True, 'original_name': 'kw_curated_smiles', 'description': 'SMILES code of the drug admistered in the sample', 'type': 'text', 'is_filter': False, 'is_column': True, 'is_curated': False, 'filter_size': 1, 'display_name': 'Drug (SMILES code)'}, 'hyb_protocol': {'is_array': False, 'is_keyword': False, 'original_name': 'hyb_protocol', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Hybridization protocol'}, 'platform_id': {'is_array': False, 'is_keyword': False, 'original_name': 'platform_id', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Platform ID'}, 'is_current': {'is_array': False, 'is_keyword': True, 'original_name': 'is_current', 'description': 'Whether this is the current version of the data entity or not', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 2000, 'display_name': 'Is current'}, 'characteristics_ch1_1': {'is_array': False, 'is_keyword': False, 'original_name': 'characteristics_ch1.1', 'description': 'NA', 'type': 'text', 'is_filter': False, 'is_column': False, 'is_curated': False, 'filter_size': 1, 'display_name': 'Characteristics ch1_1'}}}}, 'created_time': 1653926604432, 'last_updated_time': 1653926604432, 'enabled': True}}})\n"
          ]
        }
      ],
      "source": [
        "print(resp)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "query = \"SHOW TABLES IN gnomad\"\n",
        "res = omixatlas.query_metadata(query)\n",
        "res"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 26,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Query execution succeeded (time taken: 0.91 seconds, data scanned: 0.000 MB)\n",
            "Fetched 2 rows\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>table_name</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>ukbiobank.datasets</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>ukbiobank.variant_data</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "               table_name\n",
              "0      ukbiobank.datasets\n",
              "1  ukbiobank.variant_data"
            ]
          },
          "execution_count": 26,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "query = \"SHOW TABLES IN ukbiobank\"\n",
        "res = omixatlas.query_metadata(query)\n",
        "res"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": []
    }
  ],
  "metadata": {
    "kernelspec": {
      "display_name": "Python 3 (ipykernel)",
      "language": "python",
      "name": "python3"
    },
    "language_info": {
      "codemirror_mode": {
        "name": "ipython",
        "version": 3
      },
      "file_extension": ".py",
      "mimetype": "text/x-python",
      "name": "python",
      "nbconvert_exporter": "python",
      "pygments_lexer": "ipython3",
      "version": "3.10.2"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 2
}