{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "colab_type": "text",
        "id": "view-in-github"
      },
      "source": [
        "<a href=\"https://polly.elucidata.io/manage/workspaces?action=open_polly_notebook&source=github&path=path_place_holder&kernel=elucidata/Python 3.10&machine=medium\" target=\"_parent\"><img src=\"https://elucidatainc.github.io/PublicAssets/open_polly.svg\" alt=\"Open in Polly\"/></a>\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "# Welcome to Polly Python3 Notebook."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 2,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "/bin/bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)\n",
            "Collecting polly-python==1.1.0-base-url-auth-omixatlas\n",
            "  Downloading https://elucidatainc.github.io/PublicAssets/builds/polly-python/tests/testpolly/polly_python-1.1.0_base_url_auth_omixatlas-py3-none-any.whl (156 kB)\n",
            "\u001b[K     |████████████████████████████████| 156 kB 5.2 MB/s eta 0:00:01\n",
            "\u001b[?25hRequirement already satisfied: mixpanel==4.10.0 in /usr/local/lib/python3.10/site-packages (from polly-python==1.1.0-base-url-auth-omixatlas) (4.10.0)\n",
            "Requirement already satisfied: python-jose==3.3.0 in /usr/local/lib/python3.10/site-packages (from polly-python==1.1.0-base-url-auth-omixatlas) (3.3.0)\n",
            "Requirement already satisfied: cloudpathlib>=0.15.0 in /usr/local/lib/python3.10/site-packages (from polly-python==1.1.0-base-url-auth-omixatlas) (0.17.0)\n",
            "Requirement already satisfied: elucidatacmapPy==3.3.4 in /usr/local/lib/python3.10/site-packages (from polly-python==1.1.0-base-url-auth-omixatlas) (3.3.4)\n",
            "Requirement already satisfied: pandas>=1.3.5 in /usr/local/lib/python3.10/site-packages (from polly-python==1.1.0-base-url-auth-omixatlas) (1.4.1)\n",
            "Requirement already satisfied: Deprecated>=1.2.12 in /usr/local/lib/python3.10/site-packages (from polly-python==1.1.0-base-url-auth-omixatlas) (1.2.13)\n",
            "Requirement already satisfied: retrying==1.3.4 in /usr/local/lib/python3.10/site-packages (from polly-python==1.1.0-base-url-auth-omixatlas) (1.3.4)\n",
            "Requirement already satisfied: cryptography<=38.0.0,>=37.0.1 in /usr/local/lib/python3.10/site-packages (from polly-python==1.1.0-base-url-auth-omixatlas) (38.0.0)\n",
            "Requirement already satisfied: tqdm==4.65.0 in /usr/local/lib/python3.10/site-packages (from polly-python==1.1.0-base-url-auth-omixatlas) (4.65.0)\n",
            "Requirement already satisfied: pydantic==1.10.12 in /usr/local/lib/python3.10/site-packages (from polly-python==1.1.0-base-url-auth-omixatlas) (1.10.12)\n",
            "Requirement already satisfied: boto3>=1.24.0 in /usr/local/lib/python3.10/site-packages (from polly-python==1.1.0-base-url-auth-omixatlas) (1.34.7)\n",
            "Requirement already satisfied: pytest>=6.2.5 in /usr/local/lib/python3.10/site-packages (from polly-python==1.1.0-base-url-auth-omixatlas) (7.3.1)\n",
            "Requirement already satisfied: rst2txt==1.1.0 in /usr/local/lib/python3.10/site-packages (from polly-python==1.1.0-base-url-auth-omixatlas) (1.1.0)\n",
            "Requirement already satisfied: botocore>=1.27.0 in /usr/local/lib/python3.10/site-packages (from polly-python==1.1.0-base-url-auth-omixatlas) (1.34.7)\n",
            "Requirement already satisfied: polly-validator==0.0.2 in /usr/local/lib/python3.10/site-packages (from polly-python==1.1.0-base-url-auth-omixatlas) (0.0.2)\n",
            "Requirement already satisfied: requests==2.25.1 in /usr/local/lib/python3.10/site-packages (from polly-python==1.1.0-base-url-auth-omixatlas) (2.25.1)\n",
            "Requirement already satisfied: plotly>=5.0.0 in /usr/local/lib/python3.10/site-packages (from polly-python==1.1.0-base-url-auth-omixatlas) (5.14.1)\n",
            "Requirement already satisfied: datapane>=0.10.3 in /usr/local/lib/python3.10/site-packages (from polly-python==1.1.0-base-url-auth-omixatlas) (0.16.4)\n",
            "Requirement already satisfied: joblib>=1.2.0 in /usr/local/lib/python3.10/site-packages (from polly-python==1.1.0-base-url-auth-omixatlas) (1.2.0)\n",
            "Requirement already satisfied: numpy>=1.11.2 in /usr/local/lib/python3.10/site-packages (from elucidatacmapPy==3.3.4->polly-python==1.1.0-base-url-auth-omixatlas) (1.22.3)\n",
            "Requirement already satisfied: six in /usr/local/lib/python3.10/site-packages (from elucidatacmapPy==3.3.4->polly-python==1.1.0-base-url-auth-omixatlas) (1.16.0)\n",
            "Requirement already satisfied: h5py>=2.6.0 in /usr/local/lib/python3.10/site-packages (from elucidatacmapPy==3.3.4->polly-python==1.1.0-base-url-auth-omixatlas) (3.8.0)\n",
            "Requirement already satisfied: urllib3 in /usr/local/lib/python3.10/site-packages (from mixpanel==4.10.0->polly-python==1.1.0-base-url-auth-omixatlas) (1.26.6)\n",
            "Requirement already satisfied: rapidfuzz==2.1.1 in /usr/local/lib/python3.10/site-packages (from polly-validator==0.0.2->polly-python==1.1.0-base-url-auth-omixatlas) (2.1.1)\n",
            "Requirement already satisfied: appdirs==1.4.4 in /usr/local/lib/python3.10/site-packages (from polly-validator==0.0.2->polly-python==1.1.0-base-url-auth-omixatlas) (1.4.4)\n",
            "Requirement already satisfied: tabulate==0.8.10 in /usr/local/lib/python3.10/site-packages (from polly-validator==0.0.2->polly-python==1.1.0-base-url-auth-omixatlas) (0.8.10)\n",
            "Requirement already satisfied: pronto in /usr/local/lib/python3.10/site-packages (from polly-validator==0.0.2->polly-python==1.1.0-base-url-auth-omixatlas) (2.5.1)\n",
            "Requirement already satisfied: typing-extensions>=4.2.0 in /usr/local/lib/python3.10/site-packages (from pydantic==1.10.12->polly-python==1.1.0-base-url-auth-omixatlas) (4.5.0)\n",
            "Requirement already satisfied: ecdsa!=0.15 in /usr/local/lib/python3.10/site-packages (from python-jose==3.3.0->polly-python==1.1.0-base-url-auth-omixatlas) (0.18.0)\n",
            "Requirement already satisfied: rsa in /usr/local/lib/python3.10/site-packages (from python-jose==3.3.0->polly-python==1.1.0-base-url-auth-omixatlas) (4.7.2)\n",
            "Requirement already satisfied: pyasn1 in /usr/local/lib/python3.10/site-packages (from python-jose==3.3.0->polly-python==1.1.0-base-url-auth-omixatlas) (0.4.8)\n",
            "Requirement already satisfied: jarowinkler<1.1.0,>=1.0.3 in /usr/local/lib/python3.10/site-packages (from rapidfuzz==2.1.1->polly-validator==0.0.2->polly-python==1.1.0-base-url-auth-omixatlas) (1.0.5)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.10/site-packages (from requests==2.25.1->polly-python==1.1.0-base-url-auth-omixatlas) (2021.5.30)\n",
            "Requirement already satisfied: idna<3,>=2.5 in /usr/local/lib/python3.10/site-packages (from requests==2.25.1->polly-python==1.1.0-base-url-auth-omixatlas) (2.10)\n",
            "Requirement already satisfied: chardet<5,>=3.0.2 in /usr/local/lib/python3.10/site-packages (from requests==2.25.1->polly-python==1.1.0-base-url-auth-omixatlas) (4.0.0)\n",
            "Requirement already satisfied: docutils>=0.11 in /usr/local/lib/python3.10/site-packages (from rst2txt==1.1.0->polly-python==1.1.0-base-url-auth-omixatlas) (0.15.2)\n",
            "Requirement already satisfied: pygments in /usr/local/lib/python3.10/site-packages (from rst2txt==1.1.0->polly-python==1.1.0-base-url-auth-omixatlas) (2.10.0)\n",
            "Requirement already satisfied: jmespath<2.0.0,>=0.7.1 in /usr/local/lib/python3.10/site-packages (from boto3>=1.24.0->polly-python==1.1.0-base-url-auth-omixatlas) (0.10.0)\n",
            "Requirement already satisfied: s3transfer<0.11.0,>=0.10.0 in /usr/local/lib/python3.10/site-packages (from boto3>=1.24.0->polly-python==1.1.0-base-url-auth-omixatlas) (0.10.0)\n",
            "Requirement already satisfied: python-dateutil<3.0.0,>=2.1 in /usr/local/lib/python3.10/site-packages (from botocore>=1.27.0->polly-python==1.1.0-base-url-auth-omixatlas) (2.8.1)\n",
            "Requirement already satisfied: cffi>=1.12 in /usr/local/lib/python3.10/site-packages (from cryptography<=38.0.0,>=37.0.1->polly-python==1.1.0-base-url-auth-omixatlas) (1.15.1)\n",
            "Requirement already satisfied: pycparser in /usr/local/lib/python3.10/site-packages (from cffi>=1.12->cryptography<=38.0.0,>=37.0.1->polly-python==1.1.0-base-url-auth-omixatlas) (2.21)\n",
            "Requirement already satisfied: lxml<5.0.0,>=4.0.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (4.9.2)\n",
            "Requirement already satisfied: datacommons-pandas<0.0.4,>=0.0.3 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (0.0.3)\n",
            "Requirement already satisfied: datacommons<2.0.0,>=1.4.3 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (1.4.3)\n",
            "Requirement already satisfied: click<9.0.0,>=7.1.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (8.1.3)\n",
            "Requirement already satisfied: pyngrok<6.0.0,>=5.2.1 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (5.2.3)\n",
            "Requirement already satisfied: boltons<22.0.0,>=20.0.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (21.0.0)\n",
            "Requirement already satisfied: glom<24.0.0,>=22.1.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (23.3.0)\n",
            "Requirement already satisfied: altair<5.0.0,>=4.0.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (4.2.2)\n"
          ]
        },
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Requirement already satisfied: toolz<0.13.0,>=0.11.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (0.12.0)\n",
            "Requirement already satisfied: micawber>=0.5.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (0.5.4)\n",
            "Requirement already satisfied: dominate<3.0.0,>=2.4.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (2.7.0)\n",
            "Requirement already satisfied: PyYAML<7.0.0,>=5.4.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (5.4.1)\n",
            "Requirement already satisfied: multimethod<2.0,>=1.9 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (1.9.1)\n",
            "Requirement already satisfied: furl<3.0.0,>=2.0.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (2.1.3)\n",
            "Requirement already satisfied: nbconvert>=5.6.1 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (6.3.0)\n",
            "Requirement already satisfied: ipynbname<2022.0.0,>=2021.3.2 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (2021.3.2)\n",
            "Requirement already satisfied: vega-datasets<1.0.0,>=0.9.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (0.9.0)\n",
            "Requirement already satisfied: cheroot<10.0.0,>=9.0.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (9.0.0)\n",
            "Requirement already satisfied: requests-toolbelt<0.11.0,>=0.9.1 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (0.10.1)\n",
            "Requirement already satisfied: packaging<24.0.0,>=21.0.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (23.1)\n",
            "Requirement already satisfied: jsonschema<5.0.0,>=3.2.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (4.2.1)\n",
            "Requirement already satisfied: click-spinner<0.2.0,>=0.1.8 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (0.1.10)\n",
            "Requirement already satisfied: colorlog<7.0.0,>=4.1.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (6.7.0)\n",
            "Requirement already satisfied: posthog<3.0.0,>=1.4.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (2.5.0)\n",
            "Requirement already satisfied: importlib_resources<6.0.0,>=3.0.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (5.12.0)\n",
            "Requirement already satisfied: pyarrow<11.0.0,>=6.0.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (10.0.1)\n",
            "Requirement already satisfied: dacite<2.0.0,>=1.0.2 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (1.8.0)\n",
            "Requirement already satisfied: dulwich<0.22.0,>=0.20.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (0.21.3)\n",
            "Requirement already satisfied: jinja2 in /usr/local/lib/python3.10/site-packages (from altair<5.0.0,>=4.0.0->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (3.0.3)\n",
            "Requirement already satisfied: entrypoints in /usr/local/lib/python3.10/site-packages (from altair<5.0.0,>=4.0.0->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (0.3)\n",
            "Requirement already satisfied: jaraco.functools in /usr/local/lib/python3.10/site-packages (from cheroot<10.0.0,>=9.0.0->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (3.6.0)\n",
            "Requirement already satisfied: more-itertools>=2.6 in /usr/local/lib/python3.10/site-packages (from cheroot<10.0.0,>=9.0.0->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (9.1.0)\n",
            "Requirement already satisfied: wrapt<2,>=1.10 in /usr/local/lib/python3.10/site-packages (from Deprecated>=1.2.12->polly-python==1.1.0-base-url-auth-omixatlas) (1.15.0)\n",
            "Requirement already satisfied: orderedmultidict>=1.0.1 in /usr/local/lib/python3.10/site-packages (from furl<3.0.0,>=2.0.0->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (1.0.1)\n",
            "Requirement already satisfied: attrs in /usr/local/lib/python3.10/site-packages (from glom<24.0.0,>=22.1.0->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (21.2.0)\n",
            "Requirement already satisfied: face==20.1.1 in /usr/local/lib/python3.10/site-packages (from glom<24.0.0,>=22.1.0->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (20.1.1)\n",
            "Requirement already satisfied: ipykernel in /usr/local/lib/python3.10/site-packages (from ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (6.13.1)\n",
            "Requirement already satisfied: pyrsistent!=0.17.0,!=0.17.1,!=0.17.2,>=0.14.0 in /usr/local/lib/python3.10/site-packages (from jsonschema<5.0.0,>=3.2.0->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (0.18.0)\n",
            "Requirement already satisfied: pandocfilters>=1.4.1 in /usr/local/lib/python3.10/site-packages (from nbconvert>=5.6.1->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (1.5.0)\n",
            "Requirement already satisfied: mistune<2,>=0.8.1 in /usr/local/lib/python3.10/site-packages (from nbconvert>=5.6.1->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (0.8.4)\n",
            "Requirement already satisfied: jupyterlab-pygments in /usr/local/lib/python3.10/site-packages (from nbconvert>=5.6.1->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (0.2.2)\n",
            "Requirement already satisfied: bleach in /usr/local/lib/python3.10/site-packages (from nbconvert>=5.6.1->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (4.1.0)\n",
            "Requirement already satisfied: testpath in /usr/local/lib/python3.10/site-packages (from nbconvert>=5.6.1->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (0.5.0)\n",
            "Requirement already satisfied: defusedxml in /usr/local/lib/python3.10/site-packages (from nbconvert>=5.6.1->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (0.7.1)\n",
            "Requirement already satisfied: nbformat>=4.4 in /usr/local/lib/python3.10/site-packages (from nbconvert>=5.6.1->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (5.1.3)\n",
            "Requirement already satisfied: jupyter-core in /usr/local/lib/python3.10/site-packages (from nbconvert>=5.6.1->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (4.9.1)\n",
            "Requirement already satisfied: traitlets>=5.0 in /usr/local/lib/python3.10/site-packages (from nbconvert>=5.6.1->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (5.1.1)\n",
            "Requirement already satisfied: nbclient<0.6.0,>=0.5.0 in /usr/local/lib/python3.10/site-packages (from nbconvert>=5.6.1->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (0.5.13)\n",
            "Requirement already satisfied: MarkupSafe>=2.0 in /usr/local/lib/python3.10/site-packages (from jinja2->altair<5.0.0,>=4.0.0->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (2.0.1)\n",
            "Requirement already satisfied: nest-asyncio in /usr/local/lib/python3.10/site-packages (from nbclient<0.6.0,>=0.5.0->nbconvert>=5.6.1->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (1.5.6)\n",
            "Requirement already satisfied: jupyter-client>=6.1.5 in /usr/local/lib/python3.10/site-packages (from nbclient<0.6.0,>=0.5.0->nbconvert>=5.6.1->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (6.1.12)\n",
            "Requirement already satisfied: tornado>=4.1 in /usr/local/lib/python3.10/site-packages (from jupyter-client>=6.1.5->nbclient<0.6.0,>=0.5.0->nbconvert>=5.6.1->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (6.1)\n",
            "Requirement already satisfied: pyzmq>=13 in /usr/local/lib/python3.10/site-packages (from jupyter-client>=6.1.5->nbclient<0.6.0,>=0.5.0->nbconvert>=5.6.1->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (22.3.0)\n"
          ]
        },
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Requirement already satisfied: ipython-genutils in /usr/local/lib/python3.10/site-packages (from nbformat>=4.4->nbconvert>=5.6.1->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (0.2.0)\n",
            "Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.10/site-packages (from pandas>=1.3.5->polly-python==1.1.0-base-url-auth-omixatlas) (2021.3)\n",
            "Requirement already satisfied: tenacity>=6.2.0 in /usr/local/lib/python3.10/site-packages (from plotly>=5.0.0->polly-python==1.1.0-base-url-auth-omixatlas) (8.2.2)\n",
            "Requirement already satisfied: backoff>=1.10.0 in /usr/local/lib/python3.10/site-packages (from posthog<3.0.0,>=1.4.0->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (2.2.1)\n",
            "Requirement already satisfied: monotonic>=1.5 in /usr/local/lib/python3.10/site-packages (from posthog<3.0.0,>=1.4.0->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (1.6)\n",
            "Requirement already satisfied: pluggy<2.0,>=0.12 in /usr/local/lib/python3.10/site-packages (from pytest>=6.2.5->polly-python==1.1.0-base-url-auth-omixatlas) (1.0.0)\n",
            "Requirement already satisfied: exceptiongroup>=1.0.0rc8 in /usr/local/lib/python3.10/site-packages (from pytest>=6.2.5->polly-python==1.1.0-base-url-auth-omixatlas) (1.1.1)\n",
            "Requirement already satisfied: tomli>=1.0.0 in /usr/local/lib/python3.10/site-packages (from pytest>=6.2.5->polly-python==1.1.0-base-url-auth-omixatlas) (2.0.1)\n",
            "Requirement already satisfied: iniconfig in /usr/local/lib/python3.10/site-packages (from pytest>=6.2.5->polly-python==1.1.0-base-url-auth-omixatlas) (2.0.0)\n",
            "Requirement already satisfied: webencodings in /usr/local/lib/python3.10/site-packages (from bleach->nbconvert>=5.6.1->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (0.5.1)\n",
            "Requirement already satisfied: ipython>=7.23.1 in /usr/local/lib/python3.10/site-packages (from ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (8.2.0)\n",
            "Requirement already satisfied: debugpy>=1.0 in /usr/local/lib/python3.10/site-packages (from ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (1.6.7)\n",
            "Requirement already satisfied: psutil in /usr/local/lib/python3.10/site-packages (from ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (5.9.5)\n",
            "Requirement already satisfied: matplotlib-inline>=0.1 in /usr/local/lib/python3.10/site-packages (from ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (0.1.3)\n",
            "Requirement already satisfied: pexpect>4.3 in /usr/local/lib/python3.10/site-packages (from ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (4.8.0)\n",
            "Requirement already satisfied: prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0 in /usr/local/lib/python3.10/site-packages (from ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (3.0.22)\n",
            "Requirement already satisfied: setuptools>=18.5 in /usr/local/lib/python3.10/site-packages (from ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (58.1.0)\n",
            "Requirement already satisfied: pickleshare in /usr/local/lib/python3.10/site-packages (from ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (0.7.5)\n",
            "Requirement already satisfied: jedi>=0.16 in /usr/local/lib/python3.10/site-packages (from ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (0.18.1)\n",
            "Requirement already satisfied: stack-data in /usr/local/lib/python3.10/site-packages (from ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (0.6.2)\n",
            "Requirement already satisfied: backcall in /usr/local/lib/python3.10/site-packages (from ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (0.2.0)\n",
            "Requirement already satisfied: decorator in /usr/local/lib/python3.10/site-packages (from ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (5.1.0)\n",
            "Requirement already satisfied: parso<0.9.0,>=0.8.0 in /usr/local/lib/python3.10/site-packages (from jedi>=0.16->ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (0.8.2)\n",
            "Requirement already satisfied: ptyprocess>=0.5 in /usr/local/lib/python3.10/site-packages (from pexpect>4.3->ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (0.7.0)\n",
            "Requirement already satisfied: wcwidth in /usr/local/lib/python3.10/site-packages (from prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0->ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (0.2.5)\n",
            "Requirement already satisfied: fastobo~=0.12.1 in /usr/local/lib/python3.10/site-packages (from pronto->polly-validator==0.0.2->polly-python==1.1.0-base-url-auth-omixatlas) (0.12.2)\n",
            "Requirement already satisfied: networkx~=2.3 in /usr/local/lib/python3.10/site-packages (from pronto->polly-validator==0.0.2->polly-python==1.1.0-base-url-auth-omixatlas) (2.8.8)\n",
            "Requirement already satisfied: asttokens>=2.1.0 in /usr/local/lib/python3.10/site-packages (from stack-data->ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (2.2.1)\n",
            "Requirement already satisfied: pure-eval in /usr/local/lib/python3.10/site-packages (from stack-data->ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (0.2.2)\n",
            "Requirement already satisfied: executing>=1.2.0 in /usr/local/lib/python3.10/site-packages (from stack-data->ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==1.1.0-base-url-auth-omixatlas) (1.2.0)\n",
            "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\n",
            "\u001b[33mWARNING: You are using pip version 21.2.4; however, version 23.3.2 is available.\n",
            "You should consider upgrading via the '/usr/local/bin/python3.10 -m pip install --upgrade pip' command.\u001b[0m\n"
          ]
        }
      ],
      "source": [
        "!sudo pip3 install https://elucidatainc.github.io/PublicAssets/builds/polly-python/tests/testpolly/polly_python-1.1.0_base_url_auth_omixatlas-py3-none-any.whl"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {},
      "outputs": [],
      "source": [
        "import os\n",
        "from polly.auth import Polly\n",
        "AUTH_TOKEN=(\"OTZlNGZjNjA1ZDo6N2U0Zjg5NDM4MzI1ZjA0Mzk0NDQwNGFlZDhiZmRmODNmOTE3OTk4YQ==\") # Obtain authentication tokens\n",
        "Polly.auth(AUTH_TOKEN, env = \"polly\")\n",
        "from polly.omixatlas import OmixAtlas\n",
        "omixatlas = OmixAtlas()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 5,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/plain": [
              "{'data': [{'repo_id': '1655799559805',\n",
              "   'repo_name': 'gdx_atlas',\n",
              "   'indexes': {'files': 'gdx_atlas_files',\n",
              "    'gct_col_metadata': 'gdx_atlas_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'gdx_atlas_files',\n",
              "    'gct_col_metadata': 'gdx_atlas_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 5103}],\n",
              "   'datatypes': [{'transcriptomics': 5103}],\n",
              "   'dataset_count': 5103,\n",
              "   'disease_count': 1673,\n",
              "   'tissue_count': 428,\n",
              "   'organism_count': 0,\n",
              "   'cell_line_count': 701,\n",
              "   'cell_type_count': 390,\n",
              "   'drug_count': 664,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 0.0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1650348922627',\n",
              "   'repo_name': 'atavistik',\n",
              "   'indexes': {'files': 'atavistik_files',\n",
              "    'gct_col_metadata': 'atavistik_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'atavistik_files',\n",
              "    'gct_col_metadata': 'atavistik_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [],\n",
              "   'datatypes': [{'metabolomics': 8}],\n",
              "   'dataset_count': 9,\n",
              "   'disease_count': 0,\n",
              "   'tissue_count': 0,\n",
              "   'organism_count': 0,\n",
              "   'cell_line_count': 0,\n",
              "   'cell_type_count': 0,\n",
              "   'drug_count': 0,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 0,\n",
              "   'sample_count': 944.0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1623221686703',\n",
              "   'repo_name': 'gdc',\n",
              "   'indexes': {'files': 'gdc_files',\n",
              "    'gct_col_metadata': 'gdc_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'gdc_files',\n",
              "    'gct_col_metadata': 'gdc_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'gdc': 9869}],\n",
              "   'datatypes': [{'mirna': 4928},\n",
              "    {'transcriptomics': 3641},\n",
              "    {'copy number variation': 1300}],\n",
              "   'dataset_count': 9869,\n",
              "   'disease_count': 62,\n",
              "   'tissue_count': 39,\n",
              "   'organism_count': 1,\n",
              "   'cell_line_count': 1,\n",
              "   'cell_type_count': 1,\n",
              "   'drug_count': 21,\n",
              "   'data_type_count': 3,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 9869.0,\n",
              "   'normal_sample_count': 72},\n",
              "  {'repo_id': '1700562249173',\n",
              "   'repo_name': 'cptac_staging',\n",
              "   'indexes': {'files': 'cptac_staging_files',\n",
              "    'gct_col_metadata': 'cptac_staging_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'cptac_staging_files',\n",
              "    'gct_col_metadata': 'cptac_staging_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'cptac': 37}],\n",
              "   'datatypes': [{'transcriptomics': 18},\n",
              "    {'proteomics': 13},\n",
              "    {'raw counts transcriptomics': 6}],\n",
              "   'dataset_count': 37,\n",
              "   'disease_count': 12,\n",
              "   'tissue_count': 18,\n",
              "   'organism_count': 1,\n",
              "   'cell_line_count': 1,\n",
              "   'cell_type_count': 1,\n",
              "   'drug_count': 1,\n",
              "   'data_type_count': 3,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 4804.0,\n",
              "   'normal_sample_count': 234},\n",
              "  {'repo_id': '1610613867671',\n",
              "   'repo_name': 'ccle',\n",
              "   'indexes': {},\n",
              "   'v2_indexes': {},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [],\n",
              "   'datatypes': [],\n",
              "   'dataset_count': 0,\n",
              "   'disease_count': 0,\n",
              "   'tissue_count': 0,\n",
              "   'organism_count': 0,\n",
              "   'cell_line_count': 0,\n",
              "   'cell_type_count': 0,\n",
              "   'drug_count': 0,\n",
              "   'data_type_count': 0,\n",
              "   'data_source_count': 0,\n",
              "   'sample_count': 0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1670406285207',\n",
              "   'repo_name': 'destination_atlas_demo_sc',\n",
              "   'indexes': {'files': 'destination_atlas_demo_sc_files',\n",
              "    'h5ad_col_metadata': 'destination_atlas_demo_sc_h5ad_col_metadata'},\n",
              "   'v2_indexes': {'files': 'destination_atlas_demo_sc_files',\n",
              "    'h5ad_col_metadata': 'destination_atlas_demo_sc_h5ad_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 3}],\n",
              "   'datatypes': [{'single cell': 3}],\n",
              "   'dataset_count': 3,\n",
              "   'disease_count': 6,\n",
              "   'tissue_count': 3,\n",
              "   'organism_count': 2,\n",
              "   'cell_line_count': 2,\n",
              "   'cell_type_count': 9,\n",
              "   'drug_count': 1,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 12.0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1663062208722',\n",
              "   'repo_name': 'connector_testing_oa',\n",
              "   'indexes': {'files': 'connector_testing_oa_files',\n",
              "    'h5ad_col_metadata': 'connector_testing_oa_h5ad_col_metadata'},\n",
              "   'v2_indexes': {'files': 'connector_testing_oa_files',\n",
              "    'h5ad_col_metadata': 'connector_testing_oa_h5ad_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 6}],\n",
              "   'datatypes': [{'single cell': 6}],\n",
              "   'dataset_count': 6,\n",
              "   'disease_count': 4,\n",
              "   'tissue_count': 3,\n",
              "   'organism_count': 1,\n",
              "   'cell_line_count': 1,\n",
              "   'cell_type_count': 3,\n",
              "   'drug_count': 1,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 7.0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1687332105755',\n",
              "   'repo_name': 'gdx_atlas_with_single_factors',\n",
              "   'indexes': {'files': 'gdx_atlas_with_single_factors_files',\n",
              "    'gct_col_metadata': 'gdx_atlas_with_single_factors_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'gdx_atlas_with_single_factors_files',\n",
              "    'gct_col_metadata': 'gdx_atlas_with_single_factors_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'': 40797}, {'geo': 10261}],\n",
              "   'datatypes': [{'gdx': 20329},\n",
              "    {'pca_scores': 10261},\n",
              "    {'raw_counts': 10261},\n",
              "    {'normalized': 10207}],\n",
              "   'dataset_count': 51058,\n",
              "   'disease_count': 1718,\n",
              "   'tissue_count': 588,\n",
              "   'organism_count': 38,\n",
              "   'cell_line_count': 852,\n",
              "   'cell_type_count': 515,\n",
              "   'drug_count': 946,\n",
              "   'data_type_count': 4,\n",
              "   'data_source_count': 2,\n",
              "   'sample_count': 105083.0,\n",
              "   'normal_sample_count': 76697},\n",
              "  {'repo_id': '1677750932401',\n",
              "   'repo_name': 'bulkrnaseq_staging_oa',\n",
              "   'indexes': {'files': 'bulkrnaseq_staging_oa_files',\n",
              "    'gct_col_metadata': 'bulkrnaseq_staging_oa_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'bulkrnaseq_staging_oa_files',\n",
              "    'gct_col_metadata': 'bulkrnaseq_staging_oa_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 42463}],\n",
              "   'datatypes': [{'raw counts transcriptomics': 42463}],\n",
              "   'dataset_count': 42463,\n",
              "   'disease_count': 2765,\n",
              "   'tissue_count': 887,\n",
              "   'organism_count': 144,\n",
              "   'cell_line_count': 2857,\n",
              "   'cell_type_count': 757,\n",
              "   'drug_count': 2032,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 685014.0,\n",
              "   'normal_sample_count': 476405},\n",
              "  {'repo_id': '1694526139205',\n",
              "   'repo_name': 'bulk_star_oa',\n",
              "   'indexes': {'files': 'bulk_star_oa_files',\n",
              "    'gct_col_metadata': 'bulk_star_oa_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'bulk_star_oa_files',\n",
              "    'gct_col_metadata': 'bulk_star_oa_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 114}],\n",
              "   'datatypes': [{'raw counts transcriptomics': 114}],\n",
              "   'dataset_count': 114,\n",
              "   'disease_count': 84,\n",
              "   'tissue_count': 45,\n",
              "   'organism_count': 2,\n",
              "   'cell_line_count': 218,\n",
              "   'cell_type_count': 17,\n",
              "   'drug_count': 31,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 13316.0,\n",
              "   'normal_sample_count': 2042},\n",
              "  {'repo_id': '1631922399583',\n",
              "   'repo_name': 'connector_staging',\n",
              "   'indexes': {'files': 'connector_staging_files',\n",
              "    'gct_col_metadata': 'connector_staging_gct_col_metadata',\n",
              "    'h5ad_col_metadata': 'connector_staging_h5ad_col_metadata',\n",
              "    'h5ad_row_metadata': 'connector_staging_h5ad_row_metadata'},\n",
              "   'v2_indexes': {'files': 'connector_staging_files',\n",
              "    'gct_col_metadata': 'connector_staging_gct_col_metadata',\n",
              "    'h5ad_col_metadata': 'connector_staging_h5ad_col_metadata',\n",
              "    'h5ad_row_metadata': 'connector_staging_h5ad_row_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 788},\n",
              "    {'expression atlas': 24},\n",
              "    {'tabula sapiens': 24},\n",
              "    {'tcga': 17},\n",
              "    {'cziscience': 16},\n",
              "    {'scp': 15},\n",
              "    {'metabolomics workbench': 8},\n",
              "    {'publication': 8},\n",
              "    {'': 3},\n",
              "    {'covid-19 cell atlas': 2}],\n",
              "   'datatypes': [{'raw counts transcriptomics': 579},\n",
              "    {'single cell': 165},\n",
              "    {'transcriptomics': 140},\n",
              "    {'drug response': 120},\n",
              "    {'metabolomics': 11},\n",
              "    {'gwas': 4},\n",
              "    {'copy number variation': 3},\n",
              "    {'methylation': 3},\n",
              "    {'mirna': 3},\n",
              "    {'proteomics': 3},\n",
              "    {'mutation': 2},\n",
              "    {'': 1}],\n",
              "   'dataset_count': 1036,\n",
              "   'disease_count': 489,\n",
              "   'tissue_count': 247,\n",
              "   'organism_count': 10,\n",
              "   'cell_line_count': 140,\n",
              "   'cell_type_count': 371,\n",
              "   'drug_count': 148,\n",
              "   'data_type_count': 12,\n",
              "   'data_source_count': 18,\n",
              "   'sample_count': 338410.0,\n",
              "   'normal_sample_count': 17354},\n",
              "  {'repo_id': '9',\n",
              "   'repo_name': 'geo',\n",
              "   'indexes': {'files': 'geo_files',\n",
              "    'gct_col_metadata': 'geo_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'geo_files',\n",
              "    'gct_col_metadata': 'geo_gct_col_metadata'},\n",
              "   'linked_workspace_id': '11087',\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 110254}],\n",
              "   'datatypes': [{'transcriptomics': 110249}, {'microarray': 5}],\n",
              "   'dataset_count': 110254,\n",
              "   'disease_count': 6903,\n",
              "   'tissue_count': 1100,\n",
              "   'organism_count': 320,\n",
              "   'cell_line_count': 5184,\n",
              "   'cell_type_count': 912,\n",
              "   'drug_count': 3403,\n",
              "   'data_type_count': 2,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 2898235.0,\n",
              "   'normal_sample_count': 529077},\n",
              "  {'repo_id': '1670220926746',\n",
              "   'repo_name': 'destination_atlas_demo',\n",
              "   'indexes': {'files': 'destination_atlas_demo_files',\n",
              "    'gct_col_metadata': 'destination_atlas_demo_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'destination_atlas_demo_files',\n",
              "    'gct_col_metadata': 'destination_atlas_demo_gct_col_metadata'},\n",
              "   'data_type': 'bulk_rna_seq',\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 2}],\n",
              "   'datatypes': [{'transcriptomics': 2}],\n",
              "   'dataset_count': 2,\n",
              "   'disease_count': 2,\n",
              "   'tissue_count': 3,\n",
              "   'organism_count': 2,\n",
              "   'cell_line_count': 1,\n",
              "   'cell_type_count': 2,\n",
              "   'drug_count': 1,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 102.0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1657010176337',\n",
              "   'repo_name': 'ccle_data_lake',\n",
              "   'indexes': {'files': 'ccle_data_lake_files',\n",
              "    'gct_col_metadata': 'ccle_data_lake_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'ccle_data_lake_files',\n",
              "    'gct_col_metadata': 'ccle_data_lake_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'ccle': 11780}],\n",
              "   'datatypes': [{'copy number variation': 3397},\n",
              "    {'mutation': 3177},\n",
              "    {'transcriptomics': 2425},\n",
              "    {'mirna': 954},\n",
              "    {'metabolomics': 928},\n",
              "    {'proteomics': 899}],\n",
              "   'dataset_count': 11780,\n",
              "   'disease_count': 95,\n",
              "   'tissue_count': 48,\n",
              "   'organism_count': 1,\n",
              "   'cell_line_count': 1919,\n",
              "   'cell_type_count': 1,\n",
              "   'drug_count': 0,\n",
              "   'data_type_count': 6,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 11780.0,\n",
              "   'normal_sample_count': 20},\n",
              "  {'repo_id': '1661783374187',\n",
              "   'repo_name': 'x_therapeutics',\n",
              "   'indexes': {'files': 'x_therapeutics_files',\n",
              "    'gct_col_metadata': 'x_therapeutics_gct_col_metadata',\n",
              "    'h5ad_col_metadata': 'x_therapeutics_h5ad_col_metadata'},\n",
              "   'v2_indexes': {'files': 'x_therapeutics_files',\n",
              "    'gct_col_metadata': 'x_therapeutics_gct_col_metadata',\n",
              "    'h5ad_col_metadata': 'x_therapeutics_h5ad_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 2}],\n",
              "   'datatypes': [{'single cell': 1}, {'transcriptomics': 1}],\n",
              "   'dataset_count': 2,\n",
              "   'disease_count': 7,\n",
              "   'tissue_count': 4,\n",
              "   'organism_count': 1,\n",
              "   'cell_line_count': 1,\n",
              "   'cell_type_count': 4,\n",
              "   'drug_count': 1,\n",
              "   'data_type_count': 2,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 65.0,\n",
              "   'normal_sample_count': 11},\n",
              "  {'repo_id': '1612862312240',\n",
              "   'repo_name': 'transcriptomics__celt',\n",
              "   'indexes': {'files': 'transcriptomics__celt_files',\n",
              "    'gct_col_metadata': 'transcriptomics__celt_gct_col_metadata',\n",
              "    'h5ad_col_metadata': 'transcriptomics__celt_h5ad_col_metadata'},\n",
              "   'v2_indexes': {'files': 'transcriptomics__celt_files',\n",
              "    'gct_col_metadata': 'transcriptomics__celt_gct_col_metadata',\n",
              "    'h5ad_col_metadata': 'transcriptomics__celt_h5ad_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 119},\n",
              "    {'gene expression omnibus (geo)': 38},\n",
              "    {'publication': 10},\n",
              "    {'scp': 2},\n",
              "    {'synapse': 1},\n",
              "    {'zenodo': 1}],\n",
              "   'datatypes': [{'single cell': 147}, {'transcriptomics': 24}],\n",
              "   'dataset_count': 172,\n",
              "   'disease_count': 91,\n",
              "   'tissue_count': 45,\n",
              "   'organism_count': 2,\n",
              "   'cell_line_count': 7,\n",
              "   'cell_type_count': 56,\n",
              "   'drug_count': 11,\n",
              "   'data_type_count': 2,\n",
              "   'data_source_count': 6,\n",
              "   'sample_count': 23339.0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1609924165364',\n",
              "   'repo_name': 'cptac',\n",
              "   'indexes': {'files': 'cptac_files',\n",
              "    'gct_col_metadata': 'cptac_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'cptac_files',\n",
              "    'gct_col_metadata': 'cptac_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'cptac': 12635}, {'tcga': 1046}],\n",
              "   'datatypes': [{'proteomics': 2313},\n",
              "    {'phosphoproteomics': 1365},\n",
              "    {'transcriptomics': 1325},\n",
              "    {'copy number variation': 739},\n",
              "    {'mutation': 724},\n",
              "    {'mirna': 477},\n",
              "    {'acetylproteomics': 375},\n",
              "    {'methylation': 107},\n",
              "    {'lipidomics': 88},\n",
              "    {'metabolomics': 87}],\n",
              "   'dataset_count': 13681,\n",
              "   'disease_count': 12,\n",
              "   'tissue_count': 31,\n",
              "   'organism_count': 1,\n",
              "   'cell_line_count': 1,\n",
              "   'cell_type_count': 1,\n",
              "   'drug_count': 1,\n",
              "   'data_type_count': 10,\n",
              "   'data_source_count': 2,\n",
              "   'sample_count': 20470.0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1669621558634',\n",
              "   'repo_name': 'gsea_all_pathways',\n",
              "   'indexes': {'files': 'gsea_all_pathways_files'},\n",
              "   'v2_indexes': {'files': 'gsea_all_pathways_files'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 40189}],\n",
              "   'datatypes': [{'transcriptomics': 40189}],\n",
              "   'dataset_count': 40189,\n",
              "   'disease_count': 1795,\n",
              "   'tissue_count': 278,\n",
              "   'organism_count': 0,\n",
              "   'cell_line_count': 355,\n",
              "   'cell_type_count': 223,\n",
              "   'drug_count': 372,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 0.0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1658310195166',\n",
              "   'repo_name': 'alnylam_genespotlight_atlas',\n",
              "   'indexes': {'files': 'alnylam_genespotlight_atlas_files'},\n",
              "   'v2_indexes': {'files': 'alnylam_genespotlight_atlas_files'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 21806}],\n",
              "   'datatypes': [{'transcriptomics': 21806}],\n",
              "   'dataset_count': 21806,\n",
              "   'disease_count': 2051,\n",
              "   'tissue_count': 531,\n",
              "   'organism_count': 0,\n",
              "   'cell_line_count': 1077,\n",
              "   'cell_type_count': 525,\n",
              "   'drug_count': 618,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 0.0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1700598561420',\n",
              "   'repo_name': 'cptac_atlas',\n",
              "   'indexes': {'files': 'cptac_atlas_files',\n",
              "    'gct_col_metadata': 'cptac_atlas_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'cptac_atlas_files',\n",
              "    'gct_col_metadata': 'cptac_atlas_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'cptac': 36}],\n",
              "   'datatypes': [{'transcriptomics': 18},\n",
              "    {'proteomics': 12},\n",
              "    {'raw counts transcriptomics': 6}],\n",
              "   'dataset_count': 36,\n",
              "   'disease_count': 12,\n",
              "   'tissue_count': 18,\n",
              "   'organism_count': 1,\n",
              "   'cell_line_count': 1,\n",
              "   'cell_type_count': 1,\n",
              "   'drug_count': 1,\n",
              "   'data_type_count': 3,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 4728.0,\n",
              "   'normal_sample_count': 234},\n",
              "  {'repo_id': '1623986995264',\n",
              "   'repo_name': 'cbioportal',\n",
              "   'indexes': {'files': 'cbioportal_files',\n",
              "    'gct_col_metadata': 'cbioportal_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'cbioportal_files',\n",
              "    'gct_col_metadata': 'cbioportal_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'cbioportal': 116131}],\n",
              "   'datatypes': [{'mutation': 60405},\n",
              "    {'copy number variation': 44391},\n",
              "    {'fusion': 7076},\n",
              "    {'transcriptomics': 3703},\n",
              "    {'methylation': 556}],\n",
              "   'dataset_count': 116131,\n",
              "   'disease_count': 346,\n",
              "   'tissue_count': 141,\n",
              "   'organism_count': 1,\n",
              "   'cell_line_count': 1580,\n",
              "   'cell_type_count': 1,\n",
              "   'drug_count': 1,\n",
              "   'data_type_count': 5,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 116131.0,\n",
              "   'normal_sample_count': 543},\n",
              "  {'repo_id': '1668162477722',\n",
              "   'repo_name': 'exelixis_clinical_trial_atlas',\n",
              "   'indexes': {'files': 'exelixis_clinical_trial_atlas_files',\n",
              "    'gct_col_metadata': 'exelixis_clinical_trial_atlas_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'exelixis_clinical_trial_atlas_files',\n",
              "    'gct_col_metadata': 'exelixis_clinical_trial_atlas_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'immport': 1547}],\n",
              "   'datatypes': [{'lab measurement': 1424}, {'proteomics': 123}],\n",
              "   'dataset_count': 1547,\n",
              "   'disease_count': 2,\n",
              "   'tissue_count': 1,\n",
              "   'organism_count': 1,\n",
              "   'cell_line_count': 1,\n",
              "   'cell_type_count': 1,\n",
              "   'drug_count': 2,\n",
              "   'data_type_count': 2,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 1424.0,\n",
              "   'normal_sample_count': 123},\n",
              "  {'repo_id': '1687170899562',\n",
              "   'repo_name': 'bulk_rnaseq_partner_oa',\n",
              "   'indexes': {'files': 'bulk_rnaseq_partner_oa_files',\n",
              "    'gct_col_metadata': 'bulk_rnaseq_partner_oa_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'bulk_rnaseq_partner_oa_files',\n",
              "    'gct_col_metadata': 'bulk_rnaseq_partner_oa_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 50}],\n",
              "   'datatypes': [{'raw counts transcriptomics': 50}],\n",
              "   'dataset_count': 50,\n",
              "   'disease_count': 32,\n",
              "   'tissue_count': 26,\n",
              "   'organism_count': 2,\n",
              "   'cell_line_count': 24,\n",
              "   'cell_type_count': 18,\n",
              "   'drug_count': 12,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 2652.0,\n",
              "   'normal_sample_count': 455},\n",
              "  {'repo_id': '1675928719276',\n",
              "   'repo_name': 'bulk_rnaseq_test',\n",
              "   'indexes': {'files': 'bulk_rnaseq_test_files',\n",
              "    'gct_col_metadata': 'bulk_rnaseq_test_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'bulk_rnaseq_test_files',\n",
              "    'gct_col_metadata': 'bulk_rnaseq_test_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 1708}, {'': 1}],\n",
              "   'datatypes': [{'raw counts transcriptomics': 1703}, {'': 6}],\n",
              "   'dataset_count': 1709,\n",
              "   'disease_count': 526,\n",
              "   'tissue_count': 291,\n",
              "   'organism_count': 12,\n",
              "   'cell_line_count': 237,\n",
              "   'cell_type_count': 267,\n",
              "   'drug_count': 216,\n",
              "   'data_type_count': 2,\n",
              "   'data_source_count': 2,\n",
              "   'sample_count': 25488.0,\n",
              "   'normal_sample_count': 21461},\n",
              "  {'repo_id': '1612338998334',\n",
              "   'repo_name': 'depmap',\n",
              "   'indexes': {'files': 'depmap_files',\n",
              "    'gct_col_metadata': 'depmap_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'depmap_files',\n",
              "    'gct_col_metadata': 'depmap_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'depmap': 36939}],\n",
              "   'datatypes': [{'gene dependency': 29215},\n",
              "    {'gene effect': 4556},\n",
              "    {'drug screens': 2666},\n",
              "    {'rnai': 501}],\n",
              "   'dataset_count': 36939,\n",
              "   'disease_count': 131,\n",
              "   'tissue_count': 38,\n",
              "   'organism_count': 1,\n",
              "   'cell_line_count': 1442,\n",
              "   'cell_type_count': 1,\n",
              "   'drug_count': 0,\n",
              "   'data_type_count': 4,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 36938.0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1661436595857',\n",
              "   'repo_name': 'clinical_trials_omixatlas',\n",
              "   'indexes': {'files': 'clinical_trials_omixatlas_files',\n",
              "    'gct_col_metadata': 'clinical_trials_omixatlas_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'clinical_trials_omixatlas_files',\n",
              "    'gct_col_metadata': 'clinical_trials_omixatlas_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'clinicaltrials.gov': 5}],\n",
              "   'datatypes': [{'clinical trial': 5}],\n",
              "   'dataset_count': 5,\n",
              "   'disease_count': 1,\n",
              "   'tissue_count': 1,\n",
              "   'organism_count': 0,\n",
              "   'cell_line_count': 1,\n",
              "   'cell_type_count': 0,\n",
              "   'drug_count': 8,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1688978709121',\n",
              "   'repo_name': 'gates_geo_transcriptomics',\n",
              "   'indexes': {'files': 'gates_geo_transcriptomics_files',\n",
              "    'gct_col_metadata': 'gates_geo_transcriptomics_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'gates_geo_transcriptomics_files',\n",
              "    'gct_col_metadata': 'gates_geo_transcriptomics_gct_col_metadata'},\n",
              "   'org_id': '1655118646',\n",
              "   'data_type': 'bulk_rna_seq',\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 40}, {'array express': 1}, {'publication': 1}],\n",
              "   'datatypes': [{'raw counts transcriptomics': 42}],\n",
              "   'dataset_count': 42,\n",
              "   'disease_count': 16,\n",
              "   'tissue_count': 10,\n",
              "   'organism_count': 2,\n",
              "   'cell_line_count': 1,\n",
              "   'cell_type_count': 12,\n",
              "   'drug_count': 26,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 3,\n",
              "   'sample_count': 4553.0,\n",
              "   'normal_sample_count': 1332},\n",
              "  {'repo_id': '1675113843133',\n",
              "   'repo_name': 'clinical_trial_atlas',\n",
              "   'indexes': {'files': 'clinical_trial_atlas_files',\n",
              "    'gct_col_metadata': 'clinical_trial_atlas_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'clinical_trial_atlas_files',\n",
              "    'gct_col_metadata': 'clinical_trial_atlas_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'immport': 1547}],\n",
              "   'datatypes': [],\n",
              "   'dataset_count': 1547,\n",
              "   'disease_count': 0,\n",
              "   'tissue_count': 0,\n",
              "   'organism_count': 0,\n",
              "   'cell_line_count': 0,\n",
              "   'cell_type_count': 0,\n",
              "   'drug_count': 0,\n",
              "   'data_type_count': 0,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 1424.0,\n",
              "   'normal_sample_count': 123},\n",
              "  {'repo_id': '1690991358329',\n",
              "   'repo_name': 'evolveimmune_oa',\n",
              "   'indexes': {'files': 'evolveimmune_oa_files',\n",
              "    'gct_col_metadata': 'evolveimmune_oa_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'evolveimmune_oa_files',\n",
              "    'gct_col_metadata': 'evolveimmune_oa_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 8}],\n",
              "   'datatypes': [{'microarray': 7}, {'raw counts transcriptomics': 1}],\n",
              "   'dataset_count': 8,\n",
              "   'disease_count': 5,\n",
              "   'tissue_count': 3,\n",
              "   'organism_count': 1,\n",
              "   'cell_line_count': 1,\n",
              "   'cell_type_count': 1,\n",
              "   'drug_count': 13,\n",
              "   'data_type_count': 2,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 1108.0,\n",
              "   'normal_sample_count': 80},\n",
              "  {'repo_id': '1688978884650',\n",
              "   'repo_name': 'gates_single_cell_rnaseq',\n",
              "   'indexes': {'files': 'gates_single_cell_rnaseq_files',\n",
              "    'h5ad_col_metadata': 'gates_single_cell_rnaseq_h5ad_col_metadata'},\n",
              "   'v2_indexes': {'files': 'gates_single_cell_rnaseq_files',\n",
              "    'h5ad_col_metadata': 'gates_single_cell_rnaseq_h5ad_col_metadata'},\n",
              "   'org_id': '1655118646',\n",
              "   'data_type': 'single_cell',\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 2}],\n",
              "   'datatypes': [{'single cell': 2}],\n",
              "   'dataset_count': 2,\n",
              "   'disease_count': 4,\n",
              "   'tissue_count': 2,\n",
              "   'organism_count': 2,\n",
              "   'cell_line_count': 1,\n",
              "   'cell_type_count': 12,\n",
              "   'drug_count': 2,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 8.0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1668142600760',\n",
              "   'repo_name': 'geo_transcriptomics_omixatlas',\n",
              "   'indexes': {'files': 'geo_transcriptomics_omixatlas_files',\n",
              "    'gct_col_metadata': 'geo_transcriptomics_omixatlas_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'geo_transcriptomics_omixatlas_files',\n",
              "    'gct_col_metadata': 'geo_transcriptomics_omixatlas_gct_col_metadata'},\n",
              "   'data_type': 'bulk_rna_seq',\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 38011}],\n",
              "   'datatypes': [{'raw counts transcriptomics': 38011}],\n",
              "   'dataset_count': 38011,\n",
              "   'disease_count': 2662,\n",
              "   'tissue_count': 880,\n",
              "   'organism_count': 119,\n",
              "   'cell_line_count': 2617,\n",
              "   'cell_type_count': 745,\n",
              "   'drug_count': 1823,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 605632.0,\n",
              "   'normal_sample_count': 418610},\n",
              "  {'repo_id': '1658898751957',\n",
              "   'repo_name': 'gsea_atlas',\n",
              "   'indexes': {'files': 'gsea_atlas_files',\n",
              "    'gct_col_metadata': 'gsea_atlas_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'gsea_atlas_files',\n",
              "    'gct_col_metadata': 'gsea_atlas_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 4995}],\n",
              "   'datatypes': [{'transcriptomics': 4978},\n",
              "    {'raw counts transcriptomics': 17}],\n",
              "   'dataset_count': 4995,\n",
              "   'disease_count': 1676,\n",
              "   'tissue_count': 426,\n",
              "   'organism_count': 0,\n",
              "   'cell_line_count': 637,\n",
              "   'cell_type_count': 379,\n",
              "   'drug_count': 673,\n",
              "   'data_type_count': 2,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 0.0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1612862450692',\n",
              "   'repo_name': 'transcriptomics__cyt',\n",
              "   'indexes': {'files': 'transcriptomics__cyt_files',\n",
              "    'gct_col_metadata': 'transcriptomics__cyt_gct_col_metadata',\n",
              "    'h5ad_col_metadata': 'transcriptomics__cyt_h5ad_col_metadata'},\n",
              "   'v2_indexes': {'files': 'transcriptomics__cyt_files',\n",
              "    'gct_col_metadata': 'transcriptomics__cyt_gct_col_metadata',\n",
              "    'h5ad_col_metadata': 'transcriptomics__cyt_h5ad_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'gene expression omnibus (geo)': 26},\n",
              "    {'arrayexpress': 2},\n",
              "    {'ebi': 1}],\n",
              "   'datatypes': [{'transcriptomics': 29}, {'single cell': 15}],\n",
              "   'dataset_count': 44,\n",
              "   'disease_count': 0,\n",
              "   'tissue_count': 22,\n",
              "   'organism_count': 2,\n",
              "   'cell_line_count': 0,\n",
              "   'cell_type_count': 3,\n",
              "   'drug_count': 0,\n",
              "   'data_type_count': 2,\n",
              "   'data_source_count': 3,\n",
              "   'sample_count': 0.0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1649172174566',\n",
              "   'repo_name': 'exelixis',\n",
              "   'indexes': {'files': 'exelixis_files',\n",
              "    'gct_col_metadata': 'exelixis_gct_col_metadata',\n",
              "    'h5ad_col_metadata': 'exelixis_h5ad_col_metadata'},\n",
              "   'v2_indexes': {'files': 'exelixis_files',\n",
              "    'gct_col_metadata': 'exelixis_gct_col_metadata',\n",
              "    'h5ad_col_metadata': 'exelixis_h5ad_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 161},\n",
              "    {'expression atlas': 3},\n",
              "    {'htan': 3},\n",
              "    {'publication': 3},\n",
              "    {'scp': 2},\n",
              "    {'tabula sapiens': 1}],\n",
              "   'datatypes': [{'single cell': 81},\n",
              "    {'transcriptomics': 62},\n",
              "    {'microarray': 18},\n",
              "    {'raw counts transcriptomics': 8},\n",
              "    {'genome variation profiling': 3},\n",
              "    {'': 1}],\n",
              "   'dataset_count': 173,\n",
              "   'disease_count': 104,\n",
              "   'tissue_count': 66,\n",
              "   'organism_count': 2,\n",
              "   'cell_line_count': 14,\n",
              "   'cell_type_count': 262,\n",
              "   'drug_count': 34,\n",
              "   'data_type_count': 6,\n",
              "   'data_source_count': 6,\n",
              "   'sample_count': 18202.0,\n",
              "   'normal_sample_count': 759},\n",
              "  {'repo_id': '14',\n",
              "   'repo_name': 'gtex',\n",
              "   'indexes': {'files': 'gtex_files',\n",
              "    'gct_col_metadata': 'gtex_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'gtex_files',\n",
              "    'gct_col_metadata': 'gtex_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'gtex': 30}],\n",
              "   'datatypes': [{'transcriptomics': 30}],\n",
              "   'dataset_count': 30,\n",
              "   'disease_count': 1,\n",
              "   'tissue_count': 30,\n",
              "   'organism_count': 1,\n",
              "   'cell_line_count': 1,\n",
              "   'cell_type_count': 1,\n",
              "   'drug_count': 1,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 17382.0,\n",
              "   'normal_sample_count': 17385},\n",
              "  {'repo_id': '1698228502424',\n",
              "   'repo_name': 'genomics_data',\n",
              "   'indexes': {'files': 'genomics_data_files',\n",
              "    'gct_col_metadata': 'genomics_data_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'genomics_data_files',\n",
              "    'gct_col_metadata': 'genomics_data_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'proprietary': 3}],\n",
              "   'datatypes': [{'mutation': 3}],\n",
              "   'dataset_count': 3,\n",
              "   'disease_count': 1,\n",
              "   'tissue_count': 1,\n",
              "   'organism_count': 0,\n",
              "   'cell_line_count': 0,\n",
              "   'cell_type_count': 0,\n",
              "   'drug_count': 1,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 3.0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1665567898397',\n",
              "   'repo_name': 'public_oa_playground',\n",
              "   'indexes': {'files': 'public_oa_playground_files',\n",
              "    'gct_col_metadata': 'public_oa_playground_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'public_oa_playground_files',\n",
              "    'gct_col_metadata': 'public_oa_playground_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 11}],\n",
              "   'datatypes': [{'raw counts transcriptomics': 8}, {'transcriptomics': 3}],\n",
              "   'dataset_count': 11,\n",
              "   'disease_count': 15,\n",
              "   'tissue_count': 6,\n",
              "   'organism_count': 2,\n",
              "   'cell_line_count': 1,\n",
              "   'cell_type_count': 10,\n",
              "   'drug_count': 7,\n",
              "   'data_type_count': 2,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 392.0,\n",
              "   'normal_sample_count': 257},\n",
              "  {'repo_id': '1699596271840',\n",
              "   'repo_name': 'lilly_pain_model_oa',\n",
              "   'indexes': {'files': 'lilly_pain_model_oa_files',\n",
              "    'gct_col_metadata': 'lilly_pain_model_oa_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'lilly_pain_model_oa_files',\n",
              "    'gct_col_metadata': 'lilly_pain_model_oa_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 52}],\n",
              "   'datatypes': [{'raw counts transcriptomics': 51}],\n",
              "   'dataset_count': 52,\n",
              "   'disease_count': 24,\n",
              "   'tissue_count': 15,\n",
              "   'organism_count': 2,\n",
              "   'cell_line_count': 2,\n",
              "   'cell_type_count': 12,\n",
              "   'drug_count': 16,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 1222.0,\n",
              "   'normal_sample_count': 530},\n",
              "  {'repo_id': '1700635906293',\n",
              "   'repo_name': 'hybrid_test_repo',\n",
              "   'v2_indexes': {'files': 'hybrid_test_repo_files',\n",
              "    'gct_col_metadata': 'hybrid_test_repo_gct_col_metadata',\n",
              "    'h5ad_col_metadata': 'hybrid_test_repo_h5ad_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 2}],\n",
              "   'datatypes': [{'single cell': 2}],\n",
              "   'dataset_count': 2,\n",
              "   'disease_count': 2,\n",
              "   'tissue_count': 3,\n",
              "   'organism_count': 2,\n",
              "   'cell_line_count': 2,\n",
              "   'cell_type_count': 10,\n",
              "   'drug_count': 1,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 13.0,\n",
              "   'normal_sample_count': 4},\n",
              "  {'repo_id': '1663679352131',\n",
              "   'repo_name': 'metabolomics_testing_oa',\n",
              "   'indexes': {'files': 'metabolomics_testing_oa_files',\n",
              "    'gct_col_metadata': 'metabolomics_testing_oa_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'metabolomics_testing_oa_files',\n",
              "    'gct_col_metadata': 'metabolomics_testing_oa_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'metabolomics workbench': 1}],\n",
              "   'datatypes': [{'metabolomics': 1}],\n",
              "   'dataset_count': 1,\n",
              "   'disease_count': 0,\n",
              "   'tissue_count': 0,\n",
              "   'organism_count': 0,\n",
              "   'cell_line_count': 0,\n",
              "   'cell_type_count': 0,\n",
              "   'drug_count': 0,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 16.0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1671797992418',\n",
              "   'repo_name': 'mock_cod_rm',\n",
              "   'indexes': {},\n",
              "   'v2_indexes': {},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [],\n",
              "   'datatypes': [],\n",
              "   'dataset_count': 0,\n",
              "   'disease_count': 0,\n",
              "   'tissue_count': 0,\n",
              "   'organism_count': 0,\n",
              "   'cell_line_count': 0,\n",
              "   'cell_type_count': 0,\n",
              "   'drug_count': 0,\n",
              "   'data_type_count': 0,\n",
              "   'data_source_count': 0,\n",
              "   'sample_count': 0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1684951053334',\n",
              "   'repo_name': 'sarepta_spotfire',\n",
              "   'indexes': {},\n",
              "   'v2_indexes': {},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [],\n",
              "   'datatypes': [],\n",
              "   'dataset_count': 0,\n",
              "   'disease_count': 0,\n",
              "   'tissue_count': 0,\n",
              "   'organism_count': 0,\n",
              "   'cell_line_count': 0,\n",
              "   'cell_type_count': 0,\n",
              "   'drug_count': 0,\n",
              "   'data_type_count': 0,\n",
              "   'data_source_count': 0,\n",
              "   'sample_count': 0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1647341066415',\n",
              "   'repo_name': 'geo_raw_counts',\n",
              "   'indexes': {'files': 'geo_raw_counts_files',\n",
              "    'gct_col_metadata': 'geo_raw_counts_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'geo_raw_counts_files',\n",
              "    'gct_col_metadata': 'geo_raw_counts_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 17755}],\n",
              "   'datatypes': [{'raw counts transcriptomics': 17754},\n",
              "    {'transcriptomics': 1}],\n",
              "   'dataset_count': 17755,\n",
              "   'disease_count': 1771,\n",
              "   'tissue_count': 668,\n",
              "   'organism_count': 14,\n",
              "   'cell_line_count': 1785,\n",
              "   'cell_type_count': 612,\n",
              "   'drug_count': 513,\n",
              "   'data_type_count': 2,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 394848.0,\n",
              "   'normal_sample_count': 271388},\n",
              "  {'repo_id': '1661162544511',\n",
              "   'repo_name': 'mock_therapeutics',\n",
              "   'indexes': {'files': 'mock_therapeutics_files',\n",
              "    'gct_col_metadata': 'mock_therapeutics_gct_col_metadata',\n",
              "    'h5ad_col_metadata': 'mock_therapeutics_h5ad_col_metadata'},\n",
              "   'v2_indexes': {'files': 'mock_therapeutics_files',\n",
              "    'gct_col_metadata': 'mock_therapeutics_gct_col_metadata',\n",
              "    'h5ad_col_metadata': 'mock_therapeutics_h5ad_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'immport': 447},\n",
              "    {'geo': 101},\n",
              "    {'depmap': 40},\n",
              "    {'tcga': 15},\n",
              "    {'pharmacodb': 9},\n",
              "    {'cbioportal': 5},\n",
              "    {'scp': 1}],\n",
              "   'datatypes': [{'lab measurement': 426},\n",
              "    {'transcriptomics': 95},\n",
              "    {'gene dependency': 40},\n",
              "    {'proteomics': 21},\n",
              "    {'single cell': 21},\n",
              "    {'drug response': 9},\n",
              "    {'mutation': 5},\n",
              "    {'raw counts transcriptomics': 1}],\n",
              "   'dataset_count': 618,\n",
              "   'disease_count': 125,\n",
              "   'tissue_count': 96,\n",
              "   'organism_count': 6,\n",
              "   'cell_line_count': 389,\n",
              "   'cell_type_count': 51,\n",
              "   'drug_count': 71,\n",
              "   'data_type_count': 8,\n",
              "   'data_source_count': 7,\n",
              "   'sample_count': 31695.0,\n",
              "   'normal_sample_count': 2115},\n",
              "  {'repo_id': '1663307451377',\n",
              "   'repo_name': 'premium_oa',\n",
              "   'indexes': {'files': 'premium_oa_files',\n",
              "    'gct_col_metadata': 'premium_oa_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'premium_oa_files',\n",
              "    'gct_col_metadata': 'premium_oa_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'premium ': 182}],\n",
              "   'datatypes': [{'dnaseq': 99}, {'rnaseq': 83}],\n",
              "   'dataset_count': 182,\n",
              "   'disease_count': 6,\n",
              "   'tissue_count': 3,\n",
              "   'organism_count': 0,\n",
              "   'cell_line_count': 0,\n",
              "   'cell_type_count': 0,\n",
              "   'drug_count': 30,\n",
              "   'data_type_count': 2,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 182.0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1669611204575',\n",
              "   'repo_name': 'mock_diy',\n",
              "   'indexes': {},\n",
              "   'v2_indexes': {},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [],\n",
              "   'datatypes': [],\n",
              "   'dataset_count': 0,\n",
              "   'disease_count': 0,\n",
              "   'tissue_count': 0,\n",
              "   'organism_count': 0,\n",
              "   'cell_line_count': 0,\n",
              "   'cell_type_count': 0,\n",
              "   'drug_count': 0,\n",
              "   'data_type_count': 0,\n",
              "   'data_source_count': 0,\n",
              "   'sample_count': 0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1619514006627',\n",
              "   'repo_name': 'hpa',\n",
              "   'indexes': {'files': 'hpa_files',\n",
              "    'gct_col_metadata': 'hpa_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'hpa_files',\n",
              "    'gct_col_metadata': 'hpa_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'hpa': 1755}],\n",
              "   'datatypes': [{'transcriptomics': 1565},\n",
              "    {'gene expression reliability': 190}],\n",
              "   'dataset_count': 1755,\n",
              "   'disease_count': 1,\n",
              "   'tissue_count': 76,\n",
              "   'organism_count': 3,\n",
              "   'cell_line_count': 70,\n",
              "   'cell_type_count': 149,\n",
              "   'drug_count': 1,\n",
              "   'data_type_count': 2,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 1755.0,\n",
              "   'normal_sample_count': 678},\n",
              "  {'repo_id': '1700774302601',\n",
              "   'repo_name': 'nsclc_demo_atlas',\n",
              "   'indexes': {'files': 'nsclc_demo_atlas_files',\n",
              "    'h5ad_col_metadata': 'nsclc_demo_atlas_h5ad_col_metadata'},\n",
              "   'v2_indexes': {'files': 'nsclc_demo_atlas_files',\n",
              "    'h5ad_col_metadata': 'nsclc_demo_atlas_h5ad_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 3}, {'publication': 2}, {'expression atlas': 1}],\n",
              "   'datatypes': [{'single cell': 6}],\n",
              "   'dataset_count': 6,\n",
              "   'disease_count': 8,\n",
              "   'tissue_count': 9,\n",
              "   'organism_count': 1,\n",
              "   'cell_line_count': 1,\n",
              "   'cell_type_count': 0,\n",
              "   'drug_count': 1,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 3,\n",
              "   'sample_count': 572.0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1681193278029',\n",
              "   'repo_name': 'test_schema_builder_oa',\n",
              "   'indexes': {},\n",
              "   'v2_indexes': {},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [],\n",
              "   'datatypes': [],\n",
              "   'dataset_count': 0,\n",
              "   'disease_count': 0,\n",
              "   'tissue_count': 0,\n",
              "   'organism_count': 0,\n",
              "   'cell_line_count': 0,\n",
              "   'cell_type_count': 0,\n",
              "   'drug_count': 0,\n",
              "   'data_type_count': 0,\n",
              "   'data_source_count': 0,\n",
              "   'sample_count': 0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1661493085346',\n",
              "   'repo_name': 'solutionstrainingoa_hs',\n",
              "   'indexes': {'files': 'solutionstrainingoa_hs_files',\n",
              "    'gct_col_metadata': 'solutionstrainingoa_hs_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'solutionstrainingoa_hs_files',\n",
              "    'gct_col_metadata': 'solutionstrainingoa_hs_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 20}],\n",
              "   'datatypes': [],\n",
              "   'dataset_count': 20,\n",
              "   'disease_count': 0,\n",
              "   'tissue_count': 0,\n",
              "   'organism_count': 0,\n",
              "   'cell_line_count': 0,\n",
              "   'cell_type_count': 0,\n",
              "   'drug_count': 0,\n",
              "   'data_type_count': 0,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 324.0,\n",
              "   'normal_sample_count': 200},\n",
              "  {'repo_id': '1639402777465',\n",
              "   'repo_name': 'rcsb_structures',\n",
              "   'indexes': {'files': 'rcsb_structures_files'},\n",
              "   'v2_indexes': {'files': 'rcsb_structures_files'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'rcsb': 183646}],\n",
              "   'datatypes': [{'structural biology': 183646}],\n",
              "   'dataset_count': 183646,\n",
              "   'disease_count': 1,\n",
              "   'tissue_count': 209,\n",
              "   'organism_count': 7571,\n",
              "   'cell_line_count': 1,\n",
              "   'cell_type_count': 0,\n",
              "   'drug_count': 1,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 183646.0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1670216423600',\n",
              "   'repo_name': 'sc_staging',\n",
              "   'indexes': {'files': 'sc_staging_files',\n",
              "    'h5ad_col_metadata': 'sc_staging_h5ad_col_metadata'},\n",
              "   'v2_indexes': {'files': 'sc_staging_files',\n",
              "    'h5ad_col_metadata': 'sc_staging_h5ad_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'scp': 61},\n",
              "    {'geo': 41},\n",
              "    {'expression atlas': 35},\n",
              "    {'publication': 23},\n",
              "    {'covid-19 cell atlas': 16},\n",
              "    {'cziscience': 3},\n",
              "    {'array_express': 2},\n",
              "    {'human cell atlas': 1}],\n",
              "   'datatypes': [{'single cell': 182}],\n",
              "   'dataset_count': 182,\n",
              "   'disease_count': 85,\n",
              "   'tissue_count': 72,\n",
              "   'organism_count': 6,\n",
              "   'cell_line_count': 6,\n",
              "   'cell_type_count': 244,\n",
              "   'drug_count': 6,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 8,\n",
              "   'sample_count': 3619963.0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1652954957531',\n",
              "   'repo_name': 'tcga_pancan_atlas',\n",
              "   'indexes': {'files': 'tcga_pancan_atlas_files',\n",
              "    'gct_col_metadata': 'tcga_pancan_atlas_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'tcga_pancan_atlas_files',\n",
              "    'gct_col_metadata': 'tcga_pancan_atlas_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'tcga pancan xena': 11060}],\n",
              "   'datatypes': [{'transcriptomics': 11060}],\n",
              "   'dataset_count': 11060,\n",
              "   'disease_count': 35,\n",
              "   'tissue_count': 56,\n",
              "   'organism_count': 1,\n",
              "   'cell_line_count': 1,\n",
              "   'cell_type_count': 1,\n",
              "   'drug_count': 630,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 11069.0,\n",
              "   'normal_sample_count': 734},\n",
              "  {'repo_id': '1615965444377',\n",
              "   'repo_name': 'liveromix_atlas',\n",
              "   'indexes': {'files': 'liveromix_atlas_files',\n",
              "    'gct_col_metadata': 'liveromix_atlas_gct_col_metadata',\n",
              "    'h5ad_col_metadata': 'liveromix_atlas_h5ad_col_metadata'},\n",
              "   'v2_indexes': {'files': 'liveromix_atlas_files',\n",
              "    'gct_col_metadata': 'liveromix_atlas_gct_col_metadata',\n",
              "    'h5ad_col_metadata': 'liveromix_atlas_h5ad_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 3992},\n",
              "    {'lincs': 2275},\n",
              "    {'tcga': 361},\n",
              "    {'metabolomics workbench': 55},\n",
              "    {'metabolights': 36},\n",
              "    {'ccle': 31},\n",
              "    {'cptac': 3},\n",
              "    {'depmap': 3},\n",
              "    {'hpa': 2},\n",
              "    {'gtex': 1}],\n",
              "   'datatypes': [{'transcriptomics': 6188},\n",
              "    {'mutation': 384},\n",
              "    {'metabolomics': 89},\n",
              "    {'single cell': 84},\n",
              "    {'proteomics': 4},\n",
              "    {'lipidomics': 3},\n",
              "    {'mirna': 2},\n",
              "    {'drug screens': 1},\n",
              "    {'gene dependency': 1},\n",
              "    {'gene effect': 1},\n",
              "    {'methylation': 1},\n",
              "    {'phosphoproteomics': 1}],\n",
              "   'dataset_count': 6759,\n",
              "   'disease_count': 752,\n",
              "   'tissue_count': 60,\n",
              "   'organism_count': 22,\n",
              "   'cell_line_count': 347,\n",
              "   'cell_type_count': 310,\n",
              "   'drug_count': 4704,\n",
              "   'data_type_count': 12,\n",
              "   'data_source_count': 10,\n",
              "   'sample_count': 895689.0,\n",
              "   'normal_sample_count': 110234},\n",
              "  {'repo_id': '1651657628382',\n",
              "   'repo_name': 'metabolomix_gwas',\n",
              "   'indexes': {},\n",
              "   'v2_indexes': {},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [],\n",
              "   'datatypes': [],\n",
              "   'dataset_count': 0,\n",
              "   'disease_count': 0,\n",
              "   'tissue_count': 0,\n",
              "   'organism_count': 0,\n",
              "   'cell_line_count': 0,\n",
              "   'cell_type_count': 0,\n",
              "   'drug_count': 0,\n",
              "   'data_type_count': 0,\n",
              "   'data_source_count': 0,\n",
              "   'sample_count': 0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1673603304965',\n",
              "   'repo_name': 'single_cell_rnaseq_omixatlas',\n",
              "   'indexes': {'files': 'single_cell_rnaseq_omixatlas_files',\n",
              "    'h5ad_col_metadata': 'single_cell_rnaseq_omixatlas_h5ad_col_metadata'},\n",
              "   'v2_indexes': {'files': 'single_cell_rnaseq_omixatlas_files',\n",
              "    'h5ad_col_metadata': 'single_cell_rnaseq_omixatlas_h5ad_col_metadata'},\n",
              "   'data_type': 'single_cell',\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 704},\n",
              "    {'publication': 458},\n",
              "    {'scp': 173},\n",
              "    {'expression atlas': 163},\n",
              "    {'covid-19 cell atlas': 32},\n",
              "    {'tabula sapiens': 24},\n",
              "    {'array express': 7},\n",
              "    {'htan': 3},\n",
              "    {'arrayexpress': 1}],\n",
              "   'datatypes': [{'single cell': 1565}],\n",
              "   'dataset_count': 1565,\n",
              "   'disease_count': 504,\n",
              "   'tissue_count': 343,\n",
              "   'organism_count': 6,\n",
              "   'cell_line_count': 232,\n",
              "   'cell_type_count': 841,\n",
              "   'drug_count': 165,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 9,\n",
              "   'sample_count': 384141.0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1656403986124',\n",
              "   'repo_name': 'processed_atavistik',\n",
              "   'indexes': {},\n",
              "   'v2_indexes': {},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [],\n",
              "   'datatypes': [],\n",
              "   'dataset_count': 0,\n",
              "   'disease_count': 0,\n",
              "   'tissue_count': 0,\n",
              "   'organism_count': 0,\n",
              "   'cell_line_count': 0,\n",
              "   'cell_type_count': 0,\n",
              "   'drug_count': 0,\n",
              "   'data_type_count': 0,\n",
              "   'data_source_count': 0,\n",
              "   'sample_count': 0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1695301356844',\n",
              "   'repo_name': 'solutions_staging_oa_bulk',\n",
              "   'indexes': {'files': 'solutions_staging_oa_bulk_files',\n",
              "    'gct_col_metadata': 'solutions_staging_oa_bulk_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'solutions_staging_oa_bulk_files',\n",
              "    'gct_col_metadata': 'solutions_staging_oa_bulk_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'array express': 1},\n",
              "    {'european nucleotide archive': 1},\n",
              "    {'geo': 1}],\n",
              "   'datatypes': [{'raw counts transcriptomics': 3}],\n",
              "   'dataset_count': 3,\n",
              "   'disease_count': 4,\n",
              "   'tissue_count': 3,\n",
              "   'organism_count': 1,\n",
              "   'cell_line_count': 1,\n",
              "   'cell_type_count': 2,\n",
              "   'drug_count': 1,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 3,\n",
              "   'sample_count': 387.0,\n",
              "   'normal_sample_count': 356},\n",
              "  {'repo_id': '1688460933980',\n",
              "   'repo_name': 'single_cell_flex_project',\n",
              "   'indexes': {'files': 'single_cell_flex_project_files',\n",
              "    'h5ad_col_metadata': 'single_cell_flex_project_h5ad_col_metadata'},\n",
              "   'v2_indexes': {'files': 'single_cell_flex_project_files',\n",
              "    'h5ad_col_metadata': 'single_cell_flex_project_h5ad_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 11}, {'publication': 2}],\n",
              "   'datatypes': [{'': 13}],\n",
              "   'dataset_count': 13,\n",
              "   'disease_count': 17,\n",
              "   'tissue_count': 6,\n",
              "   'organism_count': 2,\n",
              "   'cell_line_count': 2,\n",
              "   'cell_type_count': 36,\n",
              "   'drug_count': 4,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 2,\n",
              "   'sample_count': 299.0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1663044321285',\n",
              "   'repo_name': 'pride_oa',\n",
              "   'indexes': {'files': 'pride_oa_files',\n",
              "    'gct_col_metadata': 'pride_oa_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'pride_oa_files',\n",
              "    'gct_col_metadata': 'pride_oa_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'pride': 164}],\n",
              "   'datatypes': [{'proteomics': 164}],\n",
              "   'dataset_count': 164,\n",
              "   'disease_count': 102,\n",
              "   'tissue_count': 67,\n",
              "   'organism_count': 17,\n",
              "   'cell_line_count': 14,\n",
              "   'cell_type_count': 24,\n",
              "   'drug_count': 42,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 0.0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '32',\n",
              "   'repo_name': 'lincs',\n",
              "   'indexes': {'files': 'lincs_files',\n",
              "    'gct_col_metadata': 'lincs_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'lincs_files',\n",
              "    'gct_col_metadata': 'lincs_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'lincs': 159280}],\n",
              "   'datatypes': [{'transcriptomics': 159280}],\n",
              "   'dataset_count': 159280,\n",
              "   'disease_count': 34,\n",
              "   'tissue_count': 21,\n",
              "   'organism_count': 0,\n",
              "   'cell_line_count': 84,\n",
              "   'cell_type_count': 2,\n",
              "   'drug_count': 20347,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 41345458.0,\n",
              "   'normal_sample_count': 187539},\n",
              "  {'repo_id': '23',\n",
              "   'repo_name': 'metabolomics',\n",
              "   'indexes': {'files': 'metabolomics_files',\n",
              "    'gct_col_metadata': 'metabolomics_gct_col_metadata',\n",
              "    'h5ad_col_metadata': 'metabolomics_h5ad_col_metadata'},\n",
              "   'v2_indexes': {'files': 'metabolomics_files',\n",
              "    'gct_col_metadata': 'metabolomics_gct_col_metadata',\n",
              "    'h5ad_col_metadata': 'metabolomics_h5ad_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'metabolomics workbench': 1243},\n",
              "    {'metabolights': 536},\n",
              "    {'publication': 2},\n",
              "    {'geo': 1}],\n",
              "   'datatypes': [{'metabolomics': 1703},\n",
              "    {'lipidomics': 78},\n",
              "    {'single cell': 1}],\n",
              "   'dataset_count': 1782,\n",
              "   'disease_count': 214,\n",
              "   'tissue_count': 134,\n",
              "   'organism_count': 184,\n",
              "   'cell_line_count': 24,\n",
              "   'cell_type_count': 23,\n",
              "   'drug_count': 67,\n",
              "   'data_type_count': 3,\n",
              "   'data_source_count': 4,\n",
              "   'sample_count': 169117.0,\n",
              "   'normal_sample_count': 67479},\n",
              "  {'repo_id': '1622113130397',\n",
              "   'repo_name': 'pcd',\n",
              "   'indexes': {'files': 'pcd_files',\n",
              "    'gct_col_metadata': 'pcd_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'pcd_files',\n",
              "    'gct_col_metadata': 'pcd_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'pharmacodb': 19648}],\n",
              "   'datatypes': [{'drug response': 19648}],\n",
              "   'dataset_count': 19648,\n",
              "   'disease_count': 133,\n",
              "   'tissue_count': 34,\n",
              "   'organism_count': 1,\n",
              "   'cell_line_count': 1400,\n",
              "   'cell_type_count': 1,\n",
              "   'drug_count': 749,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 715084.0,\n",
              "   'normal_sample_count': 13574},\n",
              "  {'repo_id': '1621422280385',\n",
              "   'repo_name': 'immport',\n",
              "   'indexes': {'files': 'immport_files',\n",
              "    'gct_col_metadata': 'immport_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'immport_files',\n",
              "    'gct_col_metadata': 'immport_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'immport': 200118}],\n",
              "   'datatypes': [{'lab measurement': 142296},\n",
              "    {'proteomics': 36990},\n",
              "    {'titer': 10613},\n",
              "    {'pcr': 7786},\n",
              "    {'cytometry': 2433}],\n",
              "   'dataset_count': 200118,\n",
              "   'disease_count': 81,\n",
              "   'tissue_count': 10,\n",
              "   'organism_count': 2,\n",
              "   'cell_line_count': 1,\n",
              "   'cell_type_count': 2,\n",
              "   'drug_count': 47,\n",
              "   'data_type_count': 5,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 168429.0,\n",
              "   'normal_sample_count': 25887},\n",
              "  {'repo_id': '1695356716209',\n",
              "   'repo_name': 'solutions_staging_oa_sc',\n",
              "   'indexes': {'files': 'solutions_staging_oa_sc_files',\n",
              "    'h5ad_col_metadata': 'solutions_staging_oa_sc_h5ad_col_metadata'},\n",
              "   'v2_indexes': {'files': 'solutions_staging_oa_sc_files',\n",
              "    'h5ad_col_metadata': 'solutions_staging_oa_sc_h5ad_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 11},\n",
              "    {'expression atlas': 2},\n",
              "    {'publication': 2},\n",
              "    {'array express': 1}],\n",
              "   'datatypes': [{'single cell': 15}, {'': 1}],\n",
              "   'dataset_count': 16,\n",
              "   'disease_count': 18,\n",
              "   'tissue_count': 17,\n",
              "   'organism_count': 2,\n",
              "   'cell_line_count': 1,\n",
              "   'cell_type_count': 41,\n",
              "   'drug_count': 7,\n",
              "   'data_type_count': 2,\n",
              "   'data_source_count': 4,\n",
              "   'sample_count': 225.0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1687170818396',\n",
              "   'repo_name': 'single_cell_rnaseq_partner_oa',\n",
              "   'indexes': {'files': 'single_cell_rnaseq_partner_oa_files',\n",
              "    'h5ad_col_metadata': 'single_cell_rnaseq_partner_oa_h5ad_col_metadata'},\n",
              "   'v2_indexes': {'files': 'single_cell_rnaseq_partner_oa_files',\n",
              "    'h5ad_col_metadata': 'single_cell_rnaseq_partner_oa_h5ad_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 10}],\n",
              "   'datatypes': [{'single cell': 10}],\n",
              "   'dataset_count': 10,\n",
              "   'disease_count': 18,\n",
              "   'tissue_count': 11,\n",
              "   'organism_count': 2,\n",
              "   'cell_line_count': 2,\n",
              "   'cell_type_count': 44,\n",
              "   'drug_count': 2,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 92.0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1695706082415',\n",
              "   'repo_name': 'polly_discover_atlas',\n",
              "   'v2_indexes': {'files': 'polly_discover_atlas_files',\n",
              "    'gct_col_metadata': 'polly_discover_atlas_gct_col_metadata',\n",
              "    'comparisons': 'polly_discover_atlas_comparisons',\n",
              "    'comparison_data': 'polly_discover_atlas_comparison_data',\n",
              "    'pathway_data': 'polly_discover_atlas_pathway_data'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 123}],\n",
              "   'datatypes': [{'raw counts transcriptomics': 123}],\n",
              "   'dataset_count': 123,\n",
              "   'disease_count': 172,\n",
              "   'tissue_count': 84,\n",
              "   'organism_count': 1,\n",
              "   'cell_line_count': 29,\n",
              "   'cell_type_count': 62,\n",
              "   'drug_count': 23,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 7721.0,\n",
              "   'normal_sample_count': 3123},\n",
              "  {'repo_id': '15',\n",
              "   'repo_name': 'tcga',\n",
              "   'indexes': {'files': 'tcga_files',\n",
              "    'gct_col_metadata': 'tcga_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'tcga_files',\n",
              "    'gct_col_metadata': 'tcga_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'tcga': 55062}],\n",
              "   'datatypes': [{'transcriptomics': 11093},\n",
              "    {'mirna': 10468},\n",
              "    {'copy number variation': 10411},\n",
              "    {'mutation': 8810},\n",
              "    {'methylation': 8027},\n",
              "    {'proteomics': 6253}],\n",
              "   'dataset_count': 55062,\n",
              "   'disease_count': 34,\n",
              "   'tissue_count': 58,\n",
              "   'organism_count': 1,\n",
              "   'cell_line_count': 0,\n",
              "   'cell_type_count': 0,\n",
              "   'drug_count': 812,\n",
              "   'data_type_count': 6,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 55062.0,\n",
              "   'normal_sample_count': 1620},\n",
              "  {'repo_id': '17',\n",
              "   'repo_name': 'sc_data_lake',\n",
              "   'indexes': {'files': 'sc_data_lake_files',\n",
              "    'gct_col_metadata': 'sc_data_lake_gct_col_metadata',\n",
              "    'h5ad_col_metadata': 'sc_data_lake_h5ad_col_metadata',\n",
              "    'h5ad_row_metadata': 'sc_data_lake_h5ad_row_metadata'},\n",
              "   'v2_indexes': {'files': 'sc_data_lake_files',\n",
              "    'gct_col_metadata': 'sc_data_lake_gct_col_metadata',\n",
              "    'h5ad_col_metadata': 'sc_data_lake_h5ad_col_metadata',\n",
              "    'h5ad_row_metadata': 'sc_data_lake_h5ad_row_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 4159},\n",
              "    {'publication': 770},\n",
              "    {'scp': 351},\n",
              "    {'expression atlas': 239},\n",
              "    {'covid-19 cell atlas': 79},\n",
              "    {'tabula sapiens': 25},\n",
              "    {'htan': 21},\n",
              "    {'array express': 12},\n",
              "    {'zenodo': 3}],\n",
              "   'datatypes': [{'single cell': 5659}],\n",
              "   'dataset_count': 5659,\n",
              "   'disease_count': 979,\n",
              "   'tissue_count': 608,\n",
              "   'organism_count': 19,\n",
              "   'cell_line_count': 419,\n",
              "   'cell_type_count': 1141,\n",
              "   'drug_count': 427,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 9,\n",
              "   'sample_count': 1750637.0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1683746537232',\n",
              "   'repo_name': 'microarray_oa',\n",
              "   'indexes': {'files': 'microarray_oa_files',\n",
              "    'gct_col_metadata': 'microarray_oa_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'microarray_oa_files',\n",
              "    'gct_col_metadata': 'microarray_oa_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 1633}],\n",
              "   'datatypes': [{'microarray': 1603}, {'transcriptomics': 30}],\n",
              "   'dataset_count': 1633,\n",
              "   'disease_count': 464,\n",
              "   'tissue_count': 212,\n",
              "   'organism_count': 8,\n",
              "   'cell_line_count': 112,\n",
              "   'cell_type_count': 174,\n",
              "   'drug_count': 236,\n",
              "   'data_type_count': 2,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 74671.0,\n",
              "   'normal_sample_count': 60048},\n",
              "  {'repo_id': '1681962148429',\n",
              "   'repo_name': 'polly_gdx_atlas',\n",
              "   'indexes': {'files': 'polly_gdx_atlas_files',\n",
              "    'gct_col_metadata': 'polly_gdx_atlas_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'polly_gdx_atlas_files',\n",
              "    'gct_col_metadata': 'polly_gdx_atlas_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'': 18262}, {'geo': 9141}],\n",
              "   'datatypes': [{'normalized': 9149},\n",
              "    {'raw_counts': 9141},\n",
              "    {'pca_scores': 9113}],\n",
              "   'dataset_count': 27403,\n",
              "   'disease_count': 1475,\n",
              "   'tissue_count': 512,\n",
              "   'organism_count': 34,\n",
              "   'cell_line_count': 714,\n",
              "   'cell_type_count': 492,\n",
              "   'drug_count': 612,\n",
              "   'data_type_count': 3,\n",
              "   'data_source_count': 2,\n",
              "   'sample_count': 56394.0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1693319144058',\n",
              "   'repo_name': 'sc_raw_counts',\n",
              "   'indexes': {'files': 'sc_raw_counts_files',\n",
              "    'h5ad_col_metadata': 'sc_raw_counts_h5ad_col_metadata'},\n",
              "   'v2_indexes': {'files': 'sc_raw_counts_files',\n",
              "    'h5ad_col_metadata': 'sc_raw_counts_h5ad_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'publication': 41},\n",
              "    {'geo': 37},\n",
              "    {'expression atlas': 11},\n",
              "    {'scp': 4}],\n",
              "   'datatypes': [{'single cell': 89}, {'': 4}],\n",
              "   'dataset_count': 93,\n",
              "   'disease_count': 68,\n",
              "   'tissue_count': 58,\n",
              "   'organism_count': 2,\n",
              "   'cell_line_count': 4,\n",
              "   'cell_type_count': 0,\n",
              "   'drug_count': 6,\n",
              "   'data_type_count': 2,\n",
              "   'data_source_count': 4,\n",
              "   'sample_count': 13710.0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1681969964105',\n",
              "   'repo_name': 'owkin_bulk_rnaseq_oa',\n",
              "   'indexes': {'files': 'owkin_bulk_rnaseq_oa_files',\n",
              "    'gct_col_metadata': 'owkin_bulk_rnaseq_oa_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'owkin_bulk_rnaseq_oa_files',\n",
              "    'gct_col_metadata': 'owkin_bulk_rnaseq_oa_gct_col_metadata'},\n",
              "   'org_id': '1663587652',\n",
              "   'data_type': 'bulk_rna_seq',\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 136}, {'array express': 1}],\n",
              "   'datatypes': [{'raw counts transcriptomics': 119}, {'transcriptomics': 12}],\n",
              "   'dataset_count': 137,\n",
              "   'disease_count': 88,\n",
              "   'tissue_count': 50,\n",
              "   'organism_count': 2,\n",
              "   'cell_line_count': 160,\n",
              "   'cell_type_count': 18,\n",
              "   'drug_count': 38,\n",
              "   'data_type_count': 2,\n",
              "   'data_source_count': 2,\n",
              "   'sample_count': 16942.0,\n",
              "   'normal_sample_count': 2281},\n",
              "  {'repo_id': '1699512702071',\n",
              "   'repo_name': 'transcriptomics_demo_oa',\n",
              "   'indexes': {'files': 'transcriptomics_demo_oa_files',\n",
              "    'gct_col_metadata': 'transcriptomics_demo_oa_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'transcriptomics_demo_oa_files',\n",
              "    'gct_col_metadata': 'transcriptomics_demo_oa_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 107}],\n",
              "   'datatypes': [{'raw counts transcriptomics': 91}, {'microarray': 16}],\n",
              "   'dataset_count': 107,\n",
              "   'disease_count': 54,\n",
              "   'tissue_count': 46,\n",
              "   'organism_count': 2,\n",
              "   'cell_line_count': 43,\n",
              "   'cell_type_count': 35,\n",
              "   'drug_count': 48,\n",
              "   'data_type_count': 2,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 7234.0,\n",
              "   'normal_sample_count': 985},\n",
              "  {'repo_id': '1699373778648',\n",
              "   'repo_name': 'test_oa_gates',\n",
              "   'indexes': {'files': 'test_oa_gates_files',\n",
              "    'gct_col_metadata': 'test_oa_gates_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'test_oa_gates_files',\n",
              "    'gct_col_metadata': 'test_oa_gates_gct_col_metadata'},\n",
              "   'org_id': '1',\n",
              "   'data_type': 'bulk_rna_seq',\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 8}, {'array express': 1}, {'publication': 1}],\n",
              "   'datatypes': [{'raw counts transcriptomics': 10}],\n",
              "   'dataset_count': 10,\n",
              "   'disease_count': 9,\n",
              "   'tissue_count': 6,\n",
              "   'organism_count': 2,\n",
              "   'cell_line_count': 1,\n",
              "   'cell_type_count': 2,\n",
              "   'drug_count': 6,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 3,\n",
              "   'sample_count': 2366.0,\n",
              "   'normal_sample_count': 1006},\n",
              "  {'repo_id': '1681538984044',\n",
              "   'repo_name': 'test_april_15',\n",
              "   'indexes': {'files': 'test_april_15_files',\n",
              "    'gct_col_metadata': 'test_april_15_gct_col_metadata',\n",
              "    'biom_col_metadata': 'test_april_15_biom_col_metadata',\n",
              "    'biom_row_metadata': 'test_april_15_biom_row_metadata'},\n",
              "   'v2_indexes': {'files': 'test_april_15_files',\n",
              "    'gct_col_metadata': 'test_april_15_gct_col_metadata',\n",
              "    'biom_col_metadata': 'test_april_15_biom_col_metadata',\n",
              "    'biom_row_metadata': 'test_april_15_biom_row_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 6}, {'': 3}],\n",
              "   'datatypes': [{'transcriptomics': 4},\n",
              "    {'gwas': 2},\n",
              "    {'raw counts transcriptomics': 2},\n",
              "    {'': 1}],\n",
              "   'dataset_count': 9,\n",
              "   'disease_count': 6,\n",
              "   'tissue_count': 2,\n",
              "   'organism_count': 2,\n",
              "   'cell_line_count': 4,\n",
              "   'cell_type_count': 9,\n",
              "   'drug_count': 6,\n",
              "   'data_type_count': 4,\n",
              "   'data_source_count': 2,\n",
              "   'sample_count': 58.0,\n",
              "   'normal_sample_count': 58},\n",
              "  {'repo_id': '1615464367934',\n",
              "   'repo_name': 'teddy',\n",
              "   'indexes': {'files': 'teddy_files',\n",
              "    'gct_col_metadata': 'teddy_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'teddy_files',\n",
              "    'gct_col_metadata': 'teddy_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'metabolomics workbench': 34662}],\n",
              "   'datatypes': [{'lipidomics': 23102}, {'metabolomics': 11560}],\n",
              "   'dataset_count': 34662,\n",
              "   'disease_count': 2,\n",
              "   'tissue_count': 1,\n",
              "   'organism_count': 1,\n",
              "   'cell_line_count': 1,\n",
              "   'cell_type_count': 1,\n",
              "   'drug_count': 1,\n",
              "   'data_type_count': 2,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 34662.0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1696578532437',\n",
              "   'repo_name': 'test_sc_oa_ss',\n",
              "   'indexes': {},\n",
              "   'v2_indexes': {},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [],\n",
              "   'datatypes': [],\n",
              "   'dataset_count': 0,\n",
              "   'disease_count': 0,\n",
              "   'tissue_count': 0,\n",
              "   'organism_count': 0,\n",
              "   'cell_line_count': 0,\n",
              "   'cell_type_count': 0,\n",
              "   'drug_count': 0,\n",
              "   'data_type_count': 0,\n",
              "   'data_source_count': 0,\n",
              "   'sample_count': 0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1653545583559',\n",
              "   'repo_name': 'tcga_raw_counts',\n",
              "   'indexes': {'files': 'tcga_raw_counts_files',\n",
              "    'gct_col_metadata': 'tcga_raw_counts_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'tcga_raw_counts_files',\n",
              "    'gct_col_metadata': 'tcga_raw_counts_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'tcga': 11123}],\n",
              "   'datatypes': [{'raw counts transcriptomics': 11123}],\n",
              "   'dataset_count': 11123,\n",
              "   'disease_count': 35,\n",
              "   'tissue_count': 57,\n",
              "   'organism_count': 1,\n",
              "   'cell_line_count': 1,\n",
              "   'cell_type_count': 1,\n",
              "   'drug_count': 645,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 11123.0,\n",
              "   'normal_sample_count': 730},\n",
              "  {'repo_id': '1661846032543',\n",
              "   'repo_name': 'testing_rna_seq_conn',\n",
              "   'indexes': {'files': 'testing_rna_seq_conn_files',\n",
              "    'gct_col_metadata': 'testing_rna_seq_conn_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'testing_rna_seq_conn_files',\n",
              "    'gct_col_metadata': 'testing_rna_seq_conn_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 19}, {'premium ': 2}, {'out of box ': 1}],\n",
              "   'datatypes': [{'raw counts transcriptomics': 21}, {'rnaseq': 3}],\n",
              "   'dataset_count': 24,\n",
              "   'disease_count': 20,\n",
              "   'tissue_count': 9,\n",
              "   'organism_count': 3,\n",
              "   'cell_line_count': 5,\n",
              "   'cell_type_count': 15,\n",
              "   'drug_count': 8,\n",
              "   'data_type_count': 2,\n",
              "   'data_source_count': 3,\n",
              "   'sample_count': 208.0,\n",
              "   'normal_sample_count': 128},\n",
              "  {'repo_id': '1670313151684',\n",
              "   'repo_name': 'transmed_oa',\n",
              "   'indexes': {'files': 'transmed_oa_files'},\n",
              "   'v2_indexes': {'files': 'transmed_oa_files'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [],\n",
              "   'datatypes': [],\n",
              "   'dataset_count': 29,\n",
              "   'disease_count': 17,\n",
              "   'tissue_count': 1,\n",
              "   'organism_count': 1,\n",
              "   'cell_line_count': 0,\n",
              "   'cell_type_count': 0,\n",
              "   'drug_count': 0,\n",
              "   'data_type_count': 0,\n",
              "   'data_source_count': 0,\n",
              "   'sample_count': 29.0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1684170038878',\n",
              "   'repo_name': 'user_bulkrna',\n",
              "   'indexes': {'files': 'user_bulkrna_files',\n",
              "    'gct_col_metadata': 'user_bulkrna_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'user_bulkrna_files',\n",
              "    'gct_col_metadata': 'user_bulkrna_gct_col_metadata'},\n",
              "   'org_id': '1684168083',\n",
              "   'data_type': 'bulk_rna_seq',\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 3}],\n",
              "   'datatypes': [{'raw counts transcriptomics': 3}],\n",
              "   'dataset_count': 3,\n",
              "   'disease_count': 6,\n",
              "   'tissue_count': 2,\n",
              "   'organism_count': 1,\n",
              "   'cell_line_count': 1,\n",
              "   'cell_type_count': 2,\n",
              "   'drug_count': 2,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 965.0,\n",
              "   'normal_sample_count': 485},\n",
              "  {'repo_id': '1689784020686',\n",
              "   'repo_name': 'unicle_war_room_oa',\n",
              "   'indexes': {'files': 'unicle_war_room_oa_files',\n",
              "    'h5ad_col_metadata': 'unicle_war_room_oa_h5ad_col_metadata'},\n",
              "   'v2_indexes': {'files': 'unicle_war_room_oa_files',\n",
              "    'h5ad_col_metadata': 'unicle_war_room_oa_h5ad_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 114},\n",
              "    {'scp': 18},\n",
              "    {'expression atlas': 12},\n",
              "    {'publication': 4},\n",
              "    {'array express': 2}],\n",
              "   'datatypes': [{'single cell': 149}, {'': 1}],\n",
              "   'dataset_count': 150,\n",
              "   'disease_count': 82,\n",
              "   'tissue_count': 69,\n",
              "   'organism_count': 3,\n",
              "   'cell_line_count': 7,\n",
              "   'cell_type_count': 280,\n",
              "   'drug_count': 19,\n",
              "   'data_type_count': 2,\n",
              "   'data_source_count': 5,\n",
              "   'sample_count': 10922.0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1675431983317',\n",
              "   'repo_name': 'trex_singlecell',\n",
              "   'indexes': {'files': 'trex_singlecell_files',\n",
              "    'h5ad_col_metadata': 'trex_singlecell_h5ad_col_metadata'},\n",
              "   'v2_indexes': {'files': 'trex_singlecell_files',\n",
              "    'h5ad_col_metadata': 'trex_singlecell_h5ad_col_metadata'},\n",
              "   'org_id': '1657793152',\n",
              "   'data_type': 'single_cell',\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 36},\n",
              "    {'scp': 8},\n",
              "    {'expression atlas': 5},\n",
              "    {'publication': 3},\n",
              "    {'array express': 1}],\n",
              "   'datatypes': [{'single cell': 53}],\n",
              "   'dataset_count': 53,\n",
              "   'disease_count': 24,\n",
              "   'tissue_count': 45,\n",
              "   'organism_count': 3,\n",
              "   'cell_line_count': 1,\n",
              "   'cell_type_count': 115,\n",
              "   'drug_count': 10,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 5,\n",
              "   'sample_count': 1155.0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1695478781204',\n",
              "   'repo_name': 'testing_refactoring_sr',\n",
              "   'v2_indexes': {'files': 'testing_refactoring_sr_files',\n",
              "    'gct_col_metadata': 'testing_refactoring_sr_gct_col_metadata'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 2}],\n",
              "   'datatypes': [{'raw counts transcriptomics': 2}],\n",
              "   'dataset_count': 2,\n",
              "   'disease_count': 3,\n",
              "   'tissue_count': 2,\n",
              "   'organism_count': 2,\n",
              "   'cell_line_count': 3,\n",
              "   'cell_type_count': 2,\n",
              "   'drug_count': 2,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 19.0,\n",
              "   'normal_sample_count': 6},\n",
              "  {'repo_id': '1662980652271',\n",
              "   'repo_name': 'td_oa_test',\n",
              "   'indexes': {'files': 'td_oa_test_files'},\n",
              "   'v2_indexes': {'files': 'td_oa_test_files'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'branchbio': 21}],\n",
              "   'datatypes': [{'single cell multiomics': 21}],\n",
              "   'dataset_count': 21,\n",
              "   'disease_count': 1,\n",
              "   'tissue_count': 1,\n",
              "   'organism_count': 1,\n",
              "   'cell_line_count': 1,\n",
              "   'cell_type_count': 1,\n",
              "   'drug_count': 0,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 21.0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1694754907858',\n",
              "   'repo_name': 'test_bulk_oa_ss',\n",
              "   'indexes': {'files': 'test_bulk_oa_ss_files'},\n",
              "   'v2_indexes': {'files': 'test_bulk_oa_ss_files'},\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 25}],\n",
              "   'datatypes': [{'': 25}],\n",
              "   'dataset_count': 25,\n",
              "   'disease_count': 15,\n",
              "   'tissue_count': 17,\n",
              "   'organism_count': 3,\n",
              "   'cell_line_count': 6,\n",
              "   'cell_type_count': 18,\n",
              "   'drug_count': 6,\n",
              "   'data_type_count': 1,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 611.0,\n",
              "   'normal_sample_count': 0},\n",
              "  {'repo_id': '1672833360780',\n",
              "   'repo_name': 'trex',\n",
              "   'indexes': {'files': 'trex_files',\n",
              "    'gct_col_metadata': 'trex_gct_col_metadata'},\n",
              "   'v2_indexes': {'files': 'trex_files',\n",
              "    'gct_col_metadata': 'trex_gct_col_metadata'},\n",
              "   'org_id': '1657793152',\n",
              "   'data_type': 'bulk_rna_seq',\n",
              "   'data_management_version': 'v1',\n",
              "   'sources': [{'geo': 349}],\n",
              "   'datatypes': [{'raw counts transcriptomics': 348}, {'microarray': 1}],\n",
              "   'dataset_count': 349,\n",
              "   'disease_count': 200,\n",
              "   'tissue_count': 79,\n",
              "   'organism_count': 3,\n",
              "   'cell_line_count': 18,\n",
              "   'cell_type_count': 123,\n",
              "   'drug_count': 72,\n",
              "   'data_type_count': 2,\n",
              "   'data_source_count': 1,\n",
              "   'sample_count': 10239.0,\n",
              "   'normal_sample_count': 6627}]}"
            ]
          },
          "execution_count": 5,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "omixatlas.get_all_omixatlas(query_api_version='v2', count_by_source=True, count_by_data_type=True)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 6,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/plain": [
              "{'data': {'repo_name': 'trex',\n",
              "  'repo_id': '1672833360780',\n",
              "  'indexes': {'files': 'trex_files',\n",
              "   'gct_col_metadata': 'trex_gct_col_metadata'},\n",
              "  'v2_indexes': {'files': 'trex_files',\n",
              "   'gct_col_metadata': 'trex_gct_col_metadata'},\n",
              "  'linked_workspace_id': None,\n",
              "  'sources': [{'geo': 349}],\n",
              "  'datatypes': [{'raw counts transcriptomics': 348}, {'microarray': 1}],\n",
              "  'dataset_count': 349,\n",
              "  'disease_count': 200,\n",
              "  'tissue_count': 79,\n",
              "  'organism_count': 3,\n",
              "  'cell_line_count': 18,\n",
              "  'cell_type_count': 123,\n",
              "  'drug_count': 72,\n",
              "  'data_type_count': 2,\n",
              "  'data_source_count': 1,\n",
              "  'sample_count': 10239.0,\n",
              "  'normal_sample_count': 6627}}"
            ]
          },
          "execution_count": 6,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "omixatlas.omixatlas_summary(\"1672833360780\", query_api_version='v2', count_by_source=True, count_by_data_type=True)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 7,
      "metadata": {},
      "outputs": [
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "DeprecationWarning: This method will soon be deprecated. Please use new function download_dataset for downloading data files directly.\n",
            "For usage: help(<omixatlas_object>.download_dataset) \n",
            "For more details: https://docs.elucidata.io/polly-python/OmixAtlas/Download%20Data.html#omixatlas.OmixAtlas.download_dataset\n"
          ]
        },
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Downloaded data successfully\n"
          ]
        },
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "--2023-12-26 09:04:34--  https://discover-prod-datalake-v1.s3.amazonaws.com/GEO_data_lake/data/Microarray/GSE100003/GCT/GSE100003_GPL15207_curated.gct?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=ASIAVRYB5UBIMIYH6VMY%2F20231226%2Fus-west-2%2Fs3%2Faws4_request&X-Amz-Date=20231226T090434Z&X-Amz-Expires=3600&X-Amz-SignedHeaders=host&X-Amz-Security-Token=IQoJb3JpZ2luX2VjEAEaCXVzLXdlc3QtMiJHMEUCIFQ2s%2BhSop0dS%2FxjfWWtH7XGCzNMlCb5C%2BHs9KOsl%2FAEAiEAjWUV9ndAtzlfBOtMs%2FS%2Bco72b1sUk4UErCvMJbYKs%2FQqlgMIiv%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARAEGgwzODE3MTkyNTcxNjgiDPHQRbvGqJM737EF5irqAqP%2BQXbTov4YfckRDdG%2FHQcrUx0uxG6QY%2FSI6IkZP2ogIrnUIEK%2FQYSu1GG%2BrleixubfUmzVLvaEBVJSoXQaIz%2FRhGTBVgXpbNTd89oU4CmoPWoXiANWcBLeLms0OfWBssqxDIy7RibalKD9I6MxVoSbnrhk8bSwlE2m5rHrtralPvAGjFXqdD13uqjqSJ3rOAbQaB%2BhsKZu6gcZSaXwsKguoK0DvO717P1PmfDCgCPcWJgugSDXfyoMxZHClC6fixenBi1%2F%2BL3vKDjjmgunK%2FXXPOqBzUmM5bDBlGrgFzlZEueqz%2FOAc4BsFxBwBArr4ZPvB5AvtUGNGi3J8LagfrziYP%2BoAJwsVlO5wMQP0ZN8eYy%2BdnYaqYSr%2BV2vOG4bDhpssG9ZyFCavvoog5mxKhRGCLNn6oiqoFdHtmExg9r1F8wD1yDOzbOUwcHXwzBOXbO7ShvNm%2FdQKADDQbXX4Y47pvYmYNaohuZJMNOqqqwGOp0BHEd%2F%2Bgb7CI0Uc%2BPPCvWPg2GoR6vXuIgLddNPLzj0DJOL5WWMLJVe9QQLYfTlGM0uzdX5E2vihKawuHhkn6dHs2IozS3r%2BDh%2FoRsvcgisH6To7cMIf5srVOsZQnkth8D05bytZJzRDBYdwGd6wIBe8WbMG8pT5YKic%2FHEkw3QagL%2FjjAYoatqUCG8GGhkFIe6IHFUIPvgXpaN%2BhWB6A%3D%3D&X-Amz-Signature=063b10bcf6e3b36eced287e2feff820962422c7fc29e428c398d7faa68f2fc97\n",
            "Resolving discover-prod-datalake-v1.s3.amazonaws.com (discover-prod-datalake-v1.s3.amazonaws.com)... 52.218.232.74, 52.218.228.19, 52.92.193.73, ...\n",
            "Connecting to discover-prod-datalake-v1.s3.amazonaws.com (discover-prod-datalake-v1.s3.amazonaws.com)|52.218.232.74|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 934488 (913K) [binary/octet-stream]\n",
            "Saving to: 'GSE100003_GPL15207.gct'\n",
            "\n",
            "     0K .......... .......... .......... .......... ..........  5% 66.9M 0s\n",
            "    50K .......... .......... .......... .......... .......... 10%  132M 0s\n",
            "   100K .......... .......... .......... .......... .......... 16% 95.4M 0s\n",
            "   150K .......... .......... .......... .......... .......... 21%  124M 0s\n",
            "   200K .......... .......... .......... .......... .......... 27%  178M 0s\n",
            "   250K .......... .......... .......... .......... .......... 32%  188M 0s\n",
            "   300K .......... .......... .......... .......... .......... 38%  190M 0s\n",
            "   350K .......... .......... .......... .......... .......... 43%  134M 0s\n",
            "   400K .......... .......... .......... .......... .......... 49%  118M 0s\n",
            "   450K .......... .......... .......... .......... .......... 54%  156M 0s\n",
            "   500K .......... .......... .......... .......... .......... 60%  134M 0s\n",
            "   550K .......... .......... .......... .......... .......... 65%  167M 0s\n",
            "   600K .......... .......... .......... .......... .......... 71%  166M 0s\n",
            "   650K .......... .......... .......... .......... .......... 76%  166M 0s\n",
            "   700K .......... .......... .......... .......... .......... 82%  132M 0s\n",
            "   750K .......... .......... .......... .......... .......... 87%  113M 0s\n",
            "   800K .......... .......... .......... .......... .......... 93%  160M 0s\n",
            "   850K .......... .......... .......... .......... .......... 98%  137M 0s\n",
            "   900K .......... ..                                         100%  318M=0.007s\n",
            "\n",
            "2023-12-26 09:04:34 (134 MB/s) - 'GSE100003_GPL15207.gct' saved [934488/934488]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "dataset_id = \"GSE100003_GPL15207\" #dataset which user wants to download.\n",
        "repo_key = \"geo\" #repo_id OR repo_name from which dataset should be downloaded from.\n",
        "file_name = f\"{dataset_id}.gct\"\n",
        "data = omixatlas.download_data(repo_key, dataset_id)\n",
        "url = data.get('data').get('attributes').get('download_url')\n",
        "status = os.system(f\"wget -O '{file_name}' '{url}'\")\n",
        "if status == 0:\n",
        "    print(\"Downloaded data successfully\")\n",
        "else:\n",
        "    raise Exception(\"Download not successful\")"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 2,
      "metadata": {},
      "outputs": [
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "INFO:root:Download successful to path=/import\n"
          ]
        }
      ],
      "source": [
        "from polly.workspaces import Workspaces\n",
        "local_path = os.getcwd()\n",
        "w = Workspaces()\n",
        "# local_path = \"/\"\n",
        "w.download_from_workspaces(14732, \"polly://Imagess/class_diagram.drawio.png\", local_path)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 3,
      "metadata": {},
      "outputs": [
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "INFO:root:Upload successful on workspace-id=14732.\n"
          ]
        }
      ],
      "source": [
        "w.upload_to_workspaces(\n",
        "    14732, \"/\", local_path+\"/class_diagram.drawio.png\"\n",
        ")"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": []
    }
  ],
  "metadata": {
    "kernelspec": {
      "display_name": "Python 3 (ipykernel)",
      "language": "python",
      "name": "python3"
    },
    "language_info": {
      "codemirror_mode": {
        "name": "ipython",
        "version": 3
      },
      "file_extension": ".py",
      "mimetype": "text/x-python",
      "name": "python",
      "nbconvert_exporter": "python",
      "pygments_lexer": "ipython3",
      "version": "3.10.2"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 2
}