{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "colab_type": "text",
    "id": "view-in-github"
   },
   "source": [
    "<a href=\"https://testpolly.elucidata.io/manage/workspaces?action=open_polly_notebook&source=github&path=ElucidataInc/PublicAssets/blob/master/internal-user/polly-python-query-examples.ipynb&kernel=elucidata/Python 3&machine=small\" target=\"_blank\"><img src=\"https://elucidatainc.github.io/PublicAssets/open_polly.svg\" alt=\"Open in Polly\"/></a>\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Welcome to polly-python Notebook consisting of sample queries for after data infrastructure upgrade\n",
    "\n",
    "\n",
    "For more details on how to use Notebooks on Polly, please visit [Polly Notebooks](https://docs.elucidata.io/Scaling%20compute/Polly%20Notebooks.html).\n",
    "\n",
    "For more details on API access to your OmixAtlas, please visit [Accessing OmixAtlas using polly-python through Polly Notebooks](https://docs.elucidata.io/OmixAtlas/Polly%20Python.html)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# please do not modify\n",
    "from IPython.display import display_html\n",
    "def restartkernel() :\n",
    "    display_html(\"<script>Jupyter.notebook.kernel.restart()</script>\",raw=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Install polly-python"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: polly-python in /usr/local/lib/python3.6/dist-packages (0.0.7)\n",
      "Requirement already satisfied: pytz==2021.1 in /usr/local/lib/python3.6/dist-packages (from polly-python) (2021.1)\n",
      "Requirement already satisfied: six==1.16.0 in /usr/local/lib/python3.6/dist-packages (from polly-python) (1.16.0)\n",
      "Requirement already satisfied: boto3>=1.17.73 in /usr/local/lib/python3.6/dist-packages (from polly-python) (1.20.48)\n",
      "Requirement already satisfied: urllib3==1.26.6 in /usr/local/lib/python3.6/dist-packages (from polly-python) (1.26.6)\n",
      "Requirement already satisfied: idna==2.10 in /usr/local/lib/python3.6/dist-packages (from polly-python) (2.10)\n",
      "Requirement already satisfied: chardet==4.0.0 in /usr/local/lib/python3.6/dist-packages (from polly-python) (4.0.0)\n",
      "Requirement already satisfied: elucidatacloudpathlib==0.6.6 in /usr/local/lib/python3.6/dist-packages (from polly-python) (0.6.6)\n",
      "Requirement already satisfied: python-dateutil==2.8.1 in /usr/local/lib/python3.6/dist-packages (from polly-python) (2.8.1)\n",
      "Requirement already satisfied: pandas in /usr/local/lib/python3.6/dist-packages (from polly-python) (1.0.1)\n",
      "Requirement already satisfied: retrying==1.3.3 in /usr/local/lib/python3.6/dist-packages (from polly-python) (1.3.3)\n",
      "Requirement already satisfied: certifi==2021.5.30 in /usr/local/lib/python3.6/dist-packages (from polly-python) (2021.5.30)\n",
      "Requirement already satisfied: postpy2==0.0.6 in /usr/local/lib/python3.6/dist-packages (from polly-python) (0.0.6)\n",
      "Requirement already satisfied: botocore>=1.20.73 in /usr/local/lib/python3.6/dist-packages (from polly-python) (1.23.48)\n",
      "Requirement already satisfied: requests==2.25.1 in /usr/local/lib/python3.6/dist-packages (from polly-python) (2.25.1)\n",
      "Requirement already satisfied: python-magic==0.4.24 in /usr/local/lib/python3.6/dist-packages (from polly-python) (0.4.24)\n",
      "Requirement already satisfied: jmespath<1.0.0,>=0.7.1 in /usr/local/lib/python3.6/dist-packages (from boto3>=1.17.73->polly-python) (0.9.5)\n",
      "Requirement already satisfied: s3transfer<0.6.0,>=0.5.0 in /usr/local/lib/python3.6/dist-packages (from boto3>=1.17.73->polly-python) (0.5.1)\n",
      "Requirement already satisfied: importlib-metadata; python_version < \"3.8\" in /usr/local/lib/python3.6/dist-packages (from elucidatacloudpathlib==0.6.6->polly-python) (1.5.0)\n",
      "Requirement already satisfied: numpy>=1.13.3 in /usr/local/lib/python3.6/dist-packages (from pandas->polly-python) (1.18.1)\n",
      "Requirement already satisfied: zipp>=0.5 in /usr/local/lib/python3.6/dist-packages (from importlib-metadata; python_version < \"3.8\"->elucidatacloudpathlib==0.6.6->polly-python) (3.1.0)\n",
      "\u001b[33mWARNING: You are using pip version 19.2.3, however version 21.3.1 is available.\n",
      "You should consider upgrading via the 'pip install --upgrade pip' command.\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "!sudo pip3 install polly-python # to install polly-python"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<script>Jupyter.notebook.kernel.restart()</script>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "restartkernel() #Pause for a few seconds before the kernel is refreshed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<script type=\"text/javascript\"> Jupyter.notebook.kernel.execute(\"url = '\" + window.location + \"'\", {}, {}); </script>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# please do not modify\n",
    "from IPython.display import HTML\n",
    "HTML('''<script type=\"text/javascript\"> Jupyter.notebook.kernel.execute(\"url = '\" + window.location + \"'\", {}, {}); </script>''')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Import required dependencies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "from polly.auth import Polly\n",
    "from polly.omixatlas import OmixAtlas\n",
    "from polly.workspaces import Workspaces"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Auth with token on Polly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "AUTH_TOKEN=(os.environ['POLLY_REFRESH_TOKEN']) # Obtain authentication tokens\n",
    "Polly.auth(AUTH_TOKEN)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.set_option('display.max_columns', 500)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Defining omixatlas object\n",
    "omixatlas = OmixAtlas()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Get basic properties of all the omixatlas users have access to\n",
    "all_omix_atlas = pd.DataFrame.from_dict(omixatlas.get_all_omixatlas()['data'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>repo_name</th>\n",
       "      <th>repo_id</th>\n",
       "      <th>indexes</th>\n",
       "      <th>diseases</th>\n",
       "      <th>organisms</th>\n",
       "      <th>sources</th>\n",
       "      <th>datatypes</th>\n",
       "      <th>dataset_count</th>\n",
       "      <th>disease_count</th>\n",
       "      <th>tissue_count</th>\n",
       "      <th>organism_count</th>\n",
       "      <th>cell_line_count</th>\n",
       "      <th>cell_type_count</th>\n",
       "      <th>drug_count</th>\n",
       "      <th>data_type_count</th>\n",
       "      <th>data_source_count</th>\n",
       "      <th>sample_count</th>\n",
       "      <th>normal_sample_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>auron_data_lake</td>\n",
       "      <td>1613041985263</td>\n",
       "      <td>{'gct_metadata': 'auron_data_lake_gct_metadata...</td>\n",
       "      <td>[neuroblastoma, normal, leukemia, acute myeloi...</td>\n",
       "      <td>[homo sapiens]</td>\n",
       "      <td>[auron, ccle, gdc, geo, public data (https://w...</td>\n",
       "      <td>[transcriptomics, genomics]</td>\n",
       "      <td>14</td>\n",
       "      <td>4</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>1277</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>hpa</td>\n",
       "      <td>1619514006627</td>\n",
       "      <td>{'gct_metadata': 'hpa_gct_metadata', 'h5ad_met...</td>\n",
       "      <td>[normal]</td>\n",
       "      <td>[homo sapiens, sus (pig), mus musculus]</td>\n",
       "      <td>[hpa]</td>\n",
       "      <td>[transcriptomics, gene expression reliability]</td>\n",
       "      <td>1755</td>\n",
       "      <td>1</td>\n",
       "      <td>78</td>\n",
       "      <td>3</td>\n",
       "      <td>70</td>\n",
       "      <td>150</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>678</td>\n",
       "      <td>678</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>gdc</td>\n",
       "      <td>1623221686703</td>\n",
       "      <td>{'gct_metadata': 'gdc_gct_metadata', 'h5ad_met...</td>\n",
       "      <td>[leukemia, myeloid, acute, precursor t-cell ly...</td>\n",
       "      <td>[homo sapiens]</td>\n",
       "      <td>[gdc]</td>\n",
       "      <td>[mirna expression, transcriptomics, copy numbe...</td>\n",
       "      <td>9869</td>\n",
       "      <td>75</td>\n",
       "      <td>41</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>5353</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>geo</td>\n",
       "      <td>9</td>\n",
       "      <td>{'gct_metadata': 'geo_gct_metadata', 'h5ad_met...</td>\n",
       "      <td>[normal, neoplasms, breast neoplasms, obesity,...</td>\n",
       "      <td>[homo sapiens, mus musculus, rattus norvegicus...</td>\n",
       "      <td>[geo]</td>\n",
       "      <td>[transcriptomics, raw counts transcriptomics]</td>\n",
       "      <td>10000</td>\n",
       "      <td>3030</td>\n",
       "      <td>985</td>\n",
       "      <td>68</td>\n",
       "      <td>4970</td>\n",
       "      <td>780</td>\n",
       "      <td>1379</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1280558</td>\n",
       "      <td>840747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>enterprise_atlas</td>\n",
       "      <td>1638441282192</td>\n",
       "      <td>{'gct_metadata': 'enterprise_atlas_gct_metadat...</td>\n",
       "      <td>[breast neoplasms, endometrial neoplasms, endo...</td>\n",
       "      <td>[homo sapiens, mus musculus]</td>\n",
       "      <td>[geo, proprietary]</td>\n",
       "      <td>[transcriptomics, single cell, mutation, mirna]</td>\n",
       "      <td>176</td>\n",
       "      <td>35</td>\n",
       "      <td>25</td>\n",
       "      <td>2</td>\n",
       "      <td>96</td>\n",
       "      <td>40</td>\n",
       "      <td>22</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>8952</td>\n",
       "      <td>3937</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>cptac</td>\n",
       "      <td>1609924165364</td>\n",
       "      <td>{'gct_metadata': 'cptac_gct_metadata', 'h5ad_m...</td>\n",
       "      <td>[carcinoma, hepatocellular, glucose-galactose ...</td>\n",
       "      <td>[homo sapiens]</td>\n",
       "      <td>[cptac]</td>\n",
       "      <td>[proteomics, phosphoproteomics, transcriptomic...</td>\n",
       "      <td>6554</td>\n",
       "      <td>13</td>\n",
       "      <td>33</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "      <td>2440</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>liveromix_atlas</td>\n",
       "      <td>1615965444377</td>\n",
       "      <td>{'gct_metadata': 'liveromix_atlas_gct_metadata...</td>\n",
       "      <td>[normal, carcinoma, hepatocellular, obesity, n...</td>\n",
       "      <td>[homo sapiens, mus musculus, rattus norvegicus...</td>\n",
       "      <td>[geo, lincs, tcga, metabolomics workbench, met...</td>\n",
       "      <td>[transcriptomics, mutation, metabolomics, sing...</td>\n",
       "      <td>6761</td>\n",
       "      <td>752</td>\n",
       "      <td>60</td>\n",
       "      <td>22</td>\n",
       "      <td>347</td>\n",
       "      <td>310</td>\n",
       "      <td>4704</td>\n",
       "      <td>12</td>\n",
       "      <td>10</td>\n",
       "      <td>143535</td>\n",
       "      <td>110202</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>cbioportal</td>\n",
       "      <td>1623986995264</td>\n",
       "      <td>{'gct_metadata': 'cbioportal_gct_metadata', 'h...</td>\n",
       "      <td>[prostatic neoplasms, carcinoma, ductal, breas...</td>\n",
       "      <td>[homo sapiens]</td>\n",
       "      <td>[cbioportal]</td>\n",
       "      <td>[mutation, copy number variation, fusion, tran...</td>\n",
       "      <td>10000</td>\n",
       "      <td>330</td>\n",
       "      <td>262</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>58994</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>pcd</td>\n",
       "      <td>1622113130397</td>\n",
       "      <td>{'gct_metadata': 'pcd_gct_metadata', 'h5ad_met...</td>\n",
       "      <td>[none, adenocarcinoma, carcinoma, squamous cel...</td>\n",
       "      <td>[homo sapiens]</td>\n",
       "      <td>[pharmacodb]</td>\n",
       "      <td>[drug response]</td>\n",
       "      <td>10000</td>\n",
       "      <td>133</td>\n",
       "      <td>34</td>\n",
       "      <td>1</td>\n",
       "      <td>1400</td>\n",
       "      <td>1</td>\n",
       "      <td>749</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>653500</td>\n",
       "      <td>262</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>lincs</td>\n",
       "      <td>32</td>\n",
       "      <td>{'gct_metadata': 'lincs_gct_metadata', 'h5ad_m...</td>\n",
       "      <td>[prostatic neoplasms, normal, colorectal neopl...</td>\n",
       "      <td>[homo sapiens]</td>\n",
       "      <td>[geo, geo, lincs]</td>\n",
       "      <td>[transcriptomics]</td>\n",
       "      <td>10000</td>\n",
       "      <td>34</td>\n",
       "      <td>21</td>\n",
       "      <td>1</td>\n",
       "      <td>84</td>\n",
       "      <td>1</td>\n",
       "      <td>20347</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1371945</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>metabolomics</td>\n",
       "      <td>23</td>\n",
       "      <td>{'gct_metadata': 'metabolomics_gct_metadata', ...</td>\n",
       "      <td>[normal, neoplasms, diabetes mellitus, obesity...</td>\n",
       "      <td>[homo sapiens, mus musculus, rattus norvegicus...</td>\n",
       "      <td>[metabolomics workbench, metabolights, publica...</td>\n",
       "      <td>[metabolomics, lipidomics, single cell]</td>\n",
       "      <td>1777</td>\n",
       "      <td>214</td>\n",
       "      <td>134</td>\n",
       "      <td>184</td>\n",
       "      <td>24</td>\n",
       "      <td>23</td>\n",
       "      <td>67</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>90677</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>immport</td>\n",
       "      <td>1621422280385</td>\n",
       "      <td>{'gct_metadata': 'immport_gct_metadata', 'h5ad...</td>\n",
       "      <td>[normal, diabetes mellitus, type 1, kidney dis...</td>\n",
       "      <td>[homo sapiens, mus musculus]</td>\n",
       "      <td>[immport]</td>\n",
       "      <td>[lab measurement, proteomics, titer, pcr, cyto...</td>\n",
       "      <td>10000</td>\n",
       "      <td>86</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>47</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>51044</td>\n",
       "      <td>25886</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>depmap_ccle</td>\n",
       "      <td>20</td>\n",
       "      <td>{'gct_metadata': 'depmap_ccle_gct_metadata', '...</td>\n",
       "      <td>[adrenal cancer, bile duct cancer, bladder can...</td>\n",
       "      <td>[homo sapiens]</td>\n",
       "      <td>[ccle]</td>\n",
       "      <td>[]</td>\n",
       "      <td>33</td>\n",
       "      <td>32</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1177</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>sc_data_lake</td>\n",
       "      <td>17</td>\n",
       "      <td>{'gct_metadata': 'sc_data_lake_gct_metadata', ...</td>\n",
       "      <td>[normal, neoplasms, breast neoplasms, melanoma...</td>\n",
       "      <td>[mus musculus, homo sapiens, macaca fascicular...</td>\n",
       "      <td>[geo, humancellatlas, gene expression omnibus ...</td>\n",
       "      <td>[single cell]</td>\n",
       "      <td>3175</td>\n",
       "      <td>582</td>\n",
       "      <td>435</td>\n",
       "      <td>5</td>\n",
       "      <td>74</td>\n",
       "      <td>552</td>\n",
       "      <td>86</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>215639</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>tcga</td>\n",
       "      <td>15</td>\n",
       "      <td>{'gct_metadata': 'tcga_gct_metadata', 'h5ad_me...</td>\n",
       "      <td>[breast neoplasms, clear-cell metastatic renal...</td>\n",
       "      <td>[homo sapiens]</td>\n",
       "      <td>[tcga]</td>\n",
       "      <td>[transcriptomics, mirna, copy number variation...</td>\n",
       "      <td>10000</td>\n",
       "      <td>41</td>\n",
       "      <td>62</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>606</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>etx</td>\n",
       "      <td>1641883311001</td>\n",
       "      <td>{'gct_metadata': 'etx_gct_metadata', 'h5ad_met...</td>\n",
       "      <td>[non-alcoholic fatty liver disease, obesity, d...</td>\n",
       "      <td>[homo sapiens]</td>\n",
       "      <td>[geo]</td>\n",
       "      <td>[transcriptomics]</td>\n",
       "      <td>56</td>\n",
       "      <td>47</td>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>14</td>\n",
       "      <td>20</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2136</td>\n",
       "      <td>869</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>ngj</td>\n",
       "      <td>1618578954468</td>\n",
       "      <td>{'gct_metadata': 'ngj_gct_metadata', 'h5ad_met...</td>\n",
       "      <td>[none, endometriosis, smith-magenis, tay–sachs...</td>\n",
       "      <td>[homo sapiens]</td>\n",
       "      <td>[propreitary]</td>\n",
       "      <td>[mutation, transcriptomics, mirna]</td>\n",
       "      <td>22</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>1220</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>gtex</td>\n",
       "      <td>14</td>\n",
       "      <td>{'gct_metadata': 'gtex_gct_metadata', 'h5ad_me...</td>\n",
       "      <td>[normal]</td>\n",
       "      <td>[homo sapiens]</td>\n",
       "      <td>[gtex]</td>\n",
       "      <td>[transcriptomics]</td>\n",
       "      <td>30</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>17381</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>gnomad</td>\n",
       "      <td>1628836648493</td>\n",
       "      <td>{'gct_metadata': 'gnomad_gct_metadata', 'h5ad_...</td>\n",
       "      <td>[none]</td>\n",
       "      <td>[homo sapiens]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[gwas]</td>\n",
       "      <td>10000</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           repo_name        repo_id  \\\n",
       "0    auron_data_lake  1613041985263   \n",
       "1                hpa  1619514006627   \n",
       "2                gdc  1623221686703   \n",
       "3                geo              9   \n",
       "4   enterprise_atlas  1638441282192   \n",
       "5              cptac  1609924165364   \n",
       "6    liveromix_atlas  1615965444377   \n",
       "7         cbioportal  1623986995264   \n",
       "8                pcd  1622113130397   \n",
       "9              lincs             32   \n",
       "10      metabolomics             23   \n",
       "11           immport  1621422280385   \n",
       "12       depmap_ccle             20   \n",
       "13      sc_data_lake             17   \n",
       "14              tcga             15   \n",
       "15               etx  1641883311001   \n",
       "16               ngj  1618578954468   \n",
       "17              gtex             14   \n",
       "18            gnomad  1628836648493   \n",
       "\n",
       "                                              indexes  \\\n",
       "0   {'gct_metadata': 'auron_data_lake_gct_metadata...   \n",
       "1   {'gct_metadata': 'hpa_gct_metadata', 'h5ad_met...   \n",
       "2   {'gct_metadata': 'gdc_gct_metadata', 'h5ad_met...   \n",
       "3   {'gct_metadata': 'geo_gct_metadata', 'h5ad_met...   \n",
       "4   {'gct_metadata': 'enterprise_atlas_gct_metadat...   \n",
       "5   {'gct_metadata': 'cptac_gct_metadata', 'h5ad_m...   \n",
       "6   {'gct_metadata': 'liveromix_atlas_gct_metadata...   \n",
       "7   {'gct_metadata': 'cbioportal_gct_metadata', 'h...   \n",
       "8   {'gct_metadata': 'pcd_gct_metadata', 'h5ad_met...   \n",
       "9   {'gct_metadata': 'lincs_gct_metadata', 'h5ad_m...   \n",
       "10  {'gct_metadata': 'metabolomics_gct_metadata', ...   \n",
       "11  {'gct_metadata': 'immport_gct_metadata', 'h5ad...   \n",
       "12  {'gct_metadata': 'depmap_ccle_gct_metadata', '...   \n",
       "13  {'gct_metadata': 'sc_data_lake_gct_metadata', ...   \n",
       "14  {'gct_metadata': 'tcga_gct_metadata', 'h5ad_me...   \n",
       "15  {'gct_metadata': 'etx_gct_metadata', 'h5ad_met...   \n",
       "16  {'gct_metadata': 'ngj_gct_metadata', 'h5ad_met...   \n",
       "17  {'gct_metadata': 'gtex_gct_metadata', 'h5ad_me...   \n",
       "18  {'gct_metadata': 'gnomad_gct_metadata', 'h5ad_...   \n",
       "\n",
       "                                             diseases  \\\n",
       "0   [neuroblastoma, normal, leukemia, acute myeloi...   \n",
       "1                                            [normal]   \n",
       "2   [leukemia, myeloid, acute, precursor t-cell ly...   \n",
       "3   [normal, neoplasms, breast neoplasms, obesity,...   \n",
       "4   [breast neoplasms, endometrial neoplasms, endo...   \n",
       "5   [carcinoma, hepatocellular, glucose-galactose ...   \n",
       "6   [normal, carcinoma, hepatocellular, obesity, n...   \n",
       "7   [prostatic neoplasms, carcinoma, ductal, breas...   \n",
       "8   [none, adenocarcinoma, carcinoma, squamous cel...   \n",
       "9   [prostatic neoplasms, normal, colorectal neopl...   \n",
       "10  [normal, neoplasms, diabetes mellitus, obesity...   \n",
       "11  [normal, diabetes mellitus, type 1, kidney dis...   \n",
       "12  [adrenal cancer, bile duct cancer, bladder can...   \n",
       "13  [normal, neoplasms, breast neoplasms, melanoma...   \n",
       "14  [breast neoplasms, clear-cell metastatic renal...   \n",
       "15  [non-alcoholic fatty liver disease, obesity, d...   \n",
       "16  [none, endometriosis, smith-magenis, tay–sachs...   \n",
       "17                                           [normal]   \n",
       "18                                             [none]   \n",
       "\n",
       "                                            organisms  \\\n",
       "0                                      [homo sapiens]   \n",
       "1             [homo sapiens, sus (pig), mus musculus]   \n",
       "2                                      [homo sapiens]   \n",
       "3   [homo sapiens, mus musculus, rattus norvegicus...   \n",
       "4                        [homo sapiens, mus musculus]   \n",
       "5                                      [homo sapiens]   \n",
       "6   [homo sapiens, mus musculus, rattus norvegicus...   \n",
       "7                                      [homo sapiens]   \n",
       "8                                      [homo sapiens]   \n",
       "9                                      [homo sapiens]   \n",
       "10  [homo sapiens, mus musculus, rattus norvegicus...   \n",
       "11                       [homo sapiens, mus musculus]   \n",
       "12                                     [homo sapiens]   \n",
       "13  [mus musculus, homo sapiens, macaca fascicular...   \n",
       "14                                     [homo sapiens]   \n",
       "15                                     [homo sapiens]   \n",
       "16                                     [homo sapiens]   \n",
       "17                                     [homo sapiens]   \n",
       "18                                     [homo sapiens]   \n",
       "\n",
       "                                              sources  \\\n",
       "0   [auron, ccle, gdc, geo, public data (https://w...   \n",
       "1                                               [hpa]   \n",
       "2                                               [gdc]   \n",
       "3                                               [geo]   \n",
       "4                                  [geo, proprietary]   \n",
       "5                                             [cptac]   \n",
       "6   [geo, lincs, tcga, metabolomics workbench, met...   \n",
       "7                                        [cbioportal]   \n",
       "8                                        [pharmacodb]   \n",
       "9                                   [geo, geo, lincs]   \n",
       "10  [metabolomics workbench, metabolights, publica...   \n",
       "11                                          [immport]   \n",
       "12                                             [ccle]   \n",
       "13  [geo, humancellatlas, gene expression omnibus ...   \n",
       "14                                             [tcga]   \n",
       "15                                              [geo]   \n",
       "16                                      [propreitary]   \n",
       "17                                             [gtex]   \n",
       "18                                                 []   \n",
       "\n",
       "                                            datatypes  dataset_count  \\\n",
       "0                         [transcriptomics, genomics]             14   \n",
       "1      [transcriptomics, gene expression reliability]           1755   \n",
       "2   [mirna expression, transcriptomics, copy numbe...           9869   \n",
       "3       [transcriptomics, raw counts transcriptomics]          10000   \n",
       "4     [transcriptomics, single cell, mutation, mirna]            176   \n",
       "5   [proteomics, phosphoproteomics, transcriptomic...           6554   \n",
       "6   [transcriptomics, mutation, metabolomics, sing...           6761   \n",
       "7   [mutation, copy number variation, fusion, tran...          10000   \n",
       "8                                     [drug response]          10000   \n",
       "9                                   [transcriptomics]          10000   \n",
       "10            [metabolomics, lipidomics, single cell]           1777   \n",
       "11  [lab measurement, proteomics, titer, pcr, cyto...          10000   \n",
       "12                                                 []             33   \n",
       "13                                      [single cell]           3175   \n",
       "14  [transcriptomics, mirna, copy number variation...          10000   \n",
       "15                                  [transcriptomics]             56   \n",
       "16                 [mutation, transcriptomics, mirna]             22   \n",
       "17                                  [transcriptomics]             30   \n",
       "18                                             [gwas]          10000   \n",
       "\n",
       "    disease_count  tissue_count  organism_count  cell_line_count  \\\n",
       "0               4             7               1                0   \n",
       "1               1            78               3               70   \n",
       "2              75            41               1                1   \n",
       "3            3030           985              68             4970   \n",
       "4              35            25               2               96   \n",
       "5              13            33               1                1   \n",
       "6             752            60              22              347   \n",
       "7             330           262               1                1   \n",
       "8             133            34               1             1400   \n",
       "9              34            21               1               84   \n",
       "10            214           134             184               24   \n",
       "11             86            10               2                1   \n",
       "12             32             0               1                0   \n",
       "13            582           435               5               74   \n",
       "14             41            62               1                1   \n",
       "15             47            13               1                4   \n",
       "16              5             5               1                1   \n",
       "17              1            30               1                1   \n",
       "18              1             1               1                1   \n",
       "\n",
       "    cell_type_count  drug_count  data_type_count  data_source_count  \\\n",
       "0                 0           0                2                  6   \n",
       "1               150           1                2                  1   \n",
       "2                 1           1                3                  1   \n",
       "3               780        1379                2                  1   \n",
       "4                40          22                4                  2   \n",
       "5                 1           1               10                  1   \n",
       "6               310        4704               12                 10   \n",
       "7                 1           1                5                  1   \n",
       "8                 1         749                1                  1   \n",
       "9                 1       20347                1                  2   \n",
       "10               23          67                3                  4   \n",
       "11                2          47                5                  1   \n",
       "12                0           0                0                  1   \n",
       "13              552          86                1                  6   \n",
       "14                1         606                6                  1   \n",
       "15               14          20                1                  1   \n",
       "16                2           1                3                  1   \n",
       "17                1           1                1                  1   \n",
       "18                0           1                1                  0   \n",
       "\n",
       "    sample_count  normal_sample_count  \n",
       "0           1277                    0  \n",
       "1            678                  678  \n",
       "2           5353                    0  \n",
       "3        1280558               840747  \n",
       "4           8952                 3937  \n",
       "5           2440                    0  \n",
       "6         143535               110202  \n",
       "7          58994                    0  \n",
       "8         653500                  262  \n",
       "9        1371945                    0  \n",
       "10         90677                    0  \n",
       "11         51044                25886  \n",
       "12          1177                    0  \n",
       "13        215639                    0  \n",
       "14             0                    0  \n",
       "15          2136                  869  \n",
       "16          1220                    0  \n",
       "17         17381                    0  \n",
       "18             0                    0  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_omix_atlas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "cell_line_count                                                     4970\n",
       "cell_type_count                                                      780\n",
       "data_source_count                                                      1\n",
       "data_type_count                                                        2\n",
       "dataset_count                                                      10000\n",
       "datatypes                  [transcriptomics, raw counts transcriptomics]\n",
       "disease_count                                                       3030\n",
       "diseases               [normal, neoplasms, breast neoplasms, obesity,...\n",
       "drug_count                                                          1379\n",
       "indexes                {'gct_metadata': 'geo_gct_metadata', 'h5ad_met...\n",
       "normal_sample_count                                               840747\n",
       "organism_count                                                        68\n",
       "organisms              [homo sapiens, mus musculus, rattus norvegicus...\n",
       "repo_id                                                                9\n",
       "repo_name                                                            geo\n",
       "sample_count                                                     1280558\n",
       "sources                                                            [geo]\n",
       "tissue_count                                                         985\n",
       "Name: data, dtype: object"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Get basic properties of all the omixatlas users have access to\n",
    "omix_atlas_summary = pd.DataFrame.from_dict(omixatlas.omixatlas_summary(9))['data']\n",
    "omix_atlas_summary"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Checking for schema of a repository"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Query execution succeeded (time taken: 0.44 seconds, data scanned: 0.000 MB)\n",
      "Fetched 3 rows\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>table_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>geo.datasets</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>geo.samples</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>geo.features</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     table_name\n",
       "0  geo.datasets\n",
       "1   geo.samples\n",
       "2  geo.features"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query = \"SHOW TABLES IN geo\"\n",
    "results=omixatlas.query_metadata(query, query_api_version=\"v2\")\n",
    "results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Querying for schema of dataset level metadata of GEO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Query execution succeeded (time taken: 1.35 seconds, data scanned: 0.000 MB)\n",
      "Fetched 52 rows\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>column_name</th>\n",
       "      <th>column_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>growth_protocol_ch1</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>src_uri</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>sample_id</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>dose_ch1</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>cohort_name</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>tissue</td>\n",
       "      <td>array&lt;string&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>src_dataset_id</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>drug</td>\n",
       "      <td>array&lt;string&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>extract_protocol_ch1</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>characteristics_ch2</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>cell_line</td>\n",
       "      <td>array&lt;string&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>characteristics_ch1</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>extract_protocol_ch2</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>age_ch1</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>drug_smiles_code</td>\n",
       "      <td>array&lt;string&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>sex_ch1</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>treatment_ch1</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>gene</td>\n",
       "      <td>array&lt;string&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>taxid_ch1</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>cohort_id</td>\n",
       "      <td>bigint</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>version</td>\n",
       "      <td>bigint</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>label_protocol_ch2</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>label_protocol_ch1</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>treatment_protocol_ch2</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>characteristics_ch2_1</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>treatment_protocol_ch1</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>instrument_model</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>name</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>bmi_ch1</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>data_processing</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>description</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>title</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>timestamp_</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>data_id</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>cell_type</td>\n",
       "      <td>array&lt;string&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>src_repo</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>id_key</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>disease</td>\n",
       "      <td>array&lt;string&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>gender_ch1</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>smoking_status_ch1</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>label_ch2</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>control</td>\n",
       "      <td>bigint</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>label_ch1</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>time_point_ch1</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>characteristics_ch1_3</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>characteristics_ch1_2</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>gene_modified</td>\n",
       "      <td>array&lt;string&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>genetic_modification</td>\n",
       "      <td>array&lt;string&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>hyb_protocol</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>platform_id</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>is_current</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>characteristics_ch1_1</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               column_name    column_type\n",
       "0      growth_protocol_ch1         string\n",
       "1                  src_uri         string\n",
       "2                sample_id         string\n",
       "3                 dose_ch1         string\n",
       "4              cohort_name         string\n",
       "5                   tissue  array<string>\n",
       "6           src_dataset_id         string\n",
       "7                     drug  array<string>\n",
       "8     extract_protocol_ch1         string\n",
       "9      characteristics_ch2         string\n",
       "10               cell_line  array<string>\n",
       "11     characteristics_ch1         string\n",
       "12    extract_protocol_ch2         string\n",
       "13                 age_ch1         string\n",
       "14        drug_smiles_code  array<string>\n",
       "15                 sex_ch1         string\n",
       "16           treatment_ch1         string\n",
       "17                    gene  array<string>\n",
       "18               taxid_ch1         string\n",
       "19               cohort_id         bigint\n",
       "20                 version         bigint\n",
       "21      label_protocol_ch2         string\n",
       "22      label_protocol_ch1         string\n",
       "23  treatment_protocol_ch2         string\n",
       "24   characteristics_ch2_1         string\n",
       "25  treatment_protocol_ch1         string\n",
       "26        instrument_model         string\n",
       "27                    name         string\n",
       "28                 bmi_ch1         string\n",
       "29         data_processing         string\n",
       "30             description         string\n",
       "31                   title         string\n",
       "32              timestamp_         string\n",
       "33                 data_id         string\n",
       "34               cell_type  array<string>\n",
       "35                src_repo         string\n",
       "36                  id_key         string\n",
       "37                 disease  array<string>\n",
       "38              gender_ch1         string\n",
       "39      smoking_status_ch1         string\n",
       "40               label_ch2         string\n",
       "41                 control         bigint\n",
       "42               label_ch1         string\n",
       "43          time_point_ch1         string\n",
       "44   characteristics_ch1_3         string\n",
       "45   characteristics_ch1_2         string\n",
       "46           gene_modified  array<string>\n",
       "47    genetic_modification  array<string>\n",
       "48            hyb_protocol         string\n",
       "49             platform_id         string\n",
       "50              is_current         string\n",
       "51   characteristics_ch1_1         string"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query = \"DESCRIBE geo.datasets\"\n",
    "#query = \"DESCRIBE geo.samples\"\n",
    "results=omixatlas.query_metadata(query, query_api_version=\"v2\")\n",
    "results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Querying for dataset-id, description, number of samples etc. from GEO where platform is 'RNASeq', number of samples >20."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Query execution succeeded (time taken: 1.84 seconds, data scanned: 3.611 MB)\n",
      "Fetched 1787 rows\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dataset_id</th>\n",
       "      <th>description</th>\n",
       "      <th>total_num_samples</th>\n",
       "      <th>disease</th>\n",
       "      <th>tissue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GSE86039_GPL19057</td>\n",
       "      <td>A distinct gene module uncouples dysfunction f...</td>\n",
       "      <td>384</td>\n",
       "      <td>[Neoplasms, Virus Diseases, T cell immunodefic...</td>\n",
       "      <td>[None]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GSE86030_GPL19057</td>\n",
       "      <td>A distinct gene module uncouples dysfunction f...</td>\n",
       "      <td>384</td>\n",
       "      <td>[Neoplasms, Virus Diseases, T cell immunodefic...</td>\n",
       "      <td>[None]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GSE101334_GPL19057</td>\n",
       "      <td>Single cell RNA-seq of smooth muscle actin cel...</td>\n",
       "      <td>384</td>\n",
       "      <td>[Normal]</td>\n",
       "      <td>[lung]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GSE102160_GPL17021</td>\n",
       "      <td>Single cell RNA Seq data of BMDM from mouse in...</td>\n",
       "      <td>384</td>\n",
       "      <td>[Salmonella Infections]</td>\n",
       "      <td>[None]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GSE86028_GPL19057</td>\n",
       "      <td>A distinct gene module uncouples dysfunction f...</td>\n",
       "      <td>384</td>\n",
       "      <td>[Neoplasms, Virus Diseases, T cell immunodefic...</td>\n",
       "      <td>[None]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1782</th>\n",
       "      <td>GSE62889_GPL11154</td>\n",
       "      <td>Genome-wide expression profile of the Tet-On H...</td>\n",
       "      <td>21</td>\n",
       "      <td>[Colonic Neoplasms]</td>\n",
       "      <td>[colon]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1783</th>\n",
       "      <td>GSE104305_GPL17021</td>\n",
       "      <td>Differential gene expression induced by anti-c...</td>\n",
       "      <td>21</td>\n",
       "      <td>[Prostatic Neoplasms, Androgen-Insensitivity S...</td>\n",
       "      <td>[prostate gland]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1784</th>\n",
       "      <td>GSE114340_GPL13112</td>\n",
       "      <td>Commensal Microbiota Promote Lung Tumorigenesi...</td>\n",
       "      <td>21</td>\n",
       "      <td>[Adenocarcinoma of Lung, Diabetes Mellitus, Ty...</td>\n",
       "      <td>[lung, spleen]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1785</th>\n",
       "      <td>GSE90046_GPL11002</td>\n",
       "      <td>C/EBPÎ² deficiency reshapes microglial gene ex...</td>\n",
       "      <td>21</td>\n",
       "      <td>[Parkinson Disease, Alzheimer Disease, Multipl...</td>\n",
       "      <td>[None]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1786</th>\n",
       "      <td>GSE104882_GPL21103</td>\n",
       "      <td>Diet-Induced Obesity Leads to Infarct Expansio...</td>\n",
       "      <td>21</td>\n",
       "      <td>[Obesity, Stroke, Brain Ischemia]</td>\n",
       "      <td>[brain]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1787 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              dataset_id                                        description  \\\n",
       "0      GSE86039_GPL19057  A distinct gene module uncouples dysfunction f...   \n",
       "1      GSE86030_GPL19057  A distinct gene module uncouples dysfunction f...   \n",
       "2     GSE101334_GPL19057  Single cell RNA-seq of smooth muscle actin cel...   \n",
       "3     GSE102160_GPL17021  Single cell RNA Seq data of BMDM from mouse in...   \n",
       "4      GSE86028_GPL19057  A distinct gene module uncouples dysfunction f...   \n",
       "...                  ...                                                ...   \n",
       "1782   GSE62889_GPL11154  Genome-wide expression profile of the Tet-On H...   \n",
       "1783  GSE104305_GPL17021  Differential gene expression induced by anti-c...   \n",
       "1784  GSE114340_GPL13112  Commensal Microbiota Promote Lung Tumorigenesi...   \n",
       "1785   GSE90046_GPL11002  C/EBPÎ² deficiency reshapes microglial gene ex...   \n",
       "1786  GSE104882_GPL21103  Diet-Induced Obesity Leads to Infarct Expansio...   \n",
       "\n",
       "      total_num_samples                                            disease  \\\n",
       "0                   384  [Neoplasms, Virus Diseases, T cell immunodefic...   \n",
       "1                   384  [Neoplasms, Virus Diseases, T cell immunodefic...   \n",
       "2                   384                                           [Normal]   \n",
       "3                   384                            [Salmonella Infections]   \n",
       "4                   384  [Neoplasms, Virus Diseases, T cell immunodefic...   \n",
       "...                 ...                                                ...   \n",
       "1782                 21                                [Colonic Neoplasms]   \n",
       "1783                 21  [Prostatic Neoplasms, Androgen-Insensitivity S...   \n",
       "1784                 21  [Adenocarcinoma of Lung, Diabetes Mellitus, Ty...   \n",
       "1785                 21  [Parkinson Disease, Alzheimer Disease, Multipl...   \n",
       "1786                 21                  [Obesity, Stroke, Brain Ischemia]   \n",
       "\n",
       "                tissue  \n",
       "0               [None]  \n",
       "1               [None]  \n",
       "2               [lung]  \n",
       "3               [None]  \n",
       "4               [None]  \n",
       "...                ...  \n",
       "1782           [colon]  \n",
       "1783  [prostate gland]  \n",
       "1784    [lung, spleen]  \n",
       "1785            [None]  \n",
       "1786           [brain]  \n",
       "\n",
       "[1787 rows x 5 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query = \"\"\"SELECT dataset_id, description, total_num_samples, disease, tissue FROM geo.datasets \n",
    "            WHERE platform = 'RNASeq' AND total_num_samples > 20 ORDER BY total_num_samples desc\"\"\"\n",
    "results=omixatlas.query_metadata(query, query_api_version=\"v2\")\n",
    "results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Querying for datasets from GEO where disease array has any item consisting of keyword 'neoplasm'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Query execution succeeded (time taken: 2.19 seconds, data scanned: 0.258 MB)\n",
      "Fetched 10740 rows\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>disease</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>[Neoplasms]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>[Neuroblastoma, Brain Neoplasms, Cognitive Dys...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>[Neoplasms]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>[Skin Neoplasms]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>[Pancreatic Neoplasms]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10735</th>\n",
       "      <td>[Carcinoma, Hepatocellular, Adenocarcinoma of ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10736</th>\n",
       "      <td>[Leukemia, Familial Mediterranean Fever, Bone ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10737</th>\n",
       "      <td>[Neoplasms]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10738</th>\n",
       "      <td>[Melanoma, Neoplasms]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10739</th>\n",
       "      <td>[Neoplasms]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10740 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                 disease\n",
       "0                                            [Neoplasms]\n",
       "1      [Neuroblastoma, Brain Neoplasms, Cognitive Dys...\n",
       "2                                            [Neoplasms]\n",
       "3                                       [Skin Neoplasms]\n",
       "4                                 [Pancreatic Neoplasms]\n",
       "...                                                  ...\n",
       "10735  [Carcinoma, Hepatocellular, Adenocarcinoma of ...\n",
       "10736  [Leukemia, Familial Mediterranean Fever, Bone ...\n",
       "10737                                        [Neoplasms]\n",
       "10738                              [Melanoma, Neoplasms]\n",
       "10739                                        [Neoplasms]\n",
       "\n",
       "[10740 rows x 1 columns]"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Querying for datasets from GEO where disease array has any item consisting of keyword 'neoplasm'\n",
    "query = \"SELECT disease FROM geo.datasets WHERE CARDINALITY(FILTER(disease, d -> LOWER(d) like '%neoplasm%')) > 0\"\n",
    "results=omixatlas.query_metadata(query, query_api_version=\"v2\")\n",
    "results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Querying for datasets and number of samples in a dataset from GEO where disease array has an exact term 'Obesity'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Query execution succeeded (time taken: 1.52 seconds, data scanned: 0.766 MB)\n",
      "Fetched 966 rows\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dataset_id</th>\n",
       "      <th>total_num_samples</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GSE87565_GPL18480</td>\n",
       "      <td>12.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GSE64699_GPL10558</td>\n",
       "      <td>28.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GSE151268_GPL16570</td>\n",
       "      <td>20.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GSE161644_GPL1261</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GSE66766_GPL1261</td>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>961</th>\n",
       "      <td>GSE117339_GPL10558</td>\n",
       "      <td>214.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>962</th>\n",
       "      <td>GSE62832_GPL6244</td>\n",
       "      <td>36.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>963</th>\n",
       "      <td>GSE62936_GPL16791</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>964</th>\n",
       "      <td>GSE16113_GPL4133</td>\n",
       "      <td>70.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>965</th>\n",
       "      <td>GSE46843_GPL7202</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>966 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             dataset_id  total_num_samples\n",
       "0     GSE87565_GPL18480               12.0\n",
       "1     GSE64699_GPL10558               28.0\n",
       "2    GSE151268_GPL16570               20.0\n",
       "3     GSE161644_GPL1261                3.0\n",
       "4      GSE66766_GPL1261               10.0\n",
       "..                  ...                ...\n",
       "961  GSE117339_GPL10558              214.0\n",
       "962    GSE62832_GPL6244               36.0\n",
       "963   GSE62936_GPL16791               13.0\n",
       "964    GSE16113_GPL4133               70.0\n",
       "965    GSE46843_GPL7202               11.0\n",
       "\n",
       "[966 rows x 2 columns]"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query = \"SELECT dataset_id, total_num_samples FROM geo.datasets WHERE CONTAINS(disease, 'Obesity')\"\n",
    "results=omixatlas.query_metadata(query, query_api_version=\"v2\")\n",
    "results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get the abundance of different tissues across all datasets in GEO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Query execution succeeded (time taken: 1.91 seconds, data scanned: 0.130 MB)\n",
      "Fetched 932 rows\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tissue_name</th>\n",
       "      <th>tissue_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>17765</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>liver</td>\n",
       "      <td>3598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>quadriceps</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>prefrontal cortex</td>\n",
       "      <td>219</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>forebrain</td>\n",
       "      <td>103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>927</th>\n",
       "      <td>flexor digitorum longus</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>928</th>\n",
       "      <td>antler</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>929</th>\n",
       "      <td>myelin sheath</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>930</th>\n",
       "      <td>olfactory organ</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>931</th>\n",
       "      <td>longissimus thoracis</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>932 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 tissue_name  tissue_count\n",
       "0                       None         17765\n",
       "1                      liver          3598\n",
       "2                 quadriceps           100\n",
       "3          prefrontal cortex           219\n",
       "4                  forebrain           103\n",
       "..                       ...           ...\n",
       "927  flexor digitorum longus             1\n",
       "928                   antler             1\n",
       "929            myelin sheath             1\n",
       "930          olfactory organ             1\n",
       "931     longissimus thoracis             1\n",
       "\n",
       "[932 rows x 2 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query = \"\"\"SELECT tissue_name, COUNT((tissue_name)) AS tissue_count FROM geo.datasets \n",
    "            CROSS JOIN UNNEST(tissue) AS t(tissue_name) GROUP BY tissue_name\"\"\"\n",
    "results=omixatlas.query_metadata(query, query_api_version=\"v2\")\n",
    "results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Fetch the list of all the dataset IDs where expression value of gene 'brd4' is reported"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Query execution succeeded (time taken: 3.40 seconds, data scanned: 139.192 MB)\n",
      "Fetched 723 rows\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>src_dataset_id</th>\n",
       "      <th>name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GSE23860_GPL6238</td>\n",
       "      <td>Brd4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GSE98742_GPL10558</td>\n",
       "      <td>BRD4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GSE94728_GPL16791</td>\n",
       "      <td>BRD4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GSE83405_GPL16331</td>\n",
       "      <td>Brd4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GSE82020_GPL10787</td>\n",
       "      <td>Brd4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>718</th>\n",
       "      <td>GSE43578_GPL6246</td>\n",
       "      <td>Brd4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>719</th>\n",
       "      <td>GSE71620_GPL11532</td>\n",
       "      <td>BRD4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>720</th>\n",
       "      <td>GSE56161_GPL10787</td>\n",
       "      <td>Brd4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>721</th>\n",
       "      <td>GSE230_GPL8300</td>\n",
       "      <td>BRD4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>722</th>\n",
       "      <td>GSE71074_GPL11154</td>\n",
       "      <td>BRD4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>723 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        src_dataset_id  name\n",
       "0     GSE23860_GPL6238  Brd4\n",
       "1    GSE98742_GPL10558  BRD4\n",
       "2    GSE94728_GPL16791  BRD4\n",
       "3    GSE83405_GPL16331  Brd4\n",
       "4    GSE82020_GPL10787  Brd4\n",
       "..                 ...   ...\n",
       "718   GSE43578_GPL6246  Brd4\n",
       "719  GSE71620_GPL11532  BRD4\n",
       "720  GSE56161_GPL10787  Brd4\n",
       "721     GSE230_GPL8300  BRD4\n",
       "722  GSE71074_GPL11154  BRD4\n",
       "\n",
       "[723 rows x 2 columns]"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query = \"SELECT src_dataset_id, name FROM geo.features WHERE LOWER(name) = 'brd4'\"\n",
    "results=omixatlas.query_metadata(query, query_api_version=\"v2\")\n",
    "results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get the name of samples, dataset ID and extract_protocol_ch1 where spectroscopy is mentioned in the extract_protocol_ch1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Query execution succeeded (time taken: 4.20 seconds, data scanned: 160.930 MB)\n",
      "Fetched 752 rows\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>src_dataset_id</th>\n",
       "      <th>extract_protocol_ch1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GSM1697400</td>\n",
       "      <td>GSE69306_GPL19559</td>\n",
       "      <td>Gastrointestinal tissues (whole stomach and ab...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GSM96567</td>\n",
       "      <td>GSE4230_GPL1261</td>\n",
       "      <td>Embryonic kidney tissue from each individual w...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GSM2210787</td>\n",
       "      <td>GSE83602_GPL21382</td>\n",
       "      <td>RNA isolation was performed using a miRNeasy M...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GSM1697388</td>\n",
       "      <td>GSE69306_GPL19559</td>\n",
       "      <td>Gastrointestinal tissues (whole stomach and ab...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GSM436442</td>\n",
       "      <td>GSE17504_GPL6244</td>\n",
       "      <td>Total RNA was isolated from hESF from subjects...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>747</th>\n",
       "      <td>GSM2757387</td>\n",
       "      <td>GSE103205_GPL2872</td>\n",
       "      <td>Total RNA was isolated from snap-frozen hepati...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>748</th>\n",
       "      <td>GSM1192607</td>\n",
       "      <td>GSE49030_GPL1261</td>\n",
       "      <td>For RNA isolation hippocampi were dissected fr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>749</th>\n",
       "      <td>GSM3143578</td>\n",
       "      <td>GSE114528_GPL19057</td>\n",
       "      <td>RNA was extracted from sorted EVP, TA and D ce...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>750</th>\n",
       "      <td>GSM2210798</td>\n",
       "      <td>GSE83602_GPL21382</td>\n",
       "      <td>RNA isolation was performed using a miRNeasy M...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>751</th>\n",
       "      <td>GSM1697446</td>\n",
       "      <td>GSE69306_GPL19559</td>\n",
       "      <td>Gastrointestinal tissues (whole stomach and ab...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>752 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           name      src_dataset_id  \\\n",
       "0    GSM1697400   GSE69306_GPL19559   \n",
       "1      GSM96567     GSE4230_GPL1261   \n",
       "2    GSM2210787   GSE83602_GPL21382   \n",
       "3    GSM1697388   GSE69306_GPL19559   \n",
       "4     GSM436442    GSE17504_GPL6244   \n",
       "..          ...                 ...   \n",
       "747  GSM2757387   GSE103205_GPL2872   \n",
       "748  GSM1192607    GSE49030_GPL1261   \n",
       "749  GSM3143578  GSE114528_GPL19057   \n",
       "750  GSM2210798   GSE83602_GPL21382   \n",
       "751  GSM1697446   GSE69306_GPL19559   \n",
       "\n",
       "                                  extract_protocol_ch1  \n",
       "0    Gastrointestinal tissues (whole stomach and ab...  \n",
       "1    Embryonic kidney tissue from each individual w...  \n",
       "2    RNA isolation was performed using a miRNeasy M...  \n",
       "3    Gastrointestinal tissues (whole stomach and ab...  \n",
       "4    Total RNA was isolated from hESF from subjects...  \n",
       "..                                                 ...  \n",
       "747  Total RNA was isolated from snap-frozen hepati...  \n",
       "748  For RNA isolation hippocampi were dissected fr...  \n",
       "749  RNA was extracted from sorted EVP, TA and D ce...  \n",
       "750  RNA isolation was performed using a miRNeasy M...  \n",
       "751  Gastrointestinal tissues (whole stomach and ab...  \n",
       "\n",
       "[752 rows x 3 columns]"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query = \"\"\"SELECT name, src_dataset_id, extract_protocol_ch1 FROM geo.samples\n",
    "         WHERE LOWER(extract_protocol_ch1) LIKE '%spectroscopy%'\"\"\"\n",
    "results=omixatlas.query_metadata(query, query_api_version=\"v2\")\n",
    "results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get the name of gene and number of samples where information about particular gene is curated "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Query execution succeeded (time taken: 2.00 seconds, data scanned: 0.093 MB)\n",
      "Fetched 326 rows\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>[]</td>\n",
       "      <td>1331220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>[none]</td>\n",
       "      <td>27319</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>[CD4]</td>\n",
       "      <td>631</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>[CD8A]</td>\n",
       "      <td>503</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>[CD14]</td>\n",
       "      <td>448</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>321</th>\n",
       "      <td>[ZEB2]</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>322</th>\n",
       "      <td>[ARID3B]</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>323</th>\n",
       "      <td>[STARD3]</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>324</th>\n",
       "      <td>[IL4]</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>325</th>\n",
       "      <td>[MIR26A1]</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>326 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          gene    count\n",
       "0           []  1331220\n",
       "1       [none]    27319\n",
       "2        [CD4]      631\n",
       "3       [CD8A]      503\n",
       "4       [CD14]      448\n",
       "..         ...      ...\n",
       "321     [ZEB2]        1\n",
       "322   [ARID3B]        1\n",
       "323   [STARD3]        1\n",
       "324      [IL4]        1\n",
       "325  [MIR26A1]        1\n",
       "\n",
       "[326 rows x 2 columns]"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query = \"\"\"SELECT gene, COUNT(*) AS count FROM geo.samples GROUP BY gene ORDER BY count DESC \"\"\"\n",
    "results=omixatlas.query_metadata(query, query_api_version=\"v2\")\n",
    "results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get number of datasets with all combinations of disease and tissues in GEO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Query execution succeeded (time taken: 1.76 seconds, data scanned: 0.388 MB)\n",
      "Fetched 27758 rows\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tissue_name</th>\n",
       "      <th>disease_name</th>\n",
       "      <th>COUNT_COL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>Normal</td>\n",
       "      <td>5854</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>liver</td>\n",
       "      <td>Normal</td>\n",
       "      <td>1344</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>None</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>1268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>None</td>\n",
       "      <td>Breast Neoplasms</td>\n",
       "      <td>1016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>bone marrow</td>\n",
       "      <td>Normal</td>\n",
       "      <td>985</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27753</th>\n",
       "      <td>peripheral blood</td>\n",
       "      <td>Myalgia</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27754</th>\n",
       "      <td>peripheral blood</td>\n",
       "      <td>Blood Coagulation Disorders, Inherited</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27755</th>\n",
       "      <td>alimentary canal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27756</th>\n",
       "      <td>left ventricle</td>\n",
       "      <td>Long Qt Syndrome 3</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27757</th>\n",
       "      <td>placenta</td>\n",
       "      <td>Pulmonary Disease, Chronic Obstructive</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>27758 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            tissue_name                            disease_name  COUNT_COL\n",
       "0                  None                                  Normal       5854\n",
       "1                 liver                                  Normal       1344\n",
       "2                  None                               Neoplasms       1268\n",
       "3                  None                        Breast Neoplasms       1016\n",
       "4           bone marrow                                  Normal        985\n",
       "...                 ...                                     ...        ...\n",
       "27753  peripheral blood                                 Myalgia          1\n",
       "27754  peripheral blood  Blood Coagulation Disorders, Inherited          1\n",
       "27755  alimentary canal                                  Normal          1\n",
       "27756    left ventricle                      Long Qt Syndrome 3          1\n",
       "27757          placenta  Pulmonary Disease, Chronic Obstructive          1\n",
       "\n",
       "[27758 rows x 3 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query = \"\"\"SELECT tissue_name, disease_name, COUNT(*) AS COUNT_COL FROM geo.datasets \n",
    "    CROSS JOIN UNNEST(tissue) AS t(tissue_name) \n",
    "    CROSS JOIN UNNEST (disease) AS t(disease_name)\n",
    "    GROUP BY tissue_name, disease_name\n",
    "    ORDER BY COUNT_COL desc\"\"\"\n",
    "results=omixatlas.query_metadata(query, query_api_version=\"v2\")\n",
    "results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Fetch data matrix for a dataset_ID of interest"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Query execution succeeded (time taken: 1.26 seconds, data scanned: 0.922 MB)\n",
      "Fetched 20736 rows\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rid</th>\n",
       "      <th>gsm96563</th>\n",
       "      <th>gsm96564</th>\n",
       "      <th>gsm96565</th>\n",
       "      <th>gsm96566</th>\n",
       "      <th>gsm96567</th>\n",
       "      <th>gsm96568</th>\n",
       "      <th>gsm96569</th>\n",
       "      <th>gsm96570</th>\n",
       "      <th>row_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Hbb-y</td>\n",
       "      <td>12.7775</td>\n",
       "      <td>12.6816</td>\n",
       "      <td>12.8115</td>\n",
       "      <td>12.7138</td>\n",
       "      <td>4.6548</td>\n",
       "      <td>4.5429</td>\n",
       "      <td>4.4265</td>\n",
       "      <td>4.3696</td>\n",
       "      <td>hbb_y</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Bbip1</td>\n",
       "      <td>12.6214</td>\n",
       "      <td>12.5121</td>\n",
       "      <td>12.3345</td>\n",
       "      <td>12.6402</td>\n",
       "      <td>9.5737</td>\n",
       "      <td>9.6951</td>\n",
       "      <td>10.6927</td>\n",
       "      <td>10.3277</td>\n",
       "      <td>bbip1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Actb</td>\n",
       "      <td>13.9787</td>\n",
       "      <td>13.8807</td>\n",
       "      <td>13.9282</td>\n",
       "      <td>14.0340</td>\n",
       "      <td>12.8278</td>\n",
       "      <td>13.0275</td>\n",
       "      <td>13.6720</td>\n",
       "      <td>12.5043</td>\n",
       "      <td>actb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Fth1</td>\n",
       "      <td>11.6626</td>\n",
       "      <td>11.2601</td>\n",
       "      <td>11.9009</td>\n",
       "      <td>11.6864</td>\n",
       "      <td>12.9458</td>\n",
       "      <td>13.1211</td>\n",
       "      <td>12.8372</td>\n",
       "      <td>13.0097</td>\n",
       "      <td>fth1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Rpgrip1</td>\n",
       "      <td>12.7654</td>\n",
       "      <td>12.7127</td>\n",
       "      <td>12.2721</td>\n",
       "      <td>12.6098</td>\n",
       "      <td>10.0882</td>\n",
       "      <td>10.1332</td>\n",
       "      <td>10.6733</td>\n",
       "      <td>10.4763</td>\n",
       "      <td>rpgrip1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20731</th>\n",
       "      <td>Snapc5</td>\n",
       "      <td>4.9242</td>\n",
       "      <td>5.0460</td>\n",
       "      <td>4.7005</td>\n",
       "      <td>4.6873</td>\n",
       "      <td>4.8461</td>\n",
       "      <td>4.8823</td>\n",
       "      <td>4.5441</td>\n",
       "      <td>4.8709</td>\n",
       "      <td>snapc5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20732</th>\n",
       "      <td>A430035B10Rik</td>\n",
       "      <td>6.4076</td>\n",
       "      <td>6.3439</td>\n",
       "      <td>6.1458</td>\n",
       "      <td>6.4500</td>\n",
       "      <td>6.4574</td>\n",
       "      <td>6.3879</td>\n",
       "      <td>6.4610</td>\n",
       "      <td>6.6280</td>\n",
       "      <td>a430035b10rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20733</th>\n",
       "      <td>Vmn1r172</td>\n",
       "      <td>5.7065</td>\n",
       "      <td>5.5740</td>\n",
       "      <td>5.4857</td>\n",
       "      <td>5.4360</td>\n",
       "      <td>5.4923</td>\n",
       "      <td>5.5236</td>\n",
       "      <td>5.5771</td>\n",
       "      <td>5.3925</td>\n",
       "      <td>vmn1r172</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20734</th>\n",
       "      <td>Gm16845</td>\n",
       "      <td>6.1398</td>\n",
       "      <td>6.1311</td>\n",
       "      <td>6.3071</td>\n",
       "      <td>5.8718</td>\n",
       "      <td>6.3268</td>\n",
       "      <td>6.2849</td>\n",
       "      <td>6.2742</td>\n",
       "      <td>6.2801</td>\n",
       "      <td>gm16845</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20735</th>\n",
       "      <td>D19Ertd200e</td>\n",
       "      <td>5.1695</td>\n",
       "      <td>5.1267</td>\n",
       "      <td>5.4580</td>\n",
       "      <td>5.2017</td>\n",
       "      <td>4.9188</td>\n",
       "      <td>4.8641</td>\n",
       "      <td>5.2946</td>\n",
       "      <td>5.1589</td>\n",
       "      <td>d19ertd200e</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>20736 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 rid  gsm96563  gsm96564  gsm96565  gsm96566  gsm96567  \\\n",
       "0              Hbb-y   12.7775   12.6816   12.8115   12.7138    4.6548   \n",
       "1              Bbip1   12.6214   12.5121   12.3345   12.6402    9.5737   \n",
       "2               Actb   13.9787   13.8807   13.9282   14.0340   12.8278   \n",
       "3               Fth1   11.6626   11.2601   11.9009   11.6864   12.9458   \n",
       "4            Rpgrip1   12.7654   12.7127   12.2721   12.6098   10.0882   \n",
       "...              ...       ...       ...       ...       ...       ...   \n",
       "20731         Snapc5    4.9242    5.0460    4.7005    4.6873    4.8461   \n",
       "20732  A430035B10Rik    6.4076    6.3439    6.1458    6.4500    6.4574   \n",
       "20733       Vmn1r172    5.7065    5.5740    5.4857    5.4360    5.4923   \n",
       "20734        Gm16845    6.1398    6.1311    6.3071    5.8718    6.3268   \n",
       "20735    D19Ertd200e    5.1695    5.1267    5.4580    5.2017    4.9188   \n",
       "\n",
       "       gsm96568  gsm96569  gsm96570         row_id  \n",
       "0        4.5429    4.4265    4.3696          hbb_y  \n",
       "1        9.6951   10.6927   10.3277          bbip1  \n",
       "2       13.0275   13.6720   12.5043           actb  \n",
       "3       13.1211   12.8372   13.0097           fth1  \n",
       "4       10.1332   10.6733   10.4763        rpgrip1  \n",
       "...         ...       ...       ...            ...  \n",
       "20731    4.8823    4.5441    4.8709         snapc5  \n",
       "20732    6.3879    6.4610    6.6280  a430035b10rik  \n",
       "20733    5.5236    5.5771    5.3925       vmn1r172  \n",
       "20734    6.2849    6.2742    6.2801        gm16845  \n",
       "20735    4.8641    5.2946    5.1589    d19ertd200e  \n",
       "\n",
       "[20736 rows x 10 columns]"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query = \"SELECT * FROM data_matrices.geo__GSE4230_GPL1261\"\n",
    "results=omixatlas.query_metadata(query, query_api_version=\"v2\")\n",
    "results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Fetch data matrix for selected genes for a dataset_ID of interest"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Query execution succeeded (time taken: 1.04 seconds, data scanned: 0.922 MB)\n",
      "Fetched 4 rows\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rid</th>\n",
       "      <th>gsm96563</th>\n",
       "      <th>gsm96564</th>\n",
       "      <th>gsm96565</th>\n",
       "      <th>gsm96566</th>\n",
       "      <th>gsm96567</th>\n",
       "      <th>gsm96568</th>\n",
       "      <th>gsm96569</th>\n",
       "      <th>gsm96570</th>\n",
       "      <th>row_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Hbb-y</td>\n",
       "      <td>12.7775</td>\n",
       "      <td>12.6816</td>\n",
       "      <td>12.8115</td>\n",
       "      <td>12.7138</td>\n",
       "      <td>4.6548</td>\n",
       "      <td>4.5429</td>\n",
       "      <td>4.4265</td>\n",
       "      <td>4.3696</td>\n",
       "      <td>hbb_y</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Bbip1</td>\n",
       "      <td>12.6214</td>\n",
       "      <td>12.5121</td>\n",
       "      <td>12.3345</td>\n",
       "      <td>12.6402</td>\n",
       "      <td>9.5737</td>\n",
       "      <td>9.6951</td>\n",
       "      <td>10.6927</td>\n",
       "      <td>10.3277</td>\n",
       "      <td>bbip1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Actb</td>\n",
       "      <td>13.9787</td>\n",
       "      <td>13.8807</td>\n",
       "      <td>13.9282</td>\n",
       "      <td>14.0340</td>\n",
       "      <td>12.8278</td>\n",
       "      <td>13.0275</td>\n",
       "      <td>13.6720</td>\n",
       "      <td>12.5043</td>\n",
       "      <td>actb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Fth1</td>\n",
       "      <td>11.6626</td>\n",
       "      <td>11.2601</td>\n",
       "      <td>11.9009</td>\n",
       "      <td>11.6864</td>\n",
       "      <td>12.9458</td>\n",
       "      <td>13.1211</td>\n",
       "      <td>12.8372</td>\n",
       "      <td>13.0097</td>\n",
       "      <td>fth1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     rid  gsm96563  gsm96564  gsm96565  gsm96566  gsm96567  gsm96568  \\\n",
       "0  Hbb-y   12.7775   12.6816   12.8115   12.7138    4.6548    4.5429   \n",
       "1  Bbip1   12.6214   12.5121   12.3345   12.6402    9.5737    9.6951   \n",
       "2   Actb   13.9787   13.8807   13.9282   14.0340   12.8278   13.0275   \n",
       "3   Fth1   11.6626   11.2601   11.9009   11.6864   12.9458   13.1211   \n",
       "\n",
       "   gsm96569  gsm96570 row_id  \n",
       "0    4.4265    4.3696  hbb_y  \n",
       "1   10.6927   10.3277  bbip1  \n",
       "2   13.6720   12.5043   actb  \n",
       "3   12.8372   13.0097   fth1  "
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene = ('hbb-y', 'fth1', 'bbip1', 'actb')\n",
    "query = f\"SELECT * FROM data_matrices.geo__GSE4230_GPL1261 WHERE LOWER(rid) IN {gene}\"\n",
    "results=omixatlas.query_metadata(query, query_api_version=\"v2\")\n",
    "results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get dataset_IDs where disease is obesity and organism can be either human or rat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Query execution succeeded (time taken: 1.73 seconds, data scanned: 0.732 MB)\n",
      "Fetched 336 rows\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dataset_id</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>disease_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GSE36295_GPL6244</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>Obesity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GSE22540_GPL10582</td>\n",
       "      <td>Rattus norvegicus</td>\n",
       "      <td>Obesity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GSE119955_GPL4135</td>\n",
       "      <td>Rattus norvegicus</td>\n",
       "      <td>Obesity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GSE76490_GPL11154</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>Obesity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GSE159809_GPL13497</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>Obesity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>331</th>\n",
       "      <td>GSE113819_GPL17586</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>Obesity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>332</th>\n",
       "      <td>GSE22857_GPL1456</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>Obesity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333</th>\n",
       "      <td>GSE73385_GPL17586</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>Obesity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334</th>\n",
       "      <td>GSE102293_GPL570</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>Obesity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>335</th>\n",
       "      <td>GSE22070_GPL6947</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>Obesity</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>336 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             dataset_id      organism_name disease_name\n",
       "0      GSE36295_GPL6244       Homo sapiens      Obesity\n",
       "1     GSE22540_GPL10582  Rattus norvegicus      Obesity\n",
       "2     GSE119955_GPL4135  Rattus norvegicus      Obesity\n",
       "3     GSE76490_GPL11154       Homo sapiens      Obesity\n",
       "4    GSE159809_GPL13497       Homo sapiens      Obesity\n",
       "..                  ...                ...          ...\n",
       "331  GSE113819_GPL17586       Homo sapiens      Obesity\n",
       "332    GSE22857_GPL1456       Homo sapiens      Obesity\n",
       "333   GSE73385_GPL17586       Homo sapiens      Obesity\n",
       "334    GSE102293_GPL570       Homo sapiens      Obesity\n",
       "335    GSE22070_GPL6947       Homo sapiens      Obesity\n",
       "\n",
       "[336 rows x 3 columns]"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query = \"\"\"SELECT dataset_id, organism_name, disease_name FROM geo.datasets \n",
    "    CROSS JOIN UNNEST(organism) AS t(organism_name) \n",
    "    CROSS JOIN UNNEST (disease) AS t(disease_name) \n",
    "    WHERE LOWER (disease_name) LIKE 'obesity' AND (LOWER(organism_name) LIKE 'homo sapiens' OR LOWER (organism_name) LIKE 'rattus norvegicus')\"\"\"\n",
    "results=omixatlas.query_metadata(query, query_api_version=\"v2\")\n",
    "results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## JOIN Queries"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get microarray datasets and their samples where the ‘HeLa’ cell line was used on a microarray platform"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Query execution succeeded (time taken: 2.76 seconds, data scanned: 15.896 MB)\n",
      "Fetched 1139 rows\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dataset_id</th>\n",
       "      <th>sample_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GSE101098_GPL16686</td>\n",
       "      <td>GSM2699149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GSE101098_GPL16686</td>\n",
       "      <td>GSM2699151</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GSE101098_GPL16686</td>\n",
       "      <td>GSM2699150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GSE101098_GPL16686</td>\n",
       "      <td>GSM2699148</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GSE101098_GPL16686</td>\n",
       "      <td>GSM2699152</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1134</th>\n",
       "      <td>GSE96671_GPL4133</td>\n",
       "      <td>GSM2537861</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1135</th>\n",
       "      <td>GSE96671_GPL4133</td>\n",
       "      <td>GSM2537864</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1136</th>\n",
       "      <td>GSE96671_GPL4133</td>\n",
       "      <td>GSM2537856</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1137</th>\n",
       "      <td>GSE96671_GPL4133</td>\n",
       "      <td>GSM2537858</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1138</th>\n",
       "      <td>GSE96671_GPL4133</td>\n",
       "      <td>GSM2537868</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1139 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              dataset_id sample_name\n",
       "0     GSE101098_GPL16686  GSM2699149\n",
       "1     GSE101098_GPL16686  GSM2699151\n",
       "2     GSE101098_GPL16686  GSM2699150\n",
       "3     GSE101098_GPL16686  GSM2699148\n",
       "4     GSE101098_GPL16686  GSM2699152\n",
       "...                  ...         ...\n",
       "1134    GSE96671_GPL4133  GSM2537861\n",
       "1135    GSE96671_GPL4133  GSM2537864\n",
       "1136    GSE96671_GPL4133  GSM2537856\n",
       "1137    GSE96671_GPL4133  GSM2537858\n",
       "1138    GSE96671_GPL4133  GSM2537868\n",
       "\n",
       "[1139 rows x 2 columns]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Get microarray datasets and their samples where the ‘HeLa’ cell line was used on a microarray platform\n",
    "\n",
    "query = \"\"\"  SELECT datasets.dataset_id AS dataset_id, sample_name AS sample_name FROM geo.datasets AS datasets\n",
    "    JOIN (\n",
    "        SELECT src_dataset_id AS dataset_id, name AS sample_name FROM geo.samples WHERE CONTAINS(cell_line, 'HeLa')\n",
    "         ) AS hela_samples\n",
    "    ON datasets.dataset_id = hela_samples.dataset_id WHERE LOWER(platform) = 'microarray'\n",
    "    ORDER BY datasets.dataset_id\"\"\"\n",
    "results=omixatlas.query_metadata(query, query_api_version=\"v2\")\n",
    "results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Queries using a sub-query"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Query execution succeeded (time taken: 2.73 seconds, data scanned: 15.896 MB)\n",
      "Fetched 1139 rows\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dataset_id</th>\n",
       "      <th>sample_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GSE75365_GPL19803</td>\n",
       "      <td>GSM1953950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GSE93568_GPL13252</td>\n",
       "      <td>GSM2454773</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GSE26781_GPL6883</td>\n",
       "      <td>GSM659010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GSE164406_GPL21185</td>\n",
       "      <td>GSM4998752</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GSE35823_GPL571</td>\n",
       "      <td>GSM875741</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1134</th>\n",
       "      <td>GSE75365_GPL19803</td>\n",
       "      <td>GSM1953948</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1135</th>\n",
       "      <td>GSE53103_GPL570</td>\n",
       "      <td>GSM1282422</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1136</th>\n",
       "      <td>GSE36509_GPL6244</td>\n",
       "      <td>GSM895994</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1137</th>\n",
       "      <td>GSE53103_GPL570</td>\n",
       "      <td>GSM1282441</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1138</th>\n",
       "      <td>GSE53103_GPL570</td>\n",
       "      <td>GSM1282401</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1139 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              dataset_id sample_name\n",
       "0      GSE75365_GPL19803  GSM1953950\n",
       "1      GSE93568_GPL13252  GSM2454773\n",
       "2       GSE26781_GPL6883   GSM659010\n",
       "3     GSE164406_GPL21185  GSM4998752\n",
       "4        GSE35823_GPL571   GSM875741\n",
       "...                  ...         ...\n",
       "1134   GSE75365_GPL19803  GSM1953948\n",
       "1135     GSE53103_GPL570  GSM1282422\n",
       "1136    GSE36509_GPL6244   GSM895994\n",
       "1137     GSE53103_GPL570  GSM1282441\n",
       "1138     GSE53103_GPL570  GSM1282401\n",
       "\n",
       "[1139 rows x 2 columns]"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query = \"\"\"SELECT src_dataset_id AS dataset_id, name AS sample_name FROM geo.samples\n",
    "    WHERE CONTAINS (cell_line, 'HeLa') AND src_dataset_id \n",
    "    IN (SELECT dataset_id FROM geo.datasets WHERE LOWER (platform) = 'microarray')\"\"\"\n",
    "results=omixatlas.query_metadata(query, query_api_version=\"v2\")\n",
    "results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
