{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Welcome to Polly Python3 Notebook."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/bin/bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)\n",
      "Defaulting to user installation because normal site-packages is not writeable\n",
      "Looking in indexes: https://pypi.org/simple, http://54.245.179.143:80/\n",
      "Requirement already satisfied: polly-python in /usr/local/lib/python3.10/site-packages (0.3.0)\n",
      "Collecting polly-python\n",
      "  Downloading polly_python-0.4.0-py3-none-any.whl (130 kB)\n",
      "\u001b[K     |████████████████████████████████| 130 kB 18.1 MB/s eta 0:00:01\n",
      "\u001b[?25hCollecting botocore>=1.27.0\n",
      "  Downloading botocore-1.31.9-py3-none-any.whl (11.0 MB)\n",
      "\u001b[K     |████████████████████████████████| 11.0 MB 30.5 MB/s eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: python-jose==3.3.0 in /usr/local/lib/python3.10/site-packages (from polly-python) (3.3.0)\n",
      "Requirement already satisfied: tqdm==4.65.0 in /usr/local/lib/python3.10/site-packages (from polly-python) (4.65.0)\n",
      "Requirement already satisfied: elucidatacmapPy==3.3.4 in /usr/local/lib/python3.10/site-packages (from polly-python) (3.3.4)\n",
      "Requirement already satisfied: beautifulsoup4 in /usr/local/lib/python3.10/site-packages (from polly-python) (4.12.2)\n",
      "Requirement already satisfied: plotly>=5.0.0 in /usr/local/lib/python3.10/site-packages (from polly-python) (5.15.0)\n",
      "Requirement already satisfied: pytest>=6.2.5 in /usr/local/lib/python3.10/site-packages (from polly-python) (7.4.0)\n",
      "Requirement already satisfied: requests==2.25.1 in /usr/local/lib/python3.10/site-packages (from polly-python) (2.25.1)\n",
      "Requirement already satisfied: mixpanel==4.10.0 in /usr/local/lib/python3.10/site-packages (from polly-python) (4.10.0)\n",
      "Collecting boto3>=1.24.0\n",
      "  Downloading boto3-1.28.9-py3-none-any.whl (135 kB)\n",
      "\u001b[K     |████████████████████████████████| 135 kB 73.6 MB/s eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: datapane>=0.10.3 in /usr/local/lib/python3.10/site-packages (from polly-python) (0.16.5)\n",
      "Requirement already satisfied: retrying==1.3.4 in /usr/local/lib/python3.10/site-packages (from polly-python) (1.3.4)\n",
      "Collecting cryptography<=38.0.0,>=37.0.1\n",
      "  Downloading cryptography-38.0.0-cp36-abi3-manylinux_2_28_x86_64.whl (4.2 MB)\n",
      "\u001b[K     |████████████████████████████████| 4.2 MB 67.0 MB/s eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: elucidatacloudpathlib==0.6.6 in /usr/local/lib/python3.10/site-packages (from polly-python) (0.6.6)\n",
      "Requirement already satisfied: Deprecated>=1.2.12 in /usr/local/lib/python3.10/site-packages (from polly-python) (1.2.14)\n",
      "Collecting polly-validator==0.0.2\n",
      "  Downloading polly_validator-0.0.2-py3-none-any.whl (29 kB)\n",
      "Requirement already satisfied: rst2txt==1.1.0 in /usr/local/lib/python3.10/site-packages (from polly-python) (1.1.0)\n",
      "Requirement already satisfied: joblib>=1.2.0 in /usr/local/lib/python3.10/site-packages (from polly-python) (1.3.1)\n",
      "Requirement already satisfied: pydantic>=1.8.2 in /usr/local/lib/python3.10/site-packages (from polly-python) (1.10.11)\n",
      "Requirement already satisfied: pandas>=1.3.5 in /usr/local/lib/python3.10/site-packages (from polly-python) (1.4.1)\n",
      "Requirement already satisfied: h5py>=2.6.0 in /usr/local/lib/python3.10/site-packages (from elucidatacmapPy==3.3.4->polly-python) (3.9.0)\n",
      "Requirement already satisfied: six in /usr/local/lib/python3.10/site-packages (from elucidatacmapPy==3.3.4->polly-python) (1.16.0)\n",
      "Requirement already satisfied: numpy>=1.11.2 in /usr/local/lib/python3.10/site-packages (from elucidatacmapPy==3.3.4->polly-python) (1.22.3)\n",
      "Requirement already satisfied: urllib3 in /usr/local/lib/python3.10/site-packages (from mixpanel==4.10.0->polly-python) (1.26.6)\n",
      "Requirement already satisfied: appdirs==1.4.4 in /usr/local/lib/python3.10/site-packages (from polly-validator==0.0.2->polly-python) (1.4.4)\n",
      "Requirement already satisfied: rapidfuzz==2.1.1 in /usr/local/lib/python3.10/site-packages (from polly-validator==0.0.2->polly-python) (2.1.1)\n",
      "Requirement already satisfied: pronto in /usr/local/lib/python3.10/site-packages (from polly-validator==0.0.2->polly-python) (2.5.1)\n",
      "Requirement already satisfied: tabulate==0.8.10 in /usr/local/lib/python3.10/site-packages (from polly-validator==0.0.2->polly-python) (0.8.10)\n",
      "Requirement already satisfied: pyasn1 in /usr/local/lib/python3.10/site-packages (from python-jose==3.3.0->polly-python) (0.4.8)\n",
      "Requirement already satisfied: rsa in /usr/local/lib/python3.10/site-packages (from python-jose==3.3.0->polly-python) (4.7.2)\n",
      "Requirement already satisfied: ecdsa!=0.15 in /usr/local/lib/python3.10/site-packages (from python-jose==3.3.0->polly-python) (0.18.0)\n",
      "Requirement already satisfied: jarowinkler<1.1.0,>=1.0.3 in /usr/local/lib/python3.10/site-packages (from rapidfuzz==2.1.1->polly-validator==0.0.2->polly-python) (1.0.5)\n",
      "Requirement already satisfied: chardet<5,>=3.0.2 in /usr/local/lib/python3.10/site-packages (from requests==2.25.1->polly-python) (4.0.0)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.10/site-packages (from requests==2.25.1->polly-python) (2021.5.30)\n",
      "Requirement already satisfied: idna<3,>=2.5 in /usr/local/lib/python3.10/site-packages (from requests==2.25.1->polly-python) (2.10)\n",
      "Requirement already satisfied: pygments in /usr/local/lib/python3.10/site-packages (from rst2txt==1.1.0->polly-python) (2.10.0)\n",
      "Requirement already satisfied: docutils>=0.11 in /usr/local/lib/python3.10/site-packages (from rst2txt==1.1.0->polly-python) (0.15.2)\n",
      "Requirement already satisfied: jmespath<2.0.0,>=0.7.1 in /usr/local/lib/python3.10/site-packages (from boto3>=1.24.0->polly-python) (0.10.0)\n",
      "Collecting s3transfer<0.7.0,>=0.6.0\n",
      "  Downloading s3transfer-0.6.1-py3-none-any.whl (79 kB)\n",
      "\u001b[K     |████████████████████████████████| 79 kB 17.5 MB/s eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: python-dateutil<3.0.0,>=2.1 in /usr/local/lib/python3.10/site-packages (from botocore>=1.27.0->polly-python) (2.8.1)\n",
      "Requirement already satisfied: cffi>=1.12 in /usr/local/lib/python3.10/site-packages (from cryptography<=38.0.0,>=37.0.1->polly-python) (1.15.1)\n",
      "Requirement already satisfied: pycparser in /usr/local/lib/python3.10/site-packages (from cffi>=1.12->cryptography<=38.0.0,>=37.0.1->polly-python) (2.21)\n",
      "Requirement already satisfied: dulwich<0.22.0,>=0.20.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python) (0.21.5)\n",
      "Requirement already satisfied: datacommons<2.0.0,>=1.4.3 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python) (1.4.3)\n",
      "Requirement already satisfied: lxml<5.0.0,>=4.0.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python) (4.9.2)\n",
      "Requirement already satisfied: glom<24.0.0,>=22.1.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python) (23.3.0)\n",
      "Requirement already satisfied: altair<5.0.0,>=4.0.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python) (4.2.2)\n",
      "Requirement already satisfied: click<9.0.0,>=7.1.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python) (8.1.3)\n",
      "Requirement already satisfied: nbconvert>=5.6.1 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python) (6.3.0)\n",
      "Requirement already satisfied: dacite<2.0.0,>=1.0.2 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python) (1.8.1)\n",
      "Requirement already satisfied: click-spinner<0.2.0,>=0.1.8 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python) (0.1.10)\n",
      "Requirement already satisfied: cheroot<10.0.0,>=9.0.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python) (9.0.0)\n",
      "Requirement already satisfied: furl<3.0.0,>=2.0.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python) (2.1.3)\n",
      "Requirement already satisfied: multimethod<2.0,>=1.9 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python) (1.9.1)\n",
      "Requirement already satisfied: PyYAML<7.0.0,>=5.4.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python) (5.4.1)\n",
      "Requirement already satisfied: micawber>=0.5.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python) (0.5.5)\n",
      "Requirement already satisfied: vega-datasets<1.0.0,>=0.9.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python) (0.9.0)\n",
      "Requirement already satisfied: posthog<3.0.0,>=1.4.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python) (2.5.0)\n",
      "Requirement already satisfied: typing-extensions<5.0.0,>=4.4.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python) (4.7.1)\n",
      "Requirement already satisfied: pyngrok<6.0.0,>=5.2.1 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python) (5.2.3)\n",
      "Requirement already satisfied: requests-toolbelt<1.1.0,>=0.9.1 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python) (1.0.0)\n",
      "Requirement already satisfied: boltons<22.0.0,>=20.0.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python) (21.0.0)\n",
      "Requirement already satisfied: packaging<24.0.0,>=21.0.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python) (23.1)\n",
      "Requirement already satisfied: pyarrow<11.0.0,>=6.0.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python) (10.0.1)\n",
      "Requirement already satisfied: toolz<0.13.0,>=0.11.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python) (0.12.0)\n",
      "Requirement already satisfied: jsonschema<5.0.0,>=3.2.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python) (4.2.1)\n",
      "Requirement already satisfied: importlib_resources<6.0.0,>=3.0.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python) (5.12.0)\n",
      "Requirement already satisfied: datacommons-pandas<0.0.4,>=0.0.3 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python) (0.0.3)\n",
      "Requirement already satisfied: ipynbname<2022.0.0,>=2021.3.2 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python) (2021.3.2)\n",
      "Requirement already satisfied: dominate<3.0.0,>=2.4.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python) (2.8.0)\n",
      "Requirement already satisfied: colorlog<7.0.0,>=4.1.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python) (6.7.0)\n",
      "Requirement already satisfied: entrypoints in /usr/local/lib/python3.10/site-packages (from altair<5.0.0,>=4.0.0->datapane>=0.10.3->polly-python) (0.3)\n",
      "Requirement already satisfied: jinja2 in /usr/local/lib/python3.10/site-packages (from altair<5.0.0,>=4.0.0->datapane>=0.10.3->polly-python) (3.0.3)\n",
      "Requirement already satisfied: jaraco.functools in /usr/local/lib/python3.10/site-packages (from cheroot<10.0.0,>=9.0.0->datapane>=0.10.3->polly-python) (3.8.0)\n",
      "Requirement already satisfied: more-itertools>=2.6 in /usr/local/lib/python3.10/site-packages (from cheroot<10.0.0,>=9.0.0->datapane>=0.10.3->polly-python) (9.1.0)\n",
      "Requirement already satisfied: wrapt<2,>=1.10 in /usr/local/lib/python3.10/site-packages (from Deprecated>=1.2.12->polly-python) (1.15.0)\n",
      "Requirement already satisfied: orderedmultidict>=1.0.1 in /usr/local/lib/python3.10/site-packages (from furl<3.0.0,>=2.0.0->datapane>=0.10.3->polly-python) (1.0.1)\n",
      "Requirement already satisfied: attrs in /usr/local/lib/python3.10/site-packages (from glom<24.0.0,>=22.1.0->datapane>=0.10.3->polly-python) (21.2.0)\n",
      "Requirement already satisfied: face==20.1.1 in /usr/local/lib/python3.10/site-packages (from glom<24.0.0,>=22.1.0->datapane>=0.10.3->polly-python) (20.1.1)\n",
      "Requirement already satisfied: ipykernel in /usr/local/lib/python3.10/site-packages (from ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python) (6.13.1)\n",
      "Requirement already satisfied: pyrsistent!=0.17.0,!=0.17.1,!=0.17.2,>=0.14.0 in /usr/local/lib/python3.10/site-packages (from jsonschema<5.0.0,>=3.2.0->datapane>=0.10.3->polly-python) (0.18.0)\n",
      "Requirement already satisfied: defusedxml in /usr/local/lib/python3.10/site-packages (from nbconvert>=5.6.1->datapane>=0.10.3->polly-python) (0.7.1)\n",
      "Requirement already satisfied: jupyterlab-pygments in /usr/local/lib/python3.10/site-packages (from nbconvert>=5.6.1->datapane>=0.10.3->polly-python) (0.2.2)\n",
      "Requirement already satisfied: nbformat>=4.4 in /usr/local/lib/python3.10/site-packages (from nbconvert>=5.6.1->datapane>=0.10.3->polly-python) (5.1.3)\n",
      "Requirement already satisfied: bleach in /usr/local/lib/python3.10/site-packages (from nbconvert>=5.6.1->datapane>=0.10.3->polly-python) (4.1.0)\n",
      "Requirement already satisfied: nbclient<0.6.0,>=0.5.0 in /usr/local/lib/python3.10/site-packages (from nbconvert>=5.6.1->datapane>=0.10.3->polly-python) (0.5.13)\n",
      "Requirement already satisfied: pandocfilters>=1.4.1 in /usr/local/lib/python3.10/site-packages (from nbconvert>=5.6.1->datapane>=0.10.3->polly-python) (1.5.0)\n",
      "Requirement already satisfied: traitlets>=5.0 in /usr/local/lib/python3.10/site-packages (from nbconvert>=5.6.1->datapane>=0.10.3->polly-python) (5.1.1)\n",
      "Requirement already satisfied: mistune<2,>=0.8.1 in /usr/local/lib/python3.10/site-packages (from nbconvert>=5.6.1->datapane>=0.10.3->polly-python) (0.8.4)\n",
      "Requirement already satisfied: testpath in /usr/local/lib/python3.10/site-packages (from nbconvert>=5.6.1->datapane>=0.10.3->polly-python) (0.5.0)\n",
      "Requirement already satisfied: jupyter-core in /usr/local/lib/python3.10/site-packages (from nbconvert>=5.6.1->datapane>=0.10.3->polly-python) (4.9.1)\n",
      "Requirement already satisfied: MarkupSafe>=2.0 in /usr/local/lib/python3.10/site-packages (from jinja2->altair<5.0.0,>=4.0.0->datapane>=0.10.3->polly-python) (2.0.1)\n",
      "Requirement already satisfied: jupyter-client>=6.1.5 in /usr/local/lib/python3.10/site-packages (from nbclient<0.6.0,>=0.5.0->nbconvert>=5.6.1->datapane>=0.10.3->polly-python) (6.1.12)\n",
      "Requirement already satisfied: nest-asyncio in /usr/local/lib/python3.10/site-packages (from nbclient<0.6.0,>=0.5.0->nbconvert>=5.6.1->datapane>=0.10.3->polly-python) (1.5.6)\n",
      "Requirement already satisfied: pyzmq>=13 in /usr/local/lib/python3.10/site-packages (from jupyter-client>=6.1.5->nbclient<0.6.0,>=0.5.0->nbconvert>=5.6.1->datapane>=0.10.3->polly-python) (22.3.0)\n",
      "Requirement already satisfied: tornado>=4.1 in /usr/local/lib/python3.10/site-packages (from jupyter-client>=6.1.5->nbclient<0.6.0,>=0.5.0->nbconvert>=5.6.1->datapane>=0.10.3->polly-python) (6.1)\n",
      "Requirement already satisfied: ipython-genutils in /usr/local/lib/python3.10/site-packages (from nbformat>=4.4->nbconvert>=5.6.1->datapane>=0.10.3->polly-python) (0.2.0)\n",
      "Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.10/site-packages (from pandas>=1.3.5->polly-python) (2021.3)\n",
      "Requirement already satisfied: tenacity>=6.2.0 in /usr/local/lib/python3.10/site-packages (from plotly>=5.0.0->polly-python) (8.2.2)\n",
      "Requirement already satisfied: backoff>=1.10.0 in /usr/local/lib/python3.10/site-packages (from posthog<3.0.0,>=1.4.0->datapane>=0.10.3->polly-python) (2.2.1)\n",
      "Requirement already satisfied: monotonic>=1.5 in /usr/local/lib/python3.10/site-packages (from posthog<3.0.0,>=1.4.0->datapane>=0.10.3->polly-python) (1.6)\n",
      "Requirement already satisfied: tomli>=1.0.0 in /usr/local/lib/python3.10/site-packages (from pytest>=6.2.5->polly-python) (2.0.1)\n",
      "Requirement already satisfied: pluggy<2.0,>=0.12 in /usr/local/lib/python3.10/site-packages (from pytest>=6.2.5->polly-python) (1.2.0)\n",
      "Requirement already satisfied: exceptiongroup>=1.0.0rc8 in /usr/local/lib/python3.10/site-packages (from pytest>=6.2.5->polly-python) (1.1.2)\n",
      "Requirement already satisfied: iniconfig in /usr/local/lib/python3.10/site-packages (from pytest>=6.2.5->polly-python) (2.0.0)\n",
      "Requirement already satisfied: soupsieve>1.2 in /usr/local/lib/python3.10/site-packages (from beautifulsoup4->polly-python) (2.4.1)\n",
      "Requirement already satisfied: webencodings in /usr/local/lib/python3.10/site-packages (from bleach->nbconvert>=5.6.1->datapane>=0.10.3->polly-python) (0.5.1)\n",
      "Requirement already satisfied: psutil in /usr/local/lib/python3.10/site-packages (from ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python) (5.9.5)\n",
      "Requirement already satisfied: matplotlib-inline>=0.1 in /usr/local/lib/python3.10/site-packages (from ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python) (0.1.3)\n",
      "Requirement already satisfied: ipython>=7.23.1 in /usr/local/lib/python3.10/site-packages (from ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python) (8.2.0)\n",
      "Requirement already satisfied: debugpy>=1.0 in /usr/local/lib/python3.10/site-packages (from ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python) (1.6.7)\n",
      "Requirement already satisfied: pickleshare in /usr/local/lib/python3.10/site-packages (from ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python) (0.7.5)\n",
      "Requirement already satisfied: setuptools>=18.5 in /usr/local/lib/python3.10/site-packages (from ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python) (58.1.0)\n",
      "Requirement already satisfied: decorator in /usr/local/lib/python3.10/site-packages (from ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python) (5.1.0)\n",
      "Requirement already satisfied: backcall in /usr/local/lib/python3.10/site-packages (from ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python) (0.2.0)\n",
      "Requirement already satisfied: prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0 in /usr/local/lib/python3.10/site-packages (from ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python) (3.0.22)\n",
      "Requirement already satisfied: jedi>=0.16 in /usr/local/lib/python3.10/site-packages (from ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python) (0.18.1)\n",
      "Requirement already satisfied: pexpect>4.3 in /usr/local/lib/python3.10/site-packages (from ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python) (4.8.0)\n",
      "Requirement already satisfied: stack-data in /usr/local/lib/python3.10/site-packages (from ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python) (0.6.2)\n",
      "Requirement already satisfied: parso<0.9.0,>=0.8.0 in /usr/local/lib/python3.10/site-packages (from jedi>=0.16->ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python) (0.8.2)\n",
      "Requirement already satisfied: ptyprocess>=0.5 in /usr/local/lib/python3.10/site-packages (from pexpect>4.3->ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python) (0.7.0)\n",
      "Requirement already satisfied: wcwidth in /usr/local/lib/python3.10/site-packages (from prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0->ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python) (0.2.5)\n",
      "Requirement already satisfied: fastobo~=0.12.1 in /usr/local/lib/python3.10/site-packages (from pronto->polly-validator==0.0.2->polly-python) (0.12.2)\n",
      "Requirement already satisfied: networkx~=2.3 in /usr/local/lib/python3.10/site-packages (from pronto->polly-validator==0.0.2->polly-python) (2.8.8)\n",
      "Requirement already satisfied: pure-eval in /usr/local/lib/python3.10/site-packages (from stack-data->ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python) (0.2.2)\n",
      "Requirement already satisfied: asttokens>=2.1.0 in /usr/local/lib/python3.10/site-packages (from stack-data->ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python) (2.2.1)\n",
      "Requirement already satisfied: executing>=1.2.0 in /usr/local/lib/python3.10/site-packages (from stack-data->ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python) (1.2.0)\n",
      "Installing collected packages: botocore, s3transfer, polly-validator, cryptography, boto3, polly-python\n",
      "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
      "awscli 1.22.74 requires botocore==1.24.19, but you have botocore 1.31.9 which is incompatible.\n",
      "awscli 1.22.74 requires s3transfer<0.6.0,>=0.5.0, but you have s3transfer 0.6.1 which is incompatible.\u001b[0m\n",
      "Successfully installed boto3-1.28.9 botocore-1.31.9 cryptography-38.0.0 polly-python-0.4.0 polly-validator-0.0.2 s3transfer-0.6.1\n",
      "\u001b[33mWARNING: You are using pip version 21.2.4; however, version 23.2.1 is available.\n",
      "You should consider upgrading via the '/usr/local/bin/python3.10 -m pip install --upgrade pip' command.\u001b[0m\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "pip install polly-python --upgrade"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/bin/bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)\n",
      "\u001b[32m\u001b[1mA new version of Polly CLI is available. To update, execute the command npm update -g @elucidatainc/pollycli\u001b[22m\u001b[39m\n",
      "Sync complete!\n"
     ]
    }
   ],
   "source": [
    "# sync file\n",
    "!polly files sync -s \"polly://new_omix_for_refac_2 /\" -d \"new_omix_for_refac_2/\" -y"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "from polly.auth import Polly\n",
    "API_KEY=\"eWJtdThpbDE5YTo6QzFvWFhYR1QxeDZtV29tTlhYOEthNUFxdWdCN2N2ZGMzR2RhcGh2WQ==\"\n",
    "Polly.auth(API_KEY, env=\"testpolly\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "from polly.omixatlas import OmixAtlas\n",
    "omixatlas = OmixAtlas()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Repository service functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Schema update didn’t go through because there’s a mistake in the schema. That mistake is summarised in the table below: \n",
      "╒════╤══════════╤════════════╤═══════════════════════╤════════════════╤═══════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════╕\n",
      "│    │ Source   │ Datatype   │ field_name            │ attribute      │ message                                                                                                                                                                                                                                                   │\n",
      "╞════╪══════════╪════════════╪═══════════════════════╪════════════════╪═══════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════╡\n",
      "│  0 │ all      │ all        │ DATA_MATRIX_AVAILABLE │ Not Applicable │ The field name DATA_MATRIX_AVAILABLE did not match the defined naming conditions. Lowercase only, Start with alphabets,  Cannot include special characters except `_`, Cannot be longer than 255 characters, Cannot be SQL reserved DDL and DML keywords. │\n",
      "╘════╧══════════╧════════════╧═══════════════════════╧════════════════╧═══════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════╛\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Source</th>\n",
       "      <th>Datatype</th>\n",
       "      <th>field_name</th>\n",
       "      <th>attribute</th>\n",
       "      <th>message</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>DATA_MATRIX_AVAILABLE</td>\n",
       "      <td>Not Applicable</td>\n",
       "      <td>The field name DATA_MATRIX_AVAILABLE did not m...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Source Datatype             field_name       attribute  \\\n",
       "0    all      all  DATA_MATRIX_AVAILABLE  Not Applicable   \n",
       "\n",
       "                                             message  \n",
       "0  The field name DATA_MATRIX_AVAILABLE did not m...  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# test case 6\n",
    "# field name upper case in datamatrix field of source = all, datatype = all\n",
    "\n",
    "with open(\"/import/new_omix_for_refac_2/new_omix_for_refac.json\",\"r\") as file:\n",
    "    schema_data = json.load(file)\n",
    "\n",
    "omixatlas.validate_schema(schema_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Schema has no errors\n",
      "The data needs to be re-ingested to accommodate the changes made in the schema. This may take from a few minutes to a few hours depending on the number of datasets in the atlas. The repository will be locked for this time and any further changes in the schema or properties of  omixatlas can not be done. Please refer to the ingestion monitoring dashboard to check the status of re-indexing.\n"
     ]
    }
   ],
   "source": [
    "# test 7 ->\n",
    "# Reingestion-> changing an attribute value of a field (one of original_name , type, is_keyword, is_array)\n",
    "# changing original name of `data_matrix_field`\n",
    "\n",
    "repo_key = \"1687428145396\"\n",
    "\n",
    "schema_file_path_for_insertion = \"/import/new_omix_for_refac_2/new_omix_for_refac_2.json\"\n",
    "\n",
    "with open(schema_file_path_for_insertion, \"r\") as schema_file:\n",
    "    schema_data_for_insertion = json.load(schema_file)\n",
    "\n",
    "omixatlas.update_schema(repo_key, schema_data_for_insertion)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# OmixAtlas service functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'data': [{'repo_id': '1687428192072',\n",
       "   'repo_name': 'new_omix_for_refac_3',\n",
       "   'indexes': {},\n",
       "   'v2_indexes': {},\n",
       "   'sources': [],\n",
       "   'datatypes': [],\n",
       "   'dataset_count': 0,\n",
       "   'disease_count': 0,\n",
       "   'tissue_count': 0,\n",
       "   'organism_count': 0,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 0,\n",
       "   'data_source_count': 0,\n",
       "   'sample_count': 0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1687428145396',\n",
       "   'repo_name': 'new_omix_for_refac_2',\n",
       "   'indexes': {},\n",
       "   'v2_indexes': {},\n",
       "   'sources': [],\n",
       "   'datatypes': [],\n",
       "   'dataset_count': 0,\n",
       "   'disease_count': 0,\n",
       "   'tissue_count': 0,\n",
       "   'organism_count': 0,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 0,\n",
       "   'data_source_count': 0,\n",
       "   'sample_count': 0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1688711659012',\n",
       "   'repo_name': 'july_test',\n",
       "   'v2_indexes': {},\n",
       "   'sources': [],\n",
       "   'datatypes': [],\n",
       "   'dataset_count': 0,\n",
       "   'disease_count': 0,\n",
       "   'tissue_count': 0,\n",
       "   'organism_count': 0,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 0,\n",
       "   'data_source_count': 0,\n",
       "   'sample_count': 0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1681886680238',\n",
       "   'repo_name': 'hybrid_testing_repo',\n",
       "   'indexes': {},\n",
       "   'v2_indexes': {},\n",
       "   'sources': [],\n",
       "   'datatypes': [],\n",
       "   'dataset_count': 0,\n",
       "   'disease_count': 0,\n",
       "   'tissue_count': 0,\n",
       "   'organism_count': 0,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 0,\n",
       "   'data_source_count': 0,\n",
       "   'sample_count': 0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1667375323145',\n",
       "   'repo_name': 'bbio_omixatlas_only_h5seurat',\n",
       "   'indexes': {'files': 'bbio_omixatlas_only_h5seurat_files'},\n",
       "   'v2_indexes': {'files': 'bbio_omixatlas_only_h5seurat_files'},\n",
       "   'sources': [{'branchbio': 4}],\n",
       "   'datatypes': [{'single cell multiomics': 4}],\n",
       "   'dataset_count': 4,\n",
       "   'disease_count': 2,\n",
       "   'tissue_count': 3,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 1,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 4.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1678794482481',\n",
       "   'repo_name': 'csv_oa_test',\n",
       "   'v2_indexes': {'files': 'csv_oa_test_files'},\n",
       "   'linked_workspace_id': '15184',\n",
       "   'sources': [{'src_1': 3}, {'all': 2}],\n",
       "   'datatypes': [{'all': 4}, {'datatype_1': 1}],\n",
       "   'dataset_count': 5,\n",
       "   'disease_count': 0,\n",
       "   'tissue_count': 0,\n",
       "   'organism_count': 0,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 2,\n",
       "   'data_source_count': 2,\n",
       "   'sample_count': 0.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1622526550765',\n",
       "   'repo_name': 'liver_atlas',\n",
       "   'indexes': {'files': 'liver_atlas_files',\n",
       "    'gct_col_metadata': 'liver_atlas_gct_col_metadata',\n",
       "    'h5ad_col_metadata': 'liver_atlas_h5ad_col_metadata',\n",
       "    'h5ad_row_metadata': 'liver_atlas_h5ad_row_metadata'},\n",
       "   'v2_indexes': {'files': 'liver_atlas_files',\n",
       "    'gct_col_metadata': 'liver_atlas_gct_col_metadata',\n",
       "    'h5ad_col_metadata': 'liver_atlas_h5ad_col_metadata',\n",
       "    'h5ad_row_metadata': 'liver_atlas_h5ad_row_metadata'},\n",
       "   'sources': [{'geo': 5}],\n",
       "   'datatypes': [{'transcriptomics': 3}, {'trancriptomics': 1}],\n",
       "   'dataset_count': 5,\n",
       "   'disease_count': 2,\n",
       "   'tissue_count': 4,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 1,\n",
       "   'cell_type_count': 3,\n",
       "   'drug_count': 3,\n",
       "   'data_type_count': 2,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 86.0,\n",
       "   'normal_sample_count': 15260},\n",
       "  {'repo_id': '1663919116821',\n",
       "   'repo_name': 'ai_in_drug_discovery_oa',\n",
       "   'indexes': {'files': 'ai_in_drug_discovery_oa_files',\n",
       "    'gct_col_metadata': 'ai_in_drug_discovery_oa_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'ai_in_drug_discovery_oa_files',\n",
       "    'gct_col_metadata': 'ai_in_drug_discovery_oa_gct_col_metadata'},\n",
       "   'sources': [{'geo': 129}],\n",
       "   'datatypes': [{'transcriptomics': 129}],\n",
       "   'dataset_count': 129,\n",
       "   'disease_count': 68,\n",
       "   'tissue_count': 43,\n",
       "   'organism_count': 6,\n",
       "   'cell_line_count': 28,\n",
       "   'cell_type_count': 61,\n",
       "   'drug_count': 55,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 3649.0,\n",
       "   'normal_sample_count': 1574},\n",
       "  {'repo_id': '1669033066992',\n",
       "   'repo_name': 'destination_atlas',\n",
       "   'indexes': {'files': 'destination_atlas_files',\n",
       "    'gct_col_metadata': 'destination_atlas_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'destination_atlas_files',\n",
       "    'gct_col_metadata': 'destination_atlas_gct_col_metadata'},\n",
       "   'sources': [{'geo': 33}],\n",
       "   'datatypes': [{'raw counts transcriptomics': 30},\n",
       "    {'transcriptomics': 2},\n",
       "    {'raw_counts_transcriptomics': 1}],\n",
       "   'dataset_count': 33,\n",
       "   'disease_count': 35,\n",
       "   'tissue_count': 27,\n",
       "   'organism_count': 3,\n",
       "   'cell_line_count': 25,\n",
       "   'cell_type_count': 23,\n",
       "   'drug_count': 56,\n",
       "   'data_type_count': 3,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 712.0,\n",
       "   'normal_sample_count': 467},\n",
       "  {'repo_id': '1685621126191',\n",
       "   'repo_name': 'gdx_demo_atlas',\n",
       "   'indexes': {'files': 'gdx_demo_atlas_files',\n",
       "    'gct_col_metadata': 'gdx_demo_atlas_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'gdx_demo_atlas_files',\n",
       "    'gct_col_metadata': 'gdx_demo_atlas_gct_col_metadata'},\n",
       "   'data_type': 'bulk_rna_seq',\n",
       "   'sources': [{'geo': 18}],\n",
       "   'datatypes': [{'': 12}, {'raw counts transcriptomics': 6}],\n",
       "   'dataset_count': 18,\n",
       "   'disease_count': 25,\n",
       "   'tissue_count': 36,\n",
       "   'organism_count': 9,\n",
       "   'cell_line_count': 3,\n",
       "   'cell_type_count': 6,\n",
       "   'drug_count': 3,\n",
       "   'data_type_count': 2,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 332.0,\n",
       "   'normal_sample_count': 17},\n",
       "  {'repo_id': '1679482518578',\n",
       "   'repo_name': 'details_page',\n",
       "   'indexes': {'files': 'details_page_files',\n",
       "    'gct_col_metadata': 'details_page_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'details_page_files',\n",
       "    'gct_col_metadata': 'details_page_gct_col_metadata'},\n",
       "   'sources': [{'geo': 27}],\n",
       "   'datatypes': [{'raw counts transcriptomics': 27}],\n",
       "   'dataset_count': 27,\n",
       "   'disease_count': 36,\n",
       "   'tissue_count': 28,\n",
       "   'organism_count': 3,\n",
       "   'cell_line_count': 6,\n",
       "   'cell_type_count': 17,\n",
       "   'drug_count': 8,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 4190.0,\n",
       "   'normal_sample_count': 2062},\n",
       "  {'repo_id': '1687411548568',\n",
       "   'repo_name': 'junee_test',\n",
       "   'v2_indexes': {'files': 'junee_test_files',\n",
       "    'gct_col_metadata': 'junee_test_gct_col_metadata'},\n",
       "   'sources': [{'geo': 21}],\n",
       "   'datatypes': [{'raw counts transcriptomics': 21}],\n",
       "   'dataset_count': 21,\n",
       "   'disease_count': 19,\n",
       "   'tissue_count': 0,\n",
       "   'organism_count': 4,\n",
       "   'cell_line_count': 10,\n",
       "   'cell_type_count': 13,\n",
       "   'drug_count': 3,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 381.0,\n",
       "   'normal_sample_count': 94},\n",
       "  {'repo_id': '1659450268526',\n",
       "   'repo_name': 'multiple_oa_2',\n",
       "   'indexes': {'files': 'multiple_oa_2_files',\n",
       "    'gct_col_metadata': 'multiple_oa_2_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'multiple_oa_2_files',\n",
       "    'gct_col_metadata': 'multiple_oa_2_gct_col_metadata'},\n",
       "   'sources': [{'cbioportal': 226},\n",
       "    {'geo': 166},\n",
       "    {'depmap': 5},\n",
       "    {'tyra': 5},\n",
       "    {'ccle': 4},\n",
       "    {'cptac': 1}],\n",
       "   'datatypes': [{'copy number variation': 192},\n",
       "    {'transcriptomics': 166},\n",
       "    {'mutation': 33},\n",
       "    {'fusion': 4},\n",
       "    {'gene dependency': 4},\n",
       "    {'drug screens': 1},\n",
       "    {'metabolomics': 1},\n",
       "    {'mirna': 1}],\n",
       "   'dataset_count': 407,\n",
       "   'disease_count': 161,\n",
       "   'tissue_count': 127,\n",
       "   'organism_count': 7,\n",
       "   'cell_line_count': 159,\n",
       "   'cell_type_count': 62,\n",
       "   'drug_count': 71,\n",
       "   'data_type_count': 8,\n",
       "   'data_source_count': 6,\n",
       "   'sample_count': 276264.0,\n",
       "   'normal_sample_count': 183062},\n",
       "  {'repo_id': '1679648155740',\n",
       "   'repo_name': 'bulkrnaseq_staging_oa',\n",
       "   'v2_indexes': {},\n",
       "   'sources': [],\n",
       "   'datatypes': [],\n",
       "   'dataset_count': 0,\n",
       "   'disease_count': 0,\n",
       "   'tissue_count': 0,\n",
       "   'organism_count': 0,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 0,\n",
       "   'data_source_count': 0,\n",
       "   'sample_count': 0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1687428220754',\n",
       "   'repo_name': 'new_omix_for_refac_4',\n",
       "   'indexes': {},\n",
       "   'v2_indexes': {},\n",
       "   'sources': [],\n",
       "   'datatypes': [],\n",
       "   'dataset_count': 0,\n",
       "   'disease_count': 0,\n",
       "   'tissue_count': 0,\n",
       "   'organism_count': 0,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 0,\n",
       "   'data_source_count': 0,\n",
       "   'sample_count': 0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1677039559242',\n",
       "   'repo_name': 'diy_poa',\n",
       "   'indexes': {'files': 'diy_poa_files',\n",
       "    'h5ad_col_metadata': 'diy_poa_h5ad_col_metadata'},\n",
       "   'v2_indexes': {'files': 'diy_poa_files',\n",
       "    'h5ad_col_metadata': 'diy_poa_h5ad_col_metadata'},\n",
       "   'sources': [{'geo': 3}],\n",
       "   'datatypes': [{'single cell': 3}],\n",
       "   'dataset_count': 3,\n",
       "   'disease_count': 4,\n",
       "   'tissue_count': 3,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 1,\n",
       "   'cell_type_count': 26,\n",
       "   'drug_count': 1,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 8.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1663151941472',\n",
       "   'repo_name': 'c_oa',\n",
       "   'indexes': {'files': 'c_oa_files'},\n",
       "   'v2_indexes': {'files': 'c_oa_files'},\n",
       "   'linked_workspace_id': 14918,\n",
       "   'sources': [{'': 10}],\n",
       "   'datatypes': [{'drug response': 10}],\n",
       "   'dataset_count': 10,\n",
       "   'disease_count': 3,\n",
       "   'tissue_count': 1,\n",
       "   'organism_count': 0,\n",
       "   'cell_line_count': 5,\n",
       "   'cell_type_count': 1,\n",
       "   'drug_count': 1,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 10.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1643016586529',\n",
       "   'repo_name': 'geo',\n",
       "   'indexes': {'files': 'geo_files',\n",
       "    'gct_col_metadata': 'geo_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'geo_files',\n",
       "    'gct_col_metadata': 'geo_gct_col_metadata'},\n",
       "   'linked_workspace_id': '15196',\n",
       "   'sources': [{'geo': 5}, {'source': 1}, {'vfv': 1}],\n",
       "   'datatypes': [{'transcriptomics': 6}, {'': 1}],\n",
       "   'dataset_count': 7,\n",
       "   'disease_count': 12,\n",
       "   'tissue_count': 8,\n",
       "   'organism_count': 3,\n",
       "   'cell_line_count': 5,\n",
       "   'cell_type_count': 6,\n",
       "   'drug_count': 2,\n",
       "   'data_type_count': 2,\n",
       "   'data_source_count': 3,\n",
       "   'sample_count': 1741.0,\n",
       "   'normal_sample_count': 154446},\n",
       "  {'repo_id': '1673411159486',\n",
       "   'repo_name': 'bulk_rnaseq_atlas',\n",
       "   'indexes': {'files': 'bulk_rnaseq_atlas_files',\n",
       "    'gct_col_metadata': 'bulk_rnaseq_atlas_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'bulk_rnaseq_atlas_files',\n",
       "    'gct_col_metadata': 'bulk_rnaseq_atlas_gct_col_metadata'},\n",
       "   'sources': [{'geo': 74}],\n",
       "   'datatypes': [{'raw_counts_transcriptomics': 57}, {'transcriptomics': 17}],\n",
       "   'dataset_count': 74,\n",
       "   'disease_count': 88,\n",
       "   'tissue_count': 71,\n",
       "   'organism_count': 2,\n",
       "   'cell_line_count': 21,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 10,\n",
       "   'data_type_count': 2,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 6808.0,\n",
       "   'normal_sample_count': 5402},\n",
       "  {'repo_id': '1689071945246',\n",
       "   'repo_name': 'gdx_test_s',\n",
       "   'v2_indexes': {'files': 'gdx_test_s_files',\n",
       "    'gct_col_metadata': 'gdx_test_s_gct_col_metadata'},\n",
       "   'sources': [{'geo': 1}],\n",
       "   'datatypes': [{'': 1}],\n",
       "   'dataset_count': 1,\n",
       "   'disease_count': 1,\n",
       "   'tissue_count': 1,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 1,\n",
       "   'cell_type_count': 2,\n",
       "   'drug_count': 1,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 6.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1683112669959',\n",
       "   'repo_name': 'may_repo_test_single',\n",
       "   'v2_indexes': {'h5ad_col_metadata': 'may_repo_test_single_h5ad_col_metadata'},\n",
       "   'org_id': '1',\n",
       "   'data_type': 'single_cell',\n",
       "   'sources': [],\n",
       "   'datatypes': [],\n",
       "   'dataset_count': 0,\n",
       "   'disease_count': 0,\n",
       "   'tissue_count': 0,\n",
       "   'organism_count': 0,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 0,\n",
       "   'data_source_count': 0,\n",
       "   'sample_count': 0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1664439952209',\n",
       "   'repo_name': 'clinical_oa_2',\n",
       "   'indexes': {'files': 'clinical_oa_2_files',\n",
       "    'gct_col_metadata': 'clinical_oa_2_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'clinical_oa_2_files',\n",
       "    'gct_col_metadata': 'clinical_oa_2_gct_col_metadata'},\n",
       "   'org_id': '1',\n",
       "   'sources': [{'teikobio': 16}, {'geo': 6}],\n",
       "   'datatypes': [{'flow cytometry': 16}, {'raw counts transcriptomics': 6}],\n",
       "   'dataset_count': 22,\n",
       "   'disease_count': 7,\n",
       "   'tissue_count': 7,\n",
       "   'organism_count': 2,\n",
       "   'cell_line_count': 1,\n",
       "   'cell_type_count': 6,\n",
       "   'drug_count': 1,\n",
       "   'data_type_count': 2,\n",
       "   'data_source_count': 2,\n",
       "   'sample_count': 149.0,\n",
       "   'normal_sample_count': 58},\n",
       "  {'repo_id': '1687327504866',\n",
       "   'repo_name': 'new_omix_for_refac',\n",
       "   'indexes': {},\n",
       "   'v2_indexes': {},\n",
       "   'sources': [],\n",
       "   'datatypes': [],\n",
       "   'dataset_count': 0,\n",
       "   'disease_count': 0,\n",
       "   'tissue_count': 0,\n",
       "   'organism_count': 0,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 0,\n",
       "   'data_source_count': 0,\n",
       "   'sample_count': 0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '31',\n",
       "   'repo_name': 'lincs',\n",
       "   'indexes': {'files': 'lincs_files',\n",
       "    'gct_col_metadata': 'lincs_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'lincs_files',\n",
       "    'gct_col_metadata': 'lincs_gct_col_metadata'},\n",
       "   'sources': [{'geo, lincs': 700}, {'geo': 95}, {'cbioportal': 2}],\n",
       "   'datatypes': [{'transcriptomics': 795}, {'': 2}],\n",
       "   'dataset_count': 797,\n",
       "   'disease_count': 30,\n",
       "   'tissue_count': 20,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 67,\n",
       "   'cell_type_count': 1,\n",
       "   'drug_count': 571,\n",
       "   'data_type_count': 2,\n",
       "   'data_source_count': 3,\n",
       "   'sample_count': 198202.0,\n",
       "   'normal_sample_count': 9062},\n",
       "  {'repo_id': '1654268055800',\n",
       "   'repo_name': 'ingestion_test_1',\n",
       "   'indexes': {'files': 'ingestion_test_1_files',\n",
       "    'gct_col_metadata': 'ingestion_test_1_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'ingestion_test_1_files',\n",
       "    'gct_col_metadata': 'ingestion_test_1_gct_col_metadata'},\n",
       "   'sources': [{'geo': 76}, {'cbioportal': 13}, {'ccle': 3}, {'tcga': 1}],\n",
       "   'datatypes': [{'transcriptomics': 65},\n",
       "    {'': 22},\n",
       "    {'copy_number_variation': 5},\n",
       "    {'single_cell': 1}],\n",
       "   'dataset_count': 93,\n",
       "   'disease_count': 0,\n",
       "   'tissue_count': 0,\n",
       "   'organism_count': 0,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 4,\n",
       "   'data_source_count': 4,\n",
       "   'sample_count': 0.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1657777478388',\n",
       "   'repo_name': 'ingestion_testing_oa',\n",
       "   'indexes': {'files': 'ingestion_testing_oa_files',\n",
       "    'gct_col_metadata': 'ingestion_testing_oa_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'ingestion_testing_oa_files',\n",
       "    'gct_col_metadata': 'ingestion_testing_oa_gct_col_metadata'},\n",
       "   'sources': [{'pride': 1}],\n",
       "   'datatypes': [{'proteomics': 1}],\n",
       "   'dataset_count': 1,\n",
       "   'disease_count': 1,\n",
       "   'tissue_count': 1,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 1,\n",
       "   'drug_count': 2,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 3.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1647941576494',\n",
       "   'repo_name': 'depmap_omixatlas',\n",
       "   'indexes': {'files': 'depmap_omixatlas_files',\n",
       "    'gct_col_metadata': 'depmap_omixatlas_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'depmap_omixatlas_files',\n",
       "    'gct_col_metadata': 'depmap_omixatlas_gct_col_metadata'},\n",
       "   'sources': [{'depmap': 546}],\n",
       "   'datatypes': [{'gene dependency': 487}, {'drug screens': 59}],\n",
       "   'dataset_count': 546,\n",
       "   'disease_count': 89,\n",
       "   'tissue_count': 37,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 410,\n",
       "   'cell_type_count': 1,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 2,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 546.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1659621145863',\n",
       "   'repo_name': 'hmp_omixatlas',\n",
       "   'indexes': {'files': 'hmp_omixatlas_files',\n",
       "    'biom_col_metadata': 'hmp_omixatlas_biom_col_metadata',\n",
       "    'biom_row_metadata': 'hmp_omixatlas_biom_row_metadata'},\n",
       "   'v2_indexes': {'files': 'hmp_omixatlas_files',\n",
       "    'biom_col_metadata': 'hmp_omixatlas_biom_col_metadata',\n",
       "    'biom_row_metadata': 'hmp_omixatlas_biom_row_metadata'},\n",
       "   'sources': [{'human microbiome project (hmp)': 32}, {'microbiome db': 1}],\n",
       "   'datatypes': [{'metagenomics': 26},\n",
       "    {'metatranscriptomics': 6},\n",
       "    {'microbiome': 1}],\n",
       "   'dataset_count': 33,\n",
       "   'disease_count': 1,\n",
       "   'tissue_count': 2,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 1,\n",
       "   'cell_type_count': 1,\n",
       "   'drug_count': 16,\n",
       "   'data_type_count': 3,\n",
       "   'data_source_count': 2,\n",
       "   'sample_count': 1248.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1666789252487',\n",
       "   'repo_name': 'bbio_omixatlas',\n",
       "   'indexes': {'files': 'bbio_omixatlas_files'},\n",
       "   'v2_indexes': {'files': 'bbio_omixatlas_files'},\n",
       "   'sources': [{'branchbio': 1}],\n",
       "   'datatypes': [{'single cell multiomics': 1}],\n",
       "   'dataset_count': 1,\n",
       "   'disease_count': 1,\n",
       "   'tissue_count': 1,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 1,\n",
       "   'cell_type_count': 1,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 1.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1649161733379',\n",
       "   'repo_name': 'cell_viability_assay',\n",
       "   'indexes': {'files': 'cell_viability_assay_files',\n",
       "    'gct_col_metadata': 'cell_viability_assay_gct_col_metadata',\n",
       "    'h5ad_col_metadata': 'cell_viability_assay_h5ad_col_metadata'},\n",
       "   'v2_indexes': {'files': 'cell_viability_assay_files',\n",
       "    'gct_col_metadata': 'cell_viability_assay_gct_col_metadata',\n",
       "    'h5ad_col_metadata': 'cell_viability_assay_h5ad_col_metadata'},\n",
       "   'org_id': '14',\n",
       "   'data_type': 'single_cell',\n",
       "   'sources': [{'geo': 4}, {'expression atlas': 1}],\n",
       "   'datatypes': [{'single cell': 4}, {'single_cell': 1}],\n",
       "   'dataset_count': 5,\n",
       "   'disease_count': 10,\n",
       "   'tissue_count': 11,\n",
       "   'organism_count': 2,\n",
       "   'cell_line_count': 2,\n",
       "   'cell_type_count': 14,\n",
       "   'drug_count': 1,\n",
       "   'data_type_count': 2,\n",
       "   'data_source_count': 2,\n",
       "   'sample_count': 17.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1629715166552',\n",
       "   'repo_name': 'gnomad',\n",
       "   'indexes': {'files': 'gnomad_files'},\n",
       "   'v2_indexes': {'files': 'gnomad_files'},\n",
       "   'org_id': '1',\n",
       "   'sources': [],\n",
       "   'datatypes': [],\n",
       "   'dataset_count': 720,\n",
       "   'disease_count': 0,\n",
       "   'tissue_count': 0,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 0,\n",
       "   'data_source_count': 0,\n",
       "   'sample_count': 0.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1673847977346',\n",
       "   'repo_name': 'lib_repo_dev',\n",
       "   'indexes': {'files': 'lib_repo_dev_files',\n",
       "    'gct_col_metadata': 'lib_repo_dev_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'lib_repo_dev_files',\n",
       "    'gct_col_metadata': 'lib_repo_dev_gct_col_metadata'},\n",
       "   'linked_workspace_id': '99999999999999999999999',\n",
       "   'sources': [{'cbioportal': 3}, {'geo': 3}],\n",
       "   'datatypes': [{'copy number variation': 3}, {'transcriptomics': 3}],\n",
       "   'dataset_count': 6,\n",
       "   'disease_count': 7,\n",
       "   'tissue_count': 3,\n",
       "   'organism_count': 0,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 2,\n",
       "   'drug_count': 1,\n",
       "   'data_type_count': 2,\n",
       "   'data_source_count': 2,\n",
       "   'sample_count': 62.0,\n",
       "   'normal_sample_count': 390},\n",
       "  {'repo_id': '1669085436194',\n",
       "   'repo_name': 'source_atlas',\n",
       "   'indexes': {'files': 'source_atlas_files',\n",
       "    'gct_col_metadata': 'source_atlas_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'source_atlas_files',\n",
       "    'gct_col_metadata': 'source_atlas_gct_col_metadata'},\n",
       "   'data_type': 'bulk_rna_seq',\n",
       "   'sources': [{'geo': 36}],\n",
       "   'datatypes': [{'raw counts transcriptomics': 35}, {'transcriptomics': 1}],\n",
       "   'dataset_count': 36,\n",
       "   'disease_count': 28,\n",
       "   'tissue_count': 29,\n",
       "   'organism_count': 4,\n",
       "   'cell_line_count': 22,\n",
       "   'cell_type_count': 19,\n",
       "   'drug_count': 4,\n",
       "   'data_type_count': 2,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 3094.0,\n",
       "   'normal_sample_count': 2769},\n",
       "  {'repo_id': '1673014297214',\n",
       "   'repo_name': 'test_connector_pipeline',\n",
       "   'indexes': {'files': 'test_connector_pipeline_files',\n",
       "    'gct_col_metadata': 'test_connector_pipeline_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'test_connector_pipeline_files',\n",
       "    'gct_col_metadata': 'test_connector_pipeline_gct_col_metadata'},\n",
       "   'sources': [{'geo': 5}],\n",
       "   'datatypes': [{'transcriptomics': 5}],\n",
       "   'dataset_count': 5,\n",
       "   'disease_count': 9,\n",
       "   'tissue_count': 3,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 8,\n",
       "   'cell_type_count': 3,\n",
       "   'drug_count': 1,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 54.0,\n",
       "   'normal_sample_count': 29},\n",
       "  {'repo_id': '1674572370484',\n",
       "   'repo_name': 'test_pipeline',\n",
       "   'indexes': {'files': 'test_pipeline_files',\n",
       "    'gct_col_metadata': 'test_pipeline_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'test_pipeline_files',\n",
       "    'gct_col_metadata': 'test_pipeline_gct_col_metadata'},\n",
       "   'sources': [{'geo': 22}],\n",
       "   'datatypes': [{'transcriptomics': 13}, {'raw counts transcriptomics': 9}],\n",
       "   'dataset_count': 22,\n",
       "   'disease_count': 24,\n",
       "   'tissue_count': 9,\n",
       "   'organism_count': 2,\n",
       "   'cell_line_count': 8,\n",
       "   'cell_type_count': 20,\n",
       "   'drug_count': 3,\n",
       "   'data_type_count': 2,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 317.0,\n",
       "   'normal_sample_count': 100},\n",
       "  {'repo_id': '1679312499313',\n",
       "   'repo_name': 'test_pipeline_rnaseq',\n",
       "   'indexes': {'files': 'test_pipeline_rnaseq_files',\n",
       "    'gct_col_metadata': 'test_pipeline_rnaseq_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'test_pipeline_rnaseq_files',\n",
       "    'gct_col_metadata': 'test_pipeline_rnaseq_gct_col_metadata'},\n",
       "   'sources': [{'geo': 110}],\n",
       "   'datatypes': [{'raw counts transcriptomics': 84},\n",
       "    {'raw_counts_transcriptomics': 24},\n",
       "    {'transcriptomics': 2}],\n",
       "   'dataset_count': 110,\n",
       "   'disease_count': 73,\n",
       "   'tissue_count': 41,\n",
       "   'organism_count': 3,\n",
       "   'cell_line_count': 39,\n",
       "   'cell_type_count': 49,\n",
       "   'drug_count': 22,\n",
       "   'data_type_count': 3,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 1926.0,\n",
       "   'normal_sample_count': 1696},\n",
       "  {'repo_id': '1681973657512',\n",
       "   'repo_name': 'test_schema_builder_oa',\n",
       "   'v2_indexes': {},\n",
       "   'sources': [],\n",
       "   'datatypes': [],\n",
       "   'dataset_count': 0,\n",
       "   'disease_count': 0,\n",
       "   'tissue_count': 0,\n",
       "   'organism_count': 0,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 0,\n",
       "   'data_source_count': 0,\n",
       "   'sample_count': 0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1649664527538',\n",
       "   'repo_name': 'tyra_cell_viability_assay',\n",
       "   'indexes': {'files': 'tyra_cell_viability_assay_files'},\n",
       "   'v2_indexes': {'files': 'tyra_cell_viability_assay_files'},\n",
       "   'sources': [],\n",
       "   'datatypes': [],\n",
       "   'dataset_count': 3760,\n",
       "   'disease_count': 0,\n",
       "   'tissue_count': 0,\n",
       "   'organism_count': 0,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 0,\n",
       "   'data_source_count': 0,\n",
       "   'sample_count': 0.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1644400393649',\n",
       "   'repo_name': 'ukbiobank',\n",
       "   'indexes': {'files': 'ukbiobank_files'},\n",
       "   'v2_indexes': {'files': 'ukbiobank_files'},\n",
       "   'sources': [],\n",
       "   'datatypes': [{'gwas': 648}],\n",
       "   'dataset_count': 648,\n",
       "   'disease_count': 0,\n",
       "   'tissue_count': 0,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 0,\n",
       "   'sample_count': 0.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1674475984908',\n",
       "   'repo_name': 'single_cell_test',\n",
       "   'indexes': {'files': 'single_cell_test_files',\n",
       "    'h5ad_col_metadata': 'single_cell_test_h5ad_col_metadata'},\n",
       "   'v2_indexes': {'files': 'single_cell_test_files',\n",
       "    'h5ad_col_metadata': 'single_cell_test_h5ad_col_metadata'},\n",
       "   'sources': [{'array express': 2},\n",
       "    {'covid-19 cell atlas': 1},\n",
       "    {'human cell atlas': 1}],\n",
       "   'datatypes': [{'single cell': 4}],\n",
       "   'dataset_count': 4,\n",
       "   'disease_count': 4,\n",
       "   'tissue_count': 2,\n",
       "   'organism_count': 2,\n",
       "   'cell_line_count': 1,\n",
       "   'cell_type_count': 11,\n",
       "   'drug_count': 2,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 3,\n",
       "   'sample_count': 31.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1670915426568',\n",
       "   'repo_name': 'single_cell_destination',\n",
       "   'indexes': {'files': 'single_cell_destination_files',\n",
       "    'h5ad_col_metadata': 'single_cell_destination_h5ad_col_metadata'},\n",
       "   'v2_indexes': {'files': 'single_cell_destination_files',\n",
       "    'h5ad_col_metadata': 'single_cell_destination_h5ad_col_metadata'},\n",
       "   'sources': [{'geo': 7}, {'tabula sapiens': 1}],\n",
       "   'datatypes': [{'single cell': 8}],\n",
       "   'dataset_count': 8,\n",
       "   'disease_count': 7,\n",
       "   'tissue_count': 13,\n",
       "   'organism_count': 2,\n",
       "   'cell_line_count': 2,\n",
       "   'cell_type_count': 34,\n",
       "   'drug_count': 10,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 2,\n",
       "   'sample_count': 8.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '17',\n",
       "   'repo_name': 'sc_data_lake',\n",
       "   'indexes': {'h5ad_col_metadata': 'sc_data_lake_h5ad_col_metadata',\n",
       "    'h5ad_row_metadata': 'sc_data_lake_h5ad_row_metadata'},\n",
       "   'v2_indexes': {'h5ad_col_metadata': 'sc_data_lake_h5ad_col_metadata',\n",
       "    'h5ad_row_metadata': 'sc_data_lake_h5ad_row_metadata'},\n",
       "   'sources': [],\n",
       "   'datatypes': [],\n",
       "   'dataset_count': 0,\n",
       "   'disease_count': 0,\n",
       "   'tissue_count': 0,\n",
       "   'organism_count': 0,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 0,\n",
       "   'data_source_count': 0,\n",
       "   'sample_count': 0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1689052723688',\n",
       "   'repo_name': 'test_july_p',\n",
       "   'v2_indexes': {'files': 'test_july_p_files',\n",
       "    'gct_col_metadata': 'test_july_p_gct_col_metadata'},\n",
       "   'sources': [{'geo': 37}],\n",
       "   'datatypes': [{'transcriptomics': 25},\n",
       "    {'raw counts transcriptomics': 11},\n",
       "    {'': 1}],\n",
       "   'dataset_count': 37,\n",
       "   'disease_count': 1,\n",
       "   'tissue_count': 33,\n",
       "   'organism_count': 3,\n",
       "   'cell_line_count': 617,\n",
       "   'cell_type_count': 29,\n",
       "   'drug_count': 1,\n",
       "   'data_type_count': 3,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 46250.0,\n",
       "   'normal_sample_count': 32893},\n",
       "  {'repo_id': '1680106629523',\n",
       "   'repo_name': 'single_sch_oa',\n",
       "   'indexes': {'files': 'single_sch_oa_files',\n",
       "    'h5ad_col_metadata': 'single_sch_oa_h5ad_col_metadata'},\n",
       "   'v2_indexes': {'files': 'single_sch_oa_files',\n",
       "    'h5ad_col_metadata': 'single_sch_oa_h5ad_col_metadata'},\n",
       "   'sources': [{'geo': 2}, {'': 1}, {'ghi': 1}],\n",
       "   'datatypes': [{'': 2}, {'single_cell': 2}],\n",
       "   'dataset_count': 4,\n",
       "   'disease_count': 9,\n",
       "   'tissue_count': 6,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 1,\n",
       "   'cell_type_count': 41,\n",
       "   'drug_count': 13,\n",
       "   'data_type_count': 2,\n",
       "   'data_source_count': 3,\n",
       "   'sample_count': 33.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1689168428573',\n",
       "   'repo_name': 'test_so_2',\n",
       "   'v2_indexes': {'files': 'test_so_2_files',\n",
       "    'gct_col_metadata': 'test_so_2_gct_col_metadata'},\n",
       "   'sources': [{'geo': 2}],\n",
       "   'datatypes': [{'raw counts transcriptomics': 2}],\n",
       "   'dataset_count': 2,\n",
       "   'disease_count': 2,\n",
       "   'tissue_count': 4,\n",
       "   'organism_count': 0,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 3,\n",
       "   'drug_count': 1,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 0.0,\n",
       "   'normal_sample_count': 7},\n",
       "  {'repo_id': '1680442387535',\n",
       "   'repo_name': 'sch_valid_change',\n",
       "   'indexes': {},\n",
       "   'v2_indexes': {},\n",
       "   'sources': [],\n",
       "   'datatypes': [],\n",
       "   'dataset_count': 0,\n",
       "   'disease_count': 0,\n",
       "   'tissue_count': 0,\n",
       "   'organism_count': 0,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 0,\n",
       "   'data_source_count': 0,\n",
       "   'sample_count': 0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1680089043360',\n",
       "   'repo_name': 'sch_validation_oa',\n",
       "   'indexes': {'files': 'sch_validation_oa_files',\n",
       "    'gct_col_metadata': 'sch_validation_oa_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'sch_validation_oa_files',\n",
       "    'gct_col_metadata': 'sch_validation_oa_gct_col_metadata'},\n",
       "   'sources': [{'geo': 16},\n",
       "    {'': 1},\n",
       "    {'abc': 1},\n",
       "    {'ghi': 1},\n",
       "    {'random': 1},\n",
       "    {'sc_data_lake': 1}],\n",
       "   'datatypes': [{'': 19}, {'jkl': 1}, {'post': 1}],\n",
       "   'dataset_count': 21,\n",
       "   'disease_count': 24,\n",
       "   'tissue_count': 21,\n",
       "   'organism_count': 0,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 20,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 3,\n",
       "   'data_source_count': 6,\n",
       "   'sample_count': 802.0,\n",
       "   'normal_sample_count': 767},\n",
       "  {'repo_id': '1688035585607',\n",
       "   'repo_name': 'repo_july_so',\n",
       "   'indexes': {'files': 'repo_july_so_files',\n",
       "    'gct_col_metadata': 'repo_july_so_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'repo_july_so_files',\n",
       "    'gct_col_metadata': 'repo_july_so_gct_col_metadata'},\n",
       "   'org_id': '1',\n",
       "   'data_type': 'bulk_rna_seq',\n",
       "   'sources': [{'geo': 20}],\n",
       "   'datatypes': [{'raw counts transcriptomics': 19}, {'transcriptomics': 1}],\n",
       "   'dataset_count': 20,\n",
       "   'disease_count': 19,\n",
       "   'tissue_count': 15,\n",
       "   'organism_count': 4,\n",
       "   'cell_line_count': 8,\n",
       "   'cell_type_count': 13,\n",
       "   'drug_count': 5,\n",
       "   'data_type_count': 2,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 2652.0,\n",
       "   'normal_sample_count': 108},\n",
       "  {'repo_id': '1662102491212',\n",
       "   'repo_name': 'p_oa',\n",
       "   'indexes': {'files': 'p_oa_files',\n",
       "    'gct_col_metadata': 'p_oa_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'p_oa_files',\n",
       "    'gct_col_metadata': 'p_oa_gct_col_metadata'},\n",
       "   'sources': [{'tcga': 35}, {'geo': 22}],\n",
       "   'datatypes': [{'transcriptomics': 40},\n",
       "    {'copy number variation': 15},\n",
       "    {'': 2}],\n",
       "   'dataset_count': 57,\n",
       "   'disease_count': 14,\n",
       "   'tissue_count': 13,\n",
       "   'organism_count': 6,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 10,\n",
       "   'data_type_count': 3,\n",
       "   'data_source_count': 2,\n",
       "   'sample_count': 188.0,\n",
       "   'normal_sample_count': 73},\n",
       "  {'repo_id': '1671791834799',\n",
       "   'repo_name': 'single_cell_source',\n",
       "   'indexes': {'files': 'single_cell_source_files',\n",
       "    'h5ad_col_metadata': 'single_cell_source_h5ad_col_metadata'},\n",
       "   'v2_indexes': {'files': 'single_cell_source_files',\n",
       "    'h5ad_col_metadata': 'single_cell_source_h5ad_col_metadata'},\n",
       "   'data_type': 'single_cell',\n",
       "   'sources': [{'geo': 43},\n",
       "    {'publication': 8},\n",
       "    {'expression atlas': 3},\n",
       "    {'scp': 3},\n",
       "    {'human cell atlas': 1},\n",
       "    {'tabula sapiens': 1}],\n",
       "   'datatypes': [{'single cell': 59}],\n",
       "   'dataset_count': 59,\n",
       "   'disease_count': 56,\n",
       "   'tissue_count': 55,\n",
       "   'organism_count': 2,\n",
       "   'cell_line_count': 5,\n",
       "   'cell_type_count': 130,\n",
       "   'drug_count': 15,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 6,\n",
       "   'sample_count': 1246.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1676020061866',\n",
       "   'repo_name': 'test_single_cell_source_feb',\n",
       "   'indexes': {'files': 'test_single_cell_source_feb_files',\n",
       "    'h5ad_col_metadata': 'test_single_cell_source_feb_h5ad_col_metadata'},\n",
       "   'v2_indexes': {'files': 'test_single_cell_source_feb_files',\n",
       "    'h5ad_col_metadata': 'test_single_cell_source_feb_h5ad_col_metadata'},\n",
       "   'data_type': 'single_cell',\n",
       "   'sources': [{'geo': 6},\n",
       "    {'expression_atlas': 2},\n",
       "    {'ana': 1},\n",
       "    {'array_express': 1},\n",
       "    {'ghi': 1},\n",
       "    {'publication': 1},\n",
       "    {'scp': 1},\n",
       "    {'tabula sapiens': 1}],\n",
       "   'datatypes': [{'single cell': 7},\n",
       "    {'single_cell': 5},\n",
       "    {'hya': 1},\n",
       "    {'jkl': 1}],\n",
       "   'dataset_count': 14,\n",
       "   'disease_count': 19,\n",
       "   'tissue_count': 10,\n",
       "   'organism_count': 2,\n",
       "   'cell_line_count': 1,\n",
       "   'cell_type_count': 58,\n",
       "   'drug_count': 13,\n",
       "   'data_type_count': 4,\n",
       "   'data_source_count': 8,\n",
       "   'sample_count': 72.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '15',\n",
       "   'repo_name': 'tcga',\n",
       "   'indexes': {'files': 'tcga_files'},\n",
       "   'v2_indexes': {'files': 'tcga_files'},\n",
       "   'sources': [{'tcga': 678}],\n",
       "   'datatypes': [{'transcriptomics': 156},\n",
       "    {'copy number variation': 124},\n",
       "    {'mirna': 122},\n",
       "    {'methylation': 118},\n",
       "    {'proteomics': 81},\n",
       "    {'mutation': 77}],\n",
       "   'dataset_count': 678,\n",
       "   'disease_count': 34,\n",
       "   'tissue_count': 55,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 101,\n",
       "   'data_type_count': 6,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 678.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1688035601240',\n",
       "   'repo_name': 'repo_g_eo',\n",
       "   'indexes': {'files': 'repo_g_eo_files',\n",
       "    'gct_col_metadata': 'repo_g_eo_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'repo_g_eo_files',\n",
       "    'gct_col_metadata': 'repo_g_eo_gct_col_metadata'},\n",
       "   'data_type': 'bulk_rna_seq',\n",
       "   'sources': [{'geo': 123}],\n",
       "   'datatypes': [{'transcriptomics': 111}, {'raw counts transcriptomics': 12}],\n",
       "   'dataset_count': 123,\n",
       "   'disease_count': 382,\n",
       "   'tissue_count': 94,\n",
       "   'organism_count': 4,\n",
       "   'cell_line_count': 640,\n",
       "   'cell_type_count': 52,\n",
       "   'drug_count': 11,\n",
       "   'data_type_count': 2,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 217989.0,\n",
       "   'normal_sample_count': 114996}]}"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "omixatlas.get_all_omixatlas()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'data': {'repo_name': 'geo',\n",
       "  'repo_id': '1643016586529',\n",
       "  'indexes': {'files': 'geo_files',\n",
       "   'gct_col_metadata': 'geo_gct_col_metadata'},\n",
       "  'v2_indexes': {'files': 'geo_files',\n",
       "   'gct_col_metadata': 'geo_gct_col_metadata'},\n",
       "  'linked_workspace_id': '15196',\n",
       "  'sources': [{'geo': 5}, {'source': 1}, {'vfv': 1}],\n",
       "  'datatypes': [{'transcriptomics': 6}, {'': 1}],\n",
       "  'dataset_count': 7,\n",
       "  'disease_count': 12,\n",
       "  'tissue_count': 8,\n",
       "  'organism_count': 3,\n",
       "  'cell_line_count': 5,\n",
       "  'cell_type_count': 6,\n",
       "  'drug_count': 2,\n",
       "  'data_type_count': 2,\n",
       "  'data_source_count': 3,\n",
       "  'sample_count': 1741.0,\n",
       "  'normal_sample_count': 154446}}"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "omixatlas.omixatlas_summary(\"geo\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Workspaces service "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "from polly.workspaces import Workspaces\n",
    "workspace = Workspaces()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Workspace_id</th>\n",
       "      <th>Workspace_name</th>\n",
       "      <th>status</th>\n",
       "      <th>description</th>\n",
       "      <th>last_modified</th>\n",
       "      <th>tag_names</th>\n",
       "      <th>favourite</th>\n",
       "      <th>watch</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>15323</td>\n",
       "      <td>schema_read_refactoring</td>\n",
       "      <td>active</td>\n",
       "      <td></td>\n",
       "      <td>2023-07-14 08:22:25</td>\n",
       "      <td>[]</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>15293</td>\n",
       "      <td>insert_validation_schema_refactoring</td>\n",
       "      <td>active</td>\n",
       "      <td></td>\n",
       "      <td>2023-06-21 06:42:30</td>\n",
       "      <td>[]</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>15277</td>\n",
       "      <td>validation_ingestion_demo</td>\n",
       "      <td>active</td>\n",
       "      <td></td>\n",
       "      <td>2023-05-31 16:47:57</td>\n",
       "      <td>[]</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>15265</td>\n",
       "      <td>UX_GAPS_VALIDATION</td>\n",
       "      <td>active</td>\n",
       "      <td></td>\n",
       "      <td>2023-05-29 06:38:58</td>\n",
       "      <td>[]</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>15251</td>\n",
       "      <td>validation_dataset_multiple_src_dp</td>\n",
       "      <td>active</td>\n",
       "      <td></td>\n",
       "      <td>2023-05-03 16:52:21</td>\n",
       "      <td>[]</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>2596</td>\n",
       "      <td>epi_test</td>\n",
       "      <td>active</td>\n",
       "      <td>None</td>\n",
       "      <td>2019-07-31 07:03:21</td>\n",
       "      <td>[]</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>2585</td>\n",
       "      <td>quant_test</td>\n",
       "      <td>active</td>\n",
       "      <td>None</td>\n",
       "      <td>2019-07-25 13:43:29</td>\n",
       "      <td>[]</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>2573</td>\n",
       "      <td>metscape</td>\n",
       "      <td>active</td>\n",
       "      <td>None</td>\n",
       "      <td>2019-07-23 09:53:59</td>\n",
       "      <td>[]</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>2565</td>\n",
       "      <td>statemanagement_test</td>\n",
       "      <td>active</td>\n",
       "      <td>None</td>\n",
       "      <td>2019-07-16 05:59:27</td>\n",
       "      <td>[]</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>2544</td>\n",
       "      <td>Karma</td>\n",
       "      <td>active</td>\n",
       "      <td>Karma</td>\n",
       "      <td>2019-07-03 10:52:56</td>\n",
       "      <td>[]</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>62 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Workspace_id                        Workspace_name  status description  \\\n",
       "0          15323               schema_read_refactoring  active               \n",
       "1          15293  insert_validation_schema_refactoring  active               \n",
       "2          15277             validation_ingestion_demo  active               \n",
       "3          15265                    UX_GAPS_VALIDATION  active               \n",
       "4          15251    validation_dataset_multiple_src_dp  active               \n",
       "..           ...                                   ...     ...         ...   \n",
       "57          2596                              epi_test  active        None   \n",
       "58          2585                            quant_test  active        None   \n",
       "59          2573                              metscape  active        None   \n",
       "60          2565                  statemanagement_test  active        None   \n",
       "61          2544                                 Karma  active       Karma   \n",
       "\n",
       "          last_modified tag_names  favourite  watch  \n",
       "0   2023-07-14 08:22:25        []      False  False  \n",
       "1   2023-06-21 06:42:30        []      False  False  \n",
       "2   2023-05-31 16:47:57        []      False  False  \n",
       "3   2023-05-29 06:38:58        []      False  False  \n",
       "4   2023-05-03 16:52:21        []      False  False  \n",
       "..                  ...       ...        ...    ...  \n",
       "57  2019-07-31 07:03:21        []      False  False  \n",
       "58  2019-07-25 13:43:29        []      False  False  \n",
       "59  2019-07-23 09:53:59        []      False  False  \n",
       "60  2019-07-16 05:59:27        []      False  False  \n",
       "61  2019-07-03 10:52:56        []      False  False  \n",
       "\n",
       "[62 rows x 8 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "workspace.fetch_my_workspaces()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:root:Workspace Created !\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'id': 15330,\n",
       " 'name': 'example_test',\n",
       " 'status': 1,\n",
       " 'active': 't',\n",
       " 'description': 'description',\n",
       " 'created_time': '2023-07-24 11:18:04.764928',\n",
       " 'last_modified': '2023-07-24 11:18:04.764928',\n",
       " 'creator': '191',\n",
       " 'project_property': {'type': 'workspaces', 'labels': ''},\n",
       " 'organisation': 1,\n",
       " 'tag_names': []}"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "workspace.create_workspace(\"example_test\",\"description\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
