{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Welcome to Polly Python3 Notebook."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/bin/bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)\n",
      "Defaulting to user installation because normal site-packages is not writeable\n",
      "Looking in indexes: https://pypi.org/simple, http://54.245.179.143:80/\n",
      "Collecting polly-python==0.4.1-omix-refac\n",
      "  Downloading https://elucidatainc.github.io/PublicAssets/builds/polly-python/tests/testpolly/polly_python-0.4.1_omix_refac-py3-none-any.whl (142 kB)\n",
      "\u001b[K     |████████████████████████████████| 142 kB 6.1 MB/s eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: python-jose==3.3.0 in /usr/local/lib/python3.10/site-packages (from polly-python==0.4.1-omix-refac) (3.3.0)\n",
      "Requirement already satisfied: requests==2.25.1 in /usr/local/lib/python3.10/site-packages (from polly-python==0.4.1-omix-refac) (2.25.1)\n",
      "Requirement already satisfied: joblib>=1.2.0 in /usr/local/lib/python3.10/site-packages (from polly-python==0.4.1-omix-refac) (1.2.0)\n",
      "Requirement already satisfied: retrying==1.3.4 in /usr/local/lib/python3.10/site-packages (from polly-python==0.4.1-omix-refac) (1.3.4)\n",
      "Requirement already satisfied: elucidatacmapPy==3.3.4 in /usr/local/lib/python3.10/site-packages (from polly-python==0.4.1-omix-refac) (3.3.4)\n",
      "Requirement already satisfied: beautifulsoup4 in /usr/local/lib/python3.10/site-packages (from polly-python==0.4.1-omix-refac) (4.12.2)\n",
      "Collecting botocore>=1.27.0\n",
      "  Downloading botocore-1.31.5-py3-none-any.whl (11.0 MB)\n",
      "\u001b[K     |████████████████████████████████| 11.0 MB 5.8 MB/s eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: mixpanel==4.10.0 in /usr/local/lib/python3.10/site-packages (from polly-python==0.4.1-omix-refac) (4.10.0)\n",
      "Requirement already satisfied: plotly>=5.0.0 in /usr/local/lib/python3.10/site-packages (from polly-python==0.4.1-omix-refac) (5.14.1)\n",
      "Collecting boto3>=1.24.0\n",
      "  Downloading boto3-1.28.5-py3-none-any.whl (135 kB)\n",
      "\u001b[K     |████████████████████████████████| 135 kB 69.3 MB/s eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: pydantic>=1.8.2 in /usr/local/lib/python3.10/site-packages (from polly-python==0.4.1-omix-refac) (1.10.7)\n",
      "Requirement already satisfied: tqdm==4.65.0 in /usr/local/lib/python3.10/site-packages (from polly-python==0.4.1-omix-refac) (4.65.0)\n",
      "Requirement already satisfied: pandas>=1.3.5 in /usr/local/lib/python3.10/site-packages (from polly-python==0.4.1-omix-refac) (1.4.1)\n",
      "Requirement already satisfied: rst2txt==1.1.0 in /usr/local/lib/python3.10/site-packages (from polly-python==0.4.1-omix-refac) (1.1.0)\n",
      "Requirement already satisfied: elucidatacloudpathlib==0.6.6 in /usr/local/lib/python3.10/site-packages (from polly-python==0.4.1-omix-refac) (0.6.6)\n",
      "Collecting cryptography<=38.0.0,>=37.0.1\n",
      "  Downloading cryptography-38.0.0-cp36-abi3-manylinux_2_28_x86_64.whl (4.2 MB)\n",
      "\u001b[K     |████████████████████████████████| 4.2 MB 70.2 MB/s eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: Deprecated>=1.2.12 in /usr/local/lib/python3.10/site-packages (from polly-python==0.4.1-omix-refac) (1.2.13)\n",
      "Collecting polly-validator==0.0.2\n",
      "  Downloading polly_validator-0.0.2-py3-none-any.whl (29 kB)\n",
      "Requirement already satisfied: pytest>=6.2.5 in /usr/local/lib/python3.10/site-packages (from polly-python==0.4.1-omix-refac) (7.3.1)\n",
      "Requirement already satisfied: datapane>=0.10.3 in /usr/local/lib/python3.10/site-packages (from polly-python==0.4.1-omix-refac) (0.16.4)\n",
      "Requirement already satisfied: six in /usr/local/lib/python3.10/site-packages (from elucidatacmapPy==3.3.4->polly-python==0.4.1-omix-refac) (1.16.0)\n",
      "Requirement already satisfied: h5py>=2.6.0 in /usr/local/lib/python3.10/site-packages (from elucidatacmapPy==3.3.4->polly-python==0.4.1-omix-refac) (3.8.0)\n",
      "Requirement already satisfied: numpy>=1.11.2 in /usr/local/lib/python3.10/site-packages (from elucidatacmapPy==3.3.4->polly-python==0.4.1-omix-refac) (1.22.3)\n",
      "Requirement already satisfied: urllib3 in /usr/local/lib/python3.10/site-packages (from mixpanel==4.10.0->polly-python==0.4.1-omix-refac) (1.26.6)\n",
      "Requirement already satisfied: appdirs==1.4.4 in /usr/local/lib/python3.10/site-packages (from polly-validator==0.0.2->polly-python==0.4.1-omix-refac) (1.4.4)\n",
      "Requirement already satisfied: tabulate==0.8.10 in /usr/local/lib/python3.10/site-packages (from polly-validator==0.0.2->polly-python==0.4.1-omix-refac) (0.8.10)\n",
      "Requirement already satisfied: rapidfuzz==2.1.1 in /usr/local/lib/python3.10/site-packages (from polly-validator==0.0.2->polly-python==0.4.1-omix-refac) (2.1.1)\n",
      "Requirement already satisfied: pronto in /usr/local/lib/python3.10/site-packages (from polly-validator==0.0.2->polly-python==0.4.1-omix-refac) (2.5.1)\n",
      "Requirement already satisfied: ecdsa!=0.15 in /usr/local/lib/python3.10/site-packages (from python-jose==3.3.0->polly-python==0.4.1-omix-refac) (0.18.0)\n",
      "Requirement already satisfied: rsa in /usr/local/lib/python3.10/site-packages (from python-jose==3.3.0->polly-python==0.4.1-omix-refac) (4.7.2)\n",
      "Requirement already satisfied: pyasn1 in /usr/local/lib/python3.10/site-packages (from python-jose==3.3.0->polly-python==0.4.1-omix-refac) (0.4.8)\n",
      "Requirement already satisfied: jarowinkler<1.1.0,>=1.0.3 in /usr/local/lib/python3.10/site-packages (from rapidfuzz==2.1.1->polly-validator==0.0.2->polly-python==0.4.1-omix-refac) (1.0.5)\n",
      "Requirement already satisfied: chardet<5,>=3.0.2 in /usr/local/lib/python3.10/site-packages (from requests==2.25.1->polly-python==0.4.1-omix-refac) (4.0.0)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.10/site-packages (from requests==2.25.1->polly-python==0.4.1-omix-refac) (2021.5.30)\n",
      "Requirement already satisfied: idna<3,>=2.5 in /usr/local/lib/python3.10/site-packages (from requests==2.25.1->polly-python==0.4.1-omix-refac) (2.10)\n",
      "Requirement already satisfied: docutils>=0.11 in /usr/local/lib/python3.10/site-packages (from rst2txt==1.1.0->polly-python==0.4.1-omix-refac) (0.15.2)\n",
      "Requirement already satisfied: pygments in /usr/local/lib/python3.10/site-packages (from rst2txt==1.1.0->polly-python==0.4.1-omix-refac) (2.10.0)\n",
      "Collecting s3transfer<0.7.0,>=0.6.0\n",
      "  Downloading s3transfer-0.6.1-py3-none-any.whl (79 kB)\n",
      "\u001b[K     |████████████████████████████████| 79 kB 15.9 MB/s eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: jmespath<2.0.0,>=0.7.1 in /usr/local/lib/python3.10/site-packages (from boto3>=1.24.0->polly-python==0.4.1-omix-refac) (0.10.0)\n",
      "Requirement already satisfied: python-dateutil<3.0.0,>=2.1 in /usr/local/lib/python3.10/site-packages (from botocore>=1.27.0->polly-python==0.4.1-omix-refac) (2.8.1)\n",
      "Requirement already satisfied: cffi>=1.12 in /usr/local/lib/python3.10/site-packages (from cryptography<=38.0.0,>=37.0.1->polly-python==0.4.1-omix-refac) (1.15.1)\n",
      "Requirement already satisfied: pycparser in /usr/local/lib/python3.10/site-packages (from cffi>=1.12->cryptography<=38.0.0,>=37.0.1->polly-python==0.4.1-omix-refac) (2.21)\n",
      "Requirement already satisfied: furl<3.0.0,>=2.0.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==0.4.1-omix-refac) (2.1.3)\n",
      "Requirement already satisfied: nbconvert>=5.6.1 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==0.4.1-omix-refac) (6.3.0)\n",
      "Requirement already satisfied: datacommons<2.0.0,>=1.4.3 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==0.4.1-omix-refac) (1.4.3)\n",
      "Requirement already satisfied: cheroot<10.0.0,>=9.0.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==0.4.1-omix-refac) (9.0.0)\n",
      "Requirement already satisfied: dulwich<0.22.0,>=0.20.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==0.4.1-omix-refac) (0.21.3)\n",
      "Requirement already satisfied: posthog<3.0.0,>=1.4.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==0.4.1-omix-refac) (2.5.0)\n",
      "Requirement already satisfied: requests-toolbelt<0.11.0,>=0.9.1 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==0.4.1-omix-refac) (0.10.1)\n",
      "Requirement already satisfied: boltons<22.0.0,>=20.0.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==0.4.1-omix-refac) (21.0.0)\n",
      "Requirement already satisfied: lxml<5.0.0,>=4.0.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==0.4.1-omix-refac) (4.9.2)\n",
      "Requirement already satisfied: click-spinner<0.2.0,>=0.1.8 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==0.4.1-omix-refac) (0.1.10)\n",
      "Requirement already satisfied: dominate<3.0.0,>=2.4.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==0.4.1-omix-refac) (2.7.0)\n",
      "Requirement already satisfied: jsonschema<5.0.0,>=3.2.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==0.4.1-omix-refac) (4.2.1)\n",
      "Requirement already satisfied: pyngrok<6.0.0,>=5.2.1 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==0.4.1-omix-refac) (5.2.3)\n",
      "Requirement already satisfied: pyarrow<11.0.0,>=6.0.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==0.4.1-omix-refac) (10.0.1)\n",
      "Requirement already satisfied: micawber>=0.5.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==0.4.1-omix-refac) (0.5.4)\n",
      "Requirement already satisfied: importlib_resources<6.0.0,>=3.0.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==0.4.1-omix-refac) (5.12.0)\n",
      "Requirement already satisfied: click<9.0.0,>=7.1.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==0.4.1-omix-refac) (8.1.3)\n",
      "Requirement already satisfied: colorlog<7.0.0,>=4.1.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==0.4.1-omix-refac) (6.7.0)\n",
      "Requirement already satisfied: toolz<0.13.0,>=0.11.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==0.4.1-omix-refac) (0.12.0)\n",
      "Requirement already satisfied: PyYAML<7.0.0,>=5.4.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==0.4.1-omix-refac) (5.4.1)\n",
      "Requirement already satisfied: datacommons-pandas<0.0.4,>=0.0.3 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==0.4.1-omix-refac) (0.0.3)\n",
      "Requirement already satisfied: dacite<2.0.0,>=1.0.2 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==0.4.1-omix-refac) (1.8.0)\n",
      "Requirement already satisfied: multimethod<2.0,>=1.9 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==0.4.1-omix-refac) (1.9.1)\n",
      "Requirement already satisfied: vega-datasets<1.0.0,>=0.9.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==0.4.1-omix-refac) (0.9.0)\n",
      "Requirement already satisfied: altair<5.0.0,>=4.0.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==0.4.1-omix-refac) (4.2.2)\n",
      "Requirement already satisfied: typing-extensions<5.0.0,>=4.4.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==0.4.1-omix-refac) (4.5.0)\n",
      "Requirement already satisfied: glom<24.0.0,>=22.1.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==0.4.1-omix-refac) (23.3.0)\n",
      "Requirement already satisfied: packaging<24.0.0,>=21.0.0 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==0.4.1-omix-refac) (23.1)\n",
      "Requirement already satisfied: ipynbname<2022.0.0,>=2021.3.2 in /usr/local/lib/python3.10/site-packages (from datapane>=0.10.3->polly-python==0.4.1-omix-refac) (2021.3.2)\n",
      "Requirement already satisfied: jinja2 in /usr/local/lib/python3.10/site-packages (from altair<5.0.0,>=4.0.0->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (3.0.3)\n",
      "Requirement already satisfied: entrypoints in /usr/local/lib/python3.10/site-packages (from altair<5.0.0,>=4.0.0->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (0.3)\n",
      "Requirement already satisfied: more-itertools>=2.6 in /usr/local/lib/python3.10/site-packages (from cheroot<10.0.0,>=9.0.0->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (9.1.0)\n",
      "Requirement already satisfied: jaraco.functools in /usr/local/lib/python3.10/site-packages (from cheroot<10.0.0,>=9.0.0->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (3.6.0)\n",
      "Requirement already satisfied: wrapt<2,>=1.10 in /usr/local/lib/python3.10/site-packages (from Deprecated>=1.2.12->polly-python==0.4.1-omix-refac) (1.15.0)\n",
      "Requirement already satisfied: orderedmultidict>=1.0.1 in /usr/local/lib/python3.10/site-packages (from furl<3.0.0,>=2.0.0->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (1.0.1)\n",
      "Requirement already satisfied: face==20.1.1 in /usr/local/lib/python3.10/site-packages (from glom<24.0.0,>=22.1.0->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (20.1.1)\n",
      "Requirement already satisfied: attrs in /usr/local/lib/python3.10/site-packages (from glom<24.0.0,>=22.1.0->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (21.2.0)\n",
      "Requirement already satisfied: ipykernel in /usr/local/lib/python3.10/site-packages (from ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (6.13.1)\n",
      "Requirement already satisfied: pyrsistent!=0.17.0,!=0.17.1,!=0.17.2,>=0.14.0 in /usr/local/lib/python3.10/site-packages (from jsonschema<5.0.0,>=3.2.0->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (0.18.0)\n",
      "Requirement already satisfied: traitlets>=5.0 in /usr/local/lib/python3.10/site-packages (from nbconvert>=5.6.1->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (5.1.1)\n",
      "Requirement already satisfied: testpath in /usr/local/lib/python3.10/site-packages (from nbconvert>=5.6.1->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (0.5.0)\n",
      "Requirement already satisfied: jupyterlab-pygments in /usr/local/lib/python3.10/site-packages (from nbconvert>=5.6.1->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (0.2.2)\n",
      "Requirement already satisfied: nbformat>=4.4 in /usr/local/lib/python3.10/site-packages (from nbconvert>=5.6.1->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (5.1.3)\n",
      "Requirement already satisfied: pandocfilters>=1.4.1 in /usr/local/lib/python3.10/site-packages (from nbconvert>=5.6.1->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (1.5.0)\n",
      "Requirement already satisfied: mistune<2,>=0.8.1 in /usr/local/lib/python3.10/site-packages (from nbconvert>=5.6.1->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (0.8.4)\n",
      "Requirement already satisfied: jupyter-core in /usr/local/lib/python3.10/site-packages (from nbconvert>=5.6.1->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (4.9.1)\n",
      "Requirement already satisfied: defusedxml in /usr/local/lib/python3.10/site-packages (from nbconvert>=5.6.1->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (0.7.1)\n",
      "Requirement already satisfied: nbclient<0.6.0,>=0.5.0 in /usr/local/lib/python3.10/site-packages (from nbconvert>=5.6.1->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (0.5.13)\n",
      "Requirement already satisfied: bleach in /usr/local/lib/python3.10/site-packages (from nbconvert>=5.6.1->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (4.1.0)\n",
      "Requirement already satisfied: MarkupSafe>=2.0 in /usr/local/lib/python3.10/site-packages (from jinja2->altair<5.0.0,>=4.0.0->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (2.0.1)\n",
      "Requirement already satisfied: jupyter-client>=6.1.5 in /usr/local/lib/python3.10/site-packages (from nbclient<0.6.0,>=0.5.0->nbconvert>=5.6.1->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (6.1.12)\n",
      "Requirement already satisfied: nest-asyncio in /usr/local/lib/python3.10/site-packages (from nbclient<0.6.0,>=0.5.0->nbconvert>=5.6.1->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (1.5.6)\n",
      "Requirement already satisfied: pyzmq>=13 in /usr/local/lib/python3.10/site-packages (from jupyter-client>=6.1.5->nbclient<0.6.0,>=0.5.0->nbconvert>=5.6.1->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (22.3.0)\n",
      "Requirement already satisfied: tornado>=4.1 in /usr/local/lib/python3.10/site-packages (from jupyter-client>=6.1.5->nbclient<0.6.0,>=0.5.0->nbconvert>=5.6.1->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (6.1)\n",
      "Requirement already satisfied: ipython-genutils in /usr/local/lib/python3.10/site-packages (from nbformat>=4.4->nbconvert>=5.6.1->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (0.2.0)\n",
      "Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.10/site-packages (from pandas>=1.3.5->polly-python==0.4.1-omix-refac) (2021.3)\n",
      "Requirement already satisfied: tenacity>=6.2.0 in /usr/local/lib/python3.10/site-packages (from plotly>=5.0.0->polly-python==0.4.1-omix-refac) (8.2.2)\n",
      "Requirement already satisfied: monotonic>=1.5 in /usr/local/lib/python3.10/site-packages (from posthog<3.0.0,>=1.4.0->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (1.6)\n",
      "Requirement already satisfied: backoff>=1.10.0 in /usr/local/lib/python3.10/site-packages (from posthog<3.0.0,>=1.4.0->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (2.2.1)\n",
      "Requirement already satisfied: tomli>=1.0.0 in /usr/local/lib/python3.10/site-packages (from pytest>=6.2.5->polly-python==0.4.1-omix-refac) (2.0.1)\n",
      "Requirement already satisfied: iniconfig in /usr/local/lib/python3.10/site-packages (from pytest>=6.2.5->polly-python==0.4.1-omix-refac) (2.0.0)\n",
      "Requirement already satisfied: exceptiongroup>=1.0.0rc8 in /usr/local/lib/python3.10/site-packages (from pytest>=6.2.5->polly-python==0.4.1-omix-refac) (1.1.1)\n",
      "Requirement already satisfied: pluggy<2.0,>=0.12 in /usr/local/lib/python3.10/site-packages (from pytest>=6.2.5->polly-python==0.4.1-omix-refac) (1.0.0)\n",
      "Requirement already satisfied: soupsieve>1.2 in /usr/local/lib/python3.10/site-packages (from beautifulsoup4->polly-python==0.4.1-omix-refac) (2.4.1)\n",
      "Requirement already satisfied: webencodings in /usr/local/lib/python3.10/site-packages (from bleach->nbconvert>=5.6.1->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (0.5.1)\n",
      "Requirement already satisfied: ipython>=7.23.1 in /usr/local/lib/python3.10/site-packages (from ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (8.2.0)\n",
      "Requirement already satisfied: debugpy>=1.0 in /usr/local/lib/python3.10/site-packages (from ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (1.6.7)\n",
      "Requirement already satisfied: matplotlib-inline>=0.1 in /usr/local/lib/python3.10/site-packages (from ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (0.1.3)\n",
      "Requirement already satisfied: psutil in /usr/local/lib/python3.10/site-packages (from ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (5.9.5)\n",
      "Requirement already satisfied: setuptools>=18.5 in /usr/local/lib/python3.10/site-packages (from ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (58.1.0)\n",
      "Requirement already satisfied: backcall in /usr/local/lib/python3.10/site-packages (from ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (0.2.0)\n",
      "Requirement already satisfied: pickleshare in /usr/local/lib/python3.10/site-packages (from ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (0.7.5)\n",
      "Requirement already satisfied: prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0 in /usr/local/lib/python3.10/site-packages (from ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (3.0.22)\n",
      "Requirement already satisfied: pexpect>4.3 in /usr/local/lib/python3.10/site-packages (from ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (4.8.0)\n",
      "Requirement already satisfied: decorator in /usr/local/lib/python3.10/site-packages (from ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (5.1.0)\n",
      "Requirement already satisfied: stack-data in /usr/local/lib/python3.10/site-packages (from ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (0.6.2)\n",
      "Requirement already satisfied: jedi>=0.16 in /usr/local/lib/python3.10/site-packages (from ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (0.18.1)\n",
      "Requirement already satisfied: parso<0.9.0,>=0.8.0 in /usr/local/lib/python3.10/site-packages (from jedi>=0.16->ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (0.8.2)\n",
      "Requirement already satisfied: ptyprocess>=0.5 in /usr/local/lib/python3.10/site-packages (from pexpect>4.3->ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (0.7.0)\n",
      "Requirement already satisfied: wcwidth in /usr/local/lib/python3.10/site-packages (from prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0->ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (0.2.5)\n",
      "Requirement already satisfied: fastobo~=0.12.1 in /usr/local/lib/python3.10/site-packages (from pronto->polly-validator==0.0.2->polly-python==0.4.1-omix-refac) (0.12.2)\n",
      "Requirement already satisfied: networkx~=2.3 in /usr/local/lib/python3.10/site-packages (from pronto->polly-validator==0.0.2->polly-python==0.4.1-omix-refac) (2.8.8)\n",
      "Requirement already satisfied: executing>=1.2.0 in /usr/local/lib/python3.10/site-packages (from stack-data->ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (1.2.0)\n",
      "Requirement already satisfied: pure-eval in /usr/local/lib/python3.10/site-packages (from stack-data->ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (0.2.2)\n",
      "Requirement already satisfied: asttokens>=2.1.0 in /usr/local/lib/python3.10/site-packages (from stack-data->ipython>=7.23.1->ipykernel->ipynbname<2022.0.0,>=2021.3.2->datapane>=0.10.3->polly-python==0.4.1-omix-refac) (2.2.1)\n",
      "Installing collected packages: botocore, s3transfer, polly-validator, cryptography, boto3, polly-python\n",
      "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
      "awscli 1.22.74 requires botocore==1.24.19, but you have botocore 1.31.5 which is incompatible.\n",
      "awscli 1.22.74 requires s3transfer<0.6.0,>=0.5.0, but you have s3transfer 0.6.1 which is incompatible.\u001b[0m\n",
      "Successfully installed boto3-1.28.5 botocore-1.31.5 cryptography-38.0.0 polly-python-0.4.1 polly-validator-0.0.2 s3transfer-0.6.1\n",
      "\u001b[33mWARNING: You are using pip version 21.2.4; however, version 23.2 is available.\n",
      "You should consider upgrading via the '/usr/local/bin/python3.10 -m pip install --upgrade pip' command.\u001b[0m\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "pip install https://elucidatainc.github.io/PublicAssets/builds/polly-python/tests/testpolly/polly_python-0.4.1_omix_refac-py3-none-any.whl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "from polly.auth import Polly\n",
    "from polly.omixatlas import OmixAtlas\n",
    "AUTH_TOKEN = \"eyJjdHkiOiJKV1QiLCJlbmMiOiJBMjU2R0NNIiwiYWxnIjoiUlNBLU9BRVAifQ.O4frloYPaTBaeMHUCF4-FMT91xTv3NERwJAVemffOTCRYxQ0gIYMZ_x-NODf39SEjcc0Kht5tqxxJ9FHx7IQOZ8m4MrawIdZBx7fwPGq4l3MJ-bi-p9mE6wpn8_EXAJuD53ebwUIl5bTanvAt-tJui25EX7OmBFr2dQ548a_kSLKIXhraNlZncemri6fQUUPKZVY1q3-50Px3tuV7gFnEgfHmXe_CepU7VUB1KvHgQakXS4Wv60VUSWygDp6aO-pw9Cu1_3bvsbLPo9Yk6ahOtjU-qQMilW5wn7r38cUmV_IllNAnxcsGTnVvKJDSbEUCky5IJ8uaVsCGJbpkPvBow.yqEFPn86xLGHJ0GN.5BTDq5KzWY2PzCbHAd5xXtsxk5MsHRU43o8tG-w2qhItqBymE5y3CDIf77wTCx4YKxVODTNmwYKC47di2uP0gN0zJCCGE8b3CbB5jFIjRVP4rnxP6KKu_BdyNaNQB36nw-wun-Q76Qq7viH0pHtyy9sq-yme2KK96JUPLgUFXxSoeziYnhH8JUM5UFMpw31SOwpUEG4XaDE0eM7_yDPAl69-7MumTmi1xkUEI30AZzDLp8NihN0ZxtvUvIstg07OmXQKD7dTB3ImJi089rD8VwQyNinSZxdR-Z55kC093mkNT4CLfYsN8iL0HNl6tfemRC1hi9WVuqaWX3OEMby_W_19aw2h6J93zU2bnAGW1qNe1OS9Der_Dzyc5iM_whRKggKGX8Rzzz0psOqNH1B_SBX6dULewEE6HZVfrsg5jKg2Zew84_Duyth-XTe1_F0XVuGmqHMUuy8mUli6Wcc16Rfd9aTNdj9Xm8K6zHDvXkDI_zC4GjBYdD9p7IaudoIjcMaDZndWKyh64G2it4kwUJIsbD8TXtGN3Ob9oH-4nX19IdYB8ShGAEnZqPEH3r2bB2c3NDH83vgYPoKZn0z0XpMv-JpmwRcde4HVYmJuFIZ-d6c6hOf3AM2v1SyscsQdeuPaJGAAHgQJFgZ5zl1ZWACuX3OKCgXKSb4ZoFvK2eKpPDCF4iQNBVw1kf_9FjD5yjDQXcPS3KfcY1AOLpFYN3CQ9RdJER_PMj_VL5BkShJS5HS26N_7wkfPKXAvbf1UtJmHEpZL5eAZ1rFGkwFNgqXKD1BwUd5CUMBNkTCJLicL36ClxtrN7m32_0ZsTNNAGHIWM2B7410nSZg1OOlXZ2CvhuPdPEsjsDIdZwtePWMfxrVBgxWMC4sz82R-VSqxV72rAmOiv-O-ghiw2s9fvyXknUu-6XyG-LreZ2Nu-BlkIG_oOS6Gutef63GN3tnPyjIgHOc7rJ8N4UjqlmqaABSzNMHnoDQDs-MynfIhX7hGIVqILN9z0BmUzv1XTSiYGsvjL_lrQ2dmoW5rxA8k7p1nV1q1HsjWz9oJnKF5Vuu3Q1jJdNfHMpWl1E6q6oLZGFqn1ZhGc86TnzjPN6IEYzZyHPBfhlK50-LqzPSIyy4ngAuSEFT861JYW8n7SPVWY50mF4M3zLzdqwu07zqPDBsPNAcOgcevDlMt_CNlitmLRUpebVJmsTK7yTSoJKwg7GJUAUXqy_9HDppotBNg58jl0NwnYNnn_-sOjZCOTnRRwFV5tKA8PsO7-Z-E_Zbop7TLouOWy1u4UDAitVSOSdWaHBjQrWJwQAZT7_C1nP_w5CbmCNN4uswKUxQGJegZSjqdRYj2HDoT.UxuT_AZTM0tgKrQzW0TY_Q\"\n",
    "Polly.auth(AUTH_TOKEN, \"testpolly\")\n",
    "omix = OmixAtlas()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'data': {'repo_name': 'geo',\n",
       "  'repo_id': '1643016586529',\n",
       "  'indexes': {'files': 'geo_files',\n",
       "   'gct_col_metadata': 'geo_gct_col_metadata'},\n",
       "  'v2_indexes': {'files': 'geo_files',\n",
       "   'gct_col_metadata': 'geo_gct_col_metadata'},\n",
       "  'linked_workspace_id': '15196',\n",
       "  'sources': [{'geo': 5}, {'source': 1}, {'vfv': 1}],\n",
       "  'datatypes': [{'transcriptomics': 6}, {'': 1}],\n",
       "  'dataset_count': 7,\n",
       "  'disease_count': 12,\n",
       "  'tissue_count': 8,\n",
       "  'organism_count': 3,\n",
       "  'cell_line_count': 5,\n",
       "  'cell_type_count': 6,\n",
       "  'drug_count': 2,\n",
       "  'data_type_count': 2,\n",
       "  'data_source_count': 3,\n",
       "  'sample_count': 1741.0,\n",
       "  'normal_sample_count': 154446}}"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "omix.omixatlas_summary(\"geo\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'data': [{'repo_name': 'geo',\n",
       "   'repo_id': '1643016586529',\n",
       "   'indexes': {'files': 'geo_files',\n",
       "    'gct_col_metadata': 'geo_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'geo_files',\n",
       "    'gct_col_metadata': 'geo_gct_col_metadata'},\n",
       "   'linked_workspace_id': '15196',\n",
       "   'sources': [{'geo': 5}, {'source': 1}, {'vfv': 1}],\n",
       "   'datatypes': [{'transcriptomics': 6}, {'': 1}],\n",
       "   'dataset_count': 7,\n",
       "   'disease_count': 12,\n",
       "   'tissue_count': 8,\n",
       "   'organism_count': 3,\n",
       "   'cell_line_count': 5,\n",
       "   'cell_type_count': 6,\n",
       "   'drug_count': 2,\n",
       "   'data_type_count': 2,\n",
       "   'data_source_count': 3,\n",
       "   'sample_count': 1741.0,\n",
       "   'normal_sample_count': 154446},\n",
       "  {'repo_id': '1687428145396',\n",
       "   'repo_name': 'new_omix_for_refac_2',\n",
       "   'indexes': {},\n",
       "   'v2_indexes': {},\n",
       "   'sources': [],\n",
       "   'datatypes': [],\n",
       "   'dataset_count': 0,\n",
       "   'disease_count': 0,\n",
       "   'tissue_count': 0,\n",
       "   'organism_count': 0,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 0,\n",
       "   'data_source_count': 0,\n",
       "   'sample_count': 0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1687428192072',\n",
       "   'repo_name': 'new_omix_for_refac_3',\n",
       "   'indexes': {},\n",
       "   'v2_indexes': {},\n",
       "   'sources': [],\n",
       "   'datatypes': [],\n",
       "   'dataset_count': 0,\n",
       "   'disease_count': 0,\n",
       "   'tissue_count': 0,\n",
       "   'organism_count': 0,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 0,\n",
       "   'data_source_count': 0,\n",
       "   'sample_count': 0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1688711659012',\n",
       "   'repo_name': 'july_test',\n",
       "   'v2_indexes': {},\n",
       "   'sources': [],\n",
       "   'datatypes': [],\n",
       "   'dataset_count': 0,\n",
       "   'disease_count': 0,\n",
       "   'tissue_count': 0,\n",
       "   'organism_count': 0,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 0,\n",
       "   'data_source_count': 0,\n",
       "   'sample_count': 0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1667375323145',\n",
       "   'repo_name': 'bbio_omixatlas_only_h5seurat',\n",
       "   'indexes': {'files': 'bbio_omixatlas_only_h5seurat_files'},\n",
       "   'v2_indexes': {'files': 'bbio_omixatlas_only_h5seurat_files'},\n",
       "   'sources': [{'branchbio': 4}],\n",
       "   'datatypes': [{'single cell multiomics': 4}],\n",
       "   'dataset_count': 4,\n",
       "   'disease_count': 2,\n",
       "   'tissue_count': 3,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 1,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 4.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1681886680238',\n",
       "   'repo_name': 'hybrid_testing_repo',\n",
       "   'indexes': {},\n",
       "   'v2_indexes': {},\n",
       "   'sources': [],\n",
       "   'datatypes': [],\n",
       "   'dataset_count': 0,\n",
       "   'disease_count': 0,\n",
       "   'tissue_count': 0,\n",
       "   'organism_count': 0,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 0,\n",
       "   'data_source_count': 0,\n",
       "   'sample_count': 0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1685621126191',\n",
       "   'repo_name': 'gdx_demo_atlas',\n",
       "   'indexes': {'files': 'gdx_demo_atlas_files',\n",
       "    'gct_col_metadata': 'gdx_demo_atlas_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'gdx_demo_atlas_files',\n",
       "    'gct_col_metadata': 'gdx_demo_atlas_gct_col_metadata'},\n",
       "   'data_type': 'bulk_rna_seq',\n",
       "   'sources': [{'geo': 8}],\n",
       "   'datatypes': [{'': 8}],\n",
       "   'dataset_count': 8,\n",
       "   'disease_count': 24,\n",
       "   'tissue_count': 32,\n",
       "   'organism_count': 8,\n",
       "   'cell_line_count': 2,\n",
       "   'cell_type_count': 5,\n",
       "   'drug_count': 2,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 114.0,\n",
       "   'normal_sample_count': 17},\n",
       "  {'repo_id': '1678794482481',\n",
       "   'repo_name': 'csv_oa_test',\n",
       "   'v2_indexes': {'files': 'csv_oa_test_files'},\n",
       "   'linked_workspace_id': '15184',\n",
       "   'sources': [{'src_1': 3}, {'all': 2}],\n",
       "   'datatypes': [{'all': 4}, {'datatype_1': 1}],\n",
       "   'dataset_count': 5,\n",
       "   'disease_count': 0,\n",
       "   'tissue_count': 0,\n",
       "   'organism_count': 0,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 2,\n",
       "   'data_source_count': 2,\n",
       "   'sample_count': 0.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1622526550765',\n",
       "   'repo_name': 'liver_atlas',\n",
       "   'indexes': {'files': 'liver_atlas_files',\n",
       "    'gct_col_metadata': 'liver_atlas_gct_col_metadata',\n",
       "    'h5ad_col_metadata': 'liver_atlas_h5ad_col_metadata',\n",
       "    'h5ad_row_metadata': 'liver_atlas_h5ad_row_metadata'},\n",
       "   'v2_indexes': {'files': 'liver_atlas_files',\n",
       "    'gct_col_metadata': 'liver_atlas_gct_col_metadata',\n",
       "    'h5ad_col_metadata': 'liver_atlas_h5ad_col_metadata',\n",
       "    'h5ad_row_metadata': 'liver_atlas_h5ad_row_metadata'},\n",
       "   'sources': [{'geo': 5}],\n",
       "   'datatypes': [{'transcriptomics': 3}, {'trancriptomics': 1}],\n",
       "   'dataset_count': 5,\n",
       "   'disease_count': 2,\n",
       "   'tissue_count': 4,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 1,\n",
       "   'cell_type_count': 3,\n",
       "   'drug_count': 3,\n",
       "   'data_type_count': 2,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 86.0,\n",
       "   'normal_sample_count': 15260},\n",
       "  {'repo_id': '1666789252487',\n",
       "   'repo_name': 'bbio_omixatlas',\n",
       "   'indexes': {'files': 'bbio_omixatlas_files'},\n",
       "   'v2_indexes': {'files': 'bbio_omixatlas_files'},\n",
       "   'sources': [{'branchbio': 1}],\n",
       "   'datatypes': [{'single cell multiomics': 1}],\n",
       "   'dataset_count': 1,\n",
       "   'disease_count': 1,\n",
       "   'tissue_count': 1,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 1,\n",
       "   'cell_type_count': 1,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 1.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1663919116821',\n",
       "   'repo_name': 'ai_in_drug_discovery_oa',\n",
       "   'indexes': {'files': 'ai_in_drug_discovery_oa_files',\n",
       "    'gct_col_metadata': 'ai_in_drug_discovery_oa_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'ai_in_drug_discovery_oa_files',\n",
       "    'gct_col_metadata': 'ai_in_drug_discovery_oa_gct_col_metadata'},\n",
       "   'sources': [{'geo': 129}],\n",
       "   'datatypes': [{'transcriptomics': 129}],\n",
       "   'dataset_count': 129,\n",
       "   'disease_count': 68,\n",
       "   'tissue_count': 43,\n",
       "   'organism_count': 6,\n",
       "   'cell_line_count': 28,\n",
       "   'cell_type_count': 61,\n",
       "   'drug_count': 55,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 3649.0,\n",
       "   'normal_sample_count': 1574},\n",
       "  {'repo_id': '1687411548568',\n",
       "   'repo_name': 'junee_test',\n",
       "   'v2_indexes': {'files': 'junee_test_files',\n",
       "    'gct_col_metadata': 'junee_test_gct_col_metadata'},\n",
       "   'sources': [{'geo': 21}],\n",
       "   'datatypes': [{'raw counts transcriptomics': 21}],\n",
       "   'dataset_count': 21,\n",
       "   'disease_count': 19,\n",
       "   'tissue_count': 0,\n",
       "   'organism_count': 4,\n",
       "   'cell_line_count': 10,\n",
       "   'cell_type_count': 13,\n",
       "   'drug_count': 3,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 381.0,\n",
       "   'normal_sample_count': 94},\n",
       "  {'repo_id': '1679482518578',\n",
       "   'repo_name': 'details_page',\n",
       "   'indexes': {'files': 'details_page_files',\n",
       "    'gct_col_metadata': 'details_page_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'details_page_files',\n",
       "    'gct_col_metadata': 'details_page_gct_col_metadata'},\n",
       "   'sources': [{'geo': 27}],\n",
       "   'datatypes': [{'raw counts transcriptomics': 27}],\n",
       "   'dataset_count': 27,\n",
       "   'disease_count': 36,\n",
       "   'tissue_count': 28,\n",
       "   'organism_count': 3,\n",
       "   'cell_line_count': 6,\n",
       "   'cell_type_count': 17,\n",
       "   'drug_count': 8,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 4190.0,\n",
       "   'normal_sample_count': 2062},\n",
       "  {'repo_id': '1689071945246',\n",
       "   'repo_name': 'gdx_test_s',\n",
       "   'v2_indexes': {'files': 'gdx_test_s_files',\n",
       "    'gct_col_metadata': 'gdx_test_s_gct_col_metadata'},\n",
       "   'sources': [{'geo': 1}],\n",
       "   'datatypes': [{'': 1}],\n",
       "   'dataset_count': 1,\n",
       "   'disease_count': 1,\n",
       "   'tissue_count': 1,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 1,\n",
       "   'cell_type_count': 2,\n",
       "   'drug_count': 1,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 6.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1679648155740',\n",
       "   'repo_name': 'bulkrnaseq_staging_oa',\n",
       "   'v2_indexes': {},\n",
       "   'sources': [],\n",
       "   'datatypes': [],\n",
       "   'dataset_count': 0,\n",
       "   'disease_count': 0,\n",
       "   'tissue_count': 0,\n",
       "   'organism_count': 0,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 0,\n",
       "   'data_source_count': 0,\n",
       "   'sample_count': 0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1687428220754',\n",
       "   'repo_name': 'new_omix_for_refac_4',\n",
       "   'indexes': {},\n",
       "   'v2_indexes': {},\n",
       "   'sources': [],\n",
       "   'datatypes': [],\n",
       "   'dataset_count': 0,\n",
       "   'disease_count': 0,\n",
       "   'tissue_count': 0,\n",
       "   'organism_count': 0,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 0,\n",
       "   'data_source_count': 0,\n",
       "   'sample_count': 0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1669033066992',\n",
       "   'repo_name': 'destination_atlas',\n",
       "   'indexes': {'files': 'destination_atlas_files',\n",
       "    'gct_col_metadata': 'destination_atlas_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'destination_atlas_files',\n",
       "    'gct_col_metadata': 'destination_atlas_gct_col_metadata'},\n",
       "   'sources': [{'geo': 33}],\n",
       "   'datatypes': [{'raw counts transcriptomics': 30},\n",
       "    {'transcriptomics': 2},\n",
       "    {'raw_counts_transcriptomics': 1}],\n",
       "   'dataset_count': 33,\n",
       "   'disease_count': 35,\n",
       "   'tissue_count': 27,\n",
       "   'organism_count': 3,\n",
       "   'cell_line_count': 25,\n",
       "   'cell_type_count': 23,\n",
       "   'drug_count': 56,\n",
       "   'data_type_count': 3,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 712.0,\n",
       "   'normal_sample_count': 467},\n",
       "  {'repo_id': '1663151941472',\n",
       "   'repo_name': 'c_oa',\n",
       "   'indexes': {'files': 'c_oa_files'},\n",
       "   'v2_indexes': {'files': 'c_oa_files'},\n",
       "   'linked_workspace_id': 14918,\n",
       "   'sources': [{'': 10}],\n",
       "   'datatypes': [{'drug response': 10}],\n",
       "   'dataset_count': 10,\n",
       "   'disease_count': 3,\n",
       "   'tissue_count': 1,\n",
       "   'organism_count': 0,\n",
       "   'cell_line_count': 5,\n",
       "   'cell_type_count': 1,\n",
       "   'drug_count': 1,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 10.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1677039559242',\n",
       "   'repo_name': 'diy_poa',\n",
       "   'indexes': {'files': 'diy_poa_files',\n",
       "    'h5ad_col_metadata': 'diy_poa_h5ad_col_metadata'},\n",
       "   'v2_indexes': {'files': 'diy_poa_files',\n",
       "    'h5ad_col_metadata': 'diy_poa_h5ad_col_metadata'},\n",
       "   'sources': [{'geo': 3}],\n",
       "   'datatypes': [{'single cell': 3}],\n",
       "   'dataset_count': 3,\n",
       "   'disease_count': 4,\n",
       "   'tissue_count': 3,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 1,\n",
       "   'cell_type_count': 26,\n",
       "   'drug_count': 1,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 8.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '31',\n",
       "   'repo_name': 'lincs',\n",
       "   'indexes': {'files': 'lincs_files',\n",
       "    'gct_col_metadata': 'lincs_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'lincs_files',\n",
       "    'gct_col_metadata': 'lincs_gct_col_metadata'},\n",
       "   'sources': [{'geo, lincs': 700}, {'geo': 95}, {'cbioportal': 2}],\n",
       "   'datatypes': [{'transcriptomics': 795}, {'': 2}],\n",
       "   'dataset_count': 797,\n",
       "   'disease_count': 30,\n",
       "   'tissue_count': 20,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 67,\n",
       "   'cell_type_count': 1,\n",
       "   'drug_count': 571,\n",
       "   'data_type_count': 2,\n",
       "   'data_source_count': 3,\n",
       "   'sample_count': 198202.0,\n",
       "   'normal_sample_count': 9062},\n",
       "  {'repo_id': '1673411159486',\n",
       "   'repo_name': 'bulk_rnaseq_atlas',\n",
       "   'indexes': {'files': 'bulk_rnaseq_atlas_files',\n",
       "    'gct_col_metadata': 'bulk_rnaseq_atlas_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'bulk_rnaseq_atlas_files',\n",
       "    'gct_col_metadata': 'bulk_rnaseq_atlas_gct_col_metadata'},\n",
       "   'sources': [{'geo': 74}],\n",
       "   'datatypes': [{'raw_counts_transcriptomics': 57}, {'transcriptomics': 17}],\n",
       "   'dataset_count': 74,\n",
       "   'disease_count': 88,\n",
       "   'tissue_count': 71,\n",
       "   'organism_count': 2,\n",
       "   'cell_line_count': 21,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 10,\n",
       "   'data_type_count': 2,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 6808.0,\n",
       "   'normal_sample_count': 5402},\n",
       "  {'repo_id': '1659450268526',\n",
       "   'repo_name': 'multiple_oa_2',\n",
       "   'indexes': {'files': 'multiple_oa_2_files',\n",
       "    'gct_col_metadata': 'multiple_oa_2_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'multiple_oa_2_files',\n",
       "    'gct_col_metadata': 'multiple_oa_2_gct_col_metadata'},\n",
       "   'sources': [{'cbioportal': 226},\n",
       "    {'geo': 166},\n",
       "    {'depmap': 5},\n",
       "    {'tyra': 5},\n",
       "    {'ccle': 4},\n",
       "    {'cptac': 1}],\n",
       "   'datatypes': [{'copy number variation': 192},\n",
       "    {'transcriptomics': 166},\n",
       "    {'mutation': 33},\n",
       "    {'fusion': 4},\n",
       "    {'gene dependency': 4},\n",
       "    {'drug screens': 1},\n",
       "    {'metabolomics': 1},\n",
       "    {'mirna': 1}],\n",
       "   'dataset_count': 407,\n",
       "   'disease_count': 161,\n",
       "   'tissue_count': 127,\n",
       "   'organism_count': 7,\n",
       "   'cell_line_count': 159,\n",
       "   'cell_type_count': 62,\n",
       "   'drug_count': 71,\n",
       "   'data_type_count': 8,\n",
       "   'data_source_count': 6,\n",
       "   'sample_count': 276264.0,\n",
       "   'normal_sample_count': 183062},\n",
       "  {'repo_id': '1683112669959',\n",
       "   'repo_name': 'may_repo_test_single',\n",
       "   'v2_indexes': {'h5ad_col_metadata': 'may_repo_test_single_h5ad_col_metadata'},\n",
       "   'org_id': '1',\n",
       "   'data_type': 'single_cell',\n",
       "   'sources': [],\n",
       "   'datatypes': [],\n",
       "   'dataset_count': 0,\n",
       "   'disease_count': 0,\n",
       "   'tissue_count': 0,\n",
       "   'organism_count': 0,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 0,\n",
       "   'data_source_count': 0,\n",
       "   'sample_count': 0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1687327504866',\n",
       "   'repo_name': 'new_omix_for_refac',\n",
       "   'indexes': {},\n",
       "   'v2_indexes': {},\n",
       "   'sources': [],\n",
       "   'datatypes': [],\n",
       "   'dataset_count': 0,\n",
       "   'disease_count': 0,\n",
       "   'tissue_count': 0,\n",
       "   'organism_count': 0,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 0,\n",
       "   'data_source_count': 0,\n",
       "   'sample_count': 0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1664439952209',\n",
       "   'repo_name': 'clinical_oa_2',\n",
       "   'indexes': {'files': 'clinical_oa_2_files',\n",
       "    'gct_col_metadata': 'clinical_oa_2_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'clinical_oa_2_files',\n",
       "    'gct_col_metadata': 'clinical_oa_2_gct_col_metadata'},\n",
       "   'org_id': '1',\n",
       "   'sources': [{'teikobio': 16}, {'geo': 6}],\n",
       "   'datatypes': [{'flow cytometry': 16}, {'raw counts transcriptomics': 6}],\n",
       "   'dataset_count': 22,\n",
       "   'disease_count': 7,\n",
       "   'tissue_count': 7,\n",
       "   'organism_count': 2,\n",
       "   'cell_line_count': 1,\n",
       "   'cell_type_count': 6,\n",
       "   'drug_count': 1,\n",
       "   'data_type_count': 2,\n",
       "   'data_source_count': 2,\n",
       "   'sample_count': 149.0,\n",
       "   'normal_sample_count': 58},\n",
       "  {'repo_id': '1654268055800',\n",
       "   'repo_name': 'ingestion_test_1',\n",
       "   'indexes': {'files': 'ingestion_test_1_files',\n",
       "    'gct_col_metadata': 'ingestion_test_1_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'ingestion_test_1_files',\n",
       "    'gct_col_metadata': 'ingestion_test_1_gct_col_metadata'},\n",
       "   'sources': [{'geo': 76}, {'cbioportal': 13}, {'ccle': 3}, {'tcga': 1}],\n",
       "   'datatypes': [{'transcriptomics': 65},\n",
       "    {'': 22},\n",
       "    {'copy_number_variation': 5},\n",
       "    {'single_cell': 1}],\n",
       "   'dataset_count': 93,\n",
       "   'disease_count': 0,\n",
       "   'tissue_count': 0,\n",
       "   'organism_count': 0,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 4,\n",
       "   'data_source_count': 4,\n",
       "   'sample_count': 0.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1629715166552',\n",
       "   'repo_name': 'gnomad',\n",
       "   'indexes': {'files': 'gnomad_files'},\n",
       "   'v2_indexes': {'files': 'gnomad_files'},\n",
       "   'org_id': '1',\n",
       "   'sources': [],\n",
       "   'datatypes': [],\n",
       "   'dataset_count': 720,\n",
       "   'disease_count': 0,\n",
       "   'tissue_count': 0,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 0,\n",
       "   'data_source_count': 0,\n",
       "   'sample_count': 0.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1673847977346',\n",
       "   'repo_name': 'lib_repo_dev',\n",
       "   'indexes': {'files': 'lib_repo_dev_files',\n",
       "    'gct_col_metadata': 'lib_repo_dev_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'lib_repo_dev_files',\n",
       "    'gct_col_metadata': 'lib_repo_dev_gct_col_metadata'},\n",
       "   'linked_workspace_id': '99999999999999999999999',\n",
       "   'sources': [{'cbioportal': 3}, {'geo': 3}],\n",
       "   'datatypes': [{'copy number variation': 3}, {'transcriptomics': 3}],\n",
       "   'dataset_count': 6,\n",
       "   'disease_count': 7,\n",
       "   'tissue_count': 3,\n",
       "   'organism_count': 0,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 2,\n",
       "   'drug_count': 1,\n",
       "   'data_type_count': 2,\n",
       "   'data_source_count': 2,\n",
       "   'sample_count': 62.0,\n",
       "   'normal_sample_count': 390},\n",
       "  {'repo_id': '1662102491212',\n",
       "   'repo_name': 'p_oa',\n",
       "   'indexes': {'files': 'p_oa_files',\n",
       "    'gct_col_metadata': 'p_oa_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'p_oa_files',\n",
       "    'gct_col_metadata': 'p_oa_gct_col_metadata'},\n",
       "   'sources': [{'tcga': 35}, {'geo': 22}],\n",
       "   'datatypes': [{'transcriptomics': 40},\n",
       "    {'copy number variation': 15},\n",
       "    {'': 2}],\n",
       "   'dataset_count': 57,\n",
       "   'disease_count': 14,\n",
       "   'tissue_count': 13,\n",
       "   'organism_count': 6,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 10,\n",
       "   'data_type_count': 3,\n",
       "   'data_source_count': 2,\n",
       "   'sample_count': 188.0,\n",
       "   'normal_sample_count': 73},\n",
       "  {'repo_id': '1649161733379',\n",
       "   'repo_name': 'cell_viability_assay',\n",
       "   'indexes': {'files': 'cell_viability_assay_files',\n",
       "    'gct_col_metadata': 'cell_viability_assay_gct_col_metadata',\n",
       "    'h5ad_col_metadata': 'cell_viability_assay_h5ad_col_metadata'},\n",
       "   'v2_indexes': {'files': 'cell_viability_assay_files',\n",
       "    'gct_col_metadata': 'cell_viability_assay_gct_col_metadata',\n",
       "    'h5ad_col_metadata': 'cell_viability_assay_h5ad_col_metadata'},\n",
       "   'org_id': '14',\n",
       "   'data_type': 'single_cell',\n",
       "   'sources': [{'geo': 4}, {'expression atlas': 1}],\n",
       "   'datatypes': [{'single cell': 4}, {'single_cell': 1}],\n",
       "   'dataset_count': 5,\n",
       "   'disease_count': 10,\n",
       "   'tissue_count': 11,\n",
       "   'organism_count': 2,\n",
       "   'cell_line_count': 2,\n",
       "   'cell_type_count': 14,\n",
       "   'drug_count': 1,\n",
       "   'data_type_count': 2,\n",
       "   'data_source_count': 2,\n",
       "   'sample_count': 17.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1659621145863',\n",
       "   'repo_name': 'hmp_omixatlas',\n",
       "   'indexes': {'files': 'hmp_omixatlas_files',\n",
       "    'biom_col_metadata': 'hmp_omixatlas_biom_col_metadata',\n",
       "    'biom_row_metadata': 'hmp_omixatlas_biom_row_metadata'},\n",
       "   'v2_indexes': {'files': 'hmp_omixatlas_files',\n",
       "    'biom_col_metadata': 'hmp_omixatlas_biom_col_metadata',\n",
       "    'biom_row_metadata': 'hmp_omixatlas_biom_row_metadata'},\n",
       "   'sources': [{'human microbiome project (hmp)': 32}, {'microbiome db': 1}],\n",
       "   'datatypes': [{'metagenomics': 26},\n",
       "    {'metatranscriptomics': 6},\n",
       "    {'microbiome': 1}],\n",
       "   'dataset_count': 33,\n",
       "   'disease_count': 1,\n",
       "   'tissue_count': 2,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 1,\n",
       "   'cell_type_count': 1,\n",
       "   'drug_count': 16,\n",
       "   'data_type_count': 3,\n",
       "   'data_source_count': 2,\n",
       "   'sample_count': 1248.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1647941576494',\n",
       "   'repo_name': 'depmap_omixatlas',\n",
       "   'indexes': {'files': 'depmap_omixatlas_files',\n",
       "    'gct_col_metadata': 'depmap_omixatlas_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'depmap_omixatlas_files',\n",
       "    'gct_col_metadata': 'depmap_omixatlas_gct_col_metadata'},\n",
       "   'sources': [{'depmap': 546}],\n",
       "   'datatypes': [{'gene dependency': 487}, {'drug screens': 59}],\n",
       "   'dataset_count': 546,\n",
       "   'disease_count': 89,\n",
       "   'tissue_count': 37,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 410,\n",
       "   'cell_type_count': 1,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 2,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 546.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1657777478388',\n",
       "   'repo_name': 'ingestion_testing_oa',\n",
       "   'indexes': {'files': 'ingestion_testing_oa_files',\n",
       "    'gct_col_metadata': 'ingestion_testing_oa_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'ingestion_testing_oa_files',\n",
       "    'gct_col_metadata': 'ingestion_testing_oa_gct_col_metadata'},\n",
       "   'sources': [{'pride': 1}],\n",
       "   'datatypes': [{'proteomics': 1}],\n",
       "   'dataset_count': 1,\n",
       "   'disease_count': 1,\n",
       "   'tissue_count': 1,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 1,\n",
       "   'drug_count': 2,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 3.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1669085436194',\n",
       "   'repo_name': 'source_atlas',\n",
       "   'indexes': {'files': 'source_atlas_files',\n",
       "    'gct_col_metadata': 'source_atlas_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'source_atlas_files',\n",
       "    'gct_col_metadata': 'source_atlas_gct_col_metadata'},\n",
       "   'data_type': 'bulk_rna_seq',\n",
       "   'sources': [{'geo': 36}],\n",
       "   'datatypes': [{'raw counts transcriptomics': 35}, {'transcriptomics': 1}],\n",
       "   'dataset_count': 36,\n",
       "   'disease_count': 28,\n",
       "   'tissue_count': 29,\n",
       "   'organism_count': 4,\n",
       "   'cell_line_count': 22,\n",
       "   'cell_type_count': 19,\n",
       "   'drug_count': 4,\n",
       "   'data_type_count': 2,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 3094.0,\n",
       "   'normal_sample_count': 2769},\n",
       "  {'repo_id': '1674572370484',\n",
       "   'repo_name': 'test_pipeline',\n",
       "   'indexes': {'files': 'test_pipeline_files',\n",
       "    'gct_col_metadata': 'test_pipeline_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'test_pipeline_files',\n",
       "    'gct_col_metadata': 'test_pipeline_gct_col_metadata'},\n",
       "   'sources': [{'geo': 22}],\n",
       "   'datatypes': [{'transcriptomics': 13}, {'raw counts transcriptomics': 9}],\n",
       "   'dataset_count': 22,\n",
       "   'disease_count': 24,\n",
       "   'tissue_count': 9,\n",
       "   'organism_count': 2,\n",
       "   'cell_line_count': 8,\n",
       "   'cell_type_count': 20,\n",
       "   'drug_count': 3,\n",
       "   'data_type_count': 2,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 317.0,\n",
       "   'normal_sample_count': 100},\n",
       "  {'repo_id': '1673014297214',\n",
       "   'repo_name': 'test_connector_pipeline',\n",
       "   'indexes': {'files': 'test_connector_pipeline_files',\n",
       "    'gct_col_metadata': 'test_connector_pipeline_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'test_connector_pipeline_files',\n",
       "    'gct_col_metadata': 'test_connector_pipeline_gct_col_metadata'},\n",
       "   'sources': [{'geo': 5}],\n",
       "   'datatypes': [{'transcriptomics': 5}],\n",
       "   'dataset_count': 5,\n",
       "   'disease_count': 9,\n",
       "   'tissue_count': 3,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 8,\n",
       "   'cell_type_count': 3,\n",
       "   'drug_count': 1,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 54.0,\n",
       "   'normal_sample_count': 29},\n",
       "  {'repo_id': '1674475984908',\n",
       "   'repo_name': 'single_cell_test',\n",
       "   'indexes': {'files': 'single_cell_test_files',\n",
       "    'h5ad_col_metadata': 'single_cell_test_h5ad_col_metadata'},\n",
       "   'v2_indexes': {'files': 'single_cell_test_files',\n",
       "    'h5ad_col_metadata': 'single_cell_test_h5ad_col_metadata'},\n",
       "   'sources': [{'array express': 2},\n",
       "    {'covid-19 cell atlas': 1},\n",
       "    {'human cell atlas': 1}],\n",
       "   'datatypes': [{'single cell': 4}],\n",
       "   'dataset_count': 4,\n",
       "   'disease_count': 4,\n",
       "   'tissue_count': 2,\n",
       "   'organism_count': 2,\n",
       "   'cell_line_count': 1,\n",
       "   'cell_type_count': 11,\n",
       "   'drug_count': 2,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 3,\n",
       "   'sample_count': 31.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1681973657512',\n",
       "   'repo_name': 'test_schema_builder_oa',\n",
       "   'v2_indexes': {},\n",
       "   'sources': [],\n",
       "   'datatypes': [],\n",
       "   'dataset_count': 0,\n",
       "   'disease_count': 0,\n",
       "   'tissue_count': 0,\n",
       "   'organism_count': 0,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 0,\n",
       "   'data_source_count': 0,\n",
       "   'sample_count': 0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1644400393649',\n",
       "   'repo_name': 'ukbiobank',\n",
       "   'indexes': {'files': 'ukbiobank_files'},\n",
       "   'v2_indexes': {'files': 'ukbiobank_files'},\n",
       "   'sources': [],\n",
       "   'datatypes': [{'gwas': 648}],\n",
       "   'dataset_count': 648,\n",
       "   'disease_count': 0,\n",
       "   'tissue_count': 0,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 0,\n",
       "   'sample_count': 0.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1679312499313',\n",
       "   'repo_name': 'test_pipeline_rnaseq',\n",
       "   'indexes': {'files': 'test_pipeline_rnaseq_files',\n",
       "    'gct_col_metadata': 'test_pipeline_rnaseq_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'test_pipeline_rnaseq_files',\n",
       "    'gct_col_metadata': 'test_pipeline_rnaseq_gct_col_metadata'},\n",
       "   'sources': [{'geo': 110}],\n",
       "   'datatypes': [{'raw counts transcriptomics': 84},\n",
       "    {'raw_counts_transcriptomics': 24},\n",
       "    {'transcriptomics': 2}],\n",
       "   'dataset_count': 110,\n",
       "   'disease_count': 73,\n",
       "   'tissue_count': 41,\n",
       "   'organism_count': 3,\n",
       "   'cell_line_count': 39,\n",
       "   'cell_type_count': 49,\n",
       "   'drug_count': 22,\n",
       "   'data_type_count': 3,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 1926.0,\n",
       "   'normal_sample_count': 1696},\n",
       "  {'repo_id': '1649664527538',\n",
       "   'repo_name': 'tyra_cell_viability_assay',\n",
       "   'indexes': {'files': 'tyra_cell_viability_assay_files'},\n",
       "   'v2_indexes': {'files': 'tyra_cell_viability_assay_files'},\n",
       "   'sources': [],\n",
       "   'datatypes': [],\n",
       "   'dataset_count': 3760,\n",
       "   'disease_count': 0,\n",
       "   'tissue_count': 0,\n",
       "   'organism_count': 0,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 0,\n",
       "   'data_source_count': 0,\n",
       "   'sample_count': 0.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '17',\n",
       "   'repo_name': 'sc_data_lake',\n",
       "   'indexes': {'h5ad_col_metadata': 'sc_data_lake_h5ad_col_metadata',\n",
       "    'h5ad_row_metadata': 'sc_data_lake_h5ad_row_metadata'},\n",
       "   'v2_indexes': {'h5ad_col_metadata': 'sc_data_lake_h5ad_col_metadata',\n",
       "    'h5ad_row_metadata': 'sc_data_lake_h5ad_row_metadata'},\n",
       "   'sources': [],\n",
       "   'datatypes': [],\n",
       "   'dataset_count': 0,\n",
       "   'disease_count': 0,\n",
       "   'tissue_count': 0,\n",
       "   'organism_count': 0,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 0,\n",
       "   'data_source_count': 0,\n",
       "   'sample_count': 0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1670915426568',\n",
       "   'repo_name': 'single_cell_destination',\n",
       "   'indexes': {'files': 'single_cell_destination_files',\n",
       "    'h5ad_col_metadata': 'single_cell_destination_h5ad_col_metadata'},\n",
       "   'v2_indexes': {'files': 'single_cell_destination_files',\n",
       "    'h5ad_col_metadata': 'single_cell_destination_h5ad_col_metadata'},\n",
       "   'sources': [{'geo': 7}, {'tabula sapiens': 1}],\n",
       "   'datatypes': [{'single cell': 8}],\n",
       "   'dataset_count': 8,\n",
       "   'disease_count': 7,\n",
       "   'tissue_count': 13,\n",
       "   'organism_count': 2,\n",
       "   'cell_line_count': 2,\n",
       "   'cell_type_count': 34,\n",
       "   'drug_count': 10,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 2,\n",
       "   'sample_count': 8.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1689052723688',\n",
       "   'repo_name': 'test_july_p',\n",
       "   'v2_indexes': {'files': 'test_july_p_files',\n",
       "    'gct_col_metadata': 'test_july_p_gct_col_metadata'},\n",
       "   'sources': [{'geo': 36}],\n",
       "   'datatypes': [{'transcriptomics': 24},\n",
       "    {'raw counts transcriptomics': 11},\n",
       "    {'': 1}],\n",
       "   'dataset_count': 36,\n",
       "   'disease_count': 37,\n",
       "   'tissue_count': 33,\n",
       "   'organism_count': 3,\n",
       "   'cell_line_count': 615,\n",
       "   'cell_type_count': 28,\n",
       "   'drug_count': 5,\n",
       "   'data_type_count': 3,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 40106.0,\n",
       "   'normal_sample_count': 29823},\n",
       "  {'repo_id': '1689168428573',\n",
       "   'repo_name': 'test_so_2',\n",
       "   'v2_indexes': {'files': 'test_so_2_files',\n",
       "    'gct_col_metadata': 'test_so_2_gct_col_metadata'},\n",
       "   'sources': [{'geo': 2}],\n",
       "   'datatypes': [{'raw counts transcriptomics': 2}],\n",
       "   'dataset_count': 2,\n",
       "   'disease_count': 2,\n",
       "   'tissue_count': 4,\n",
       "   'organism_count': 2,\n",
       "   'cell_line_count': 1,\n",
       "   'cell_type_count': 3,\n",
       "   'drug_count': 1,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 7.0,\n",
       "   'normal_sample_count': 7},\n",
       "  {'repo_id': '1680442387535',\n",
       "   'repo_name': 'sch_valid_change',\n",
       "   'indexes': {},\n",
       "   'v2_indexes': {},\n",
       "   'sources': [],\n",
       "   'datatypes': [],\n",
       "   'dataset_count': 0,\n",
       "   'disease_count': 0,\n",
       "   'tissue_count': 0,\n",
       "   'organism_count': 0,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 0,\n",
       "   'data_source_count': 0,\n",
       "   'sample_count': 0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1680106629523',\n",
       "   'repo_name': 'single_sch_oa',\n",
       "   'indexes': {'files': 'single_sch_oa_files',\n",
       "    'h5ad_col_metadata': 'single_sch_oa_h5ad_col_metadata'},\n",
       "   'v2_indexes': {'files': 'single_sch_oa_files',\n",
       "    'h5ad_col_metadata': 'single_sch_oa_h5ad_col_metadata'},\n",
       "   'sources': [{'geo': 2}, {'': 1}, {'ghi': 1}],\n",
       "   'datatypes': [{'': 2}, {'single_cell': 2}],\n",
       "   'dataset_count': 4,\n",
       "   'disease_count': 9,\n",
       "   'tissue_count': 6,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 1,\n",
       "   'cell_type_count': 41,\n",
       "   'drug_count': 13,\n",
       "   'data_type_count': 2,\n",
       "   'data_source_count': 3,\n",
       "   'sample_count': 33.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1688035601240',\n",
       "   'repo_name': 'repo_g_eo',\n",
       "   'indexes': {'files': 'repo_g_eo_files',\n",
       "    'gct_col_metadata': 'repo_g_eo_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'repo_g_eo_files',\n",
       "    'gct_col_metadata': 'repo_g_eo_gct_col_metadata'},\n",
       "   'data_type': 'bulk_rna_seq',\n",
       "   'sources': [{'geo': 123}],\n",
       "   'datatypes': [{'transcriptomics': 111}, {'raw counts transcriptomics': 12}],\n",
       "   'dataset_count': 123,\n",
       "   'disease_count': 382,\n",
       "   'tissue_count': 94,\n",
       "   'organism_count': 4,\n",
       "   'cell_line_count': 638,\n",
       "   'cell_type_count': 52,\n",
       "   'drug_count': 11,\n",
       "   'data_type_count': 2,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 217989.0,\n",
       "   'normal_sample_count': 114996},\n",
       "  {'repo_id': '1688035585607',\n",
       "   'repo_name': 'repo_july_so',\n",
       "   'indexes': {'files': 'repo_july_so_files',\n",
       "    'gct_col_metadata': 'repo_july_so_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'repo_july_so_files',\n",
       "    'gct_col_metadata': 'repo_july_so_gct_col_metadata'},\n",
       "   'org_id': '1',\n",
       "   'data_type': 'bulk_rna_seq',\n",
       "   'sources': [{'geo': 20}],\n",
       "   'datatypes': [{'raw counts transcriptomics': 19}, {'transcriptomics': 1}],\n",
       "   'dataset_count': 20,\n",
       "   'disease_count': 19,\n",
       "   'tissue_count': 15,\n",
       "   'organism_count': 4,\n",
       "   'cell_line_count': 8,\n",
       "   'cell_type_count': 13,\n",
       "   'drug_count': 5,\n",
       "   'data_type_count': 2,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 2652.0,\n",
       "   'normal_sample_count': 108},\n",
       "  {'repo_id': '1680089043360',\n",
       "   'repo_name': 'sch_validation_oa',\n",
       "   'indexes': {'files': 'sch_validation_oa_files',\n",
       "    'gct_col_metadata': 'sch_validation_oa_gct_col_metadata'},\n",
       "   'v2_indexes': {'files': 'sch_validation_oa_files',\n",
       "    'gct_col_metadata': 'sch_validation_oa_gct_col_metadata'},\n",
       "   'sources': [{'geo': 16},\n",
       "    {'': 1},\n",
       "    {'abc': 1},\n",
       "    {'ghi': 1},\n",
       "    {'random': 1},\n",
       "    {'sc_data_lake': 1}],\n",
       "   'datatypes': [{'': 19}, {'jkl': 1}, {'post': 1}],\n",
       "   'dataset_count': 21,\n",
       "   'disease_count': 24,\n",
       "   'tissue_count': 21,\n",
       "   'organism_count': 0,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 20,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 3,\n",
       "   'data_source_count': 6,\n",
       "   'sample_count': 802.0,\n",
       "   'normal_sample_count': 767},\n",
       "  {'repo_id': '15',\n",
       "   'repo_name': 'tcga',\n",
       "   'indexes': {'files': 'tcga_files'},\n",
       "   'v2_indexes': {'files': 'tcga_files'},\n",
       "   'sources': [{'tcga': 678}],\n",
       "   'datatypes': [{'transcriptomics': 156},\n",
       "    {'copy number variation': 124},\n",
       "    {'mirna': 122},\n",
       "    {'methylation': 118},\n",
       "    {'proteomics': 81},\n",
       "    {'mutation': 77}],\n",
       "   'dataset_count': 678,\n",
       "   'disease_count': 34,\n",
       "   'tissue_count': 55,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 101,\n",
       "   'data_type_count': 6,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 678.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1676020061866',\n",
       "   'repo_name': 'test_single_cell_source_feb',\n",
       "   'indexes': {'files': 'test_single_cell_source_feb_files',\n",
       "    'h5ad_col_metadata': 'test_single_cell_source_feb_h5ad_col_metadata'},\n",
       "   'v2_indexes': {'files': 'test_single_cell_source_feb_files',\n",
       "    'h5ad_col_metadata': 'test_single_cell_source_feb_h5ad_col_metadata'},\n",
       "   'data_type': 'single_cell',\n",
       "   'sources': [{'geo': 6},\n",
       "    {'expression_atlas': 2},\n",
       "    {'ana': 1},\n",
       "    {'array_express': 1},\n",
       "    {'ghi': 1},\n",
       "    {'publication': 1},\n",
       "    {'scp': 1},\n",
       "    {'tabula_sapiens': 1}],\n",
       "   'datatypes': [{'single_cell': 7},\n",
       "    {'single cell': 5},\n",
       "    {'hya': 1},\n",
       "    {'jkl': 1}],\n",
       "   'dataset_count': 14,\n",
       "   'disease_count': 19,\n",
       "   'tissue_count': 10,\n",
       "   'organism_count': 2,\n",
       "   'cell_line_count': 1,\n",
       "   'cell_type_count': 58,\n",
       "   'drug_count': 13,\n",
       "   'data_type_count': 4,\n",
       "   'data_source_count': 8,\n",
       "   'sample_count': 72.0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_id': '1671791834799',\n",
       "   'repo_name': 'single_cell_source',\n",
       "   'indexes': {'files': 'single_cell_source_files',\n",
       "    'h5ad_col_metadata': 'single_cell_source_h5ad_col_metadata'},\n",
       "   'v2_indexes': {'files': 'single_cell_source_files',\n",
       "    'h5ad_col_metadata': 'single_cell_source_h5ad_col_metadata'},\n",
       "   'data_type': 'single_cell',\n",
       "   'sources': [{'geo': 43},\n",
       "    {'publication': 8},\n",
       "    {'expression atlas': 3},\n",
       "    {'scp': 3},\n",
       "    {'human cell atlas': 1},\n",
       "    {'tabula sapiens': 1}],\n",
       "   'datatypes': [{'single cell': 59}],\n",
       "   'dataset_count': 59,\n",
       "   'disease_count': 56,\n",
       "   'tissue_count': 55,\n",
       "   'organism_count': 2,\n",
       "   'cell_line_count': 5,\n",
       "   'cell_type_count': 130,\n",
       "   'drug_count': 15,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 6,\n",
       "   'sample_count': 1246.0,\n",
       "   'normal_sample_count': 0}]}"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "omix.get_all_omixatlas()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " OmixAtlas 1689763574024 Created  \n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Repository Id</th>\n",
       "      <th>Repository Name</th>\n",
       "      <th>Category</th>\n",
       "      <th>Display Name</th>\n",
       "      <th>Description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1689763574024</td>\n",
       "      <td>ex_repo_1</td>\n",
       "      <td>private</td>\n",
       "      <td>ex_repo_1</td>\n",
       "      <td>example for create</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Repository Id Repository Name Category Display Name         Description\n",
       "0  1689763574024       ex_repo_1  private    ex_repo_1  example for create"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "omix.create(\"ex_repo_1\",\"example for create\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " OmixAtlas 1643016586529 Updated  \n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Repository Id</th>\n",
       "      <th>Repository Name</th>\n",
       "      <th>Category</th>\n",
       "      <th>Display Name</th>\n",
       "      <th>Description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1643016586529</td>\n",
       "      <td>geo</td>\n",
       "      <td>private</td>\n",
       "      <td>updated description</td>\n",
       "      <td>this is the revised description of demo omixatlas</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Repository Id Repository Name Category         Display Name  \\\n",
       "0  1643016586529             geo  private  updated description   \n",
       "\n",
       "                                         Description  \n",
       "0  this is the revised description of demo omixatlas  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "omix.update(\"geo\",\"updated description\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Query execution succeeded (time taken: 1.55 seconds, data scanned: 0.010 MB)\n",
      "Fetched 10 rows\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dataset_id</th>\n",
       "      <th>abstract</th>\n",
       "      <th>author</th>\n",
       "      <th>bucket</th>\n",
       "      <th>curated_cell_line</th>\n",
       "      <th>curated_cell_type</th>\n",
       "      <th>curated_disease</th>\n",
       "      <th>curated_disease_type</th>\n",
       "      <th>curated_drug</th>\n",
       "      <th>curated_file_type</th>\n",
       "      <th>curated_gene</th>\n",
       "      <th>curated_organism</th>\n",
       "      <th>curated_project</th>\n",
       "      <th>curated_repo</th>\n",
       "      <th>curated_strain</th>\n",
       "      <th>curated_tissue</th>\n",
       "      <th>curation_version</th>\n",
       "      <th>data_matrix_available</th>\n",
       "      <th>data_repository</th>\n",
       "      <th>data_source</th>\n",
       "      <th>data_table_name</th>\n",
       "      <th>data_table_version</th>\n",
       "      <th>data_type</th>\n",
       "      <th>dataset_source</th>\n",
       "      <th>description</th>\n",
       "      <th>drug_smiles</th>\n",
       "      <th>experimental_design</th>\n",
       "      <th>file_location</th>\n",
       "      <th>file_type</th>\n",
       "      <th>is_current</th>\n",
       "      <th>key</th>\n",
       "      <th>missing_samples</th>\n",
       "      <th>operation</th>\n",
       "      <th>overall_design</th>\n",
       "      <th>package</th>\n",
       "      <th>platform</th>\n",
       "      <th>publication</th>\n",
       "      <th>raw_disease</th>\n",
       "      <th>region</th>\n",
       "      <th>sample_field</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>src_repo</th>\n",
       "      <th>src_uri</th>\n",
       "      <th>subseries_ids</th>\n",
       "      <th>summary</th>\n",
       "      <th>timestamp_</th>\n",
       "      <th>total_num_samples</th>\n",
       "      <th>version</th>\n",
       "      <th>year</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GSE107990_GPL10558</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>discover-test-datalake-v1</td>\n",
       "      <td>[None]</td>\n",
       "      <td>[None]</td>\n",
       "      <td>[Influenza, Human, Acquired Immunodeficiency S...</td>\n",
       "      <td></td>\n",
       "      <td>[prednisone]</td>\n",
       "      <td></td>\n",
       "      <td>[]</td>\n",
       "      <td>[Homo sapiens]</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>[None1, None2]</td>\n",
       "      <td>[None]</td>\n",
       "      <td>2023_02_02</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>geo__gse107990_gpl10558</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcriptomics</td>\n",
       "      <td>GEO</td>\n",
       "      <td>Gene expression profiles associated with humor...</td>\n",
       "      <td>[[H][C@@]12CCC3=CC(=O)C=C[C@]3(C)[C@@]1([H])C(...</td>\n",
       "      <td></td>\n",
       "      <td>https://discover-test-datalake-v1.s3-ap-southe...</td>\n",
       "      <td>gct</td>\n",
       "      <td>true</td>\n",
       "      <td>geo/data/Microarray/GSE107990/GCT/GSE107990_GP...</td>\n",
       "      <td>[]</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>geo/data</td>\n",
       "      <td>unknown_platform</td>\n",
       "      <td></td>\n",
       "      <td>[]</td>\n",
       "      <td>ap-southeast-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>geo</td>\n",
       "      <td>polly:data://geo/data/Microarray/GSE107990/GCT...</td>\n",
       "      <td>[]</td>\n",
       "      <td></td>\n",
       "      <td>1686052005503</td>\n",
       "      <td>671</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GSE128722_GPL11154</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>discover-test-datalake-v1</td>\n",
       "      <td>[U-87MG ATCC, LN-18]</td>\n",
       "      <td>[bgb]</td>\n",
       "      <td>[Glioma, disease2]</td>\n",
       "      <td></td>\n",
       "      <td>[None]</td>\n",
       "      <td></td>\n",
       "      <td>[None]</td>\n",
       "      <td>[tt]</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>[None]</td>\n",
       "      <td>[rg]</td>\n",
       "      <td>2023_02_02</td>\n",
       "      <td></td>\n",
       "      <td>geo</td>\n",
       "      <td></td>\n",
       "      <td>geo__gse128722_gpl11154</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcriptomics</td>\n",
       "      <td>vfv</td>\n",
       "      <td>bbg</td>\n",
       "      <td>[[C@H]12O[C@H]([C@@H]([C@@H]1OP(OC2)(=O)O)OC(=...</td>\n",
       "      <td></td>\n",
       "      <td>https://discover-test-datalake-v1.s3-ap-southe...</td>\n",
       "      <td>gct</td>\n",
       "      <td>true</td>\n",
       "      <td>geo/data/RNASeq/GSE128722/GCT/GSE128722_GPL111...</td>\n",
       "      <td>[]</td>\n",
       "      <td>{\"is_batch_corrected\": \"false\", \"normalized_ty...</td>\n",
       "      <td></td>\n",
       "      <td>geo/data</td>\n",
       "      <td>tbtbgb</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi...</td>\n",
       "      <td>[]</td>\n",
       "      <td>ap-southeast-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>geo</td>\n",
       "      <td>polly:data://geo/data/RNASeq/GSE128722/GCT/GSE...</td>\n",
       "      <td>[]</td>\n",
       "      <td></td>\n",
       "      <td>1685628391670</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>btb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GSE140509_GPL16791</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>discover-test-datalake-v1</td>\n",
       "      <td>[NCI-H460]</td>\n",
       "      <td>[malignant cell]</td>\n",
       "      <td>[Carcinoma, Large Cell]</td>\n",
       "      <td></td>\n",
       "      <td>[None]</td>\n",
       "      <td></td>\n",
       "      <td>[DSP, PKP1, PKP2]</td>\n",
       "      <td>[Homo sapiens]</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>[None]</td>\n",
       "      <td>[lung]</td>\n",
       "      <td>g3</td>\n",
       "      <td>true</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>geo__gse140509_gpl16791</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcriptomics</td>\n",
       "      <td>GEO</td>\n",
       "      <td>Exploring the downstream genes of PKP2</td>\n",
       "      <td>[]</td>\n",
       "      <td></td>\n",
       "      <td>https://discover-test-datalake-v1.s3-ap-southe...</td>\n",
       "      <td>gct</td>\n",
       "      <td>true</td>\n",
       "      <td>geo/data/geo/RNASeq/GSE140509/GCT/GSE140509_GP...</td>\n",
       "      <td>[]</td>\n",
       "      <td></td>\n",
       "      <td>To explore the downstream genes of PKP2,we per...</td>\n",
       "      <td>geo/data</td>\n",
       "      <td>RNAseq</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi...</td>\n",
       "      <td>[]</td>\n",
       "      <td>ap-southeast-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>geo</td>\n",
       "      <td>polly:data://geo/data/geo/RNASeq/GSE140509/GCT...</td>\n",
       "      <td>[]</td>\n",
       "      <td>Plakophilin 2 (PKP2), encodes a plakophilin pr...</td>\n",
       "      <td>1675920716537</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GSE76297_GPL17586</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>discover-test-datalake-v1</td>\n",
       "      <td>[None]</td>\n",
       "      <td>[Langerhans cell]</td>\n",
       "      <td>[Neoplastic Cells, Circulating, Obesity, Carci...</td>\n",
       "      <td></td>\n",
       "      <td>[None]</td>\n",
       "      <td></td>\n",
       "      <td>[ARID2, TP53, ARID1A, ECT2, PLK1]</td>\n",
       "      <td>[Homo sapiens]</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>[None]</td>\n",
       "      <td>[liver]</td>\n",
       "      <td></td>\n",
       "      <td>true</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>geo__gse76297_gpl17586</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcriptomics</td>\n",
       "      <td>GEO</td>\n",
       "      <td>Gene expression data of human hepatocellular c...</td>\n",
       "      <td>[]</td>\n",
       "      <td></td>\n",
       "      <td>https://discover-test-datalake-v1.s3-ap-southe...</td>\n",
       "      <td>gct</td>\n",
       "      <td>true</td>\n",
       "      <td>geo/data/31Oct_1222/GSE76297_GPL17586.gct</td>\n",
       "      <td>[]</td>\n",
       "      <td></td>\n",
       "      <td>For gene expression profiling, tumors and pair...</td>\n",
       "      <td>geo/data</td>\n",
       "      <td>Microarray</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi...</td>\n",
       "      <td>[]</td>\n",
       "      <td>ap-southeast-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>geo</td>\n",
       "      <td>polly:data://geo/data/31Oct_1222/GSE76297_GPL1...</td>\n",
       "      <td>[]</td>\n",
       "      <td>We used Affymetrix HTA2.0 microarray profiling...</td>\n",
       "      <td>1686904619239</td>\n",
       "      <td>304</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GSE100051_GPL10558</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>discover-test-datalake-v1</td>\n",
       "      <td>[None]</td>\n",
       "      <td>[None]</td>\n",
       "      <td>[Congenital Abnormalities, Diabetes Mellitus]</td>\n",
       "      <td></td>\n",
       "      <td>[None]</td>\n",
       "      <td></td>\n",
       "      <td>[ELF5, VGLL1, CDX2, EOMES, TEAD4]</td>\n",
       "      <td>[Homo sapiens]</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>[None]</td>\n",
       "      <td>[placenta]</td>\n",
       "      <td>2023_02_02</td>\n",
       "      <td>true</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>geo__gse100051_gpl10558</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcriptomics</td>\n",
       "      <td>GEO</td>\n",
       "      <td>Comparative analysis of mouse and human placen...</td>\n",
       "      <td>[]</td>\n",
       "      <td></td>\n",
       "      <td>https://discover-test-datalake-v1.s3-ap-southe...</td>\n",
       "      <td>gct</td>\n",
       "      <td>true</td>\n",
       "      <td>geo/data/geo/19Oct/GSE100051_GPL10558.gct</td>\n",
       "      <td>[]</td>\n",
       "      <td></td>\n",
       "      <td>In order to characerise the human cytotrophobl...</td>\n",
       "      <td>geo/data</td>\n",
       "      <td>Microarray</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi...</td>\n",
       "      <td>[]</td>\n",
       "      <td>ap-southeast-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>geo</td>\n",
       "      <td>polly:data://geo/data/geo/19Oct/GSE100051_GPL1...</td>\n",
       "      <td>[]</td>\n",
       "      <td>In this study, we compared the genome-wide tra...</td>\n",
       "      <td>1675920676897</td>\n",
       "      <td>54</td>\n",
       "      <td>0</td>\n",
       "      <td>2018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GSE100467_GPL11154</td>\n",
       "      <td>A comprehensive, cell specific microRNA catalo...</td>\n",
       "      <td>Matthias,,Huebenthal</td>\n",
       "      <td>discover-test-datalake-v1</td>\n",
       "      <td>[None]</td>\n",
       "      <td>[blood cell]</td>\n",
       "      <td>[Normal, disease2]</td>\n",
       "      <td></td>\n",
       "      <td>[None]</td>\n",
       "      <td></td>\n",
       "      <td>[MIR486-1]</td>\n",
       "      <td>[Homo sapiens]</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>[None]</td>\n",
       "      <td>[serum, blood, peripheral blood]</td>\n",
       "      <td>2023_02_02</td>\n",
       "      <td>true</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>geo__gse100467_gpl11154</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcriptomics</td>\n",
       "      <td>GEO</td>\n",
       "      <td>A comprehensive, cell specific microRNA catalo...</td>\n",
       "      <td>[]</td>\n",
       "      <td></td>\n",
       "      <td>https://discover-test-datalake-v1.s3-ap-southe...</td>\n",
       "      <td>gct</td>\n",
       "      <td>true</td>\n",
       "      <td>geo/data/6Dec22/GSE100467_GPL11154.gct</td>\n",
       "      <td>[GSM2683537, GSM2683917, GSM2683632, GSM268389...</td>\n",
       "      <td></td>\n",
       "      <td>miRNomes from seven distinct blood cell popula...</td>\n",
       "      <td>geo/data</td>\n",
       "      <td>RNASeq</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi...</td>\n",
       "      <td>[]</td>\n",
       "      <td>ap-southeast-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>geo</td>\n",
       "      <td>polly:data://geo/data/6Dec22/GSE100467_GPL1115...</td>\n",
       "      <td>[]</td>\n",
       "      <td>A comprehensive, cell specific microRNA catalo...</td>\n",
       "      <td>1685970450268</td>\n",
       "      <td>445</td>\n",
       "      <td>0</td>\n",
       "      <td>2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GSE102392_GPL16791</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>discover-test-datalake-v1</td>\n",
       "      <td>[None]</td>\n",
       "      <td>[respiratory epithelial cell]</td>\n",
       "      <td>[Respiratory Tract Infections]</td>\n",
       "      <td></td>\n",
       "      <td>[None]</td>\n",
       "      <td></td>\n",
       "      <td>[IL18, IL1A]</td>\n",
       "      <td>[Homo sapiens]</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>[None]</td>\n",
       "      <td>[respiratory epithelium]</td>\n",
       "      <td>2023_02_02</td>\n",
       "      <td></td>\n",
       "      <td>geo</td>\n",
       "      <td></td>\n",
       "      <td>geo__gse102392_gpl16791</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcriptomics</td>\n",
       "      <td>source</td>\n",
       "      <td>Transcriptome analysis of human airway epithel...</td>\n",
       "      <td>[]</td>\n",
       "      <td></td>\n",
       "      <td>https://discover-test-datalake-v1.s3-ap-southe...</td>\n",
       "      <td>gct</td>\n",
       "      <td>true</td>\n",
       "      <td>geo/data/RNASeq/GSE102392/GCT/GSE102392_GPL167...</td>\n",
       "      <td>[]</td>\n",
       "      <td>{\"is_batch_corrected\": \"false\", \"normalized_ty...</td>\n",
       "      <td></td>\n",
       "      <td>geo/data</td>\n",
       "      <td>RNASeq</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi...</td>\n",
       "      <td>[]</td>\n",
       "      <td>ap-southeast-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>geo</td>\n",
       "      <td>polly:data://geo/data/RNASeq/GSE102392/GCT/GSE...</td>\n",
       "      <td>[]</td>\n",
       "      <td></td>\n",
       "      <td>1685628380248</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>2018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>GSE76311_GPL17586</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>discover-test-datalake-v1</td>\n",
       "      <td>[none, test, test1]</td>\n",
       "      <td>[Langerhans cell, Another cellline]</td>\n",
       "      <td>[Carcinoma, Hepatocellular, Obesity, Leukoence...</td>\n",
       "      <td></td>\n",
       "      <td>[none]</td>\n",
       "      <td></td>\n",
       "      <td>[TP53, ARID1A, ECT2, PLK1, ARID2]</td>\n",
       "      <td>[Homo sapiens]</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>[none]</td>\n",
       "      <td>[liver]</td>\n",
       "      <td></td>\n",
       "      <td>false</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>geo__gse76311_gpl17586</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcriptomics</td>\n",
       "      <td>GEO</td>\n",
       "      <td>Common Molecular Subtypes Among Asian Hepatoce...</td>\n",
       "      <td>[]</td>\n",
       "      <td></td>\n",
       "      <td>https://discover-test-datalake-v1.s3-ap-southe...</td>\n",
       "      <td>gct</td>\n",
       "      <td>true</td>\n",
       "      <td>geo/data/31Oct_1222/GSE76311_GPL17586.gct</td>\n",
       "      <td>[]</td>\n",
       "      <td></td>\n",
       "      <td>Refer to individual Series</td>\n",
       "      <td>geo/data</td>\n",
       "      <td>Microarray</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi...</td>\n",
       "      <td>[]</td>\n",
       "      <td>ap-southeast-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>geo</td>\n",
       "      <td>polly:data://geo/data/31Oct_1222/GSE76311_GPL1...</td>\n",
       "      <td>[GSE76213, GSE76297]</td>\n",
       "      <td>This SuperSeries is composed of the SubSeries ...</td>\n",
       "      <td>1687343553467</td>\n",
       "      <td>304</td>\n",
       "      <td>0</td>\n",
       "      <td>2015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>GSE93505_GPL16791</td>\n",
       "      <td>Free Fatty Acids Rewire Cancer Metabolism in O...</td>\n",
       "      <td></td>\n",
       "      <td>discover-test-datalake-v1</td>\n",
       "      <td>[MCF-7]</td>\n",
       "      <td>[malignant cell]</td>\n",
       "      <td>[Osteoporosis, Postmenopausal, Breast Neoplasm...</td>\n",
       "      <td></td>\n",
       "      <td>[None]</td>\n",
       "      <td></td>\n",
       "      <td>[ESRRA, ESR1, MTOR]</td>\n",
       "      <td>[]</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>[None]</td>\n",
       "      <td>[breast]</td>\n",
       "      <td>2023_01_27</td>\n",
       "      <td>true</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>geo__gse93505_gpl16791</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcriptomics</td>\n",
       "      <td>GEO</td>\n",
       "      <td></td>\n",
       "      <td>[]</td>\n",
       "      <td></td>\n",
       "      <td>https://discover-test-datalake-v1.s3-ap-southe...</td>\n",
       "      <td>gct</td>\n",
       "      <td>true</td>\n",
       "      <td>geo/data/geo/RNASeq/GSE93505/GCT/GSE93505_GPL1...</td>\n",
       "      <td>[]</td>\n",
       "      <td></td>\n",
       "      <td>Examination of metabolic pathways in MCF-7 cel...</td>\n",
       "      <td>geo/data</td>\n",
       "      <td>RNAseq</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi...</td>\n",
       "      <td>[]</td>\n",
       "      <td>ap-southeast-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>geo</td>\n",
       "      <td>polly:data://geo/data/geo/RNASeq/GSE93505/GCT/...</td>\n",
       "      <td>[]</td>\n",
       "      <td>Obesity is a risk factor for postmenopausal ER...</td>\n",
       "      <td>1675920756278</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>GSE99771_GPL21493</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>discover-test-datalake-v1</td>\n",
       "      <td>[None]</td>\n",
       "      <td>[None]</td>\n",
       "      <td>[Normal]</td>\n",
       "      <td></td>\n",
       "      <td>[None]</td>\n",
       "      <td></td>\n",
       "      <td>[METTL3]</td>\n",
       "      <td>[Mus musculus]</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>[None]</td>\n",
       "      <td>[testis]</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>geo__gse99771_gpl21493</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td>GEO</td>\n",
       "      <td>Transcriptome analysis of gene expression and ...</td>\n",
       "      <td>[]</td>\n",
       "      <td></td>\n",
       "      <td>https://discover-test-datalake-v1.s3-ap-southe...</td>\n",
       "      <td>gct</td>\n",
       "      <td>true</td>\n",
       "      <td>geo/data/GSE76297_GPL17586.gct</td>\n",
       "      <td>[]</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>geo/data</td>\n",
       "      <td>RNASeq</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi...</td>\n",
       "      <td>[]</td>\n",
       "      <td>ap-southeast-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>geo</td>\n",
       "      <td>polly:data://geo/data/GSE76297_GPL17586.gct</td>\n",
       "      <td>[]</td>\n",
       "      <td></td>\n",
       "      <td>1684151425521</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>2018</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           dataset_id                                           abstract  \\\n",
       "0  GSE107990_GPL10558                                                      \n",
       "1  GSE128722_GPL11154                                                      \n",
       "2  GSE140509_GPL16791                                                      \n",
       "3   GSE76297_GPL17586                                                      \n",
       "4  GSE100051_GPL10558                                                      \n",
       "5  GSE100467_GPL11154  A comprehensive, cell specific microRNA catalo...   \n",
       "6  GSE102392_GPL16791                                                      \n",
       "7   GSE76311_GPL17586                                                      \n",
       "8   GSE93505_GPL16791  Free Fatty Acids Rewire Cancer Metabolism in O...   \n",
       "9   GSE99771_GPL21493                                                      \n",
       "\n",
       "                 author                     bucket     curated_cell_line  \\\n",
       "0                        discover-test-datalake-v1                [None]   \n",
       "1                        discover-test-datalake-v1  [U-87MG ATCC, LN-18]   \n",
       "2                        discover-test-datalake-v1            [NCI-H460]   \n",
       "3                        discover-test-datalake-v1                [None]   \n",
       "4                        discover-test-datalake-v1                [None]   \n",
       "5  Matthias,,Huebenthal  discover-test-datalake-v1                [None]   \n",
       "6                        discover-test-datalake-v1                [None]   \n",
       "7                        discover-test-datalake-v1   [none, test, test1]   \n",
       "8                        discover-test-datalake-v1               [MCF-7]   \n",
       "9                        discover-test-datalake-v1                [None]   \n",
       "\n",
       "                     curated_cell_type  \\\n",
       "0                               [None]   \n",
       "1                                [bgb]   \n",
       "2                     [malignant cell]   \n",
       "3                    [Langerhans cell]   \n",
       "4                               [None]   \n",
       "5                         [blood cell]   \n",
       "6        [respiratory epithelial cell]   \n",
       "7  [Langerhans cell, Another cellline]   \n",
       "8                     [malignant cell]   \n",
       "9                               [None]   \n",
       "\n",
       "                                     curated_disease curated_disease_type  \\\n",
       "0  [Influenza, Human, Acquired Immunodeficiency S...                        \n",
       "1                                 [Glioma, disease2]                        \n",
       "2                            [Carcinoma, Large Cell]                        \n",
       "3  [Neoplastic Cells, Circulating, Obesity, Carci...                        \n",
       "4      [Congenital Abnormalities, Diabetes Mellitus]                        \n",
       "5                                 [Normal, disease2]                        \n",
       "6                     [Respiratory Tract Infections]                        \n",
       "7  [Carcinoma, Hepatocellular, Obesity, Leukoence...                        \n",
       "8  [Osteoporosis, Postmenopausal, Breast Neoplasm...                        \n",
       "9                                           [Normal]                        \n",
       "\n",
       "   curated_drug curated_file_type                       curated_gene  \\\n",
       "0  [prednisone]                                                   []   \n",
       "1        [None]                                               [None]   \n",
       "2        [None]                                    [DSP, PKP1, PKP2]   \n",
       "3        [None]                    [ARID2, TP53, ARID1A, ECT2, PLK1]   \n",
       "4        [None]                    [ELF5, VGLL1, CDX2, EOMES, TEAD4]   \n",
       "5        [None]                                           [MIR486-1]   \n",
       "6        [None]                                         [IL18, IL1A]   \n",
       "7        [none]                    [TP53, ARID1A, ECT2, PLK1, ARID2]   \n",
       "8        [None]                                  [ESRRA, ESR1, MTOR]   \n",
       "9        [None]                                             [METTL3]   \n",
       "\n",
       "  curated_organism curated_project curated_repo  curated_strain  \\\n",
       "0   [Homo sapiens]                               [None1, None2]   \n",
       "1             [tt]                                       [None]   \n",
       "2   [Homo sapiens]                                       [None]   \n",
       "3   [Homo sapiens]                                       [None]   \n",
       "4   [Homo sapiens]                                       [None]   \n",
       "5   [Homo sapiens]                                       [None]   \n",
       "6   [Homo sapiens]                                       [None]   \n",
       "7   [Homo sapiens]                                       [none]   \n",
       "8               []                                       [None]   \n",
       "9   [Mus musculus]                                       [None]   \n",
       "\n",
       "                     curated_tissue curation_version data_matrix_available  \\\n",
       "0                            [None]       2023_02_02                         \n",
       "1                              [rg]       2023_02_02                         \n",
       "2                            [lung]               g3                  true   \n",
       "3                           [liver]                                   true   \n",
       "4                        [placenta]       2023_02_02                  true   \n",
       "5  [serum, blood, peripheral blood]       2023_02_02                  true   \n",
       "6          [respiratory epithelium]       2023_02_02                         \n",
       "7                           [liver]                                  false   \n",
       "8                          [breast]       2023_01_27                  true   \n",
       "9                          [testis]                                          \n",
       "\n",
       "  data_repository data_source          data_table_name  data_table_version  \\\n",
       "0                              geo__gse107990_gpl10558                   0   \n",
       "1             geo              geo__gse128722_gpl11154                   0   \n",
       "2                              geo__gse140509_gpl16791                   0   \n",
       "3                               geo__gse76297_gpl17586                   0   \n",
       "4                              geo__gse100051_gpl10558                   0   \n",
       "5                              geo__gse100467_gpl11154                   0   \n",
       "6             geo              geo__gse102392_gpl16791                   0   \n",
       "7                               geo__gse76311_gpl17586                   0   \n",
       "8                               geo__gse93505_gpl16791                   0   \n",
       "9                               geo__gse99771_gpl21493                   0   \n",
       "\n",
       "         data_type dataset_source  \\\n",
       "0  Transcriptomics            GEO   \n",
       "1  Transcriptomics            vfv   \n",
       "2  Transcriptomics            GEO   \n",
       "3  Transcriptomics            GEO   \n",
       "4  Transcriptomics            GEO   \n",
       "5  Transcriptomics            GEO   \n",
       "6  Transcriptomics         source   \n",
       "7  Transcriptomics            GEO   \n",
       "8  Transcriptomics            GEO   \n",
       "9                             GEO   \n",
       "\n",
       "                                         description  \\\n",
       "0  Gene expression profiles associated with humor...   \n",
       "1                                                bbg   \n",
       "2             Exploring the downstream genes of PKP2   \n",
       "3  Gene expression data of human hepatocellular c...   \n",
       "4  Comparative analysis of mouse and human placen...   \n",
       "5  A comprehensive, cell specific microRNA catalo...   \n",
       "6  Transcriptome analysis of human airway epithel...   \n",
       "7  Common Molecular Subtypes Among Asian Hepatoce...   \n",
       "8                                                      \n",
       "9  Transcriptome analysis of gene expression and ...   \n",
       "\n",
       "                                         drug_smiles experimental_design  \\\n",
       "0  [[H][C@@]12CCC3=CC(=O)C=C[C@]3(C)[C@@]1([H])C(...                       \n",
       "1  [[C@H]12O[C@H]([C@@H]([C@@H]1OP(OC2)(=O)O)OC(=...                       \n",
       "2                                                 []                       \n",
       "3                                                 []                       \n",
       "4                                                 []                       \n",
       "5                                                 []                       \n",
       "6                                                 []                       \n",
       "7                                                 []                       \n",
       "8                                                 []                       \n",
       "9                                                 []                       \n",
       "\n",
       "                                       file_location file_type is_current  \\\n",
       "0  https://discover-test-datalake-v1.s3-ap-southe...       gct       true   \n",
       "1  https://discover-test-datalake-v1.s3-ap-southe...       gct       true   \n",
       "2  https://discover-test-datalake-v1.s3-ap-southe...       gct       true   \n",
       "3  https://discover-test-datalake-v1.s3-ap-southe...       gct       true   \n",
       "4  https://discover-test-datalake-v1.s3-ap-southe...       gct       true   \n",
       "5  https://discover-test-datalake-v1.s3-ap-southe...       gct       true   \n",
       "6  https://discover-test-datalake-v1.s3-ap-southe...       gct       true   \n",
       "7  https://discover-test-datalake-v1.s3-ap-southe...       gct       true   \n",
       "8  https://discover-test-datalake-v1.s3-ap-southe...       gct       true   \n",
       "9  https://discover-test-datalake-v1.s3-ap-southe...       gct       true   \n",
       "\n",
       "                                                 key  \\\n",
       "0  geo/data/Microarray/GSE107990/GCT/GSE107990_GP...   \n",
       "1  geo/data/RNASeq/GSE128722/GCT/GSE128722_GPL111...   \n",
       "2  geo/data/geo/RNASeq/GSE140509/GCT/GSE140509_GP...   \n",
       "3          geo/data/31Oct_1222/GSE76297_GPL17586.gct   \n",
       "4          geo/data/geo/19Oct/GSE100051_GPL10558.gct   \n",
       "5             geo/data/6Dec22/GSE100467_GPL11154.gct   \n",
       "6  geo/data/RNASeq/GSE102392/GCT/GSE102392_GPL167...   \n",
       "7          geo/data/31Oct_1222/GSE76311_GPL17586.gct   \n",
       "8  geo/data/geo/RNASeq/GSE93505/GCT/GSE93505_GPL1...   \n",
       "9                     geo/data/GSE76297_GPL17586.gct   \n",
       "\n",
       "                                     missing_samples  \\\n",
       "0                                                 []   \n",
       "1                                                 []   \n",
       "2                                                 []   \n",
       "3                                                 []   \n",
       "4                                                 []   \n",
       "5  [GSM2683537, GSM2683917, GSM2683632, GSM268389...   \n",
       "6                                                 []   \n",
       "7                                                 []   \n",
       "8                                                 []   \n",
       "9                                                 []   \n",
       "\n",
       "                                           operation  \\\n",
       "0                                                      \n",
       "1  {\"is_batch_corrected\": \"false\", \"normalized_ty...   \n",
       "2                                                      \n",
       "3                                                      \n",
       "4                                                      \n",
       "5                                                      \n",
       "6  {\"is_batch_corrected\": \"false\", \"normalized_ty...   \n",
       "7                                                      \n",
       "8                                                      \n",
       "9                                                      \n",
       "\n",
       "                                      overall_design   package  \\\n",
       "0                                                     geo/data   \n",
       "1                                                     geo/data   \n",
       "2  To explore the downstream genes of PKP2,we per...  geo/data   \n",
       "3  For gene expression profiling, tumors and pair...  geo/data   \n",
       "4  In order to characerise the human cytotrophobl...  geo/data   \n",
       "5  miRNomes from seven distinct blood cell popula...  geo/data   \n",
       "6                                                     geo/data   \n",
       "7                         Refer to individual Series  geo/data   \n",
       "8  Examination of metabolic pathways in MCF-7 cel...  geo/data   \n",
       "9                                                     geo/data   \n",
       "\n",
       "           platform                                        publication  \\\n",
       "0  unknown_platform                                                      \n",
       "1            tbtbgb  https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi...   \n",
       "2            RNAseq  https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi...   \n",
       "3        Microarray  https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi...   \n",
       "4        Microarray  https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi...   \n",
       "5            RNASeq  https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi...   \n",
       "6            RNASeq  https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi...   \n",
       "7        Microarray  https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi...   \n",
       "8            RNAseq  https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi...   \n",
       "9            RNASeq  https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi...   \n",
       "\n",
       "  raw_disease          region sample_field sample_id src_repo  \\\n",
       "0          []  ap-southeast-1                             geo   \n",
       "1          []  ap-southeast-1                             geo   \n",
       "2          []  ap-southeast-1                             geo   \n",
       "3          []  ap-southeast-1                             geo   \n",
       "4          []  ap-southeast-1                             geo   \n",
       "5          []  ap-southeast-1                             geo   \n",
       "6          []  ap-southeast-1                             geo   \n",
       "7          []  ap-southeast-1                             geo   \n",
       "8          []  ap-southeast-1                             geo   \n",
       "9          []  ap-southeast-1                             geo   \n",
       "\n",
       "                                             src_uri         subseries_ids  \\\n",
       "0  polly:data://geo/data/Microarray/GSE107990/GCT...                    []   \n",
       "1  polly:data://geo/data/RNASeq/GSE128722/GCT/GSE...                    []   \n",
       "2  polly:data://geo/data/geo/RNASeq/GSE140509/GCT...                    []   \n",
       "3  polly:data://geo/data/31Oct_1222/GSE76297_GPL1...                    []   \n",
       "4  polly:data://geo/data/geo/19Oct/GSE100051_GPL1...                    []   \n",
       "5  polly:data://geo/data/6Dec22/GSE100467_GPL1115...                    []   \n",
       "6  polly:data://geo/data/RNASeq/GSE102392/GCT/GSE...                    []   \n",
       "7  polly:data://geo/data/31Oct_1222/GSE76311_GPL1...  [GSE76213, GSE76297]   \n",
       "8  polly:data://geo/data/geo/RNASeq/GSE93505/GCT/...                    []   \n",
       "9        polly:data://geo/data/GSE76297_GPL17586.gct                    []   \n",
       "\n",
       "                                             summary     timestamp_  \\\n",
       "0                                                     1686052005503   \n",
       "1                                                     1685628391670   \n",
       "2  Plakophilin 2 (PKP2), encodes a plakophilin pr...  1675920716537   \n",
       "3  We used Affymetrix HTA2.0 microarray profiling...  1686904619239   \n",
       "4  In this study, we compared the genome-wide tra...  1675920676897   \n",
       "5  A comprehensive, cell specific microRNA catalo...  1685970450268   \n",
       "6                                                     1685628380248   \n",
       "7  This SuperSeries is composed of the SubSeries ...  1687343553467   \n",
       "8  Obesity is a risk factor for postmenopausal ER...  1675920756278   \n",
       "9                                                     1684151425521   \n",
       "\n",
       "   total_num_samples  version  year  \n",
       "0                671        0        \n",
       "1                  5        0   btb  \n",
       "2                  2        0        \n",
       "3                304        0        \n",
       "4                 54        0  2018  \n",
       "5                445        0  2019  \n",
       "6                  4        0  2018  \n",
       "7                304        0  2015  \n",
       "8                  3        0        \n",
       "9                  8        0  2018  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query = \"Select * from geo.datasets LIMIT 100\"\n",
    "omix.query_metadata(query)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Query execution succeeded (time taken: 1.24 seconds, data scanned: 0.010 MB)\n"
     ]
    }
   ],
   "source": [
    "itr = omix.query_metadata_iterator(query)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
