{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "colab_type": "text",
        "id": "view-in-github"
      },
      "source": [
        "<a href=\"https://testpolly.elucidata.io/manage/workspaces?action=open_polly_notebook&source=github&path=path_place_holder&kernel=elucidata/Python 3.10&machine=large\" target=\"_parent\"><img src=\"https://elucidatainc.github.io/PublicAssets/open_polly.svg\" alt=\"Open in Polly\"/></a>\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "# Welcome to Polly Python3 Notebook."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {
        "scrolled": false
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "/bin/bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)\n",
            "Collecting polly-python==0.1.6-validator\n",
            "  Downloading https://elucidatainc.github.io/PublicAssets/builds/polly-python/tests/testpolly/polly_python-0.1.6_validator-py3-none-any.whl (67 kB)\n",
            "\u001b[K     |████████████████████████████████| 67 kB 9.2 MB/s  eta 0:00:01\n",
            "\u001b[?25hCollecting polly-validator@ https://elucidatainc.github.io/PublicAssets/builds/polly_validator-0.0.1-py3-none-any.whl\n",
            "  Downloading https://elucidatainc.github.io/PublicAssets/builds/polly_validator-0.0.1-py3-none-any.whl (19 kB)\n",
            "Collecting postpy2==0.0.6\n",
            "  Downloading postpy2-0.0.6-py3-none-any.whl (17 kB)\n",
            "Requirement already satisfied: six==1.16.0 in /usr/local/lib/python3.10/site-packages (from polly-python==0.1.6-validator) (1.16.0)\n",
            "Collecting Cerberus==1.3.2\n",
            "  Downloading Cerberus-1.3.2.tar.gz (52 kB)\n",
            "\u001b[K     |████████████████████████████████| 52 kB 3.4 MB/s  eta 0:00:01\n",
            "\u001b[?25hCollecting elucidatacloudpathlib==0.6.6\n",
            "  Downloading elucidatacloudpathlib-0.6.6-py3-none-any.whl (50 kB)\n",
            "\u001b[K     |████████████████████████████████| 50 kB 19.0 MB/s  eta 0:00:01\n",
            "\u001b[?25hRequirement already satisfied: tqdm in /usr/local/lib/python3.10/site-packages (from polly-python==0.1.6-validator) (4.64.0)\n",
            "Requirement already satisfied: boto3>=1.17.73 in /usr/local/lib/python3.10/site-packages (from polly-python==0.1.6-validator) (1.21.19)\n",
            "Requirement already satisfied: python-dateutil==2.8.1 in /usr/local/lib/python3.10/site-packages (from polly-python==0.1.6-validator) (2.8.1)\n",
            "Requirement already satisfied: pandas in /usr/local/lib/python3.10/site-packages (from polly-python==0.1.6-validator) (1.4.1)\n",
            "Collecting pytest\n",
            "  Downloading pytest-7.1.2-py3-none-any.whl (297 kB)\n",
            "\u001b[K     |████████████████████████████████| 297 kB 66.8 MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting black\n",
            "  Downloading black-22.6.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.5 MB)\n",
            "\u001b[K     |████████████████████████████████| 1.5 MB 135.6 MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting joblib\n",
            "  Downloading joblib-1.1.0-py2.py3-none-any.whl (306 kB)\n",
            "\u001b[K     |████████████████████████████████| 306 kB 151.0 MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting certifi==2021.10.8\n",
            "  Downloading certifi-2021.10.8-py2.py3-none-any.whl (149 kB)\n",
            "\u001b[K     |████████████████████████████████| 149 kB 154.4 MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting sqlparse\n",
            "  Downloading sqlparse-0.4.2-py3-none-any.whl (42 kB)\n",
            "\u001b[K     |████████████████████████████████| 42 kB 3.2 MB/s s eta 0:00:01\n",
            "\u001b[?25hRequirement already satisfied: idna==2.10 in /usr/local/lib/python3.10/site-packages (from polly-python==0.1.6-validator) (2.10)\n",
            "Collecting retrying==1.3.3\n",
            "  Downloading retrying-1.3.3.tar.gz (10 kB)\n",
            "Requirement already satisfied: chardet==4.0.0 in /usr/local/lib/python3.10/site-packages (from polly-python==0.1.6-validator) (4.0.0)\n",
            "Collecting cmapPy\n",
            "  Downloading cmapPy-4.0.1-py2.py3-none-any.whl (150 kB)\n",
            "\u001b[K     |████████████████████████████████| 150 kB 134.6 MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting Deprecated\n",
            "  Downloading Deprecated-1.2.13-py2.py3-none-any.whl (9.6 kB)\n",
            "Requirement already satisfied: urllib3==1.26.6 in /usr/local/lib/python3.10/site-packages (from polly-python==0.1.6-validator) (1.26.6)\n",
            "Collecting rst2txt\n",
            "  Downloading rst2txt-1.1.0-py2.py3-none-any.whl (12 kB)\n",
            "Collecting pytz==2021.1\n",
            "  Downloading pytz-2021.1-py2.py3-none-any.whl (510 kB)\n",
            "\u001b[K     |████████████████████████████████| 510 kB 133.4 MB/s eta 0:00:01\n",
            "\u001b[?25hRequirement already satisfied: botocore>=1.20.73 in /usr/local/lib/python3.10/site-packages (from polly-python==0.1.6-validator) (1.24.19)\n",
            "Collecting python-magic==0.4.24\n",
            "  Downloading python_magic-0.4.24-py2.py3-none-any.whl (12 kB)\n",
            "Collecting requests==2.25.1\n",
            "  Downloading requests-2.25.1-py2.py3-none-any.whl (61 kB)\n",
            "\u001b[K     |████████████████████████████████| 61 kB 22.2 MB/s  eta 0:00:01\n",
            "\u001b[?25hRequirement already satisfied: setuptools in /usr/local/lib/python3.10/site-packages (from Cerberus==1.3.2->polly-python==0.1.6-validator) (58.1.0)\n",
            "Requirement already satisfied: jmespath<1.0.0,>=0.7.1 in /usr/local/lib/python3.10/site-packages (from boto3>=1.17.73->polly-python==0.1.6-validator) (0.10.0)\n",
            "Requirement already satisfied: s3transfer<0.6.0,>=0.5.0 in /usr/local/lib/python3.10/site-packages (from boto3>=1.17.73->polly-python==0.1.6-validator) (0.5.2)\n",
            "Collecting mypy-extensions>=0.4.3\n",
            "  Downloading mypy_extensions-0.4.3-py2.py3-none-any.whl (4.5 kB)\n",
            "Collecting click>=8.0.0\n",
            "  Downloading click-8.1.3-py3-none-any.whl (96 kB)\n",
            "\u001b[K     |████████████████████████████████| 96 kB 14.2 MB/s  eta 0:00:01\n",
            "\u001b[?25hCollecting platformdirs>=2\n",
            "  Downloading platformdirs-2.5.2-py3-none-any.whl (14 kB)\n",
            "Collecting tomli>=1.1.0\n",
            "  Downloading tomli-2.0.1-py3-none-any.whl (12 kB)\n",
            "Collecting pathspec>=0.9.0\n",
            "  Downloading pathspec-0.9.0-py2.py3-none-any.whl (31 kB)\n",
            "Collecting h5py>=2.6.0\n",
            "  Downloading h5py-3.7.0-cp310-cp310-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (4.5 MB)\n",
            "\u001b[K     |████████████████████████████████| 4.5 MB 108.1 MB/s eta 0:00:01\n",
            "\u001b[?25hRequirement already satisfied: numpy>=1.11.2 in /usr/local/lib/python3.10/site-packages (from cmapPy->polly-python==0.1.6-validator) (1.22.3)\n",
            "Collecting wrapt<2,>=1.10\n",
            "  Downloading wrapt-1.14.1-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (77 kB)\n",
            "\u001b[K     |████████████████████████████████| 77 kB 6.8 MB/s eta 0:00:011\n",
            "\u001b[?25hCollecting psutil==5.9.0\n",
            "  Downloading psutil-5.9.0-cp310-cp310-manylinux_2_12_x86_64.manylinux2010_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (281 kB)\n",
            "\u001b[K     |████████████████████████████████| 281 kB 102.2 MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting pydantic==1.9.1\n",
            "  Downloading pydantic-1.9.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (12.0 MB)\n",
            "\u001b[K     |████████████████████████████████| 12.0 MB 22.6 MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting SQLAlchemy==1.4.39\n",
            "  Downloading SQLAlchemy-1.4.39-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.6 MB)\n",
            "\u001b[K     |████████████████████████████████| 1.6 MB 61.9 MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting appdirs==1.4.4\n",
            "  Downloading appdirs-1.4.4-py2.py3-none-any.whl (9.6 kB)\n",
            "Collecting tabulate==0.8.10\n",
            "  Downloading tabulate-0.8.10-py3-none-any.whl (29 kB)\n",
            "Collecting pandas\n",
            "  Downloading pandas-1.4.3-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (11.6 MB)\n",
            "\u001b[K     |████████████████████████████████| 11.6 MB 8.2 MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting rapidfuzz==2.1.1\n",
            "  Downloading rapidfuzz-2.1.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (2.0 MB)\n",
            "\u001b[K     |████████████████████████████████| 2.0 MB 134.5 MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting pronto==2.4.7\n",
            "  Downloading pronto-2.4.7-py2.py3-none-any.whl (60 kB)\n",
            "\u001b[K     |████████████████████████████████| 60 kB 1.2 MB/s eta 0:00:011\n",
            "\u001b[?25hCollecting fastobo~=0.11.1\n",
            "  Downloading fastobo-0.11.1-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (2.1 MB)\n",
            "\u001b[K     |████████████████████████████████| 2.1 MB 126.2 MB/s eta 0:00:01\n",
            "\u001b[?25hRequirement already satisfied: networkx~=2.3 in /usr/local/lib/python3.10/site-packages (from pronto==2.4.7->polly-validator@ https://elucidatainc.github.io/PublicAssets/builds/polly_validator-0.0.1-py3-none-any.whl->polly-python==0.1.6-validator) (2.8.2)\n",
            "Collecting typing-extensions>=3.7.4.3\n",
            "  Downloading typing_extensions-4.3.0-py3-none-any.whl (25 kB)\n",
            "Collecting jarowinkler<1.1.0,>=1.0.3\n",
            "  Downloading jarowinkler-1.0.5-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (103 kB)\n",
            "\u001b[K     |████████████████████████████████| 103 kB 6.0 MB/s eta 0:00:01\n",
            "\u001b[?25hRequirement already satisfied: greenlet!=0.4.17 in /usr/local/lib/python3.10/site-packages (from SQLAlchemy==1.4.39->polly-validator@ https://elucidatainc.github.io/PublicAssets/builds/polly_validator-0.0.1-py3-none-any.whl->polly-python==0.1.6-validator) (1.1.2)\n",
            "Requirement already satisfied: attrs>=19.2.0 in /usr/local/lib/python3.10/site-packages (from pytest->polly-python==0.1.6-validator) (21.2.0)\n",
            "Collecting py>=1.8.2\n",
            "  Downloading py-1.11.0-py2.py3-none-any.whl (98 kB)\n",
            "\u001b[K     |████████████████████████████████| 98 kB 14.9 MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting iniconfig\n",
            "  Downloading iniconfig-1.1.1-py2.py3-none-any.whl (5.0 kB)\n",
            "Requirement already satisfied: packaging in /usr/local/lib/python3.10/site-packages (from pytest->polly-python==0.1.6-validator) (21.3)\n"
          ]
        },
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Collecting pluggy<2.0,>=0.12\n",
            "  Downloading pluggy-1.0.0-py2.py3-none-any.whl (13 kB)\n",
            "Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in /usr/local/lib/python3.10/site-packages (from packaging->pytest->polly-python==0.1.6-validator) (3.0.6)\n",
            "Requirement already satisfied: docutils>=0.11 in /usr/local/lib/python3.10/site-packages (from rst2txt->polly-python==0.1.6-validator) (0.15.2)\n",
            "Requirement already satisfied: pygments in /usr/local/lib/python3.10/site-packages (from rst2txt->polly-python==0.1.6-validator) (2.10.0)\n",
            "Using legacy 'setup.py install' for Cerberus, since package 'wheel' is not installed.\n",
            "Using legacy 'setup.py install' for retrying, since package 'wheel' is not installed.\n",
            "Installing collected packages: typing-extensions, pytz, jarowinkler, fastobo, certifi, wrapt, tomli, tabulate, SQLAlchemy, requests, rapidfuzz, python-magic, pydantic, py, psutil, pronto, pluggy, platformdirs, pathspec, pandas, mypy-extensions, iniconfig, h5py, click, appdirs, sqlparse, rst2txt, retrying, pytest, postpy2, polly-validator, joblib, elucidatacloudpathlib, Deprecated, cmapPy, Cerberus, black, polly-python\n",
            "  Attempting uninstall: pytz\n",
            "    Found existing installation: pytz 2021.3\n",
            "    Uninstalling pytz-2021.3:\n",
            "      Successfully uninstalled pytz-2021.3\n",
            "  Attempting uninstall: certifi\n",
            "    Found existing installation: certifi 2021.5.30\n",
            "    Uninstalling certifi-2021.5.30:\n",
            "      Successfully uninstalled certifi-2021.5.30\n",
            "  Attempting uninstall: tabulate\n",
            "    Found existing installation: tabulate 0.8.9\n",
            "    Uninstalling tabulate-0.8.9:\n",
            "      Successfully uninstalled tabulate-0.8.9\n",
            "  Attempting uninstall: SQLAlchemy\n",
            "    Found existing installation: SQLAlchemy 1.4.27\n",
            "    Uninstalling SQLAlchemy-1.4.27:\n",
            "      Successfully uninstalled SQLAlchemy-1.4.27\n",
            "  Attempting uninstall: requests\n",
            "    Found existing installation: requests 2.27.1\n",
            "    Uninstalling requests-2.27.1:\n",
            "      Successfully uninstalled requests-2.27.1\n",
            "  Attempting uninstall: psutil\n",
            "    Found existing installation: psutil 5.9.1\n",
            "    Uninstalling psutil-5.9.1:\n",
            "      Successfully uninstalled psutil-5.9.1\n",
            "  Attempting uninstall: pandas\n",
            "    Found existing installation: pandas 1.4.1\n",
            "    Uninstalling pandas-1.4.1:\n",
            "      Successfully uninstalled pandas-1.4.1\n",
            "    Running setup.py install for retrying ... \u001b[?25ldone\n",
            "\u001b[?25h  Attempting uninstall: Cerberus\n",
            "    Found existing installation: Cerberus 1.3.4\n",
            "    Uninstalling Cerberus-1.3.4:\n",
            "      Successfully uninstalled Cerberus-1.3.4\n",
            "    Running setup.py install for Cerberus ... \u001b[?25ldone\n",
            "\u001b[?25hSuccessfully installed Cerberus-1.3.2 Deprecated-1.2.13 SQLAlchemy-1.4.39 appdirs-1.4.4 black-22.6.0 certifi-2021.10.8 click-8.1.3 cmapPy-4.0.1 elucidatacloudpathlib-0.6.6 fastobo-0.11.1 h5py-3.7.0 iniconfig-1.1.1 jarowinkler-1.0.5 joblib-1.1.0 mypy-extensions-0.4.3 pandas-1.4.3 pathspec-0.9.0 platformdirs-2.5.2 pluggy-1.0.0 polly-python-0.1.6 polly-validator-0.0.1 postpy2-0.0.6 pronto-2.4.7 psutil-5.9.0 py-1.11.0 pydantic-1.9.1 pytest-7.1.2 python-magic-0.4.24 pytz-2021.1 rapidfuzz-2.1.1 requests-2.25.1 retrying-1.3.3 rst2txt-1.1.0 sqlparse-0.4.2 tabulate-0.8.10 tomli-2.0.1 typing-extensions-4.3.0 wrapt-1.14.1\n",
            "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\n",
            "\u001b[33mWARNING: You are using pip version 21.2.4; however, version 22.1.2 is available.\n",
            "You should consider upgrading via the '/usr/local/bin/python3.10 -m pip install --upgrade pip' command.\u001b[0m\n"
          ]
        }
      ],
      "source": [
        "!sudo pip3 install https://elucidatainc.github.io/PublicAssets/builds/polly-python/tests/testpolly/polly_python-0.1.6_validator-py3-none-any.whl"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 2,
      "metadata": {},
      "outputs": [],
      "source": [
        "import os\n",
        "from polly.auth import Polly\n",
        "AUTH_TOKEN=(os.environ['POLLY_REFRESH_TOKEN']) # Obtain authentication tokens\n",
        "Polly.auth(AUTH_TOKEN)\n",
        "from polly.omixatlas import OmixAtlas\n",
        "omixatlas = OmixAtlas()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 3,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Retrieving ontological names...\n"
          ]
        },
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "/usr/local/lib/python3.10/site-packages/polly_validator/downloader/get_valid_names.py:49: UnicodeWarning: unsound encoding, assuming utf-8 (99% confidence)\n",
            "  diseases = pt.Ontology(f'https://raw.githubusercontent.com/bioinfo-el/ontologies/'\n"
          ]
        },
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Getting predefined values for Data Type and Data Source...\n"
          ]
        }
      ],
      "source": [
        "from polly_validator.validators import dataset_metadata_validator"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 4,
      "metadata": {},
      "outputs": [],
      "source": [
        "from polly_validator.validators import sample_metadata_validator"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 6,
      "metadata": {},
      "outputs": [
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "--2022-07-18 11:30:43--  https://discover-test-datalake-v1.s3.amazonaws.com/geo/data/RNASeq/GSE100035/GCT/GSE100035_GPL21103_curated.gct?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=ASIARBT2ZRFJIBLZRQUU%2F20220718%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Date=20220718T113043Z&X-Amz-Expires=3600&X-Amz-SignedHeaders=host&X-Amz-Security-Token=IQoJb3JpZ2luX2VjELT%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaDmFwLXNvdXRoZWFzdC0xIkgwRgIhALk%2BAr8kG0ikqrvOb6Zt4LaQWXftTyAwj4khula1ch9rAiEAjQrGJKCtU9xKhcFQZDJSJaWAkV2aSRAd7WcFfLz09z8qqgII7f%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARACGgwwNzIxOTgyMjgzMDYiDD8qoL0uLO1Xpw9u2Sr%2BASShaJx%2F%2BH%2F1xILPzd7504NLDIl9ej35%2BbMVdtfcKwnfCWMYNuAYpX4NdZvWmJ2vwJKtA%2Bt2k%2BXrv9v6YZPnvtBNUb9gnqPbKel%2B1TThJp9OQSyuaVkV%2BXXMXs33GcFIrMHA7L17Dvnw3Zv10KyYTPWaNcK5xRDQtDXoNBDXBIXpl%2BzUdqsPjT%2BrWqluSErn5VQjmFK1o%2BzCkLHOYe2FCiozW4LsHJ7eMkdQP3Fu52Hgd3%2FLlaYvhfmfTobRaWI9u25KW67G80q4EdXTGLsXJloLJAUPVJikZbxLjgCGANh4yntTHzszJfI5W9suN5TOaV69qMKE3LEeGGZtJqaKMNaI1ZYGOpkB%2FOOcckvlu8sePesgE0sHjR1zMKGvnaxkqNoANQvh94JeYWDAAG7fegJho9f2gU6oE2iaQDVBf9ZhACWeLZItmlNfFRWQuGBRhgkNQI8eRZ%2B%2FsMv7EwkOCPoImZQeBzcKzpw5N74tYgs5rHpbLcdOCuND13oVaIoVfC%2B2mE0ExLTtH%2FyYzLdpDnfhgqIkwWuj21g2WEEJqCdD&X-Amz-Signature=24ec3b969d29141df35b429f20af0462e32c6b42e8242f12f3d25e58a9132e83\n",
            "Resolving discover-test-datalake-v1.s3.amazonaws.com (discover-test-datalake-v1.s3.amazonaws.com)... 52.219.124.100\n",
            "Connecting to discover-test-datalake-v1.s3.amazonaws.com (discover-test-datalake-v1.s3.amazonaws.com)|52.219.124.100|:443... connected.\n",
            "HTTP request sent, awaiting response... "
          ]
        },
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Downloaded data successfully\n"
          ]
        },
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "200 OK\n",
            "Length: 770919 (753K) [binary/octet-stream]\n",
            "Saving to: 'GSE100035_GPL21103.gct'\n",
            "\n",
            "     0K .......... .......... .......... .......... ..........  6% 28.2M 0s\n",
            "    50K .......... .......... .......... .......... .......... 13% 55.6M 0s\n",
            "   100K .......... .......... .......... .......... .......... 19% 55.7M 0s\n",
            "   150K .......... .......... .......... .......... .......... 26% 59.2M 0s\n",
            "   200K .......... .......... .......... .......... .......... 33% 58.2M 0s\n",
            "   250K .......... .......... .......... .......... .......... 39%  181M 0s\n",
            "   300K .......... .......... .......... .......... .......... 46% 59.3M 0s\n",
            "   350K .......... .......... .......... .......... .......... 53% 62.6M 0s\n",
            "   400K .......... .......... .......... .......... .......... 59% 60.1M 0s\n",
            "   450K .......... .......... .......... .......... .......... 66%  189M 0s\n",
            "   500K .......... .......... .......... .......... .......... 73% 61.9M 0s\n",
            "   550K .......... .......... .......... .......... .......... 79% 59.6M 0s\n",
            "   600K .......... .......... .......... .......... .......... 86%  175M 0s\n",
            "   650K .......... .......... .......... .......... .......... 92% 66.3M 0s\n",
            "   700K .......... .......... .......... .......... .......... 99%  199M 0s\n",
            "   750K ..                                                    100% 5437G=0.01s\n",
            "\n",
            "2022-07-18 11:30:44 (67.1 MB/s) - 'GSE100035_GPL21103.gct' saved [770919/770919]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "dataset_id = \"GSE100035_GPL21103\"\n",
        "repo_key = \"geo\"\n",
        "file_name = f\"{dataset_id}.gct\"\n",
        "data = omixatlas.download_data(repo_key, dataset_id)\n",
        "url = data.get('data').get('attributes').get('download_url')\n",
        "status = os.system(f\"wget -O '{file_name}' '{url}'\")\n",
        "if status == 0:\n",
        "    print(\"Downloaded data successfully\")\n",
        "else:\n",
        "    raise Exception(\"Download not successful\")"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 7,
      "metadata": {},
      "outputs": [],
      "source": [
        "import pandas as pd\n",
        "import cmapPy\n",
        "from cmapPy.pandasGEXpress.parse_gct import parse\n",
        "\n",
        "gct_obj = parse(file_name) # Parse the file to create a gct object\n",
        "df_real = gct_obj.data_df # Extract the dataframe from the gct object\n",
        "col_metadata = gct_obj.col_metadata_df "
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 8,
      "metadata": {
        "scrolled": false
      },
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th>chd</th>\n",
              "      <th>title</th>\n",
              "      <th>geo_accession</th>\n",
              "      <th>status</th>\n",
              "      <th>submission_date</th>\n",
              "      <th>last_update_date</th>\n",
              "      <th>type</th>\n",
              "      <th>channel_count</th>\n",
              "      <th>source_name_ch1</th>\n",
              "      <th>organism_ch1</th>\n",
              "      <th>characteristics_ch1</th>\n",
              "      <th>...</th>\n",
              "      <th>kw_curated_tissue</th>\n",
              "      <th>kw_curated_genetic_mod_type</th>\n",
              "      <th>kw_curated_modified_gene</th>\n",
              "      <th>curated_min_age</th>\n",
              "      <th>curated_max_age</th>\n",
              "      <th>curated_age_unit</th>\n",
              "      <th>kw_curated_gender</th>\n",
              "      <th>curated_is_control</th>\n",
              "      <th>curated_cohort_id</th>\n",
              "      <th>curated_cohort_name</th>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>cid</th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>GSM2668172</th>\n",
              "      <td>female_naive_neurons_3</td>\n",
              "      <td>GSM2668172</td>\n",
              "      <td>Public on Nov 28 2017</td>\n",
              "      <td>Jun 14 2017</td>\n",
              "      <td>May 15 2019</td>\n",
              "      <td>SRA</td>\n",
              "      <td>1</td>\n",
              "      <td>FACSed DRG neurons</td>\n",
              "      <td>Mus musculus</td>\n",
              "      <td>cell type: DRG neurons</td>\n",
              "      <td>...</td>\n",
              "      <td>none</td>\n",
              "      <td>wildtype</td>\n",
              "      <td>none</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>none</td>\n",
              "      <td>Female</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>AdvGFP, gensat.org : STOCK Tg(Avil-EGFP)QD84Gs...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>GSM2668165</th>\n",
              "      <td>male_naive_neurons_7</td>\n",
              "      <td>GSM2668165</td>\n",
              "      <td>Public on Nov 28 2017</td>\n",
              "      <td>Jun 14 2017</td>\n",
              "      <td>May 15 2019</td>\n",
              "      <td>SRA</td>\n",
              "      <td>1</td>\n",
              "      <td>FACSed DRG neurons</td>\n",
              "      <td>Mus musculus</td>\n",
              "      <td>cell type: DRG neurons</td>\n",
              "      <td>...</td>\n",
              "      <td>none</td>\n",
              "      <td>wildtype</td>\n",
              "      <td>none</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>none</td>\n",
              "      <td>Male</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>AdvGFP, gensat.org : STOCK Tg(Avil-EGFP)QD84Gs...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>GSM2668171</th>\n",
              "      <td>female_naive_neurons_2</td>\n",
              "      <td>GSM2668171</td>\n",
              "      <td>Public on Nov 28 2017</td>\n",
              "      <td>Jun 14 2017</td>\n",
              "      <td>May 15 2019</td>\n",
              "      <td>SRA</td>\n",
              "      <td>1</td>\n",
              "      <td>FACSed DRG neurons</td>\n",
              "      <td>Mus musculus</td>\n",
              "      <td>cell type: DRG neurons</td>\n",
              "      <td>...</td>\n",
              "      <td>none</td>\n",
              "      <td>wildtype</td>\n",
              "      <td>none</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>none</td>\n",
              "      <td>Female</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>AdvGFP, gensat.org : STOCK Tg(Avil-EGFP)QD84Gs...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>GSM2668164</th>\n",
              "      <td>male_naive_neurons_6</td>\n",
              "      <td>GSM2668164</td>\n",
              "      <td>Public on Nov 28 2017</td>\n",
              "      <td>Jun 14 2017</td>\n",
              "      <td>May 15 2019</td>\n",
              "      <td>SRA</td>\n",
              "      <td>1</td>\n",
              "      <td>FACSed DRG neurons</td>\n",
              "      <td>Mus musculus</td>\n",
              "      <td>cell type: DRG neurons</td>\n",
              "      <td>...</td>\n",
              "      <td>none</td>\n",
              "      <td>wildtype</td>\n",
              "      <td>none</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>none</td>\n",
              "      <td>Male</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>AdvGFP, gensat.org : STOCK Tg(Avil-EGFP)QD84Gs...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>GSM2668180</th>\n",
              "      <td>female_naive_neurons_11</td>\n",
              "      <td>GSM2668180</td>\n",
              "      <td>Public on Nov 28 2017</td>\n",
              "      <td>Jun 14 2017</td>\n",
              "      <td>May 15 2019</td>\n",
              "      <td>SRA</td>\n",
              "      <td>1</td>\n",
              "      <td>FACSed DRG neurons</td>\n",
              "      <td>Mus musculus</td>\n",
              "      <td>cell type: DRG neurons</td>\n",
              "      <td>...</td>\n",
              "      <td>none</td>\n",
              "      <td>wildtype</td>\n",
              "      <td>none</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>none</td>\n",
              "      <td>Female</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>AdvGFP, gensat.org : STOCK Tg(Avil-EGFP)QD84Gs...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>GSM2668162</th>\n",
              "      <td>male_naive_neurons_4</td>\n",
              "      <td>GSM2668162</td>\n",
              "      <td>Public on Nov 28 2017</td>\n",
              "      <td>Jun 14 2017</td>\n",
              "      <td>May 15 2019</td>\n",
              "      <td>SRA</td>\n",
              "      <td>1</td>\n",
              "      <td>FACSed DRG neurons</td>\n",
              "      <td>Mus musculus</td>\n",
              "      <td>cell type: DRG neurons</td>\n",
              "      <td>...</td>\n",
              "      <td>none</td>\n",
              "      <td>wildtype</td>\n",
              "      <td>none</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>none</td>\n",
              "      <td>Male</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>AdvGFP, gensat.org : STOCK Tg(Avil-EGFP)QD84Gs...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>6 rows × 66 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "chd                           title geo_accession                 status  \\\n",
              "cid                                                                        \n",
              "GSM2668172   female_naive_neurons_3    GSM2668172  Public on Nov 28 2017   \n",
              "GSM2668165     male_naive_neurons_7    GSM2668165  Public on Nov 28 2017   \n",
              "GSM2668171   female_naive_neurons_2    GSM2668171  Public on Nov 28 2017   \n",
              "GSM2668164     male_naive_neurons_6    GSM2668164  Public on Nov 28 2017   \n",
              "GSM2668180  female_naive_neurons_11    GSM2668180  Public on Nov 28 2017   \n",
              "GSM2668162     male_naive_neurons_4    GSM2668162  Public on Nov 28 2017   \n",
              "\n",
              "chd        submission_date last_update_date type  channel_count  \\\n",
              "cid                                                               \n",
              "GSM2668172     Jun 14 2017      May 15 2019  SRA              1   \n",
              "GSM2668165     Jun 14 2017      May 15 2019  SRA              1   \n",
              "GSM2668171     Jun 14 2017      May 15 2019  SRA              1   \n",
              "GSM2668164     Jun 14 2017      May 15 2019  SRA              1   \n",
              "GSM2668180     Jun 14 2017      May 15 2019  SRA              1   \n",
              "GSM2668162     Jun 14 2017      May 15 2019  SRA              1   \n",
              "\n",
              "chd            source_name_ch1  organism_ch1     characteristics_ch1  ...  \\\n",
              "cid                                                                   ...   \n",
              "GSM2668172  FACSed DRG neurons  Mus musculus  cell type: DRG neurons  ...   \n",
              "GSM2668165  FACSed DRG neurons  Mus musculus  cell type: DRG neurons  ...   \n",
              "GSM2668171  FACSed DRG neurons  Mus musculus  cell type: DRG neurons  ...   \n",
              "GSM2668164  FACSed DRG neurons  Mus musculus  cell type: DRG neurons  ...   \n",
              "GSM2668180  FACSed DRG neurons  Mus musculus  cell type: DRG neurons  ...   \n",
              "GSM2668162  FACSed DRG neurons  Mus musculus  cell type: DRG neurons  ...   \n",
              "\n",
              "chd        kw_curated_tissue kw_curated_genetic_mod_type  \\\n",
              "cid                                                        \n",
              "GSM2668172              none                    wildtype   \n",
              "GSM2668165              none                    wildtype   \n",
              "GSM2668171              none                    wildtype   \n",
              "GSM2668164              none                    wildtype   \n",
              "GSM2668180              none                    wildtype   \n",
              "GSM2668162              none                    wildtype   \n",
              "\n",
              "chd        kw_curated_modified_gene curated_min_age curated_max_age  \\\n",
              "cid                                                                   \n",
              "GSM2668172                     none             NaN             NaN   \n",
              "GSM2668165                     none             NaN             NaN   \n",
              "GSM2668171                     none             NaN             NaN   \n",
              "GSM2668164                     none             NaN             NaN   \n",
              "GSM2668180                     none             NaN             NaN   \n",
              "GSM2668162                     none             NaN             NaN   \n",
              "\n",
              "chd        curated_age_unit kw_curated_gender curated_is_control  \\\n",
              "cid                                                                \n",
              "GSM2668172             none            Female                  0   \n",
              "GSM2668165             none              Male                  0   \n",
              "GSM2668171             none            Female                  0   \n",
              "GSM2668164             none              Male                  0   \n",
              "GSM2668180             none            Female                  0   \n",
              "GSM2668162             none              Male                  0   \n",
              "\n",
              "chd         curated_cohort_id  \\\n",
              "cid                             \n",
              "GSM2668172                  0   \n",
              "GSM2668165                  0   \n",
              "GSM2668171                  0   \n",
              "GSM2668164                  0   \n",
              "GSM2668180                  0   \n",
              "GSM2668162                  0   \n",
              "\n",
              "chd                                       curated_cohort_name  \n",
              "cid                                                            \n",
              "GSM2668172  AdvGFP, gensat.org : STOCK Tg(Avil-EGFP)QD84Gs...  \n",
              "GSM2668165  AdvGFP, gensat.org : STOCK Tg(Avil-EGFP)QD84Gs...  \n",
              "GSM2668171  AdvGFP, gensat.org : STOCK Tg(Avil-EGFP)QD84Gs...  \n",
              "GSM2668164  AdvGFP, gensat.org : STOCK Tg(Avil-EGFP)QD84Gs...  \n",
              "GSM2668180  AdvGFP, gensat.org : STOCK Tg(Avil-EGFP)QD84Gs...  \n",
              "GSM2668162  AdvGFP, gensat.org : STOCK Tg(Avil-EGFP)QD84Gs...  \n",
              "\n",
              "[6 rows x 66 columns]"
            ]
          },
          "execution_count": 8,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "col_metadata"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 9,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Query execution succeeded (time taken: 1.55 seconds, data scanned: 0.000 MB)\n",
            "Fetched 53 rows\n"
          ]
        }
      ],
      "source": [
        "query = 'DESCRIBE geo.samples'\n",
        "schema_df_sample = omixatlas.query_metadata(query, query_api_version=\"v2\")"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 12,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "╒════╤═══════════════════════════════════╤═══════════════════════════╤═════════════════════╤════════╤══════════════════╤═══════════╤═════════════╤═════════════════════════════╕\n",
            "│    │ Field                             │ Error Message             │ Error Type          │ Repo   │ src_dataset_id   │ data_id   │ sample_id   │ Original Name               │\n",
            "╞════╪═══════════════════════════════════╪═══════════════════════════╪═════════════════════╪════════╪══════════════════╪═══════════╪═════════════╪═════════════════════════════╡\n",
            "│  0 │ curated_gene_modified             │ value is not a valid list │ type_error.list     │ geo    │ NA               │ NA        │ NA          │ kw_curated_modified_gene    │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│  1 │ curated_cell_type                 │ value is not a valid list │ type_error.list     │ geo    │ NA               │ NA        │ NA          │ kw_curated_cell_type        │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│  2 │ curated_gene                      │ field required            │ value_error.missing │ geo    │ NA               │ NA        │ NA          │ kw_curated_gene             │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│  3 │ curated_drug                      │ value is not a valid list │ type_error.list     │ geo    │ NA               │ NA        │ NA          │ kw_curated_drug             │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│  4 │ curated_genetic_modification_type │ value is not a valid list │ type_error.list     │ geo    │ NA               │ NA        │ NA          │ kw_curated_genetic_mod_type │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│  5 │ curated_disease                   │ value is not a valid list │ type_error.list     │ geo    │ NA               │ NA        │ NA          │ kw_curated_disease          │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│  6 │ curated_drug_smiles_code          │ field required            │ value_error.missing │ geo    │ NA               │ NA        │ NA          │ kw_curated_smiles           │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│  7 │ curated_gene_modified             │ value is not a valid list │ type_error.list     │ geo    │ NA               │ NA        │ NA          │ kw_curated_modified_gene    │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│  8 │ curated_cell_type                 │ value is not a valid list │ type_error.list     │ geo    │ NA               │ NA        │ NA          │ kw_curated_cell_type        │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│  9 │ curated_gene                      │ field required            │ value_error.missing │ geo    │ NA               │ NA        │ NA          │ kw_curated_gene             │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│ 10 │ curated_drug                      │ value is not a valid list │ type_error.list     │ geo    │ NA               │ NA        │ NA          │ kw_curated_drug             │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│ 11 │ curated_genetic_modification_type │ value is not a valid list │ type_error.list     │ geo    │ NA               │ NA        │ NA          │ kw_curated_genetic_mod_type │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│ 12 │ curated_disease                   │ value is not a valid list │ type_error.list     │ geo    │ NA               │ NA        │ NA          │ kw_curated_disease          │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│ 13 │ curated_drug_smiles_code          │ field required            │ value_error.missing │ geo    │ NA               │ NA        │ NA          │ kw_curated_smiles           │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│ 14 │ curated_gene_modified             │ value is not a valid list │ type_error.list     │ geo    │ NA               │ NA        │ NA          │ kw_curated_modified_gene    │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│ 15 │ curated_cell_type                 │ value is not a valid list │ type_error.list     │ geo    │ NA               │ NA        │ NA          │ kw_curated_cell_type        │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│ 16 │ curated_gene                      │ field required            │ value_error.missing │ geo    │ NA               │ NA        │ NA          │ kw_curated_gene             │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│ 17 │ curated_drug                      │ value is not a valid list │ type_error.list     │ geo    │ NA               │ NA        │ NA          │ kw_curated_drug             │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│ 18 │ curated_genetic_modification_type │ value is not a valid list │ type_error.list     │ geo    │ NA               │ NA        │ NA          │ kw_curated_genetic_mod_type │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│ 19 │ curated_disease                   │ value is not a valid list │ type_error.list     │ geo    │ NA               │ NA        │ NA          │ kw_curated_disease          │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│ 20 │ curated_drug_smiles_code          │ field required            │ value_error.missing │ geo    │ NA               │ NA        │ NA          │ kw_curated_smiles           │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│ 21 │ curated_gene_modified             │ value is not a valid list │ type_error.list     │ geo    │ NA               │ NA        │ NA          │ kw_curated_modified_gene    │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│ 22 │ curated_cell_type                 │ value is not a valid list │ type_error.list     │ geo    │ NA               │ NA        │ NA          │ kw_curated_cell_type        │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│ 23 │ curated_gene                      │ field required            │ value_error.missing │ geo    │ NA               │ NA        │ NA          │ kw_curated_gene             │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│ 24 │ curated_drug                      │ value is not a valid list │ type_error.list     │ geo    │ NA               │ NA        │ NA          │ kw_curated_drug             │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│ 25 │ curated_genetic_modification_type │ value is not a valid list │ type_error.list     │ geo    │ NA               │ NA        │ NA          │ kw_curated_genetic_mod_type │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│ 26 │ curated_disease                   │ value is not a valid list │ type_error.list     │ geo    │ NA               │ NA        │ NA          │ kw_curated_disease          │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│ 27 │ curated_drug_smiles_code          │ field required            │ value_error.missing │ geo    │ NA               │ NA        │ NA          │ kw_curated_smiles           │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│ 28 │ curated_gene_modified             │ value is not a valid list │ type_error.list     │ geo    │ NA               │ NA        │ NA          │ kw_curated_modified_gene    │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│ 29 │ curated_cell_type                 │ value is not a valid list │ type_error.list     │ geo    │ NA               │ NA        │ NA          │ kw_curated_cell_type        │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│ 30 │ curated_gene                      │ field required            │ value_error.missing │ geo    │ NA               │ NA        │ NA          │ kw_curated_gene             │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│ 31 │ curated_drug                      │ value is not a valid list │ type_error.list     │ geo    │ NA               │ NA        │ NA          │ kw_curated_drug             │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│ 32 │ curated_genetic_modification_type │ value is not a valid list │ type_error.list     │ geo    │ NA               │ NA        │ NA          │ kw_curated_genetic_mod_type │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│ 33 │ curated_disease                   │ value is not a valid list │ type_error.list     │ geo    │ NA               │ NA        │ NA          │ kw_curated_disease          │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│ 34 │ curated_drug_smiles_code          │ field required            │ value_error.missing │ geo    │ NA               │ NA        │ NA          │ kw_curated_smiles           │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│ 35 │ curated_gene_modified             │ value is not a valid list │ type_error.list     │ geo    │ NA               │ NA        │ NA          │ kw_curated_modified_gene    │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│ 36 │ curated_cell_type                 │ value is not a valid list │ type_error.list     │ geo    │ NA               │ NA        │ NA          │ kw_curated_cell_type        │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│ 37 │ curated_gene                      │ field required            │ value_error.missing │ geo    │ NA               │ NA        │ NA          │ kw_curated_gene             │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│ 38 │ curated_drug                      │ value is not a valid list │ type_error.list     │ geo    │ NA               │ NA        │ NA          │ kw_curated_drug             │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│ 39 │ curated_genetic_modification_type │ value is not a valid list │ type_error.list     │ geo    │ NA               │ NA        │ NA          │ kw_curated_genetic_mod_type │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│ 40 │ curated_disease                   │ value is not a valid list │ type_error.list     │ geo    │ NA               │ NA        │ NA          │ kw_curated_disease          │\n",
            "├────┼───────────────────────────────────┼───────────────────────────┼─────────────────────┼────────┼──────────────────┼───────────┼─────────────┼─────────────────────────────┤\n",
            "│ 41 │ curated_drug_smiles_code          │ field required            │ value_error.missing │ geo    │ NA               │ NA        │ NA          │ kw_curated_smiles           │\n",
            "╘════╧═══════════════════════════════════╧═══════════════════════════╧═════════════════════╧════════╧══════════════════╧═══════════╧═════════════╧═════════════════════════════╛\n"
          ]
        }
      ],
      "source": [
        "err_sample, status_sample = sample_metadata_validator.check_metadata_for_errors(repo='geo',\n",
        "                                                                              schema_df=schema_df_sample,\n",
        "                                                                              metadata_list=col_metadata.to_dict('records'),\n",
        "                                                                              validate_on='value',\n",
        "                                                                              env=\"testpolly\",\n",
        "                                                                              auth_token=AUTH_TOKEN,\n",
        "                                                                                print_table=True)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 13,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/plain": [
              "[{'title': 'female_naive_neurons_3',\n",
              "  'geo_accession': 'GSM2668172',\n",
              "  'status': 'Public on Nov 28 2017',\n",
              "  'submission_date': 'Jun 14 2017',\n",
              "  'last_update_date': 'May 15 2019',\n",
              "  'type': 'SRA',\n",
              "  'channel_count': 1,\n",
              "  'source_name_ch1': 'FACSed DRG neurons',\n",
              "  'organism_ch1': 'Mus musculus',\n",
              "  'characteristics_ch1': 'cell type: DRG neurons',\n",
              "  'characteristics_ch1.1': 'strain: AdvGFP, gensat.org: STOCK Tg(Avil-EGFP)QD84Gsat/Mmucd',\n",
              "  'characteristics_ch1.2': 'Sex: female',\n",
              "  'characteristics_ch1.3': 'injury condition: naïve',\n",
              "  'characteristics_ch1.4': 'estrus identification: no',\n",
              "  'characteristics_ch1.5': 'contaminant: satellite glial cells',\n",
              "  'molecule_ch1': 'total RNA',\n",
              "  'extract_protocol_ch1': 'RNA extractions were performed in a single, gender-matched batch using a Qiagen RNeasy micokit, following manufacturer’s protocols with some minor modifications. RNA quality was assessed using an Agilent Bioanalyser PicoChiP (for RIN values of sequencing samples see Supplementary Table 6).',\n",
              "  'extract_protocol_ch1.1': 'RNA was sent for batch-controlled library preparation (SMARTer Ultra Low Input HV kit, 634820, Clontech) and sequencing at the High-Throughput Genomics Group at the Wellcome Trust Centre for Human Genetics at Oxford University. After amplification, samples were multiplexed in replicate flow cells on an Illumina HiSeq4500 platform to yield 75 bp reads of at least 25 M read depth.',\n",
              "  'taxid_ch1': 10090,\n",
              "  'description': 'S215_female',\n",
              "  'description.1': 'Supplementary Table 1 (cufflinks), Supplementary Table 3 (deseq2), Supplementary Table 6 (QC)',\n",
              "  'data_processing': 'alignment using STAR to mm10 (standard parameters, except with --outFilterMultimapNmax 1)',\n",
              "  'data_processing.1': 'merging of bam files across lanes using samtools',\n",
              "  'data_processing.2': 'cufflinks for expression level determination (including fragment bias, length and multi-read corrections)',\n",
              "  'data_processing.3': 'cufflinks expression cut-off deemed mathematically and biologically appropriate: FPKM >= 1',\n",
              "  'data_processing.4': 'Deseq2 for differential expression level analysis (standard parameters)',\n",
              "  'data_processing.5': 'Genome_build: mm10',\n",
              "  'data_processing.6': 'Supplementary_files_format_and_content: Supplementary Table 1: cufflinks-derived FPKM values for FACSed naïve DRG neurons with & without expression cut-off',\n",
              "  'data_processing.7': 'Supplementary_files_format_and_content: Supplementary Table 2: cufflinks-derived FPKM values for MACSed DRG neurons after PSNL with and without expression cut-off',\n",
              "  'data_processing.8': 'Supplementary_files_format_and_content: Supplementary Table 3: DESeq2 comparing male vs. female DRG neurons',\n",
              "  'data_processing.9': 'Supplementary_files_format_and_content: Supplementary Table 6: RNA RIN values and alignment stats',\n",
              "  'platform_id': 'GPL21103',\n",
              "  'contact_name': 'Franziska,,Denk',\n",
              "  'contact_institute': \"King's College London\",\n",
              "  'contact_address': \"Guy's Campus\",\n",
              "  'contact_city': 'London',\n",
              "  'contact_zip.postal_code': 'SE1 1UL',\n",
              "  'contact_country': 'United Kingdom',\n",
              "  'data_row_count': 0,\n",
              "  'instrument_model': 'Illumina HiSeq 4000',\n",
              "  'library_selection': 'cDNA',\n",
              "  'library_source': 'transcriptomic',\n",
              "  'library_strategy': 'RNA-Seq',\n",
              "  'relation': 'BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN07237227',\n",
              "  'relation.1': 'SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX2918410',\n",
              "  'supplementary_file_1': 'NONE',\n",
              "  'Sex.ch1': 'female',\n",
              "  'cell.type.ch1': 'DRG neurons',\n",
              "  'contaminant.ch1': 'satellite glial cells',\n",
              "  'estrus.identification.ch1': 'no',\n",
              "  'injury.condition.ch1': 'na<U+FFFD><U+FFFD>ve',\n",
              "  'strain.ch1': 'AdvGFP, gensat.org: STOCK Tg(Avil-EGFP)QD84Gsat/Mmucd',\n",
              "  'kw_curated_cell_line': 'none',\n",
              "  'kw_curated_cell_type': 'glial cell',\n",
              "  'kw_curated_disease': 'Normal',\n",
              "  'kw_curated_drug': 'none',\n",
              "  'kw_curated_tissue': 'none',\n",
              "  'kw_curated_genetic_mod_type': 'wildtype',\n",
              "  'kw_curated_modified_gene': 'none',\n",
              "  'curated_min_age': nan,\n",
              "  'curated_max_age': nan,\n",
              "  'curated_age_unit': 'none',\n",
              "  'kw_curated_gender': 'Female',\n",
              "  'curated_is_control': 0,\n",
              "  'curated_cohort_id': 0,\n",
              "  'curated_cohort_name': 'AdvGFP, gensat.org : STOCK Tg(Avil-EGFP)QD84Gsat/Mmucd; naïve; satellite glial cells; FACSed DRG neurons'},\n",
              " {'title': 'male_naive_neurons_7',\n",
              "  'geo_accession': 'GSM2668165',\n",
              "  'status': 'Public on Nov 28 2017',\n",
              "  'submission_date': 'Jun 14 2017',\n",
              "  'last_update_date': 'May 15 2019',\n",
              "  'type': 'SRA',\n",
              "  'channel_count': 1,\n",
              "  'source_name_ch1': 'FACSed DRG neurons',\n",
              "  'organism_ch1': 'Mus musculus',\n",
              "  'characteristics_ch1': 'cell type: DRG neurons',\n",
              "  'characteristics_ch1.1': 'strain: AdvGFP, gensat.org: STOCK Tg(Avil-EGFP)QD84Gsat/Mmucd',\n",
              "  'characteristics_ch1.2': 'Sex: male',\n",
              "  'characteristics_ch1.3': 'injury condition: naïve',\n",
              "  'characteristics_ch1.4': 'estrus identification: no',\n",
              "  'characteristics_ch1.5': 'contaminant: satellite glial cells',\n",
              "  'molecule_ch1': 'total RNA',\n",
              "  'extract_protocol_ch1': 'RNA extractions were performed in a single, gender-matched batch using a Qiagen RNeasy micokit, following manufacturer’s protocols with some minor modifications. RNA quality was assessed using an Agilent Bioanalyser PicoChiP (for RIN values of sequencing samples see Supplementary Table 6).',\n",
              "  'extract_protocol_ch1.1': 'RNA was sent for batch-controlled library preparation (SMARTer Ultra Low Input HV kit, 634820, Clontech) and sequencing at the High-Throughput Genomics Group at the Wellcome Trust Centre for Human Genetics at Oxford University. After amplification, samples were multiplexed in replicate flow cells on an Illumina HiSeq4500 platform to yield 75 bp reads of at least 25 M read depth.',\n",
              "  'taxid_ch1': 10090,\n",
              "  'description': 'S219_male',\n",
              "  'description.1': 'Supplementary Table 1 (cufflinks), Supplementary Table 3 (deseq2), Supplementary Table 6 (QC)',\n",
              "  'data_processing': 'alignment using STAR to mm10 (standard parameters, except with --outFilterMultimapNmax 1)',\n",
              "  'data_processing.1': 'merging of bam files across lanes using samtools',\n",
              "  'data_processing.2': 'cufflinks for expression level determination (including fragment bias, length and multi-read corrections)',\n",
              "  'data_processing.3': 'cufflinks expression cut-off deemed mathematically and biologically appropriate: FPKM >= 1',\n",
              "  'data_processing.4': 'Deseq2 for differential expression level analysis (standard parameters)',\n",
              "  'data_processing.5': 'Genome_build: mm10',\n",
              "  'data_processing.6': 'Supplementary_files_format_and_content: Supplementary Table 1: cufflinks-derived FPKM values for FACSed naïve DRG neurons with & without expression cut-off',\n",
              "  'data_processing.7': 'Supplementary_files_format_and_content: Supplementary Table 2: cufflinks-derived FPKM values for MACSed DRG neurons after PSNL with and without expression cut-off',\n",
              "  'data_processing.8': 'Supplementary_files_format_and_content: Supplementary Table 3: DESeq2 comparing male vs. female DRG neurons',\n",
              "  'data_processing.9': 'Supplementary_files_format_and_content: Supplementary Table 6: RNA RIN values and alignment stats',\n",
              "  'platform_id': 'GPL21103',\n",
              "  'contact_name': 'Franziska,,Denk',\n",
              "  'contact_institute': \"King's College London\",\n",
              "  'contact_address': \"Guy's Campus\",\n",
              "  'contact_city': 'London',\n",
              "  'contact_zip.postal_code': 'SE1 1UL',\n",
              "  'contact_country': 'United Kingdom',\n",
              "  'data_row_count': 0,\n",
              "  'instrument_model': 'Illumina HiSeq 4000',\n",
              "  'library_selection': 'cDNA',\n",
              "  'library_source': 'transcriptomic',\n",
              "  'library_strategy': 'RNA-Seq',\n",
              "  'relation': 'BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN07237220',\n",
              "  'relation.1': 'SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX2918403',\n",
              "  'supplementary_file_1': 'NONE',\n",
              "  'Sex.ch1': 'male',\n",
              "  'cell.type.ch1': 'DRG neurons',\n",
              "  'contaminant.ch1': 'satellite glial cells',\n",
              "  'estrus.identification.ch1': 'no',\n",
              "  'injury.condition.ch1': 'na<U+FFFD><U+FFFD>ve',\n",
              "  'strain.ch1': 'AdvGFP, gensat.org: STOCK Tg(Avil-EGFP)QD84Gsat/Mmucd',\n",
              "  'kw_curated_cell_line': 'none',\n",
              "  'kw_curated_cell_type': 'glial cell',\n",
              "  'kw_curated_disease': 'Normal',\n",
              "  'kw_curated_drug': 'none',\n",
              "  'kw_curated_tissue': 'none',\n",
              "  'kw_curated_genetic_mod_type': 'wildtype',\n",
              "  'kw_curated_modified_gene': 'none',\n",
              "  'curated_min_age': nan,\n",
              "  'curated_max_age': nan,\n",
              "  'curated_age_unit': 'none',\n",
              "  'kw_curated_gender': 'Male',\n",
              "  'curated_is_control': 0,\n",
              "  'curated_cohort_id': 0,\n",
              "  'curated_cohort_name': 'AdvGFP, gensat.org : STOCK Tg(Avil-EGFP)QD84Gsat/Mmucd; naïve; satellite glial cells; FACSed DRG neurons'},\n",
              " {'title': 'female_naive_neurons_2',\n",
              "  'geo_accession': 'GSM2668171',\n",
              "  'status': 'Public on Nov 28 2017',\n",
              "  'submission_date': 'Jun 14 2017',\n",
              "  'last_update_date': 'May 15 2019',\n",
              "  'type': 'SRA',\n",
              "  'channel_count': 1,\n",
              "  'source_name_ch1': 'FACSed DRG neurons',\n",
              "  'organism_ch1': 'Mus musculus',\n",
              "  'characteristics_ch1': 'cell type: DRG neurons',\n",
              "  'characteristics_ch1.1': 'strain: AdvGFP, gensat.org: STOCK Tg(Avil-EGFP)QD84Gsat/Mmucd',\n",
              "  'characteristics_ch1.2': 'Sex: female',\n",
              "  'characteristics_ch1.3': 'injury condition: naïve',\n",
              "  'characteristics_ch1.4': 'estrus identification: no',\n",
              "  'characteristics_ch1.5': 'contaminant: satellite glial cells',\n",
              "  'molecule_ch1': 'total RNA',\n",
              "  'extract_protocol_ch1': 'RNA extractions were performed in a single, gender-matched batch using a Qiagen RNeasy micokit, following manufacturer’s protocols with some minor modifications. RNA quality was assessed using an Agilent Bioanalyser PicoChiP (for RIN values of sequencing samples see Supplementary Table 6).',\n",
              "  'extract_protocol_ch1.1': 'RNA was sent for batch-controlled library preparation (SMARTer Ultra Low Input HV kit, 634820, Clontech) and sequencing at the High-Throughput Genomics Group at the Wellcome Trust Centre for Human Genetics at Oxford University. After amplification, samples were multiplexed in replicate flow cells on an Illumina HiSeq4500 platform to yield 75 bp reads of at least 25 M read depth.',\n",
              "  'taxid_ch1': 10090,\n",
              "  'description': 'S214_female',\n",
              "  'description.1': 'Supplementary Table 1 (cufflinks), Supplementary Table 3 (deseq2), Supplementary Table 6 (QC)',\n",
              "  'data_processing': 'alignment using STAR to mm10 (standard parameters, except with --outFilterMultimapNmax 1)',\n",
              "  'data_processing.1': 'merging of bam files across lanes using samtools',\n",
              "  'data_processing.2': 'cufflinks for expression level determination (including fragment bias, length and multi-read corrections)',\n",
              "  'data_processing.3': 'cufflinks expression cut-off deemed mathematically and biologically appropriate: FPKM >= 1',\n",
              "  'data_processing.4': 'Deseq2 for differential expression level analysis (standard parameters)',\n",
              "  'data_processing.5': 'Genome_build: mm10',\n",
              "  'data_processing.6': 'Supplementary_files_format_and_content: Supplementary Table 1: cufflinks-derived FPKM values for FACSed naïve DRG neurons with & without expression cut-off',\n",
              "  'data_processing.7': 'Supplementary_files_format_and_content: Supplementary Table 2: cufflinks-derived FPKM values for MACSed DRG neurons after PSNL with and without expression cut-off',\n",
              "  'data_processing.8': 'Supplementary_files_format_and_content: Supplementary Table 3: DESeq2 comparing male vs. female DRG neurons',\n",
              "  'data_processing.9': 'Supplementary_files_format_and_content: Supplementary Table 6: RNA RIN values and alignment stats',\n",
              "  'platform_id': 'GPL21103',\n",
              "  'contact_name': 'Franziska,,Denk',\n",
              "  'contact_institute': \"King's College London\",\n",
              "  'contact_address': \"Guy's Campus\",\n",
              "  'contact_city': 'London',\n",
              "  'contact_zip.postal_code': 'SE1 1UL',\n",
              "  'contact_country': 'United Kingdom',\n",
              "  'data_row_count': 0,\n",
              "  'instrument_model': 'Illumina HiSeq 4000',\n",
              "  'library_selection': 'cDNA',\n",
              "  'library_source': 'transcriptomic',\n",
              "  'library_strategy': 'RNA-Seq',\n",
              "  'relation': 'BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN07237214',\n",
              "  'relation.1': 'SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX2918409',\n",
              "  'supplementary_file_1': 'NONE',\n",
              "  'Sex.ch1': 'female',\n",
              "  'cell.type.ch1': 'DRG neurons',\n",
              "  'contaminant.ch1': 'satellite glial cells',\n",
              "  'estrus.identification.ch1': 'no',\n",
              "  'injury.condition.ch1': 'na<U+FFFD><U+FFFD>ve',\n",
              "  'strain.ch1': 'AdvGFP, gensat.org: STOCK Tg(Avil-EGFP)QD84Gsat/Mmucd',\n",
              "  'kw_curated_cell_line': 'none',\n",
              "  'kw_curated_cell_type': 'glial cell',\n",
              "  'kw_curated_disease': 'Normal',\n",
              "  'kw_curated_drug': 'none',\n",
              "  'kw_curated_tissue': 'none',\n",
              "  'kw_curated_genetic_mod_type': 'wildtype',\n",
              "  'kw_curated_modified_gene': 'none',\n",
              "  'curated_min_age': nan,\n",
              "  'curated_max_age': nan,\n",
              "  'curated_age_unit': 'none',\n",
              "  'kw_curated_gender': 'Female',\n",
              "  'curated_is_control': 0,\n",
              "  'curated_cohort_id': 0,\n",
              "  'curated_cohort_name': 'AdvGFP, gensat.org : STOCK Tg(Avil-EGFP)QD84Gsat/Mmucd; naïve; satellite glial cells; FACSed DRG neurons'},\n",
              " {'title': 'male_naive_neurons_6',\n",
              "  'geo_accession': 'GSM2668164',\n",
              "  'status': 'Public on Nov 28 2017',\n",
              "  'submission_date': 'Jun 14 2017',\n",
              "  'last_update_date': 'May 15 2019',\n",
              "  'type': 'SRA',\n",
              "  'channel_count': 1,\n",
              "  'source_name_ch1': 'FACSed DRG neurons',\n",
              "  'organism_ch1': 'Mus musculus',\n",
              "  'characteristics_ch1': 'cell type: DRG neurons',\n",
              "  'characteristics_ch1.1': 'strain: AdvGFP, gensat.org: STOCK Tg(Avil-EGFP)QD84Gsat/Mmucd',\n",
              "  'characteristics_ch1.2': 'Sex: male',\n",
              "  'characteristics_ch1.3': 'injury condition: naïve',\n",
              "  'characteristics_ch1.4': 'estrus identification: no',\n",
              "  'characteristics_ch1.5': 'contaminant: satellite glial cells',\n",
              "  'molecule_ch1': 'total RNA',\n",
              "  'extract_protocol_ch1': 'RNA extractions were performed in a single, gender-matched batch using a Qiagen RNeasy micokit, following manufacturer’s protocols with some minor modifications. RNA quality was assessed using an Agilent Bioanalyser PicoChiP (for RIN values of sequencing samples see Supplementary Table 6).',\n",
              "  'extract_protocol_ch1.1': 'RNA was sent for batch-controlled library preparation (SMARTer Ultra Low Input HV kit, 634820, Clontech) and sequencing at the High-Throughput Genomics Group at the Wellcome Trust Centre for Human Genetics at Oxford University. After amplification, samples were multiplexed in replicate flow cells on an Illumina HiSeq4500 platform to yield 75 bp reads of at least 25 M read depth.',\n",
              "  'taxid_ch1': 10090,\n",
              "  'description': 'S218_male',\n",
              "  'description.1': 'Supplementary Table 1 (cufflinks), Supplementary Table 3 (deseq2), Supplementary Table 6 (QC)',\n",
              "  'data_processing': 'alignment using STAR to mm10 (standard parameters, except with --outFilterMultimapNmax 1)',\n",
              "  'data_processing.1': 'merging of bam files across lanes using samtools',\n",
              "  'data_processing.2': 'cufflinks for expression level determination (including fragment bias, length and multi-read corrections)',\n",
              "  'data_processing.3': 'cufflinks expression cut-off deemed mathematically and biologically appropriate: FPKM >= 1',\n",
              "  'data_processing.4': 'Deseq2 for differential expression level analysis (standard parameters)',\n",
              "  'data_processing.5': 'Genome_build: mm10',\n",
              "  'data_processing.6': 'Supplementary_files_format_and_content: Supplementary Table 1: cufflinks-derived FPKM values for FACSed naïve DRG neurons with & without expression cut-off',\n",
              "  'data_processing.7': 'Supplementary_files_format_and_content: Supplementary Table 2: cufflinks-derived FPKM values for MACSed DRG neurons after PSNL with and without expression cut-off',\n",
              "  'data_processing.8': 'Supplementary_files_format_and_content: Supplementary Table 3: DESeq2 comparing male vs. female DRG neurons',\n",
              "  'data_processing.9': 'Supplementary_files_format_and_content: Supplementary Table 6: RNA RIN values and alignment stats',\n",
              "  'platform_id': 'GPL21103',\n",
              "  'contact_name': 'Franziska,,Denk',\n",
              "  'contact_institute': \"King's College London\",\n",
              "  'contact_address': \"Guy's Campus\",\n",
              "  'contact_city': 'London',\n",
              "  'contact_zip.postal_code': 'SE1 1UL',\n",
              "  'contact_country': 'United Kingdom',\n",
              "  'data_row_count': 0,\n",
              "  'instrument_model': 'Illumina HiSeq 4000',\n",
              "  'library_selection': 'cDNA',\n",
              "  'library_source': 'transcriptomic',\n",
              "  'library_strategy': 'RNA-Seq',\n",
              "  'relation': 'BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN07237221',\n",
              "  'relation.1': 'SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX2918402',\n",
              "  'supplementary_file_1': 'NONE',\n",
              "  'Sex.ch1': 'male',\n",
              "  'cell.type.ch1': 'DRG neurons',\n",
              "  'contaminant.ch1': 'satellite glial cells',\n",
              "  'estrus.identification.ch1': 'no',\n",
              "  'injury.condition.ch1': 'na<U+FFFD><U+FFFD>ve',\n",
              "  'strain.ch1': 'AdvGFP, gensat.org: STOCK Tg(Avil-EGFP)QD84Gsat/Mmucd',\n",
              "  'kw_curated_cell_line': 'none',\n",
              "  'kw_curated_cell_type': 'glial cell',\n",
              "  'kw_curated_disease': 'Normal',\n",
              "  'kw_curated_drug': 'none',\n",
              "  'kw_curated_tissue': 'none',\n",
              "  'kw_curated_genetic_mod_type': 'wildtype',\n",
              "  'kw_curated_modified_gene': 'none',\n",
              "  'curated_min_age': nan,\n",
              "  'curated_max_age': nan,\n",
              "  'curated_age_unit': 'none',\n",
              "  'kw_curated_gender': 'Male',\n",
              "  'curated_is_control': 0,\n",
              "  'curated_cohort_id': 0,\n",
              "  'curated_cohort_name': 'AdvGFP, gensat.org : STOCK Tg(Avil-EGFP)QD84Gsat/Mmucd; naïve; satellite glial cells; FACSed DRG neurons'},\n",
              " {'title': 'female_naive_neurons_11',\n",
              "  'geo_accession': 'GSM2668180',\n",
              "  'status': 'Public on Nov 28 2017',\n",
              "  'submission_date': 'Jun 14 2017',\n",
              "  'last_update_date': 'May 15 2019',\n",
              "  'type': 'SRA',\n",
              "  'channel_count': 1,\n",
              "  'source_name_ch1': 'FACSed DRG neurons',\n",
              "  'organism_ch1': 'Mus musculus',\n",
              "  'characteristics_ch1': 'cell type: DRG neurons',\n",
              "  'characteristics_ch1.1': 'strain: AdvGFP, gensat.org: STOCK Tg(Avil-EGFP)QD84Gsat/Mmucd',\n",
              "  'characteristics_ch1.2': 'Sex: female',\n",
              "  'characteristics_ch1.3': 'injury condition: naïve',\n",
              "  'characteristics_ch1.4': 'estrus identification: no',\n",
              "  'characteristics_ch1.5': 'contaminant: satellite glial cells',\n",
              "  'molecule_ch1': 'total RNA',\n",
              "  'extract_protocol_ch1': 'RNA extractions were performed in a single, gender-matched batch using a Qiagen RNeasy micokit, following manufacturer’s protocols with some minor modifications. RNA quality was assessed using an Agilent Bioanalyser PicoChiP (for RIN values of sequencing samples see Supplementary Table 6).',\n",
              "  'extract_protocol_ch1.1': 'RNA was sent for batch-controlled library preparation (SMARTer Ultra Low Input HV kit, 634820, Clontech) and sequencing at the High-Throughput Genomics Group at the Wellcome Trust Centre for Human Genetics at Oxford University. After amplification, samples were multiplexed in replicate flow cells on an Illumina HiSeq4500 platform to yield 75 bp reads of at least 25 M read depth.',\n",
              "  'taxid_ch1': 10090,\n",
              "  'description': 'S230_female',\n",
              "  'description.1': 'Supplementary Table 1 (cufflinks), Supplementary Table 3 (deseq2), Supplementary Table 6 (QC)',\n",
              "  'data_processing': 'alignment using STAR to mm10 (standard parameters, except with --outFilterMultimapNmax 1)',\n",
              "  'data_processing.1': 'merging of bam files across lanes using samtools',\n",
              "  'data_processing.2': 'cufflinks for expression level determination (including fragment bias, length and multi-read corrections)',\n",
              "  'data_processing.3': 'cufflinks expression cut-off deemed mathematically and biologically appropriate: FPKM >= 1',\n",
              "  'data_processing.4': 'Deseq2 for differential expression level analysis (standard parameters)',\n",
              "  'data_processing.5': 'Genome_build: mm10',\n",
              "  'data_processing.6': 'Supplementary_files_format_and_content: Supplementary Table 1: cufflinks-derived FPKM values for FACSed naïve DRG neurons with & without expression cut-off',\n",
              "  'data_processing.7': 'Supplementary_files_format_and_content: Supplementary Table 2: cufflinks-derived FPKM values for MACSed DRG neurons after PSNL with and without expression cut-off',\n",
              "  'data_processing.8': 'Supplementary_files_format_and_content: Supplementary Table 3: DESeq2 comparing male vs. female DRG neurons',\n",
              "  'data_processing.9': 'Supplementary_files_format_and_content: Supplementary Table 6: RNA RIN values and alignment stats',\n",
              "  'platform_id': 'GPL21103',\n",
              "  'contact_name': 'Franziska,,Denk',\n",
              "  'contact_institute': \"King's College London\",\n",
              "  'contact_address': \"Guy's Campus\",\n",
              "  'contact_city': 'London',\n",
              "  'contact_zip.postal_code': 'SE1 1UL',\n",
              "  'contact_country': 'United Kingdom',\n",
              "  'data_row_count': 0,\n",
              "  'instrument_model': 'Illumina HiSeq 4000',\n",
              "  'library_selection': 'cDNA',\n",
              "  'library_source': 'transcriptomic',\n",
              "  'library_strategy': 'RNA-Seq',\n",
              "  'relation': 'BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN07237236',\n",
              "  'relation.1': 'SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX2918418',\n",
              "  'supplementary_file_1': 'NONE',\n",
              "  'Sex.ch1': 'female',\n",
              "  'cell.type.ch1': 'DRG neurons',\n",
              "  'contaminant.ch1': 'satellite glial cells',\n",
              "  'estrus.identification.ch1': 'no',\n",
              "  'injury.condition.ch1': 'na<U+FFFD><U+FFFD>ve',\n",
              "  'strain.ch1': 'AdvGFP, gensat.org: STOCK Tg(Avil-EGFP)QD84Gsat/Mmucd',\n",
              "  'kw_curated_cell_line': 'none',\n",
              "  'kw_curated_cell_type': 'glial cell',\n",
              "  'kw_curated_disease': 'Normal',\n",
              "  'kw_curated_drug': 'none',\n",
              "  'kw_curated_tissue': 'none',\n",
              "  'kw_curated_genetic_mod_type': 'wildtype',\n",
              "  'kw_curated_modified_gene': 'none',\n",
              "  'curated_min_age': nan,\n",
              "  'curated_max_age': nan,\n",
              "  'curated_age_unit': 'none',\n",
              "  'kw_curated_gender': 'Female',\n",
              "  'curated_is_control': 0,\n",
              "  'curated_cohort_id': 0,\n",
              "  'curated_cohort_name': 'AdvGFP, gensat.org : STOCK Tg(Avil-EGFP)QD84Gsat/Mmucd; naïve; satellite glial cells; FACSed DRG neurons'},\n",
              " {'title': 'male_naive_neurons_4',\n",
              "  'geo_accession': 'GSM2668162',\n",
              "  'status': 'Public on Nov 28 2017',\n",
              "  'submission_date': 'Jun 14 2017',\n",
              "  'last_update_date': 'May 15 2019',\n",
              "  'type': 'SRA',\n",
              "  'channel_count': 1,\n",
              "  'source_name_ch1': 'FACSed DRG neurons',\n",
              "  'organism_ch1': 'Mus musculus',\n",
              "  'characteristics_ch1': 'cell type: DRG neurons',\n",
              "  'characteristics_ch1.1': 'strain: AdvGFP, gensat.org: STOCK Tg(Avil-EGFP)QD84Gsat/Mmucd',\n",
              "  'characteristics_ch1.2': 'Sex: male',\n",
              "  'characteristics_ch1.3': 'injury condition: naïve',\n",
              "  'characteristics_ch1.4': 'estrus identification: no',\n",
              "  'characteristics_ch1.5': 'contaminant: satellite glial cells',\n",
              "  'molecule_ch1': 'total RNA',\n",
              "  'extract_protocol_ch1': 'RNA extractions were performed in a single, gender-matched batch using a Qiagen RNeasy micokit, following manufacturer’s protocols with some minor modifications. RNA quality was assessed using an Agilent Bioanalyser PicoChiP (for RIN values of sequencing samples see Supplementary Table 6).',\n",
              "  'extract_protocol_ch1.1': 'RNA was sent for batch-controlled library preparation (SMARTer Ultra Low Input HV kit, 634820, Clontech) and sequencing at the High-Throughput Genomics Group at the Wellcome Trust Centre for Human Genetics at Oxford University. After amplification, samples were multiplexed in replicate flow cells on an Illumina HiSeq4500 platform to yield 75 bp reads of at least 25 M read depth.',\n",
              "  'taxid_ch1': 10090,\n",
              "  'description': 'S212_male',\n",
              "  'description.1': 'Supplementary Table 1 (cufflinks), Supplementary Table 3 (deseq2), Supplementary Table 6 (QC)',\n",
              "  'data_processing': 'alignment using STAR to mm10 (standard parameters, except with --outFilterMultimapNmax 1)',\n",
              "  'data_processing.1': 'merging of bam files across lanes using samtools',\n",
              "  'data_processing.2': 'cufflinks for expression level determination (including fragment bias, length and multi-read corrections)',\n",
              "  'data_processing.3': 'cufflinks expression cut-off deemed mathematically and biologically appropriate: FPKM >= 1',\n",
              "  'data_processing.4': 'Deseq2 for differential expression level analysis (standard parameters)',\n",
              "  'data_processing.5': 'Genome_build: mm10',\n",
              "  'data_processing.6': 'Supplementary_files_format_and_content: Supplementary Table 1: cufflinks-derived FPKM values for FACSed naïve DRG neurons with & without expression cut-off',\n",
              "  'data_processing.7': 'Supplementary_files_format_and_content: Supplementary Table 2: cufflinks-derived FPKM values for MACSed DRG neurons after PSNL with and without expression cut-off',\n",
              "  'data_processing.8': 'Supplementary_files_format_and_content: Supplementary Table 3: DESeq2 comparing male vs. female DRG neurons',\n",
              "  'data_processing.9': 'Supplementary_files_format_and_content: Supplementary Table 6: RNA RIN values and alignment stats',\n",
              "  'platform_id': 'GPL21103',\n",
              "  'contact_name': 'Franziska,,Denk',\n",
              "  'contact_institute': \"King's College London\",\n",
              "  'contact_address': \"Guy's Campus\",\n",
              "  'contact_city': 'London',\n",
              "  'contact_zip.postal_code': 'SE1 1UL',\n",
              "  'contact_country': 'United Kingdom',\n",
              "  'data_row_count': 0,\n",
              "  'instrument_model': 'Illumina HiSeq 4000',\n",
              "  'library_selection': 'cDNA',\n",
              "  'library_source': 'transcriptomic',\n",
              "  'library_strategy': 'RNA-Seq',\n",
              "  'relation': 'BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN07237223',\n",
              "  'relation.1': 'SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX2918400',\n",
              "  'supplementary_file_1': 'NONE',\n",
              "  'Sex.ch1': 'male',\n",
              "  'cell.type.ch1': 'DRG neurons',\n",
              "  'contaminant.ch1': 'satellite glial cells',\n",
              "  'estrus.identification.ch1': 'no',\n",
              "  'injury.condition.ch1': 'na<U+FFFD><U+FFFD>ve',\n",
              "  'strain.ch1': 'AdvGFP, gensat.org: STOCK Tg(Avil-EGFP)QD84Gsat/Mmucd',\n",
              "  'kw_curated_cell_line': 'none',\n",
              "  'kw_curated_cell_type': 'glial cell',\n",
              "  'kw_curated_disease': 'Normal',\n",
              "  'kw_curated_drug': 'none',\n",
              "  'kw_curated_tissue': 'none',\n",
              "  'kw_curated_genetic_mod_type': 'wildtype',\n",
              "  'kw_curated_modified_gene': 'none',\n",
              "  'curated_min_age': nan,\n",
              "  'curated_max_age': nan,\n",
              "  'curated_age_unit': 'none',\n",
              "  'kw_curated_gender': 'Male',\n",
              "  'curated_is_control': 0,\n",
              "  'curated_cohort_id': 0,\n",
              "  'curated_cohort_name': 'AdvGFP, gensat.org : STOCK Tg(Avil-EGFP)QD84Gsat/Mmucd; naïve; satellite glial cells; FACSed DRG neurons'}]"
            ]
          },
          "execution_count": 13,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "col_metadata.to_dict('records')"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": []
    }
  ],
  "metadata": {
    "kernelspec": {
      "display_name": "Python 3 (ipykernel)",
      "language": "python",
      "name": "python3"
    },
    "language_info": {
      "codemirror_mode": {
        "name": "ipython",
        "version": 3
      },
      "file_extension": ".py",
      "mimetype": "text/x-python",
      "name": "python",
      "nbconvert_exporter": "python",
      "pygments_lexer": "ipython3",
      "version": "3.10.2"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 2
}