{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Welcome to Polly Python3 Notebook."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### to install polly python on jupyter notebook"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[33mWARNING: You are using pip version 19.2.3, however version 21.3.1 is available.\r\n",
      "You should consider upgrading via the 'pip install --upgrade pip' command.\u001b[0m\r\n"
     ]
    }
   ],
   "source": [
    "!sudo pip3 install polly-python --quiet "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### importing modules"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from polly.auth import Polly\n",
    "from polly.omixatlas import OmixAtlas"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### authentication"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "AUTH_TOKEN=os.environ['POLLY_REFRESH_TOKEN'] \n",
    "polly = Polly()                              \n",
    "polly.auth(AUTH_TOKEN)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### making object to use class functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "omixatlas = OmixAtlas() "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### get_all_omixatlas\n",
    "### input \n",
    "None\n",
    "### output\n",
    "it will return a list of obejcts which contains the information about\n",
    "<a id=\"get_all_omixatlas\"></a>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'data': [{'repo_name': 'atavistik',\n",
       "   'repo_id': '1650348922627',\n",
       "   'indexes': {'gct_metadata': 'atavistik_gct_metadata',\n",
       "    'h5ad_metadata': 'atavistik_h5ad_metadata',\n",
       "    'csv': 'atavistik_csv',\n",
       "    'files': 'atavistik_files',\n",
       "    'json': 'atavistik_json',\n",
       "    'ipynb': 'atavistik_ipynb',\n",
       "    'gct_data': 'atavistik_gct_data',\n",
       "    'h5ad_data': 'atavistik_h5ad_data'},\n",
       "   'diseases': ['none'],\n",
       "   'organisms': ['none'],\n",
       "   'sources': ['none'],\n",
       "   'datatypes': [],\n",
       "   'dataset_count': 1,\n",
       "   'disease_count': 1,\n",
       "   'tissue_count': 1,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 0,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_name': 'gdx',\n",
       "   'repo_id': '1646370413059',\n",
       "   'indexes': {'gct_metadata': 'gdx_gct_metadata',\n",
       "    'h5ad_metadata': 'gdx_h5ad_metadata',\n",
       "    'csv': 'gdx_csv',\n",
       "    'files': 'gdx_files',\n",
       "    'json': 'gdx_json',\n",
       "    'ipynb': 'gdx_ipynb',\n",
       "    'gct_data': 'gdx_gct_data',\n",
       "    'h5ad_data': 'gdx_h5ad_data'},\n",
       "   'diseases': [],\n",
       "   'organisms': [],\n",
       "   'sources': [],\n",
       "   'datatypes': [],\n",
       "   'dataset_count': 0,\n",
       "   'disease_count': 0,\n",
       "   'tissue_count': 0,\n",
       "   'organism_count': 0,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 0,\n",
       "   'data_source_count': 0,\n",
       "   'sample_count': 0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_name': 'auron_data_lake',\n",
       "   'repo_id': '1613041985263',\n",
       "   'indexes': {'gct_metadata': 'auron_data_lake_gct_metadata',\n",
       "    'h5ad_metadata': 'auron_data_lake_h5ad_metadata',\n",
       "    'csv': 'auron_data_lake_csv',\n",
       "    'files': 'auron_data_lake_files',\n",
       "    'json': 'auron_data_lake_json',\n",
       "    'ipynb': 'auron_data_lake_ipynb',\n",
       "    'gct_data': 'auron_data_lake_gct_data',\n",
       "    'h5ad_data': 'auron_data_lake_h5ad_data'},\n",
       "   'diseases': ['small cell lung carcinoma',\n",
       "    'carcinoma, small cell',\n",
       "    'carcinoma, neuroendocrine',\n",
       "    'normal',\n",
       "    '[normal]',\n",
       "    'neuroendocrine tumors',\n",
       "    'carcinoma, hepatocellular',\n",
       "    'carcinoma, non-small-cell lung',\n",
       "    'leukemia, myeloid, acute',\n",
       "    'neoplasm metastasis',\n",
       "    'neoplasms, squamous cell',\n",
       "    'prostatic neoplasms',\n",
       "    'urinary bladder neoplasms',\n",
       "    '[aml]',\n",
       "    '[leukemia, acute myeloid, neuroblastoma]',\n",
       "    '[leukemia, acute myeloid, normal]',\n",
       "    '[leukemia, acute myeloid]',\n",
       "    '[neuroblastoma, skin cancer]',\n",
       "    'adenocarcinoma of lung',\n",
       "    'adenoma, chromophobe'],\n",
       "   'organisms': ['homo sapiens', 'mus musculus', '[homo sapiens]'],\n",
       "   'sources': ['geo', 'auron', 'pubmed', 'tcga'],\n",
       "   'datatypes': ['raw counts transcriptomics', 'transcriptomics'],\n",
       "   'dataset_count': 33,\n",
       "   'disease_count': 53,\n",
       "   'tissue_count': 68,\n",
       "   'organism_count': 3,\n",
       "   'cell_line_count': 22,\n",
       "   'cell_type_count': 32,\n",
       "   'drug_count': 503,\n",
       "   'data_type_count': 2,\n",
       "   'data_source_count': 4,\n",
       "   'sample_count': 12457,\n",
       "   'normal_sample_count': 1416},\n",
       "  {'repo_name': 'hpa',\n",
       "   'repo_id': '1619514006627',\n",
       "   'indexes': {'gct_metadata': 'hpa_gct_metadata',\n",
       "    'h5ad_metadata': 'hpa_h5ad_metadata',\n",
       "    'csv': 'hpa_csv',\n",
       "    'files': 'hpa_files',\n",
       "    'json': 'hpa_json',\n",
       "    'ipynb': 'hpa_ipynb',\n",
       "    'gct_data': 'hpa_gct_data',\n",
       "    'h5ad_data': 'hpa_h5ad_data'},\n",
       "   'diseases': ['normal'],\n",
       "   'organisms': ['homo sapiens', 'sus (pig)', 'mus musculus'],\n",
       "   'sources': ['hpa'],\n",
       "   'datatypes': ['transcriptomics', 'gene expression reliability'],\n",
       "   'dataset_count': 1755,\n",
       "   'disease_count': 1,\n",
       "   'tissue_count': 78,\n",
       "   'organism_count': 3,\n",
       "   'cell_line_count': 70,\n",
       "   'cell_type_count': 150,\n",
       "   'drug_count': 1,\n",
       "   'data_type_count': 2,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 678,\n",
       "   'normal_sample_count': 678},\n",
       "  {'repo_name': 'teddy',\n",
       "   'repo_id': '1615464367934',\n",
       "   'indexes': {'gct_metadata': 'teddy_gct_metadata',\n",
       "    'h5ad_metadata': 'teddy_h5ad_metadata',\n",
       "    'csv': 'teddy_csv',\n",
       "    'files': 'teddy_files',\n",
       "    'json': 'teddy_json',\n",
       "    'ipynb': 'teddy_ipynb',\n",
       "    'gct_data': 'teddy_gct_data',\n",
       "    'h5ad_data': 'teddy_h5ad_data'},\n",
       "   'diseases': ['none', 'diabetes mellitus, type 1'],\n",
       "   'organisms': ['homo sapiens'],\n",
       "   'sources': ['metabolomics workbench'],\n",
       "   'datatypes': ['lipidomics', 'metabolomics'],\n",
       "   'dataset_count': 34518,\n",
       "   'disease_count': 2,\n",
       "   'tissue_count': 1,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 1,\n",
       "   'cell_type_count': 1,\n",
       "   'drug_count': 1,\n",
       "   'data_type_count': 2,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 11565,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_name': 'valo_onco',\n",
       "   'repo_id': '1649184014614',\n",
       "   'indexes': {'gct_metadata': 'valo_onco_gct_metadata',\n",
       "    'h5ad_metadata': 'valo_onco_h5ad_metadata',\n",
       "    'csv': 'valo_onco_csv',\n",
       "    'files': 'valo_onco_files',\n",
       "    'json': 'valo_onco_json',\n",
       "    'ipynb': 'valo_onco_ipynb',\n",
       "    'gct_data': 'valo_onco_gct_data',\n",
       "    'h5ad_data': 'valo_onco_h5ad_data'},\n",
       "   'diseases': ['normal',\n",
       "    'breast neoplasms',\n",
       "    'neoplasms',\n",
       "    'brain neoplasms',\n",
       "    'carcinoma, non-small-cell lung',\n",
       "    'colonic neoplasm',\n",
       "    'kidney neoplasms',\n",
       "    'leukemia',\n",
       "    'melanoma',\n",
       "    'ovarian neoplasms',\n",
       "    'prostatic neoplasms',\n",
       "    'stomach neoplasm',\n",
       "    'central nervous system neoplasms',\n",
       "    'hematologic neoplasms',\n",
       "    'leukemia, myeloid, acute',\n",
       "    'leukemia, t-cell',\n",
       "    'multiple myeloma',\n",
       "    'neoplasms, squamous cell',\n",
       "    'precursor t-cell lymphoblastic leukemia-lymphoma',\n",
       "    'zika virus infection'],\n",
       "   'organisms': ['homo sapiens', 'mus musculus'],\n",
       "   'sources': ['geo'],\n",
       "   'datatypes': ['transcriptomics', 'raw counts transcriptomics'],\n",
       "   'dataset_count': 19,\n",
       "   'disease_count': 20,\n",
       "   'tissue_count': 14,\n",
       "   'organism_count': 2,\n",
       "   'cell_line_count': 28,\n",
       "   'cell_type_count': 17,\n",
       "   'drug_count': 3,\n",
       "   'data_type_count': 2,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 202,\n",
       "   'normal_sample_count': 32},\n",
       "  {'repo_name': 'gdc',\n",
       "   'repo_id': '1623221686703',\n",
       "   'indexes': {'gct_metadata': 'gdc_gct_metadata',\n",
       "    'h5ad_metadata': 'gdc_h5ad_metadata',\n",
       "    'csv': 'gdc_csv',\n",
       "    'files': 'gdc_files',\n",
       "    'json': 'gdc_json',\n",
       "    'ipynb': 'gdc_ipynb',\n",
       "    'gct_data': 'gdc_gct_data',\n",
       "    'h5ad_data': 'gdc_h5ad_data'},\n",
       "   'diseases': ['leukemia, myeloid, acute',\n",
       "    'precursor t-cell lymphoblastic leukemia-lymphoma',\n",
       "    'multiple myeloma',\n",
       "    'precursor b-cell lymphoblastic leukemia-lymphoma',\n",
       "    'lymphoma, large b-cell, diffuse',\n",
       "    'wilms tumor',\n",
       "    'burkitt lymphoma, nos (includes all variants)',\n",
       "    'squamous cell carcinoma, nonkeratinizing, nos',\n",
       "    'rhabdoid tumor',\n",
       "    'adenocarcinoma, nos',\n",
       "    'squamous cell carcinoma, keratinizing, nos',\n",
       "    'none',\n",
       "    'neuroblastoma',\n",
       "    'infiltrating duct carcinoma, nos',\n",
       "    'adenocarcinoma, metastatic, nos',\n",
       "    'myelodysplastic syndromes',\n",
       "    'osteosarcoma, nos',\n",
       "    'not reported',\n",
       "    'unknown',\n",
       "    'leukemia, biphenotypic, acute'],\n",
       "   'organisms': ['homo sapiens'],\n",
       "   'sources': ['gdc'],\n",
       "   'datatypes': ['mirna expression',\n",
       "    'transcriptomics',\n",
       "    'copy number variation'],\n",
       "   'dataset_count': 9908,\n",
       "   'disease_count': 67,\n",
       "   'tissue_count': 41,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 1,\n",
       "   'cell_type_count': 1,\n",
       "   'drug_count': 1,\n",
       "   'data_type_count': 3,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 5353,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_name': 'geo',\n",
       "   'repo_id': '9',\n",
       "   'indexes': {'gct_metadata': 'geo_gct_metadata',\n",
       "    'h5ad_metadata': 'geo_h5ad_metadata',\n",
       "    'csv': 'geo_csv',\n",
       "    'files': 'geo_files',\n",
       "    'ipynb': 'geo_ipynb',\n",
       "    'gct_data': 'geo_gct_data',\n",
       "    'h5ad_data': 'geo_h5ad_data'},\n",
       "   'diseases': ['normal',\n",
       "    'neoplasms',\n",
       "    'breast neoplasms',\n",
       "    'obesity',\n",
       "    'carcinoma, hepatocellular',\n",
       "    'prostatic neoplasms',\n",
       "    'melanoma',\n",
       "    'leukemia, myeloid, acute',\n",
       "    'colorectal neoplasms',\n",
       "    'glioblastoma',\n",
       "    'colonic neoplasms',\n",
       "    'diabetes mellitus',\n",
       "    'diabetes mellitus, type 2',\n",
       "    'adenocarcinoma of lung',\n",
       "    'triple negative breast neoplasms',\n",
       "    'ovarian neoplasms',\n",
       "    'leukemia',\n",
       "    'autoimmune diseases',\n",
       "    'neuroblastoma',\n",
       "    'neoplasm metastasis'],\n",
       "   'organisms': ['homo sapiens',\n",
       "    'mus musculus',\n",
       "    'rattus norvegicus',\n",
       "    'caenorhabditis elegans',\n",
       "    'escherichia coli',\n",
       "    'arabidopsis thaliana',\n",
       "    'danio rerio',\n",
       "    'saccharomyces cerevisiae',\n",
       "    'macaca mulatta',\n",
       "    'macaca fascicularis',\n",
       "    'pan troglodytes',\n",
       "    'gallus gallus',\n",
       "    'homo sapiens,mus musculus',\n",
       "    'chlorocebus aethiops',\n",
       "    'chlorocebus sabaeus',\n",
       "    'mesocricetus auratus',\n",
       "    'papio hamadryas',\n",
       "    'bos taurus',\n",
       "    'microcebus murinus',\n",
       "    'rattus rattus'],\n",
       "   'sources': ['geo'],\n",
       "   'datatypes': ['transcriptomics',\n",
       "    'single cell',\n",
       "    'raw counts transcriptomics',\n",
       "    'snp array'],\n",
       "   'dataset_count': 63532,\n",
       "   'disease_count': 3374,\n",
       "   'tissue_count': 1004,\n",
       "   'organism_count': 108,\n",
       "   'cell_line_count': 5245,\n",
       "   'cell_type_count': 791,\n",
       "   'drug_count': 1598,\n",
       "   'data_type_count': 4,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 1556378,\n",
       "   'normal_sample_count': 1017118},\n",
       "  {'repo_name': 'rcsb_structures',\n",
       "   'repo_id': '1639402777465',\n",
       "   'indexes': {'gct_metadata': 'rcsb_structures_gct_metadata',\n",
       "    'h5ad_metadata': 'rcsb_structures_h5ad_metadata',\n",
       "    'csv': 'rcsb_structures_csv',\n",
       "    'files': 'rcsb_structures_files',\n",
       "    'json': 'rcsb_structures_json',\n",
       "    'ipynb': 'rcsb_structures_ipynb',\n",
       "    'gct_data': 'rcsb_structures_gct_data',\n",
       "    'h5ad_data': 'rcsb_structures_h5ad_data'},\n",
       "   'diseases': ['normal'],\n",
       "   'organisms': ['homo sapiens',\n",
       "    'none',\n",
       "    'escherichia coli',\n",
       "    'mus musculus',\n",
       "    'rattus norvegicus',\n",
       "    'saccharomyces cerevisiae',\n",
       "    'mycobacterium tuberculosis',\n",
       "    'pseudomonas aeruginosa',\n",
       "    'arabidopsis thaliana',\n",
       "    'staphylococcus aureus',\n",
       "    'severe acute respiratory syndrome coronavirus 2',\n",
       "    'bos taurus',\n",
       "    'human immunodeficiency virus 1',\n",
       "    'bacillus subtilis',\n",
       "    'synthetic construct',\n",
       "    'thermus thermophilus',\n",
       "    'gallus gallus',\n",
       "    'drosophila melanogaster',\n",
       "    'enterobacteria phage t4',\n",
       "    'escherichia coli (strain k12)'],\n",
       "   'sources': ['rcsb'],\n",
       "   'datatypes': ['structural biology'],\n",
       "   'dataset_count': 135144,\n",
       "   'disease_count': 1,\n",
       "   'tissue_count': 186,\n",
       "   'organism_count': 6852,\n",
       "   'cell_line_count': 165,\n",
       "   'cell_type_count': 145,\n",
       "   'drug_count': 1,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_name': 'auron_single_cell_atlas',\n",
       "   'repo_id': '1649415790935',\n",
       "   'indexes': {'gct_metadata': 'auron_single_cell_atlas_gct_metadata',\n",
       "    'h5ad_metadata': 'auron_single_cell_atlas_h5ad_metadata',\n",
       "    'csv': 'auron_single_cell_atlas_csv',\n",
       "    'files': 'auron_single_cell_atlas_files',\n",
       "    'json': 'auron_single_cell_atlas_json',\n",
       "    'ipynb': 'auron_single_cell_atlas_ipynb',\n",
       "    'gct_data': 'auron_single_cell_atlas_gct_data',\n",
       "    'h5ad_data': 'auron_single_cell_atlas_h5ad_data'},\n",
       "   'diseases': [],\n",
       "   'organisms': [],\n",
       "   'sources': [],\n",
       "   'datatypes': [],\n",
       "   'dataset_count': 0,\n",
       "   'disease_count': 0,\n",
       "   'tissue_count': 0,\n",
       "   'organism_count': 0,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 0,\n",
       "   'data_source_count': 0,\n",
       "   'sample_count': 0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_name': 'transcriptomics__celt',\n",
       "   'repo_id': '1612862312240',\n",
       "   'indexes': {'gct_metadata': 'transcriptomics__celt_gct_metadata',\n",
       "    'h5ad_metadata': 'transcriptomics__celt_h5ad_metadata',\n",
       "    'csv': 'transcriptomics__celt_csv',\n",
       "    'files': 'transcriptomics__celt_files',\n",
       "    'json': 'transcriptomics__celt_json',\n",
       "    'ipynb': 'transcriptomics__celt_ipynb',\n",
       "    'gct_data': 'transcriptomics__celt_gct_data',\n",
       "    'h5ad_data': 'transcriptomics__celt_h5ad_data'},\n",
       "   'diseases': ['crohn disease',\n",
       "    'ulcerative colitis',\n",
       "    'colorectal cancer',\n",
       "    'enteric nervous system',\n",
       "    'non-small-cell lung cancer',\n",
       "    'melanoma',\n",
       "    'pediatric obesity',\n",
       "    'carcinoma, hepatocellular',\n",
       "    'inflammatory bowel disease',\n",
       "    'pediatric crohn disease',\n",
       "    'urinary bladder neoplasms',\n",
       "    'breast cancer',\n",
       "    'prostatic neoplasms',\n",
       "    'uveal melanoma',\n",
       "    'carcinoma, ductal',\n",
       "    'cholangiocarcinoma',\n",
       "    'head and neck squamous cell carcinomas',\n",
       "    'non-alcoholic fatty liver disease',\n",
       "    'normal',\n",
       "    'pancreatic neoplasms'],\n",
       "   'organisms': ['homo sapiens', 'mus musculus'],\n",
       "   'sources': ['gene expression omnibus (geo)', 'geo', 'publication'],\n",
       "   'datatypes': ['single cell', 'transcriptomics'],\n",
       "   'dataset_count': 92,\n",
       "   'disease_count': 48,\n",
       "   'tissue_count': 36,\n",
       "   'organism_count': 2,\n",
       "   'cell_line_count': 4,\n",
       "   'cell_type_count': 37,\n",
       "   'drug_count': 4,\n",
       "   'data_type_count': 2,\n",
       "   'data_source_count': 3,\n",
       "   'sample_count': 4718,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_name': 'enterprise_atlas',\n",
       "   'repo_id': '1638441282192',\n",
       "   'indexes': {'gct_metadata': 'enterprise_atlas_gct_metadata',\n",
       "    'h5ad_metadata': 'enterprise_atlas_h5ad_metadata',\n",
       "    'csv': 'enterprise_atlas_csv',\n",
       "    'files': 'enterprise_atlas_files',\n",
       "    'json': 'enterprise_atlas_json',\n",
       "    'ipynb': 'enterprise_atlas_ipynb',\n",
       "    'gct_data': 'enterprise_atlas_gct_data',\n",
       "    'h5ad_data': 'enterprise_atlas_h5ad_data'},\n",
       "   'diseases': ['breast neoplasms',\n",
       "    'endometrial neoplasms',\n",
       "    'endometriosis',\n",
       "    'none',\n",
       "    'triple negative breast neoplasms',\n",
       "    'smith-magenis',\n",
       "    'tay–sachs',\n",
       "    'carcinoma, hepatocellular',\n",
       "    'infertility',\n",
       "    'neoplasm metastasis',\n",
       "    'neoplasms, basal cell',\n",
       "    'carcinoma',\n",
       "    'hereditary breast and ovarian cancer syndrome',\n",
       "    'influenza, human',\n",
       "    'leiomyoma',\n",
       "    'neuroblastoma',\n",
       "    'normal',\n",
       "    'prostatic neoplasms',\n",
       "    'autoimmune diseases',\n",
       "    'bone marrow neoplasms'],\n",
       "   'organisms': ['homo sapiens', 'mus musculus'],\n",
       "   'sources': ['geo', 'proprietary'],\n",
       "   'datatypes': ['transcriptomics', 'single cell', 'mutation', 'mirna'],\n",
       "   'dataset_count': 173,\n",
       "   'disease_count': 36,\n",
       "   'tissue_count': 25,\n",
       "   'organism_count': 2,\n",
       "   'cell_line_count': 96,\n",
       "   'cell_type_count': 44,\n",
       "   'drug_count': 22,\n",
       "   'data_type_count': 4,\n",
       "   'data_source_count': 2,\n",
       "   'sample_count': 8962,\n",
       "   'normal_sample_count': 3939},\n",
       "  {'repo_name': 'cptac',\n",
       "   'repo_id': '1609924165364',\n",
       "   'indexes': {'gct_metadata': 'cptac_gct_metadata',\n",
       "    'h5ad_metadata': 'cptac_h5ad_metadata',\n",
       "    'csv': 'cptac_csv',\n",
       "    'files': 'cptac_files',\n",
       "    'json': 'cptac_json',\n",
       "    'ipynb': 'cptac_ipynb',\n",
       "    'gct_data': 'cptac_gct_data',\n",
       "    'h5ad_data': 'cptac_h5ad_data'},\n",
       "   'diseases': ['carcinoma, hepatocellular',\n",
       "    'glucose-galactose malabsorption',\n",
       "    'squamous cell carcinoma of head and neck',\n",
       "    'hereditary leiomyomatosis and renal cell cancer',\n",
       "    'uterine endometrial carcinoma',\n",
       "    'colonic neoplasms',\n",
       "    'breast cancer',\n",
       "    'ovarian neoplasms',\n",
       "    'adenocarcinoma',\n",
       "    'brain neoplasms',\n",
       "    'breast neoplasms',\n",
       "    'carcinoma',\n",
       "    'glioblastoma'],\n",
       "   'organisms': ['homo sapiens'],\n",
       "   'sources': ['cptac'],\n",
       "   'datatypes': ['proteomics',\n",
       "    'phosphoproteomics',\n",
       "    'transcriptomics',\n",
       "    'copy number variation',\n",
       "    'mutation',\n",
       "    'mirna expression',\n",
       "    'acetylproteomics',\n",
       "    'methylation',\n",
       "    'lipidomics',\n",
       "    'metabolomics'],\n",
       "   'dataset_count': 6542,\n",
       "   'disease_count': 13,\n",
       "   'tissue_count': 33,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 1,\n",
       "   'cell_type_count': 1,\n",
       "   'drug_count': 1,\n",
       "   'data_type_count': 10,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 2440,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_name': 'liveromix_atlas',\n",
       "   'repo_id': '1615965444377',\n",
       "   'indexes': {'gct_metadata': 'liveromix_atlas_gct_metadata',\n",
       "    'h5ad_metadata': 'liveromix_atlas_h5ad_metadata',\n",
       "    'csv': 'liveromix_atlas_csv',\n",
       "    'files': 'liveromix_atlas_files',\n",
       "    'json': 'liveromix_atlas_json',\n",
       "    'ipynb': 'liveromix_atlas_ipynb',\n",
       "    'gct_data': 'liveromix_atlas_gct_data',\n",
       "    'h5ad_data': 'liveromix_atlas_h5ad_data'},\n",
       "   'diseases': ['normal',\n",
       "    'carcinoma, hepatocellular',\n",
       "    'obesity',\n",
       "    'neoplasms',\n",
       "    'fatty liver',\n",
       "    'non-alcoholic fatty liver disease',\n",
       "    'chemical and drug induced liver injury',\n",
       "    'insulin resistance',\n",
       "    'diabetes mellitus',\n",
       "    'diabetes mellitus, type 2',\n",
       "    'coronaviridae infections',\n",
       "    'metabolic syndrome',\n",
       "    'hepatitis c',\n",
       "    'liver neoplasms',\n",
       "    'breast neoplasms',\n",
       "    'colorectal neoplasms',\n",
       "    'metabolic diseases',\n",
       "    'glucose intolerance',\n",
       "    'liver diseases',\n",
       "    'hepatitis b'],\n",
       "   'organisms': ['homo sapiens',\n",
       "    'mus musculus',\n",
       "    'rattus norvegicus',\n",
       "    'danio rerio',\n",
       "    'sus scrofa domesticus',\n",
       "    'macaca mulatta',\n",
       "    'dicentrarchus labrax',\n",
       "    'mus spretus',\n",
       "    'oncorhynchus mykiss',\n",
       "    'marmota monax',\n",
       "    'pan troglodytes',\n",
       "    'capra hircus',\n",
       "    'chlorocebus aethiops',\n",
       "    'macaca fascicularis',\n",
       "    'mus musculus domesticus',\n",
       "    'arvicanthis niloticus',\n",
       "    'cebus capucinus',\n",
       "    'neotoma lepida',\n",
       "    'papio cynocephalus',\n",
       "    'papio hamadryas'],\n",
       "   'sources': ['geo',\n",
       "    'lincs',\n",
       "    'tcga',\n",
       "    'metabolomics workbench',\n",
       "    'metabolights',\n",
       "    'ccle',\n",
       "    'cptac',\n",
       "    'depmap',\n",
       "    'hpa',\n",
       "    'gtex'],\n",
       "   'datatypes': ['transcriptomics',\n",
       "    'mutation',\n",
       "    'metabolomics',\n",
       "    'single cell',\n",
       "    'proteomics',\n",
       "    'lipidomics',\n",
       "    'mirna',\n",
       "    'drug screens',\n",
       "    'gene dependency',\n",
       "    'gene effect'],\n",
       "   'dataset_count': 6740,\n",
       "   'disease_count': 752,\n",
       "   'tissue_count': 60,\n",
       "   'organism_count': 22,\n",
       "   'cell_line_count': 347,\n",
       "   'cell_type_count': 310,\n",
       "   'drug_count': 4704,\n",
       "   'data_type_count': 12,\n",
       "   'data_source_count': 10,\n",
       "   'sample_count': 143535,\n",
       "   'normal_sample_count': 110202},\n",
       "  {'repo_name': 'cbioportal',\n",
       "   'repo_id': '1623986995264',\n",
       "   'indexes': {'gct_metadata': 'cbioportal_gct_metadata',\n",
       "    'h5ad_metadata': 'cbioportal_h5ad_metadata',\n",
       "    'csv': 'cbioportal_csv',\n",
       "    'files': 'cbioportal_files',\n",
       "    'json': 'cbioportal_json',\n",
       "    'ipynb': 'cbioportal_ipynb',\n",
       "    'gct_data': 'cbioportal_gct_data',\n",
       "    'h5ad_data': 'cbioportal_h5ad_data'},\n",
       "   'diseases': ['prostatic neoplasms',\n",
       "    'adenocarcinoma of lung',\n",
       "    'carcinoma, ductal, breast',\n",
       "    'breast neoplasms',\n",
       "    'colorectal neoplasms',\n",
       "    'leukemia, myeloid, acute',\n",
       "    'melanoma',\n",
       "    'leukemia, b-cell',\n",
       "    'glioblastoma',\n",
       "    'lymphoma, large b-cell, diffuse',\n",
       "    'carcinoma, adenoid cystic',\n",
       "    'cholangiocarcinoma',\n",
       "    'urinary bladder neoplasms',\n",
       "    'esophageal neoplasms',\n",
       "    'pancreatic neoplasms',\n",
       "    'carcinoma, non-small-cell lung',\n",
       "    'neuroblastoma',\n",
       "    'leukemia, lymphocytic, chronic, b-cell',\n",
       "    'astrocytoma',\n",
       "    'neoplasms, unknown primary'],\n",
       "   'organisms': ['homo sapiens'],\n",
       "   'sources': ['cbioportal'],\n",
       "   'datatypes': ['mutation',\n",
       "    'copy number variation',\n",
       "    'fusion',\n",
       "    'transcriptomics',\n",
       "    'methylation'],\n",
       "   'dataset_count': 116760,\n",
       "   'disease_count': 346,\n",
       "   'tissue_count': 141,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 1580,\n",
       "   'cell_type_count': 1,\n",
       "   'drug_count': 1,\n",
       "   'data_type_count': 5,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 58994,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_name': 'depmap',\n",
       "   'repo_id': '1612338998334',\n",
       "   'indexes': {'gct_metadata': 'depmap_gct_metadata',\n",
       "    'h5ad_metadata': 'depmap_h5ad_metadata',\n",
       "    'csv': 'depmap_csv',\n",
       "    'files': 'depmap_files',\n",
       "    'json': 'depmap_json',\n",
       "    'ipynb': 'depmap_ipynb',\n",
       "    'gct_data': 'depmap_gct_data',\n",
       "    'h5ad_data': 'depmap_h5ad_data'},\n",
       "   'diseases': ['urinary bladder neoplasms',\n",
       "    'adenocarcinoma of lung',\n",
       "    'breast neoplasms',\n",
       "    'melanoma',\n",
       "    'squamous cell carcinoma of head and neck',\n",
       "    'glioblastoma',\n",
       "    'colorectal neoplasms',\n",
       "    'neuroblastoma',\n",
       "    'pancreatic neoplasms',\n",
       "    'stomach neoplasms',\n",
       "    'liver neoplasms',\n",
       "    'esophageal squamous cell carcinoma',\n",
       "    'multiple myeloma',\n",
       "    'neoplasms, squamous cell',\n",
       "    'endometrial neoplasms',\n",
       "    'lung neoplasms',\n",
       "    'carcinoma, pancreatic ductal',\n",
       "    'lymphoma, non-hodgkin',\n",
       "    'small cell lung carcinoma',\n",
       "    'ovarian neoplasms'],\n",
       "   'organisms': ['homo sapiens'],\n",
       "   'sources': ['depmap'],\n",
       "   'datatypes': ['gene dependency', 'drug screens', 'gene effect', 'rnai'],\n",
       "   'dataset_count': 32248,\n",
       "   'disease_count': 130,\n",
       "   'tissue_count': 48,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 1406,\n",
       "   'cell_type_count': 1,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 4,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 4465,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_name': 'pcd',\n",
       "   'repo_id': '1622113130397',\n",
       "   'indexes': {'gct_metadata': 'pcd_gct_metadata',\n",
       "    'h5ad_metadata': 'pcd_h5ad_metadata',\n",
       "    'csv': 'pcd_csv',\n",
       "    'files': 'pcd_files',\n",
       "    'json': 'pcd_json',\n",
       "    'ipynb': 'pcd_ipynb',\n",
       "    'gct_data': 'pcd_gct_data',\n",
       "    'h5ad_data': 'pcd_h5ad_data'},\n",
       "   'diseases': ['none',\n",
       "    'adenocarcinoma',\n",
       "    'carcinoma, squamous cell',\n",
       "    'melanoma',\n",
       "    'squamous cell carcinoma of head and neck',\n",
       "    'neoplasms, squamous cell',\n",
       "    'mesothelioma',\n",
       "    'neuroblastoma',\n",
       "    'sarcoma',\n",
       "    'lymphoma, non-hodgkin',\n",
       "    'carcinoma, renal cell',\n",
       "    'melanoma, amelanotic',\n",
       "    'esophageal squamous cell carcinoma',\n",
       "    'carcinoma, adenosquamous',\n",
       "    'glioblastoma',\n",
       "    'carcinoma, endometrioid',\n",
       "    'adenocarcinoma, mucinous',\n",
       "    'adenocarcinoma of lung',\n",
       "    'chondrosarcoma',\n",
       "    'sarcoma, synovial'],\n",
       "   'organisms': ['homo sapiens'],\n",
       "   'sources': ['pharmacodb'],\n",
       "   'datatypes': ['drug response'],\n",
       "   'dataset_count': 19501,\n",
       "   'disease_count': 133,\n",
       "   'tissue_count': 34,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 1400,\n",
       "   'cell_type_count': 1,\n",
       "   'drug_count': 749,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 653500,\n",
       "   'normal_sample_count': 262},\n",
       "  {'repo_name': 'lincs',\n",
       "   'repo_id': '32',\n",
       "   'indexes': {'gct_metadata': 'lincs_gct_metadata',\n",
       "    'h5ad_metadata': 'lincs_h5ad_metadata',\n",
       "    'csv': 'lincs_csv',\n",
       "    'files': 'lincs_files',\n",
       "    'ipynb': 'lincs_ipynb',\n",
       "    'gct_data': 'lincs_gct_data',\n",
       "    'h5ad_data': 'lincs_h5ad_data'},\n",
       "   'diseases': ['prostatic neoplasms',\n",
       "    'normal',\n",
       "    'colorectal neoplasms',\n",
       "    'adenocarcinoma of lung',\n",
       "    'breast neoplasms',\n",
       "    'melanoma, amelanotic',\n",
       "    'hepatoblastoma',\n",
       "    'carcinoma, pancreatic ductal',\n",
       "    'adenocarcinoma',\n",
       "    'pancreatic neoplasms',\n",
       "    'colonic neoplasms',\n",
       "    'not available',\n",
       "    'leukemia, monocytic, acute',\n",
       "    'endometrial neoplasms',\n",
       "    'carcinoma, large cell',\n",
       "    'melanoma',\n",
       "    'adenocarcinoma, mucinous',\n",
       "    'small cell lung carcinoma',\n",
       "    'leukemia, myeloid, acute',\n",
       "    'cystadenocarcinoma, mucinous'],\n",
       "   'organisms': ['homo sapiens'],\n",
       "   'sources': ['geo, lincs'],\n",
       "   'datatypes': ['transcriptomics'],\n",
       "   'dataset_count': 158254,\n",
       "   'disease_count': 34,\n",
       "   'tissue_count': 21,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 84,\n",
       "   'cell_type_count': 2,\n",
       "   'drug_count': 20347,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 1368466,\n",
       "   'normal_sample_count': 187120},\n",
       "  {'repo_name': 'metabolomics',\n",
       "   'repo_id': '23',\n",
       "   'indexes': {'gct_metadata': 'metabolomics_gct_metadata',\n",
       "    'h5ad_metadata': 'metabolomics_h5ad_metadata',\n",
       "    'csv': 'metabolomics_csv',\n",
       "    'files': 'metabolomics_files',\n",
       "    'ipynb': 'metabolomics_ipynb',\n",
       "    'gct_data': 'metabolomics_gct_data',\n",
       "    'h5ad_data': 'metabolomics_h5ad_data'},\n",
       "   'diseases': ['normal',\n",
       "    'neoplasms',\n",
       "    'diabetes mellitus',\n",
       "    'obesity',\n",
       "    'liver diseases',\n",
       "    'cardiovascular diseases',\n",
       "    'malaria',\n",
       "    'heart diseases',\n",
       "    'inflammation',\n",
       "    'asthma',\n",
       "    'eye diseases',\n",
       "    'carcinoma, hepatocellular',\n",
       "    'diabetes mellitus, type 2',\n",
       "    'kidney diseases',\n",
       "    'metabolic syndrome',\n",
       "    'hyperglycemia',\n",
       "    'insulin resistance',\n",
       "    'wounds and injuries',\n",
       "    'fibrosis',\n",
       "    'glucose intolerance'],\n",
       "   'organisms': ['homo sapiens',\n",
       "    'mus musculus',\n",
       "    'rattus norvegicus',\n",
       "    'arabidopsis thaliana',\n",
       "    'none',\n",
       "    'bos taurus',\n",
       "    'escherichia coli',\n",
       "    'plasmodium falciparum',\n",
       "    'ovis aries',\n",
       "    'saccharomyces cerevisiae',\n",
       "    'trypanosoma brucei',\n",
       "    'canis lupus familiaris',\n",
       "    'solanum lycopersicum',\n",
       "    'zea mays',\n",
       "    'brassica napus',\n",
       "    'macaca mulatta',\n",
       "    'triticum aestivum',\n",
       "    'sus scrofa domesticus',\n",
       "    'thalassiosira pseudonana',\n",
       "    'caenorhabditis elegans'],\n",
       "   'sources': ['metabolomics workbench', 'metabolights', 'publication', 'geo'],\n",
       "   'datatypes': ['metabolomics', 'lipidomics', 'single cell'],\n",
       "   'dataset_count': 1782,\n",
       "   'disease_count': 215,\n",
       "   'tissue_count': 135,\n",
       "   'organism_count': 184,\n",
       "   'cell_line_count': 24,\n",
       "   'cell_type_count': 23,\n",
       "   'drug_count': 67,\n",
       "   'data_type_count': 3,\n",
       "   'data_source_count': 4,\n",
       "   'sample_count': 90845,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_name': 'immport',\n",
       "   'repo_id': '1621422280385',\n",
       "   'indexes': {'gct_metadata': 'immport_gct_metadata',\n",
       "    'h5ad_metadata': 'immport_h5ad_metadata',\n",
       "    'csv': 'immport_csv',\n",
       "    'files': 'immport_files',\n",
       "    'json': 'immport_json',\n",
       "    'ipynb': 'immport_ipynb',\n",
       "    'gct_data': 'immport_gct_data',\n",
       "    'h5ad_data': 'immport_h5ad_data'},\n",
       "   'diseases': ['normal',\n",
       "    'diabetes mellitus, type 1',\n",
       "    'kidney diseases',\n",
       "    'renal insufficiency',\n",
       "    'fused kidney',\n",
       "    'anthrax',\n",
       "    'neoplasms',\n",
       "    'kidney failure, chronic',\n",
       "    'virus diseases',\n",
       "    'covid-19',\n",
       "    'kaposi varicelliform eruption',\n",
       "    'dermatitis, atopic',\n",
       "    'fibrosis',\n",
       "    'lymphoma, non-hodgkin',\n",
       "    'addison disease',\n",
       "    'metabolic diseases',\n",
       "    'lupus erythematosus, systemic',\n",
       "    'smallpox',\n",
       "    'heart failure',\n",
       "    'influenza, human'],\n",
       "   'organisms': ['homo sapiens', 'mus musculus'],\n",
       "   'sources': ['immport'],\n",
       "   'datatypes': ['lab measurement', 'proteomics', 'titer', 'pcr', 'cytometry'],\n",
       "   'dataset_count': 209432,\n",
       "   'disease_count': 86,\n",
       "   'tissue_count': 10,\n",
       "   'organism_count': 2,\n",
       "   'cell_line_count': 1,\n",
       "   'cell_type_count': 2,\n",
       "   'drug_count': 47,\n",
       "   'data_type_count': 5,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 51044,\n",
       "   'normal_sample_count': 25886},\n",
       "  {'repo_name': 'transcriptomics__cyt',\n",
       "   'repo_id': '1612862450692',\n",
       "   'indexes': {'gct_metadata': 'transcriptomics__cyt_gct_metadata',\n",
       "    'h5ad_metadata': 'transcriptomics__cyt_h5ad_metadata',\n",
       "    'csv': 'transcriptomics__cyt_csv',\n",
       "    'files': 'transcriptomics__cyt_files',\n",
       "    'json': 'transcriptomics__cyt_json',\n",
       "    'ipynb': 'transcriptomics__cyt_ipynb',\n",
       "    'gct_data': 'transcriptomics__cyt_gct_data',\n",
       "    'h5ad_data': 'transcriptomics__cyt_h5ad_data'},\n",
       "   'diseases': ['normal',\n",
       "    'ulcerative colitis',\n",
       "    'crohn disease',\n",
       "    'breast carcinoma',\n",
       "    'colorectal carcinoma',\n",
       "    'melanoma',\n",
       "    'colon carcinoma',\n",
       "    'osteoarthritis',\n",
       "    'covid-19',\n",
       "    'endometrial carcinoma',\n",
       "    'gastroenteropancreatic neuroendocrine',\n",
       "    'lung adenocarcinoma',\n",
       "    'non-small cell lung carcinoma',\n",
       "    'pancreatic adenocarcinoma',\n",
       "    'renal carcinoma',\n",
       "    'salmonella enterica',\n",
       "    'salmonella infection'],\n",
       "   'organisms': ['homo sapiens', 'mus musculus', 'rattus rattus'],\n",
       "   'sources': ['gene expression omnibus (geo)', 'arrayexpress', 'ebi'],\n",
       "   'datatypes': ['transcriptomics', 'single cell'],\n",
       "   'dataset_count': 70,\n",
       "   'disease_count': 17,\n",
       "   'tissue_count': 33,\n",
       "   'organism_count': 3,\n",
       "   'cell_line_count': 0,\n",
       "   'cell_type_count': 3,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 2,\n",
       "   'data_source_count': 3,\n",
       "   'sample_count': 3021,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_name': 'ukbiobank',\n",
       "   'repo_id': '1638762466067',\n",
       "   'indexes': {'gct_metadata': 'ukbiobank_gct_metadata',\n",
       "    'h5ad_metadata': 'ukbiobank_h5ad_metadata',\n",
       "    'csv': 'ukbiobank_csv',\n",
       "    'files': 'ukbiobank_files',\n",
       "    'json': 'ukbiobank_json',\n",
       "    'ipynb': 'ukbiobank_ipynb',\n",
       "    'gct_data': 'ukbiobank_gct_data',\n",
       "    'h5ad_data': 'ukbiobank_h5ad_data'},\n",
       "   'diseases': ['none'],\n",
       "   'organisms': ['homo sapiens'],\n",
       "   'sources': [],\n",
       "   'datatypes': ['gwas'],\n",
       "   'dataset_count': 362349,\n",
       "   'disease_count': 1,\n",
       "   'tissue_count': 1,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 1,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 1,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 0,\n",
       "   'sample_count': 0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_name': 'exelixis',\n",
       "   'repo_id': '1649172174566',\n",
       "   'indexes': {'gct_metadata': 'exelixis_gct_metadata',\n",
       "    'h5ad_metadata': 'exelixis_h5ad_metadata',\n",
       "    'csv': 'exelixis_csv',\n",
       "    'files': 'exelixis_files',\n",
       "    'json': 'exelixis_json',\n",
       "    'ipynb': 'exelixis_ipynb',\n",
       "    'gct_data': 'exelixis_gct_data',\n",
       "    'h5ad_data': 'exelixis_h5ad_data'},\n",
       "   'diseases': ['carcinoma, hepatocellular'],\n",
       "   'organisms': ['homo sapiens'],\n",
       "   'sources': ['geo'],\n",
       "   'datatypes': ['transcriptomics'],\n",
       "   'dataset_count': 5,\n",
       "   'disease_count': 1,\n",
       "   'tissue_count': 2,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 1,\n",
       "   'cell_type_count': 1,\n",
       "   'drug_count': 3,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 484,\n",
       "   'normal_sample_count': 85},\n",
       "  {'repo_name': 'bbio',\n",
       "   'repo_id': '1647250357385',\n",
       "   'indexes': {'gct_metadata': 'bbio_gct_metadata',\n",
       "    'h5ad_metadata': 'bbio_h5ad_metadata',\n",
       "    'csv': 'bbio_csv',\n",
       "    'files': 'bbio_files',\n",
       "    'json': 'bbio_json',\n",
       "    'ipynb': 'bbio_ipynb',\n",
       "    'gct_data': 'bbio_gct_data',\n",
       "    'h5ad_data': 'bbio_h5ad_data'},\n",
       "   'diseases': ['leukemia, myeloid, acute', 'normal'],\n",
       "   'organisms': ['homo sapiens'],\n",
       "   'sources': ['branchbio'],\n",
       "   'datatypes': ['single cell multiomics'],\n",
       "   'dataset_count': 7,\n",
       "   'disease_count': 2,\n",
       "   'tissue_count': 2,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 1,\n",
       "   'cell_type_count': 2,\n",
       "   'drug_count': 0,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_name': 'sc_data_lake',\n",
       "   'repo_id': '17',\n",
       "   'indexes': {'gct_metadata': 'sc_data_lake_gct_metadata',\n",
       "    'h5ad_metadata': 'sc_data_lake_h5ad_metadata',\n",
       "    'csv': 'sc_data_lake_csv',\n",
       "    'files': 'sc_data_lake_files',\n",
       "    'ipynb': 'sc_data_lake_ipynb',\n",
       "    'gct_data': 'sc_data_lake_gct_data',\n",
       "    'h5ad_data': 'sc_data_lake_h5ad_data'},\n",
       "   'diseases': ['normal',\n",
       "    'neoplasms',\n",
       "    'breast neoplasms',\n",
       "    'melanoma',\n",
       "    'covid-19',\n",
       "    'obesity',\n",
       "    'colorectal neoplasms',\n",
       "    'carcinoma, non-small-cell lung',\n",
       "    'glioblastoma',\n",
       "    'autoimmune diseases',\n",
       "    'diabetes mellitus',\n",
       "    'leukemia, myeloid, acute',\n",
       "    'carcinoma, hepatocellular',\n",
       "    'alzheimer disease',\n",
       "    'adenocarcinoma of lung',\n",
       "    'multiple sclerosis',\n",
       "    'prostatic neoplasms',\n",
       "    'diabetes mellitus, type 2',\n",
       "    'non-alcoholic fatty liver disease',\n",
       "    'brain neoplasms'],\n",
       "   'organisms': ['mus musculus',\n",
       "    'homo sapiens',\n",
       "    'none',\n",
       "    'danio rerio',\n",
       "    'macaca mulatta',\n",
       "    'mouse',\n",
       "    'nan',\n",
       "    'homo sapiens,mus musculus',\n",
       "    'macaca fascicularis',\n",
       "    'rattus norvegicus',\n",
       "    'sus scrofa',\n",
       "    'drosophila melanogaster',\n",
       "    'gallus gallus',\n",
       "    'homo',\n",
       "    'homo sapiens,rattus norvegicus',\n",
       "    'mesocricetus auratus',\n",
       "    'mus musculus domesticus',\n",
       "    'other sequences'],\n",
       "   'sources': ['geo',\n",
       "    'scp',\n",
       "    'expression atlas',\n",
       "    'humancellatlas',\n",
       "    'gene expression omnibus (geo)',\n",
       "    'publication',\n",
       "    'zenodo'],\n",
       "   'datatypes': ['single cell'],\n",
       "   'dataset_count': 3129,\n",
       "   'disease_count': 786,\n",
       "   'tissue_count': 537,\n",
       "   'organism_count': 18,\n",
       "   'cell_line_count': 97,\n",
       "   'cell_type_count': 719,\n",
       "   'drug_count': 133,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 7,\n",
       "   'sample_count': 355001,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_name': 'tcga',\n",
       "   'repo_id': '15',\n",
       "   'indexes': {'gct_metadata': 'tcga_gct_metadata',\n",
       "    'h5ad_metadata': 'tcga_h5ad_metadata',\n",
       "    'csv': 'tcga_csv',\n",
       "    'files': 'tcga_files',\n",
       "    'ipynb': 'tcga_ipynb',\n",
       "    'gct_data': 'tcga_gct_data',\n",
       "    'h5ad_data': 'tcga_h5ad_data'},\n",
       "   'diseases': ['breast neoplasms',\n",
       "    'clear-cell metastatic renal cell carcinoma',\n",
       "    'squamous cell carcinoma of head and neck',\n",
       "    'endometrial neoplasms',\n",
       "    'colorectal neoplasms',\n",
       "    'adenocarcinoma of lung',\n",
       "    'glioma',\n",
       "    'stomach neoplasms',\n",
       "    'urinary bladder neoplasms',\n",
       "    'prostatic neoplasms',\n",
       "    'neoplasms, squamous cell',\n",
       "    'cystadenocarcinoma, serous',\n",
       "    'carcinoma, hepatocellular',\n",
       "    'thyroid neoplasms',\n",
       "    'melanoma',\n",
       "    'carcinoma, renal cell',\n",
       "    'uterine cervical neoplasms',\n",
       "    'sarcoma',\n",
       "    'glioblastoma',\n",
       "    'rectal neoplasms'],\n",
       "   'organisms': ['homo sapiens'],\n",
       "   'sources': ['tcga'],\n",
       "   'datatypes': ['transcriptomics',\n",
       "    'mirna',\n",
       "    'copy number variation',\n",
       "    'mutation',\n",
       "    'methylation',\n",
       "    'proteomics'],\n",
       "   'dataset_count': 55277,\n",
       "   'disease_count': 41,\n",
       "   'tissue_count': 62,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 1,\n",
       "   'cell_type_count': 1,\n",
       "   'drug_count': 703,\n",
       "   'data_type_count': 6,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_name': 'etx',\n",
       "   'repo_id': '1641883311001',\n",
       "   'indexes': {'gct_metadata': 'etx_gct_metadata',\n",
       "    'h5ad_metadata': 'etx_h5ad_metadata',\n",
       "    'csv': 'etx_csv',\n",
       "    'files': 'etx_files',\n",
       "    'json': 'etx_json',\n",
       "    'ipynb': 'etx_ipynb',\n",
       "    'gct_data': 'etx_gct_data',\n",
       "    'h5ad_data': 'etx_h5ad_data'},\n",
       "   'diseases': ['non-alcoholic fatty liver disease',\n",
       "    'obesity',\n",
       "    'diabetes mellitus, type 2',\n",
       "    'fatty liver',\n",
       "    'insulin resistance',\n",
       "    'normal',\n",
       "    'multiple organ failure',\n",
       "    'abdominal',\n",
       "    'chemical and drug induced liver injury',\n",
       "    'myotonic dystrophy',\n",
       "    'carcinoma',\n",
       "    'cholangitis, sclerosing',\n",
       "    'end stage liver disease',\n",
       "    'gallstones',\n",
       "    'hemochromatosis',\n",
       "    'hepatitis',\n",
       "    'hepatitis c',\n",
       "    'hepatitis, alcoholic',\n",
       "    'hepatitis, autoimmune',\n",
       "    'hepatocellular'],\n",
       "   'organisms': ['homo sapiens'],\n",
       "   'sources': ['geo',\n",
       "    'metabolomics workbench',\n",
       "    'arrayexpress',\n",
       "    'metabolights',\n",
       "    'ncbi'],\n",
       "   'datatypes': ['transcriptomics',\n",
       "    'metabolomics',\n",
       "    'lipidomics',\n",
       "    'methylation'],\n",
       "   'dataset_count': 98,\n",
       "   'disease_count': 28,\n",
       "   'tissue_count': 13,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 3,\n",
       "   'cell_type_count': 17,\n",
       "   'drug_count': 12,\n",
       "   'data_type_count': 4,\n",
       "   'data_source_count': 5,\n",
       "   'sample_count': 1564,\n",
       "   'normal_sample_count': 767},\n",
       "  {'repo_name': 'ngj',\n",
       "   'repo_id': '1618578954468',\n",
       "   'indexes': {'gct_metadata': 'ngj_gct_metadata',\n",
       "    'h5ad_metadata': 'ngj_h5ad_metadata',\n",
       "    'csv': 'ngj_csv',\n",
       "    'files': 'ngj_files',\n",
       "    'json': 'ngj_json',\n",
       "    'ipynb': 'ngj_ipynb',\n",
       "    'gct_data': 'ngj_gct_data',\n",
       "    'h5ad_data': 'ngj_h5ad_data'},\n",
       "   'diseases': ['none',\n",
       "    'endometriosis',\n",
       "    'smith-magenis',\n",
       "    'tay–sachs',\n",
       "    'endometriosis, pcos, healthy'],\n",
       "   'organisms': ['homo sapiens'],\n",
       "   'sources': ['propreitary'],\n",
       "   'datatypes': ['mutation', 'transcriptomics', 'mirna'],\n",
       "   'dataset_count': 18,\n",
       "   'disease_count': 5,\n",
       "   'tissue_count': 5,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 1,\n",
       "   'cell_type_count': 2,\n",
       "   'drug_count': 1,\n",
       "   'data_type_count': 3,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 1220,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_name': 'gtex',\n",
       "   'repo_id': '14',\n",
       "   'indexes': {'gct_metadata': 'gtex_gct_metadata',\n",
       "    'h5ad_metadata': 'gtex_h5ad_metadata',\n",
       "    'csv': 'gtex_csv',\n",
       "    'files': 'gtex_files',\n",
       "    'ipynb': 'gtex_ipynb',\n",
       "    'gct_data': 'gtex_gct_data',\n",
       "    'h5ad_data': 'gtex_h5ad_data'},\n",
       "   'diseases': ['normal'],\n",
       "   'organisms': ['homo sapiens'],\n",
       "   'sources': ['gtex'],\n",
       "   'datatypes': ['transcriptomics'],\n",
       "   'dataset_count': 30,\n",
       "   'disease_count': 1,\n",
       "   'tissue_count': 30,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 1,\n",
       "   'cell_type_count': 1,\n",
       "   'drug_count': 1,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 17381,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_name': 'gnomad',\n",
       "   'repo_id': '1628836648493',\n",
       "   'indexes': {'gct_metadata': 'gnomad_gct_metadata',\n",
       "    'h5ad_metadata': 'gnomad_h5ad_metadata',\n",
       "    'csv': 'gnomad_csv',\n",
       "    'files': 'gnomad_files',\n",
       "    'json': 'gnomad_json',\n",
       "    'ipynb': 'gnomad_ipynb',\n",
       "    'gct_data': 'gnomad_gct_data',\n",
       "    'h5ad_data': 'gnomad_h5ad_data'},\n",
       "   'diseases': ['none'],\n",
       "   'organisms': ['homo sapiens'],\n",
       "   'sources': [],\n",
       "   'datatypes': ['gwas'],\n",
       "   'dataset_count': 196744,\n",
       "   'disease_count': 1,\n",
       "   'tissue_count': 1,\n",
       "   'organism_count': 1,\n",
       "   'cell_line_count': 1,\n",
       "   'cell_type_count': 0,\n",
       "   'drug_count': 1,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 0,\n",
       "   'sample_count': 0,\n",
       "   'normal_sample_count': 0},\n",
       "  {'repo_name': 'geo_raw_counts',\n",
       "   'repo_id': '1647341066415',\n",
       "   'indexes': {'gct_metadata': 'geo_raw_counts_gct_metadata',\n",
       "    'h5ad_metadata': 'geo_raw_counts_h5ad_metadata',\n",
       "    'csv': 'geo_raw_counts_csv',\n",
       "    'files': 'geo_raw_counts_files',\n",
       "    'json': 'geo_raw_counts_json',\n",
       "    'ipynb': 'geo_raw_counts_ipynb',\n",
       "    'gct_data': 'geo_raw_counts_gct_data',\n",
       "    'h5ad_data': 'geo_raw_counts_h5ad_data'},\n",
       "   'diseases': ['normal',\n",
       "    'neoplasms',\n",
       "    'breast neoplasms',\n",
       "    'leukemia, myeloid, acute',\n",
       "    'obesity',\n",
       "    'melanoma',\n",
       "    'prostatic neoplasms',\n",
       "    'carcinoma, hepatocellular',\n",
       "    'glioblastoma',\n",
       "    'autoimmune diseases',\n",
       "    'colorectal neoplasms',\n",
       "    'alzheimer disease',\n",
       "    'leukemia',\n",
       "    'adenocarcinoma of lung',\n",
       "    'diabetes mellitus',\n",
       "    'colonic neoplasms',\n",
       "    'huntington disease',\n",
       "    'neuroblastoma',\n",
       "    'neurodegenerative diseases',\n",
       "    'diabetes mellitus, type 2'],\n",
       "   'organisms': ['mus musculus',\n",
       "    'homo sapiens',\n",
       "    'caenorhabditis elegans',\n",
       "    'escherichia coli',\n",
       "    'danio rerio',\n",
       "    'rattus norvegicus',\n",
       "    'arabidopsis thaliana',\n",
       "    'saccharomyces cerevisiae',\n",
       "    'human herpesvirus 4'],\n",
       "   'sources': ['geo'],\n",
       "   'datatypes': ['raw counts transcriptomics'],\n",
       "   'dataset_count': 10827,\n",
       "   'disease_count': 1306,\n",
       "   'tissue_count': 537,\n",
       "   'organism_count': 9,\n",
       "   'cell_line_count': 1316,\n",
       "   'cell_type_count': 517,\n",
       "   'drug_count': 238,\n",
       "   'data_type_count': 1,\n",
       "   'data_source_count': 1,\n",
       "   'sample_count': 149895,\n",
       "   'normal_sample_count': 123590}]}"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "omixatlas.get_all_omixatlas()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### omixatlas_summary\n",
    "### input\n",
    "repo_id or repo key\n",
    "### output\n",
    "it will return a object which contain information about repo\n",
    "<a id=\"omixatlas_summary\"></a>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'data': {'repo_name': 'atavistik',\n",
       "  'repo_id': '1650348922627',\n",
       "  'indexes': {'gct_metadata': 'atavistik_gct_metadata',\n",
       "   'h5ad_metadata': 'atavistik_h5ad_metadata',\n",
       "   'csv': 'atavistik_csv',\n",
       "   'files': 'atavistik_files',\n",
       "   'json': 'atavistik_json',\n",
       "   'ipynb': 'atavistik_ipynb',\n",
       "   'gct_data': 'atavistik_gct_data',\n",
       "   'h5ad_data': 'atavistik_h5ad_data'},\n",
       "  'diseases': ['none'],\n",
       "  'organisms': ['none'],\n",
       "  'sources': ['none'],\n",
       "  'datatypes': [],\n",
       "  'dataset_count': 1,\n",
       "  'disease_count': 1,\n",
       "  'tissue_count': 1,\n",
       "  'organism_count': 1,\n",
       "  'cell_line_count': 0,\n",
       "  'cell_type_count': 0,\n",
       "  'drug_count': 0,\n",
       "  'data_type_count': 0,\n",
       "  'data_source_count': 1,\n",
       "  'sample_count': 0,\n",
       "  'normal_sample_count': 0}}"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "omixatlas.omixatlas_summary(1650348922627)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### query_metadata\n",
    "### input :\n",
    "query (str) : sql query on omixatlas for example - “SELECT * FROM liveromix_atlas.datasets”.<br>\n",
    "experimental_features (optional): this section includes in querying metadata. <br>\n",
    "query_api_version (str) (optional): v1 or v2.<br>\n",
    "page_size (int) (optional): page size for query.<br>\n",
    "for more information - https://vito39.github.io/testing_repo_2/polly.html#polly.omixatlas.OmixAtlas.query_metadata\n",
    "### output :\n",
    "It will return a table that will contain metadata (for example- disease,total_num_cells) and datasets that \n",
    "satisfied the sql query.\n",
    "<a id=\"query_metadata\"></a>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Query execution succeeded\n",
      "Fetched 20 rows\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>publication_name</th>\n",
       "      <th>tissue</th>\n",
       "      <th>dataset_source</th>\n",
       "      <th>description</th>\n",
       "      <th>organism</th>\n",
       "      <th>year</th>\n",
       "      <th>disease</th>\n",
       "      <th>operation</th>\n",
       "      <th>platform</th>\n",
       "      <th>dataset_id</th>\n",
       "      <th>...</th>\n",
       "      <th>kw_region</th>\n",
       "      <th>kw_location</th>\n",
       "      <th>kw_timestamp</th>\n",
       "      <th>experimental_design</th>\n",
       "      <th>author</th>\n",
       "      <th>abstract</th>\n",
       "      <th>type</th>\n",
       "      <th>manually_curated</th>\n",
       "      <th>file_type</th>\n",
       "      <th>source_process</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>none</td>\n",
       "      <td>[None]</td>\n",
       "      <td>GEO</td>\n",
       "      <td>Inhibition of NF-ÎºB-dependent signaling enhan...</td>\n",
       "      <td>[Homo sapiens]</td>\n",
       "      <td>2018</td>\n",
       "      <td>[Melanoma, Uveal melanoma]</td>\n",
       "      <td>{'is_normalized': 'true', 'batch_corrected_var...</td>\n",
       "      <td>RNASeq</td>\n",
       "      <td>GSE124059_GPL11154</td>\n",
       "      <td>...</td>\n",
       "      <td>us-west-2</td>\n",
       "      <td>https://discover-prod-datalake-v1.s3-us-west-2...</td>\n",
       "      <td>1647622402236</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>30518681</td>\n",
       "      <td>[bone marrow]</td>\n",
       "      <td>GEO</td>\n",
       "      <td>Human Bone Marrow Assessment by Single Cell RN...</td>\n",
       "      <td>[Homo sapiens]</td>\n",
       "      <td>2018</td>\n",
       "      <td>[Normal]</td>\n",
       "      <td>{'is_normalized': 'true', 'batch_corrected_var...</td>\n",
       "      <td>RNASeq</td>\n",
       "      <td>GSE120444_GPL21290</td>\n",
       "      <td>...</td>\n",
       "      <td>us-west-2</td>\n",
       "      <td>https://discover-prod-datalake-v1.s3-us-west-2...</td>\n",
       "      <td>1647622405780</td>\n",
       "      <td>{'categorical_variables': {'age': [{'name': '2...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>31350386</td>\n",
       "      <td>[joint, knee]</td>\n",
       "      <td>GEO</td>\n",
       "      <td>Stabilizing Heterochromatin by DGCR8 Alleviate...</td>\n",
       "      <td>[Mus musculus]</td>\n",
       "      <td>2018</td>\n",
       "      <td>[Osteoarthritis, DiGeorge Syndrome]</td>\n",
       "      <td>{'is_normalized': 'true', 'batch_corrected_var...</td>\n",
       "      <td>RNASeq</td>\n",
       "      <td>GSE121029_GPL21273</td>\n",
       "      <td>...</td>\n",
       "      <td>us-west-2</td>\n",
       "      <td>https://discover-prod-datalake-v1.s3-us-west-2...</td>\n",
       "      <td>1647622407836</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>28903501</td>\n",
       "      <td>[liver]</td>\n",
       "      <td>GEO</td>\n",
       "      <td>Liver lncRNA gene expression in response to CA...</td>\n",
       "      <td>[Mus musculus]</td>\n",
       "      <td>Jun 17 2019</td>\n",
       "      <td>[Normal]</td>\n",
       "      <td>{'is_normalized': 'true', 'batch_corrected_var...</td>\n",
       "      <td>RNASeq</td>\n",
       "      <td>GSE120851_GPL13112</td>\n",
       "      <td>...</td>\n",
       "      <td>us-west-2</td>\n",
       "      <td>https://discover-prod-datalake-v1.s3-us-west-2...</td>\n",
       "      <td>1647622403758</td>\n",
       "      <td>{'categorical_variables': {'age': [{'name': '8...</td>\n",
       "      <td>David,J.,Waxman</td>\n",
       "      <td>Gene expression in livers of male and female m...</td>\n",
       "      <td>Non-coding RNA profiling by high throughput se...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>31365869</td>\n",
       "      <td>[tibialis anterior]</td>\n",
       "      <td>GEO</td>\n",
       "      <td>RNAi Screening for Ubiquitin Ligases that Regu...</td>\n",
       "      <td>[Mus musculus]</td>\n",
       "      <td>2018</td>\n",
       "      <td>[Myostatin-related muscle hypertrophy]</td>\n",
       "      <td>{'is_normalized': 'true', 'batch_corrected_var...</td>\n",
       "      <td>RNASeq</td>\n",
       "      <td>GSE126922_GPL17021</td>\n",
       "      <td>...</td>\n",
       "      <td>us-west-2</td>\n",
       "      <td>https://discover-prod-datalake-v1.s3-us-west-2...</td>\n",
       "      <td>1647622447941</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>30540963</td>\n",
       "      <td>[muscle, motoneuron]</td>\n",
       "      <td>GEO</td>\n",
       "      <td>RNAseq of fluidically isolated human and mouse...</td>\n",
       "      <td>[Mus musculus]</td>\n",
       "      <td>2018</td>\n",
       "      <td>[Amyotrophic Lateral Sclerosis]</td>\n",
       "      <td>{'is_normalized': 'true', 'batch_corrected_var...</td>\n",
       "      <td>RNASeq</td>\n",
       "      <td>GSE121069_GPL17021</td>\n",
       "      <td>...</td>\n",
       "      <td>us-west-2</td>\n",
       "      <td>https://discover-prod-datalake-v1.s3-us-west-2...</td>\n",
       "      <td>1647622444678</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>30707899</td>\n",
       "      <td>[None]</td>\n",
       "      <td>GEO</td>\n",
       "      <td>Loss of primary cilia drives switching from He...</td>\n",
       "      <td>[Mus musculus]</td>\n",
       "      <td>2018</td>\n",
       "      <td>[Carcinoma, Basal Cell]</td>\n",
       "      <td>{'is_normalized': 'true', 'batch_corrected_var...</td>\n",
       "      <td>RNASeq</td>\n",
       "      <td>GSE120954_GPL13112</td>\n",
       "      <td>...</td>\n",
       "      <td>us-west-2</td>\n",
       "      <td>https://discover-prod-datalake-v1.s3-us-west-2...</td>\n",
       "      <td>1647622444725</td>\n",
       "      <td>{'categorical_variables': {'agent': [{'name': ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>31160601</td>\n",
       "      <td>[None]</td>\n",
       "      <td>GEO</td>\n",
       "      <td>Transcriptional profiling of Ampk-deficient B ...</td>\n",
       "      <td>[Mus musculus]</td>\n",
       "      <td>Jun 16 2019</td>\n",
       "      <td>[Normal]</td>\n",
       "      <td>{'is_normalized': 'true', 'batch_corrected_var...</td>\n",
       "      <td>RNASeq</td>\n",
       "      <td>GSE121025_GPL24247</td>\n",
       "      <td>...</td>\n",
       "      <td>us-west-2</td>\n",
       "      <td>https://discover-prod-datalake-v1.s3-us-west-2...</td>\n",
       "      <td>1647622446965</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Fasih,Mubtasim,Ahsan</td>\n",
       "      <td>5’-AMP-Activated Protein Kinase (Ampk) is an e...</td>\n",
       "      <td>Expression profiling by high throughput sequen...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>none</td>\n",
       "      <td>[hippocampus]</td>\n",
       "      <td>GEO</td>\n",
       "      <td>Altered expression of signaling pathways regul...</td>\n",
       "      <td>[Homo sapiens]</td>\n",
       "      <td>2018</td>\n",
       "      <td>[Seizures, Epilepsy, Temporal Lobe]</td>\n",
       "      <td>{'is_normalized': 'true', 'batch_corrected_var...</td>\n",
       "      <td>RNASeq</td>\n",
       "      <td>GSE127871_GPL16791</td>\n",
       "      <td>...</td>\n",
       "      <td>us-west-2</td>\n",
       "      <td>https://discover-prod-datalake-v1.s3-us-west-2...</td>\n",
       "      <td>1647622443907</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>31591481</td>\n",
       "      <td>[None]</td>\n",
       "      <td>GEO</td>\n",
       "      <td>Oncogenic HOXB8 is driven by MYC-regulated sup...</td>\n",
       "      <td>[Homo sapiens]</td>\n",
       "      <td>Mar 01 2020</td>\n",
       "      <td>[Colonic Neoplasms]</td>\n",
       "      <td>{'is_normalized': 'true', 'batch_corrected_var...</td>\n",
       "      <td>RNASeq</td>\n",
       "      <td>GSE121207_GPL23227</td>\n",
       "      <td>...</td>\n",
       "      <td>us-west-2</td>\n",
       "      <td>https://discover-prod-datalake-v1.s3-us-west-2...</td>\n",
       "      <td>1647622457503</td>\n",
       "      <td>{'categorical_variables': {'cell_line': [{'nam...</td>\n",
       "      <td>Xingsheng,,Shu</td>\n",
       "      <td>HOXB8 acts as a transcription facor, thus we k...</td>\n",
       "      <td>Expression profiling by high throughput sequen...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>31442403</td>\n",
       "      <td>[spinal cord, bone marrow, liver]</td>\n",
       "      <td>GEO</td>\n",
       "      <td>Dietary intake regulates the circulating pro-i...</td>\n",
       "      <td>[Mus musculus]</td>\n",
       "      <td>2018</td>\n",
       "      <td>[Normal]</td>\n",
       "      <td>{'is_normalized': 'true', 'batch_corrected_var...</td>\n",
       "      <td>RNASeq</td>\n",
       "      <td>GSE126899_GPL19057</td>\n",
       "      <td>...</td>\n",
       "      <td>us-west-2</td>\n",
       "      <td>https://discover-prod-datalake-v1.s3-us-west-2...</td>\n",
       "      <td>1647622450144</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>none</td>\n",
       "      <td>[None]</td>\n",
       "      <td>GEO</td>\n",
       "      <td>RNAseq analysis of glomerular transcriptome of...</td>\n",
       "      <td>[Mus musculus]</td>\n",
       "      <td>2018</td>\n",
       "      <td>[Glomerulonephritis, Proteinuria, Renal Insuff...</td>\n",
       "      <td>{'is_normalized': 'true', 'batch_corrected_var...</td>\n",
       "      <td>RNASeq</td>\n",
       "      <td>GSE126217_GPL13112</td>\n",
       "      <td>...</td>\n",
       "      <td>us-west-2</td>\n",
       "      <td>https://discover-prod-datalake-v1.s3-us-west-2...</td>\n",
       "      <td>1647622503324</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>31444334</td>\n",
       "      <td>[None]</td>\n",
       "      <td>GEO</td>\n",
       "      <td>Single-cell profiling guided combinatorial imm...</td>\n",
       "      <td>[Mus musculus]</td>\n",
       "      <td>Sep 03 2019</td>\n",
       "      <td>[Triple Negative Breast Neoplasms]</td>\n",
       "      <td>{'is_normalized': 'true', 'batch_corrected_var...</td>\n",
       "      <td>RNASeq</td>\n",
       "      <td>GSE122336_GPL17021</td>\n",
       "      <td>...</td>\n",
       "      <td>us-west-2</td>\n",
       "      <td>https://discover-prod-datalake-v1.s3-us-west-2...</td>\n",
       "      <td>1647622511647</td>\n",
       "      <td>{'categorical_variables': {'genotype': [{'name...</td>\n",
       "      <td>Qingfei,,Wang</td>\n",
       "      <td>Trastuzumab (Herceptin™), a humanized monoclon...</td>\n",
       "      <td>Expression profiling by high throughput sequen...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>none</td>\n",
       "      <td>[breast]</td>\n",
       "      <td>GEO</td>\n",
       "      <td>RNA sequencing of GLO1-depleted MDA-MB-231 bre...</td>\n",
       "      <td>[Homo sapiens]</td>\n",
       "      <td>2018</td>\n",
       "      <td>[Breast Neoplasms]</td>\n",
       "      <td>{'is_normalized': 'true', 'batch_corrected_var...</td>\n",
       "      <td>RNASeq</td>\n",
       "      <td>GSE122229_GPL18573</td>\n",
       "      <td>...</td>\n",
       "      <td>us-west-2</td>\n",
       "      <td>https://discover-prod-datalake-v1.s3-us-west-2...</td>\n",
       "      <td>1647622505579</td>\n",
       "      <td>{'categorical_variables': {'cancer_type': [{'n...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>30670076</td>\n",
       "      <td>[None]</td>\n",
       "      <td>GEO</td>\n",
       "      <td>Prediction of functional microRNA targets by i...</td>\n",
       "      <td>[Homo sapiens]</td>\n",
       "      <td>2018</td>\n",
       "      <td>[Normal]</td>\n",
       "      <td>{'is_normalized': 'true', 'batch_corrected_var...</td>\n",
       "      <td>RNASeq</td>\n",
       "      <td>GSE124530_GPL21290</td>\n",
       "      <td>...</td>\n",
       "      <td>us-west-2</td>\n",
       "      <td>https://discover-prod-datalake-v1.s3-us-west-2...</td>\n",
       "      <td>1647622579881</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>30814741</td>\n",
       "      <td>[None]</td>\n",
       "      <td>GEO</td>\n",
       "      <td>Gene expression profiles of 35 paired HCC and ...</td>\n",
       "      <td>[Homo sapiens]</td>\n",
       "      <td>2018</td>\n",
       "      <td>[Normal, Carcinoma, Hepatocellular]</td>\n",
       "      <td>{'is_normalized': 'true', 'batch_corrected_var...</td>\n",
       "      <td>RNASeq</td>\n",
       "      <td>GSE124535_GPL20795</td>\n",
       "      <td>...</td>\n",
       "      <td>us-west-2</td>\n",
       "      <td>https://discover-prod-datalake-v1.s3-us-west-2...</td>\n",
       "      <td>1647622586359</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[kw_curated_disease]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>18537183</td>\n",
       "      <td>[None]</td>\n",
       "      <td>GEO</td>\n",
       "      <td>Gene expression after transfection of IGF-II s...</td>\n",
       "      <td>[Homo sapiens]</td>\n",
       "      <td>2018</td>\n",
       "      <td>[Carcinoma, Hepatocellular]</td>\n",
       "      <td>{'is_normalized': 'true', 'batch_corrected_var...</td>\n",
       "      <td>Microarray</td>\n",
       "      <td>GSE8714_GPL5705</td>\n",
       "      <td>...</td>\n",
       "      <td>us-west-2</td>\n",
       "      <td>https://discover-prod-datalake-v1.s3-us-west-2...</td>\n",
       "      <td>1647620477069</td>\n",
       "      <td>{'categorical_variables': {'other_0': [{'name'...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>NaN</td>\n",
       "      <td>[None]</td>\n",
       "      <td>GEO</td>\n",
       "      <td>Identification of genes regulated by Rb in wil...</td>\n",
       "      <td>[Mus musculus]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Normal]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Microarray</td>\n",
       "      <td>GSE90571_GPL10787</td>\n",
       "      <td>...</td>\n",
       "      <td>us-west-2</td>\n",
       "      <td>https://discover-prod-datalake-v1.s3-us-west-2...</td>\n",
       "      <td>1647620477228</td>\n",
       "      <td>{'categorical_variables': {'cell_type': [{'nam...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gct</td>\n",
       "      <td>connector</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>none</td>\n",
       "      <td>[tongue]</td>\n",
       "      <td>GEO</td>\n",
       "      <td>Dek overexpression under the environment expos...</td>\n",
       "      <td>[Mus musculus]</td>\n",
       "      <td>2018</td>\n",
       "      <td>[Squamous Cell Carcinoma of Head and Neck, Ton...</td>\n",
       "      <td>{'is_normalized': 'true', 'batch_corrected_var...</td>\n",
       "      <td>Microarray</td>\n",
       "      <td>GSE87587_GPL13912</td>\n",
       "      <td>...</td>\n",
       "      <td>us-west-2</td>\n",
       "      <td>https://discover-prod-datalake-v1.s3-us-west-2...</td>\n",
       "      <td>1647620477253</td>\n",
       "      <td>{'categorical_variables': {'genotype_variation...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>NaN</td>\n",
       "      <td>[lung]</td>\n",
       "      <td>GEO</td>\n",
       "      <td>Examine the expression profiles of lncRNAs in ...</td>\n",
       "      <td>[Homo sapiens]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Carcinoma, Squamous Cell]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Microarray</td>\n",
       "      <td>GSE88862_GPL16956</td>\n",
       "      <td>...</td>\n",
       "      <td>us-west-2</td>\n",
       "      <td>https://discover-prod-datalake-v1.s3-us-west-2...</td>\n",
       "      <td>1647620504342</td>\n",
       "      <td>{'categorical_variables': {'age': [{'name': '6...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gct</td>\n",
       "      <td>connector</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>20 rows × 43 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   publication_name                             tissue dataset_source  \\\n",
       "0              none                             [None]            GEO   \n",
       "1          30518681                      [bone marrow]            GEO   \n",
       "2          31350386                      [joint, knee]            GEO   \n",
       "3          28903501                            [liver]            GEO   \n",
       "4          31365869                [tibialis anterior]            GEO   \n",
       "5          30540963               [muscle, motoneuron]            GEO   \n",
       "6          30707899                             [None]            GEO   \n",
       "7          31160601                             [None]            GEO   \n",
       "8              none                      [hippocampus]            GEO   \n",
       "9          31591481                             [None]            GEO   \n",
       "10         31442403  [spinal cord, bone marrow, liver]            GEO   \n",
       "11             none                             [None]            GEO   \n",
       "12         31444334                             [None]            GEO   \n",
       "13             none                           [breast]            GEO   \n",
       "14         30670076                             [None]            GEO   \n",
       "15         30814741                             [None]            GEO   \n",
       "16         18537183                             [None]            GEO   \n",
       "17              NaN                             [None]            GEO   \n",
       "18             none                           [tongue]            GEO   \n",
       "19              NaN                             [lung]            GEO   \n",
       "\n",
       "                                          description        organism  \\\n",
       "0   Inhibition of NF-ÎºB-dependent signaling enhan...  [Homo sapiens]   \n",
       "1   Human Bone Marrow Assessment by Single Cell RN...  [Homo sapiens]   \n",
       "2   Stabilizing Heterochromatin by DGCR8 Alleviate...  [Mus musculus]   \n",
       "3   Liver lncRNA gene expression in response to CA...  [Mus musculus]   \n",
       "4   RNAi Screening for Ubiquitin Ligases that Regu...  [Mus musculus]   \n",
       "5   RNAseq of fluidically isolated human and mouse...  [Mus musculus]   \n",
       "6   Loss of primary cilia drives switching from He...  [Mus musculus]   \n",
       "7   Transcriptional profiling of Ampk-deficient B ...  [Mus musculus]   \n",
       "8   Altered expression of signaling pathways regul...  [Homo sapiens]   \n",
       "9   Oncogenic HOXB8 is driven by MYC-regulated sup...  [Homo sapiens]   \n",
       "10  Dietary intake regulates the circulating pro-i...  [Mus musculus]   \n",
       "11  RNAseq analysis of glomerular transcriptome of...  [Mus musculus]   \n",
       "12  Single-cell profiling guided combinatorial imm...  [Mus musculus]   \n",
       "13  RNA sequencing of GLO1-depleted MDA-MB-231 bre...  [Homo sapiens]   \n",
       "14  Prediction of functional microRNA targets by i...  [Homo sapiens]   \n",
       "15  Gene expression profiles of 35 paired HCC and ...  [Homo sapiens]   \n",
       "16  Gene expression after transfection of IGF-II s...  [Homo sapiens]   \n",
       "17  Identification of genes regulated by Rb in wil...  [Mus musculus]   \n",
       "18  Dek overexpression under the environment expos...  [Mus musculus]   \n",
       "19  Examine the expression profiles of lncRNAs in ...  [Homo sapiens]   \n",
       "\n",
       "           year                                            disease  \\\n",
       "0          2018                         [Melanoma, Uveal melanoma]   \n",
       "1          2018                                           [Normal]   \n",
       "2          2018                [Osteoarthritis, DiGeorge Syndrome]   \n",
       "3   Jun 17 2019                                           [Normal]   \n",
       "4          2018             [Myostatin-related muscle hypertrophy]   \n",
       "5          2018                    [Amyotrophic Lateral Sclerosis]   \n",
       "6          2018                            [Carcinoma, Basal Cell]   \n",
       "7   Jun 16 2019                                           [Normal]   \n",
       "8          2018                [Seizures, Epilepsy, Temporal Lobe]   \n",
       "9   Mar 01 2020                                [Colonic Neoplasms]   \n",
       "10         2018                                           [Normal]   \n",
       "11         2018  [Glomerulonephritis, Proteinuria, Renal Insuff...   \n",
       "12  Sep 03 2019                 [Triple Negative Breast Neoplasms]   \n",
       "13         2018                                 [Breast Neoplasms]   \n",
       "14         2018                                           [Normal]   \n",
       "15         2018                [Normal, Carcinoma, Hepatocellular]   \n",
       "16         2018                        [Carcinoma, Hepatocellular]   \n",
       "17          NaN                                           [Normal]   \n",
       "18         2018  [Squamous Cell Carcinoma of Head and Neck, Ton...   \n",
       "19          NaN                         [Carcinoma, Squamous Cell]   \n",
       "\n",
       "                                            operation    platform  \\\n",
       "0   {'is_normalized': 'true', 'batch_corrected_var...      RNASeq   \n",
       "1   {'is_normalized': 'true', 'batch_corrected_var...      RNASeq   \n",
       "2   {'is_normalized': 'true', 'batch_corrected_var...      RNASeq   \n",
       "3   {'is_normalized': 'true', 'batch_corrected_var...      RNASeq   \n",
       "4   {'is_normalized': 'true', 'batch_corrected_var...      RNASeq   \n",
       "5   {'is_normalized': 'true', 'batch_corrected_var...      RNASeq   \n",
       "6   {'is_normalized': 'true', 'batch_corrected_var...      RNASeq   \n",
       "7   {'is_normalized': 'true', 'batch_corrected_var...      RNASeq   \n",
       "8   {'is_normalized': 'true', 'batch_corrected_var...      RNASeq   \n",
       "9   {'is_normalized': 'true', 'batch_corrected_var...      RNASeq   \n",
       "10  {'is_normalized': 'true', 'batch_corrected_var...      RNASeq   \n",
       "11  {'is_normalized': 'true', 'batch_corrected_var...      RNASeq   \n",
       "12  {'is_normalized': 'true', 'batch_corrected_var...      RNASeq   \n",
       "13  {'is_normalized': 'true', 'batch_corrected_var...      RNASeq   \n",
       "14  {'is_normalized': 'true', 'batch_corrected_var...      RNASeq   \n",
       "15  {'is_normalized': 'true', 'batch_corrected_var...      RNASeq   \n",
       "16  {'is_normalized': 'true', 'batch_corrected_var...  Microarray   \n",
       "17                                                NaN  Microarray   \n",
       "18  {'is_normalized': 'true', 'batch_corrected_var...  Microarray   \n",
       "19                                                NaN  Microarray   \n",
       "\n",
       "            dataset_id  ...  kw_region  \\\n",
       "0   GSE124059_GPL11154  ...  us-west-2   \n",
       "1   GSE120444_GPL21290  ...  us-west-2   \n",
       "2   GSE121029_GPL21273  ...  us-west-2   \n",
       "3   GSE120851_GPL13112  ...  us-west-2   \n",
       "4   GSE126922_GPL17021  ...  us-west-2   \n",
       "5   GSE121069_GPL17021  ...  us-west-2   \n",
       "6   GSE120954_GPL13112  ...  us-west-2   \n",
       "7   GSE121025_GPL24247  ...  us-west-2   \n",
       "8   GSE127871_GPL16791  ...  us-west-2   \n",
       "9   GSE121207_GPL23227  ...  us-west-2   \n",
       "10  GSE126899_GPL19057  ...  us-west-2   \n",
       "11  GSE126217_GPL13112  ...  us-west-2   \n",
       "12  GSE122336_GPL17021  ...  us-west-2   \n",
       "13  GSE122229_GPL18573  ...  us-west-2   \n",
       "14  GSE124530_GPL21290  ...  us-west-2   \n",
       "15  GSE124535_GPL20795  ...  us-west-2   \n",
       "16     GSE8714_GPL5705  ...  us-west-2   \n",
       "17   GSE90571_GPL10787  ...  us-west-2   \n",
       "18   GSE87587_GPL13912  ...  us-west-2   \n",
       "19   GSE88862_GPL16956  ...  us-west-2   \n",
       "\n",
       "                                          kw_location   kw_timestamp  \\\n",
       "0   https://discover-prod-datalake-v1.s3-us-west-2...  1647622402236   \n",
       "1   https://discover-prod-datalake-v1.s3-us-west-2...  1647622405780   \n",
       "2   https://discover-prod-datalake-v1.s3-us-west-2...  1647622407836   \n",
       "3   https://discover-prod-datalake-v1.s3-us-west-2...  1647622403758   \n",
       "4   https://discover-prod-datalake-v1.s3-us-west-2...  1647622447941   \n",
       "5   https://discover-prod-datalake-v1.s3-us-west-2...  1647622444678   \n",
       "6   https://discover-prod-datalake-v1.s3-us-west-2...  1647622444725   \n",
       "7   https://discover-prod-datalake-v1.s3-us-west-2...  1647622446965   \n",
       "8   https://discover-prod-datalake-v1.s3-us-west-2...  1647622443907   \n",
       "9   https://discover-prod-datalake-v1.s3-us-west-2...  1647622457503   \n",
       "10  https://discover-prod-datalake-v1.s3-us-west-2...  1647622450144   \n",
       "11  https://discover-prod-datalake-v1.s3-us-west-2...  1647622503324   \n",
       "12  https://discover-prod-datalake-v1.s3-us-west-2...  1647622511647   \n",
       "13  https://discover-prod-datalake-v1.s3-us-west-2...  1647622505579   \n",
       "14  https://discover-prod-datalake-v1.s3-us-west-2...  1647622579881   \n",
       "15  https://discover-prod-datalake-v1.s3-us-west-2...  1647622586359   \n",
       "16  https://discover-prod-datalake-v1.s3-us-west-2...  1647620477069   \n",
       "17  https://discover-prod-datalake-v1.s3-us-west-2...  1647620477228   \n",
       "18  https://discover-prod-datalake-v1.s3-us-west-2...  1647620477253   \n",
       "19  https://discover-prod-datalake-v1.s3-us-west-2...  1647620504342   \n",
       "\n",
       "                                  experimental_design                author  \\\n",
       "0                                                 NaN                   NaN   \n",
       "1   {'categorical_variables': {'age': [{'name': '2...                   NaN   \n",
       "2                                                 NaN                   NaN   \n",
       "3   {'categorical_variables': {'age': [{'name': '8...       David,J.,Waxman   \n",
       "4                                                 NaN                   NaN   \n",
       "5                                                 NaN                   NaN   \n",
       "6   {'categorical_variables': {'agent': [{'name': ...                   NaN   \n",
       "7                                                 NaN  Fasih,Mubtasim,Ahsan   \n",
       "8                                                 NaN                   NaN   \n",
       "9   {'categorical_variables': {'cell_line': [{'nam...        Xingsheng,,Shu   \n",
       "10                                                NaN                   NaN   \n",
       "11                                                NaN                   NaN   \n",
       "12  {'categorical_variables': {'genotype': [{'name...         Qingfei,,Wang   \n",
       "13  {'categorical_variables': {'cancer_type': [{'n...                   NaN   \n",
       "14                                                NaN                   NaN   \n",
       "15                                                NaN                   NaN   \n",
       "16  {'categorical_variables': {'other_0': [{'name'...                   NaN   \n",
       "17  {'categorical_variables': {'cell_type': [{'nam...                   NaN   \n",
       "18  {'categorical_variables': {'genotype_variation...                   NaN   \n",
       "19  {'categorical_variables': {'age': [{'name': '6...                   NaN   \n",
       "\n",
       "                                             abstract  \\\n",
       "0                                                 NaN   \n",
       "1                                                 NaN   \n",
       "2                                                 NaN   \n",
       "3   Gene expression in livers of male and female m...   \n",
       "4                                                 NaN   \n",
       "5                                                 NaN   \n",
       "6                                                 NaN   \n",
       "7   5’-AMP-Activated Protein Kinase (Ampk) is an e...   \n",
       "8                                                 NaN   \n",
       "9   HOXB8 acts as a transcription facor, thus we k...   \n",
       "10                                                NaN   \n",
       "11                                                NaN   \n",
       "12  Trastuzumab (Herceptin™), a humanized monoclon...   \n",
       "13                                                NaN   \n",
       "14                                                NaN   \n",
       "15                                                NaN   \n",
       "16                                                NaN   \n",
       "17                                                NaN   \n",
       "18                                                NaN   \n",
       "19                                                NaN   \n",
       "\n",
       "                                                 type      manually_curated  \\\n",
       "0                                                 NaN                   NaN   \n",
       "1                                                 NaN                   NaN   \n",
       "2                                                 NaN                   NaN   \n",
       "3   Non-coding RNA profiling by high throughput se...                   NaN   \n",
       "4                                                 NaN                   NaN   \n",
       "5                                                 NaN                   NaN   \n",
       "6                                                 NaN                   NaN   \n",
       "7   Expression profiling by high throughput sequen...                   NaN   \n",
       "8                                                 NaN                   NaN   \n",
       "9   Expression profiling by high throughput sequen...                   NaN   \n",
       "10                                                NaN                   NaN   \n",
       "11                                                NaN                   NaN   \n",
       "12  Expression profiling by high throughput sequen...                   NaN   \n",
       "13                                                NaN                   NaN   \n",
       "14                                                NaN                   NaN   \n",
       "15                                                NaN  [kw_curated_disease]   \n",
       "16                                                NaN                   NaN   \n",
       "17                                                NaN                   NaN   \n",
       "18                                                NaN                   NaN   \n",
       "19                                                NaN                   NaN   \n",
       "\n",
       "   file_type source_process  \n",
       "0        NaN            NaN  \n",
       "1        NaN            NaN  \n",
       "2        NaN            NaN  \n",
       "3        NaN            NaN  \n",
       "4        NaN            NaN  \n",
       "5        NaN            NaN  \n",
       "6        NaN            NaN  \n",
       "7        NaN            NaN  \n",
       "8        NaN            NaN  \n",
       "9        NaN            NaN  \n",
       "10       NaN            NaN  \n",
       "11       NaN            NaN  \n",
       "12       NaN            NaN  \n",
       "13       NaN            NaN  \n",
       "14       NaN            NaN  \n",
       "15       NaN            NaN  \n",
       "16       NaN            NaN  \n",
       "17       gct      connector  \n",
       "18       NaN            NaN  \n",
       "19       gct      connector  \n",
       "\n",
       "[20 rows x 43 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query = \"SELECT * FROM geo.datasets LIMIT 20\"\n",
    "omixatlas.query_metadata(query)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### download_data\n",
    "### input\n",
    "repo_key (str): repo_id OR repo_name from where the data needs to be downloaded.<br>\n",
    "dataset_id (str): dataset_id which the user wants to download.\n",
    "### output\n",
    "message containing information about dataset\n",
    "<a id=\"download_data\"></a>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'data': {'attributes': {'last-modified': '2021-12-22 08:38:05.000000',\n",
       "   'size': '2.03 MB',\n",
       "   'download_url': 'https://discover-prod-datalake-v1.s3.amazonaws.com/GEO_data_lake/data/RNASeq/GSE124059/GCT/GSE124059_GPL11154_curated.gct?AWSAccessKeyId=ASIAVRYB5UBIF5PI74HH&Signature=L6HzDXxVubK%2BaJB%2B4WSykrvSjC0%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEBUaCXVzLXdlc3QtMiJHMEUCIDW7yqwmK09n9gGe99ajjMR%2Fhj8%2B4JW2eki%2ByfX87YdfAiEArHENmeeV%2F7C6Z5NFWHXQYk79tBKnqwgTMVOvMkRYnzAqqgIIzv%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARADGgwzODE3MTkyNTcxNjgiDMDWCMFKhrNC46gklSr%2BAVbyNPD%2FLqMzTBp8MomhXIu7SRgE4iclXtWv1OVbgA%2BkrkqSbS6IG2AAfwzmpuKyqz9rbC5q4pD%2BPmLF48dCa9Bey%2B207zDclfGGzmA0NAiBcoLpE0TvYjn1oq4q5hINeIiw5Rx9lMfNGLpOIvIjAYvYCEtsMcqk%2BlqNu2XO6CXPVD0GJ71dIQgqqiPOJb%2F%2BUrXWXlQtbCqIjyD%2FHkCrjdW2%2Bl9pfPeOvYnjpkTK378gXP72V71l4Mmpnl14KalFMae2bO2wIsQeqUZ96S2jqF4xKh92LQk9XbB9WR0%2FvOod2LxZ8wz3g%2BMo488sDmghwDVqPZ7%2BSOPjNyj9yxa2MOq1qJMGOpoBCt%2F2p%2FXoUZwo%2F8uYbC2UBhNaOcK6JJgfU%2BT%2BHpP7RuxusxevIivuPZl1Nvl%2Bncr8Bcl8K4oE5JTKoHH7inB1lROkAoVm2Rz%2FxO8d%2BspBnuVKAMH4ZaCly0vx1LGX%2FA1VdZf9aNRUN5nR%2FPmhbp1iadjcmuw3m%2BFayrOTCUUNlXt1WQsmQt3GCocyU8fi1obr2YidhQFXvO48JQ%3D%3D&Expires=1651127122'}}}"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "omixatlas.download_data(\"geo\", \"GSE124059_GPL11154\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### save_to_workspace\n",
    "### input :\n",
    "repo_id (str) : repo id.<br>\n",
    "dataset_id (str) : dataset id.<br>\n",
    "workspace_id (str) : workspace id of polly.<br>\n",
    "workspace_path (str) : path in workspace where you want to save file.<br>\n",
    "### output :\n",
    "object containing information about dataset upload.\n",
    "<a id=\"save_to_workspace\"></a>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:root:Data Saved to workspace=9009\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'data': {'type': 'workspace-jobs',\n",
       "  'id': '9/9009',\n",
       "  'attributes': {'destination-gct': '9009/data/GSE124059_GPL11154_curated.gct',\n",
       "   'destination-json': '9009/data/GSE124059_GPL11154.json'}}}"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "omixatlas.save_to_workspace('9', 'GSE124059_GPL11154', 9009, 'data')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### format_converter\n",
    "### Input:\n",
    "repo_key (str) : repo_id.<br>\n",
    "dataset_id (str) : dataset_id.<br>\n",
    "to(str) : output file format.<br>\n",
    "### Output:\n",
    "Message File converted successfully!\n",
    "<a id=\"format_converter\"></a>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:cmap_logger:Reading GCT: ACC_2019_Mutation_ACYC-FMI-19.gct\n",
      "INFO:root:File converted successfully!\n"
     ]
    }
   ],
   "source": [
    "omixatlas.format_converter(\"cbioportal\", \"ACC_2019_Mutation_ACYC-FMI-19\", \"maf\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### get_schema\n",
    "### Input:\n",
    "repo_key (str) : repo_id OR repo_name. This is a mandatory field.<br>\n",
    "schema_level (list) : The default value is [‘dataset’, ‘sample’]. The users can use [‘dataset’] OR [‘sample’] to fetch\n",
    "the schema of dataset OR sample level metadata respectively.<br>\n",
    "source (str) : is the source from where data is ingested into the Omixatlas.<br>\n",
    "data_type (str) : is the datatype for which user wants to get the schema for. The default value is ‘all’, which will \n",
    "fetch the schema of all datatypes except single cell. To fetch the schema for single cell datatype from an OmixAtlas, \n",
    "the user should use ‘single_cell’.<br>\n",
    "### Output:\n",
    "It will contain the schema for dataset, sample as dataframe.\n",
    "{\n",
    "    'dataset':pd.DataFrame,\n",
    "    'sample':pd.DataFrame\n",
    "}\n",
    "<a id=\"get_schema\"></a>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "schema = omixatlas.get_schema(\"geo\", ['dataset', 'sample'], \"all\", \"all\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Source</th>\n",
       "      <th>Datatype</th>\n",
       "      <th>Field Name</th>\n",
       "      <th>Field Description</th>\n",
       "      <th>Field Type</th>\n",
       "      <th>Is Array</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>curated_organism</td>\n",
       "      <td>Orgnism from which the samples were derived</td>\n",
       "      <td>text</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>src_uri</td>\n",
       "      <td>Unique URI derived from data file's S3 location</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>total_num_samples</td>\n",
       "      <td>Total number of samples in a dataset</td>\n",
       "      <td>integer</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>year</td>\n",
       "      <td>Year in which the dataset was published</td>\n",
       "      <td>integer</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>description</td>\n",
       "      <td>Description of the dataset</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>curated_cell_line</td>\n",
       "      <td>Cell lines from which the samples were derived...</td>\n",
       "      <td>text</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>data_table_name</td>\n",
       "      <td>Name of the data table associated with data file</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>data_table_version</td>\n",
       "      <td>Current version of the data table associated w...</td>\n",
       "      <td>integer</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>platform</td>\n",
       "      <td>Platform Accession number signifying the techn...</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>timestamp_</td>\n",
       "      <td>Unix timestamp denoting time of creation for t...</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>file_type</td>\n",
       "      <td>Data file's type</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>publication</td>\n",
       "      <td>Publication associated with the dataset</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>curated_cell_type</td>\n",
       "      <td>Types of cell present in the dataset</td>\n",
       "      <td>text</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>key</td>\n",
       "      <td>Key to the data file in source package</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>summary</td>\n",
       "      <td>Summary of the paper</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>src_repo</td>\n",
       "      <td>Name of the repository this data file originat...</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>drug_smiles</td>\n",
       "      <td>SMILES code of the drug</td>\n",
       "      <td>text</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>package</td>\n",
       "      <td>Name of the package this data file was part of</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>file_location</td>\n",
       "      <td>Unique S3 location for this data file</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>author</td>\n",
       "      <td>Name of the author who published the dataset</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>dataset_id</td>\n",
       "      <td>Unique ID assocaited with every dataset</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>curated_drug</td>\n",
       "      <td>Drugs administered in the samples belonging to...</td>\n",
       "      <td>text</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>curated_gene</td>\n",
       "      <td>Gene studied in the dataset</td>\n",
       "      <td>text</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>curated_disease</td>\n",
       "      <td>Disease associated with the dataset</td>\n",
       "      <td>text</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>abstract</td>\n",
       "      <td>Abstract of the publication assocaited with th...</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>version</td>\n",
       "      <td>Version of data file</td>\n",
       "      <td>integer</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>curated_strain</td>\n",
       "      <td>Strain of the organism from which samples were...</td>\n",
       "      <td>text</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>bucket</td>\n",
       "      <td>S3 bucket in which the data file resides</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>curated_tissue</td>\n",
       "      <td>Tissue from which the samples were derivved</td>\n",
       "      <td>text</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>dataset_source</td>\n",
       "      <td>Source from where the data was fetched</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>data_type</td>\n",
       "      <td>The type of biomolecular data captured (eg - E...</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>overall_design</td>\n",
       "      <td>Overall Design of the experiment as given by t...</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>is_current</td>\n",
       "      <td>Whether this is the current version of the dat...</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>region</td>\n",
       "      <td>Region (AWS) in which this data is located</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Source Datatype          Field Name  \\\n",
       "0     all      all    curated_organism   \n",
       "1     all      all             src_uri   \n",
       "2     all      all   total_num_samples   \n",
       "3     all      all                year   \n",
       "4     all      all         description   \n",
       "5     all      all   curated_cell_line   \n",
       "6     all      all     data_table_name   \n",
       "7     all      all  data_table_version   \n",
       "8     all      all            platform   \n",
       "9     all      all          timestamp_   \n",
       "10    all      all           file_type   \n",
       "11    all      all         publication   \n",
       "12    all      all   curated_cell_type   \n",
       "13    all      all                 key   \n",
       "14    all      all             summary   \n",
       "15    all      all            src_repo   \n",
       "16    all      all         drug_smiles   \n",
       "17    all      all             package   \n",
       "18    all      all       file_location   \n",
       "19    all      all              author   \n",
       "20    all      all          dataset_id   \n",
       "21    all      all        curated_drug   \n",
       "22    all      all        curated_gene   \n",
       "23    all      all     curated_disease   \n",
       "24    all      all            abstract   \n",
       "25    all      all             version   \n",
       "26    all      all      curated_strain   \n",
       "27    all      all              bucket   \n",
       "28    all      all      curated_tissue   \n",
       "29    all      all      dataset_source   \n",
       "30    all      all           data_type   \n",
       "31    all      all      overall_design   \n",
       "32    all      all          is_current   \n",
       "33    all      all              region   \n",
       "\n",
       "                                    Field Description Field Type  Is Array  \n",
       "0         Orgnism from which the samples were derived       text      True  \n",
       "1     Unique URI derived from data file's S3 location       text     False  \n",
       "2                Total number of samples in a dataset    integer     False  \n",
       "3             Year in which the dataset was published    integer     False  \n",
       "4                          Description of the dataset       text     False  \n",
       "5   Cell lines from which the samples were derived...       text      True  \n",
       "6    Name of the data table associated with data file       text     False  \n",
       "7   Current version of the data table associated w...    integer     False  \n",
       "8   Platform Accession number signifying the techn...       text     False  \n",
       "9   Unix timestamp denoting time of creation for t...       text     False  \n",
       "10                                   Data file's type       text     False  \n",
       "11            Publication associated with the dataset       text     False  \n",
       "12               Types of cell present in the dataset       text      True  \n",
       "13             Key to the data file in source package       text     False  \n",
       "14                               Summary of the paper       text     False  \n",
       "15  Name of the repository this data file originat...       text     False  \n",
       "16                            SMILES code of the drug       text      True  \n",
       "17     Name of the package this data file was part of       text     False  \n",
       "18              Unique S3 location for this data file       text     False  \n",
       "19       Name of the author who published the dataset       text     False  \n",
       "20            Unique ID assocaited with every dataset       text     False  \n",
       "21  Drugs administered in the samples belonging to...       text      True  \n",
       "22                        Gene studied in the dataset       text      True  \n",
       "23                Disease associated with the dataset       text      True  \n",
       "24  Abstract of the publication assocaited with th...       text     False  \n",
       "25                               Version of data file    integer     False  \n",
       "26  Strain of the organism from which samples were...       text      True  \n",
       "27           S3 bucket in which the data file resides       text     False  \n",
       "28        Tissue from which the samples were derivved       text      True  \n",
       "29             Source from where the data was fetched       text     False  \n",
       "30  The type of biomolecular data captured (eg - E...       text     False  \n",
       "31  Overall Design of the experiment as given by t...       text     False  \n",
       "32  Whether this is the current version of the dat...       text     False  \n",
       "33         Region (AWS) in which this data is located       text     False  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "schema.dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Source</th>\n",
       "      <th>Datatype</th>\n",
       "      <th>Field Name</th>\n",
       "      <th>Field Description</th>\n",
       "      <th>Field Type</th>\n",
       "      <th>Is Array</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>growth_protocol_ch1</td>\n",
       "      <td>NA</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>src_uri</td>\n",
       "      <td>Unique URI derived from source data file's S3 ...</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>sample_id</td>\n",
       "      <td>Unique ID associated with every sample</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>curated_gene_modified</td>\n",
       "      <td>Gene modified through genetic modification</td>\n",
       "      <td>text</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>dose_ch1</td>\n",
       "      <td>NA</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>curated_cohort_name</td>\n",
       "      <td>Name of the cohort to which the sample belongs</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>curated_control</td>\n",
       "      <td>Signifies whether the given sample is a contro...</td>\n",
       "      <td>integer</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>src_dataset_id</td>\n",
       "      <td>Dataset ID of the file this data entity origin...</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>extract_protocol_ch1</td>\n",
       "      <td>NA</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>characteristics_ch2</td>\n",
       "      <td>NA</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>characteristics_ch1</td>\n",
       "      <td>NA</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>extract_protocol_ch2</td>\n",
       "      <td>NA</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>age_ch1</td>\n",
       "      <td>NA</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>curated_cell_type</td>\n",
       "      <td>Types of cell present in the sample</td>\n",
       "      <td>text</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>sex_ch1</td>\n",
       "      <td>NA</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>treatment_ch1</td>\n",
       "      <td>NA</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>dataset_id</td>\n",
       "      <td>Unique ID of the dataset to which the sample b...</td>\n",
       "      <td>text</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>taxid_ch1</td>\n",
       "      <td>NA</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>curated_gene</td>\n",
       "      <td>Gene of interest in the sample</td>\n",
       "      <td>text</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>curated_drug</td>\n",
       "      <td>Drug admistered in the sample</td>\n",
       "      <td>text</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>version</td>\n",
       "      <td>Version of data entity</td>\n",
       "      <td>integer</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>label_protocol_ch2</td>\n",
       "      <td>NA</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>label_protocol_ch1</td>\n",
       "      <td>NA</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>treatment_protocol_ch2</td>\n",
       "      <td>NA</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>characteristics_ch2_1</td>\n",
       "      <td>NA</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>treatment_protocol_ch1</td>\n",
       "      <td>NA</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>instrument_model</td>\n",
       "      <td>NA</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>name</td>\n",
       "      <td>Name of the data entity (local to its source f...</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>curated_genetic_modification_type</td>\n",
       "      <td>Signifies the kind of genetic modification per...</td>\n",
       "      <td>text</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>bmi_ch1</td>\n",
       "      <td>NA</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>data_processing</td>\n",
       "      <td>NA</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>curated_cell_line</td>\n",
       "      <td>Cell line from which the sample was derived</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>description</td>\n",
       "      <td>Description of the sample</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>title</td>\n",
       "      <td>NA</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>timestamp_</td>\n",
       "      <td>Unix timestamp denoting time of creation for t...</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>curated_cohort_id</td>\n",
       "      <td>Numeric ID of the cohort to which the sample b...</td>\n",
       "      <td>integer</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>data_id</td>\n",
       "      <td>Unique ID for this data entity on Polly</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>src_repo</td>\n",
       "      <td>Name of the repository this data entity origin...</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>id_key</td>\n",
       "      <td>Name of the key that was used for creation of ...</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>gender_ch1</td>\n",
       "      <td>NA</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>smoking_status_ch1</td>\n",
       "      <td>NA</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>label_ch2</td>\n",
       "      <td>NA</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>curated_disease</td>\n",
       "      <td>Disease associated with the sample</td>\n",
       "      <td>text</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>label_ch1</td>\n",
       "      <td>NA</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>time_point_ch1</td>\n",
       "      <td>NA</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>characteristics_ch1_3</td>\n",
       "      <td>NA</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>characteristics_ch1_2</td>\n",
       "      <td>NA</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>curated_tissue</td>\n",
       "      <td>Tissue to which the sample belongs</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>curated_drug_smiles_code</td>\n",
       "      <td>SMILES code of the drug admistered in the sample</td>\n",
       "      <td>text</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>hyb_protocol</td>\n",
       "      <td>NA</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>platform_id</td>\n",
       "      <td>NA</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>is_current</td>\n",
       "      <td>Whether this is the current version of the dat...</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>all</td>\n",
       "      <td>all</td>\n",
       "      <td>characteristics_ch1_1</td>\n",
       "      <td>NA</td>\n",
       "      <td>text</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Source Datatype                         Field Name  \\\n",
       "0     all      all                growth_protocol_ch1   \n",
       "1     all      all                            src_uri   \n",
       "2     all      all                          sample_id   \n",
       "3     all      all              curated_gene_modified   \n",
       "4     all      all                           dose_ch1   \n",
       "5     all      all                curated_cohort_name   \n",
       "6     all      all                    curated_control   \n",
       "7     all      all                     src_dataset_id   \n",
       "8     all      all               extract_protocol_ch1   \n",
       "9     all      all                characteristics_ch2   \n",
       "10    all      all                characteristics_ch1   \n",
       "11    all      all               extract_protocol_ch2   \n",
       "12    all      all                            age_ch1   \n",
       "13    all      all                  curated_cell_type   \n",
       "14    all      all                            sex_ch1   \n",
       "15    all      all                      treatment_ch1   \n",
       "16    all      all                         dataset_id   \n",
       "17    all      all                          taxid_ch1   \n",
       "18    all      all                       curated_gene   \n",
       "19    all      all                       curated_drug   \n",
       "20    all      all                            version   \n",
       "21    all      all                 label_protocol_ch2   \n",
       "22    all      all                 label_protocol_ch1   \n",
       "23    all      all             treatment_protocol_ch2   \n",
       "24    all      all              characteristics_ch2_1   \n",
       "25    all      all             treatment_protocol_ch1   \n",
       "26    all      all                   instrument_model   \n",
       "27    all      all                               name   \n",
       "28    all      all  curated_genetic_modification_type   \n",
       "29    all      all                            bmi_ch1   \n",
       "30    all      all                    data_processing   \n",
       "31    all      all                  curated_cell_line   \n",
       "32    all      all                        description   \n",
       "33    all      all                              title   \n",
       "34    all      all                         timestamp_   \n",
       "35    all      all                  curated_cohort_id   \n",
       "36    all      all                            data_id   \n",
       "37    all      all                           src_repo   \n",
       "38    all      all                             id_key   \n",
       "39    all      all                         gender_ch1   \n",
       "40    all      all                 smoking_status_ch1   \n",
       "41    all      all                          label_ch2   \n",
       "42    all      all                    curated_disease   \n",
       "43    all      all                          label_ch1   \n",
       "44    all      all                     time_point_ch1   \n",
       "45    all      all              characteristics_ch1_3   \n",
       "46    all      all              characteristics_ch1_2   \n",
       "47    all      all                     curated_tissue   \n",
       "48    all      all           curated_drug_smiles_code   \n",
       "49    all      all                       hyb_protocol   \n",
       "50    all      all                        platform_id   \n",
       "51    all      all                         is_current   \n",
       "52    all      all              characteristics_ch1_1   \n",
       "\n",
       "                                    Field Description Field Type  Is Array  \n",
       "0                                                  NA       text     False  \n",
       "1   Unique URI derived from source data file's S3 ...       text     False  \n",
       "2              Unique ID associated with every sample       text     False  \n",
       "3          Gene modified through genetic modification       text      True  \n",
       "4                                                  NA       text     False  \n",
       "5      Name of the cohort to which the sample belongs       text     False  \n",
       "6   Signifies whether the given sample is a contro...    integer     False  \n",
       "7   Dataset ID of the file this data entity origin...       text     False  \n",
       "8                                                  NA       text     False  \n",
       "9                                                  NA       text     False  \n",
       "10                                                 NA       text     False  \n",
       "11                                                 NA       text     False  \n",
       "12                                                 NA       text     False  \n",
       "13                Types of cell present in the sample       text      True  \n",
       "14                                                 NA       text     False  \n",
       "15                                                 NA       text     False  \n",
       "16  Unique ID of the dataset to which the sample b...       text      True  \n",
       "17                                                 NA       text     False  \n",
       "18                     Gene of interest in the sample       text      True  \n",
       "19                      Drug admistered in the sample       text      True  \n",
       "20                             Version of data entity    integer     False  \n",
       "21                                                 NA       text     False  \n",
       "22                                                 NA       text     False  \n",
       "23                                                 NA       text     False  \n",
       "24                                                 NA       text     False  \n",
       "25                                                 NA       text     False  \n",
       "26                                                 NA       text     False  \n",
       "27  Name of the data entity (local to its source f...       text     False  \n",
       "28  Signifies the kind of genetic modification per...       text      True  \n",
       "29                                                 NA       text     False  \n",
       "30                                                 NA       text     False  \n",
       "31        Cell line from which the sample was derived       text     False  \n",
       "32                          Description of the sample       text     False  \n",
       "33                                                 NA       text     False  \n",
       "34  Unix timestamp denoting time of creation for t...       text     False  \n",
       "35  Numeric ID of the cohort to which the sample b...    integer     False  \n",
       "36            Unique ID for this data entity on Polly       text     False  \n",
       "37  Name of the repository this data entity origin...       text     False  \n",
       "38  Name of the key that was used for creation of ...       text     False  \n",
       "39                                                 NA       text     False  \n",
       "40                                                 NA       text     False  \n",
       "41                                                 NA       text     False  \n",
       "42                 Disease associated with the sample       text      True  \n",
       "43                                                 NA       text     False  \n",
       "44                                                 NA       text     False  \n",
       "45                                                 NA       text     False  \n",
       "46                                                 NA       text     False  \n",
       "47                 Tissue to which the sample belongs       text     False  \n",
       "48   SMILES code of the drug admistered in the sample       text      True  \n",
       "49                                                 NA       text     False  \n",
       "50                                                 NA       text     False  \n",
       "51  Whether this is the current version of the dat...       text     False  \n",
       "52                                                 NA       text     False  "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "schema.sample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "repo_key = 'any_key'\n",
    "payload = {\n",
    " \"repo_id\": \"any_key\",\n",
    " \"schema_type\": \"files\",\n",
    " \"created_time\": 1649765088197,\n",
    " \"enabled\": True,\n",
    " \"last_updated_time\": 1650421685751,\n",
    " \"schema\": {\n",
    "  \"all\": {\n",
    "    \"author\": {\n",
    "     \"description\": \"Name of the author who published the dataset\",\n",
    "     \"display_name\": \"Author\",\n",
    "     \"filter_size\": 1,\n",
    "     \"is_array\": False,\n",
    "     \"is_column\": False,\n",
    "     \"is_filter\": False,\n",
    "     \"is_keyword\": False,\n",
    "     \"original_name\": \"author\",\n",
    "     \"type\": \"text\"\n",
    "    },\n",
    "    \"bucket\": {\n",
    "     \"description\": \"S3 bucket in which the data file resides\",\n",
    "     \"display_name\": \"Bucket\",\n",
    "     \"filter_size\": 2000,\n",
    "     \"is_array\": False,\n",
    "     \"is_column\": False,\n",
    "     \"is_filter\": False,\n",
    "     \"is_keyword\": True,\n",
    "     \"original_name\": \"kw_bucket\",\n",
    "     \"type\": \"text\"\n",
    "    },\n",
    "    \"curated_cell_line\": {\n",
    "     \"description\": \"Cell lines from which the samples were derived for this dataset\",\n",
    "     \"display_name\": \"Cell line\",\n",
    "     \"filter_size\": 1000,\n",
    "     \"is_array\": True,\n",
    "     \"is_column\": True,\n",
    "     \"is_filter\": True,\n",
    "     \"is_keyword\": True,\n",
    "     \"original_name\": \"kw_cell_line\",\n",
    "     \"type\": \"text\"\n",
    "    },\n",
    "    \"curated_cell_type\": {\n",
    "     \"description\": \"Types of cell present in the dataset\",\n",
    "     \"display_name\": \"Cell type\",\n",
    "     \"filter_size\": 1,\n",
    "     \"is_array\": True,\n",
    "     \"is_column\": True,\n",
    "     \"is_filter\": False,\n",
    "     \"is_keyword\": True,\n",
    "     \"original_name\": \"kw_cell_type\",\n",
    "     \"type\": \"text\"\n",
    "    },\n",
    "    \"curated_disease\": {\n",
    "     \"description\": \"Disease associated with the dataset\",\n",
    "     \"display_name\": \"Disease\",\n",
    "     \"filter_size\": 3000,\n",
    "     \"is_array\": True,\n",
    "     \"is_column\": True,\n",
    "     \"is_filter\": True,\n",
    "     \"is_keyword\": True,\n",
    "     \"original_name\": \"disease\",\n",
    "     \"type\": \"text\"\n",
    "    },\n",
    "    \"curated_drug\": {\n",
    "     \"description\": \"Drugs administered in the samples belonging to the dataset\",\n",
    "     \"display_name\": \"Drug\",\n",
    "     \"filter_size\": 3000,\n",
    "     \"is_array\": True,\n",
    "     \"is_column\": True,\n",
    "     \"is_filter\": True,\n",
    "     \"is_keyword\": True,\n",
    "     \"original_name\": \"kw_drug\",\n",
    "     \"type\": \"text\"\n",
    "    },\n",
    "    \"curated_gene\": {\n",
    "     \"description\": \"Gene studied in the dataset\",\n",
    "     \"display_name\": \"Gene\",\n",
    "     \"filter_size\": 1,\n",
    "     \"is_array\": True,\n",
    "     \"is_column\": False,\n",
    "     \"is_filter\": False,\n",
    "     \"is_keyword\": True,\n",
    "     \"original_name\": \"kw_gene\",\n",
    "     \"type\": \"text\"\n",
    "    },\n",
    "    \"curated_organism\": {\n",
    "     \"description\": \"Orgnism from which the samples were derived\",\n",
    "     \"display_name\": \"Organism\",\n",
    "     \"filter_size\": 1000,\n",
    "     \"is_array\": True,\n",
    "     \"is_column\": True,\n",
    "     \"is_filter\": True,\n",
    "     \"is_keyword\": True,\n",
    "     \"original_name\": \"organism\",\n",
    "     \"type\": \"text\"\n",
    "    },\n",
    "      \"all\": {\n",
    "     \"description\": \"Orgnism from which the samples were derived\",\n",
    "     \"display_name\": \"Organism\",\n",
    "     \"filter_size\": 1000,\n",
    "     \"is_array\": True,\n",
    "     \"is_column\": True,\n",
    "     \"is_filter\": True,\n",
    "     \"is_keyword\": True,\n",
    "     \"original_name\": \"organism\",\n",
    "     \"type\": \"text\"\n",
    "    }\n",
    "   }\n",
    "  }\n",
    " }\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### insert_schema\n",
    "### Input :\n",
    "repo_key: (str) repo_id OR repo_name. This is a mandatory field.<br>\n",
    "payload: (dict) The payload is a JSON file which should be as per the structure defined for schema. Only data-admin will have the authentication to update the schema.\n",
    "<a id=\"insert_schema\"></a>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "omixatlas.insert_schema(repo_key, payload)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### update_schema\n",
    "### Input :\n",
    "repo_key (str): repo_id OR repo_name. This is a mandatory field.<br>\n",
    "payload (dict): The payload is a JSON file which should be as per the structure defined for schema. Only data-admin will have the authentication to update the schema.\n",
    "<a id=\"update_schema\"></a>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "omixatlas.update_schema(repo_key, payload)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
